
<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="ipynb_website:version" content="0.9.7" />
<meta name="viewport" content="width=device-width, initial-scale=1" />

<link rel="stylesheet" type="text/css" href="../css/jt.css">

<link rel="stylesheet" type="text/css" href="../css/toc2.css">

<link href="../site_libs/jqueryui-1.11.4/jquery-ui.css">
<link rel="stylesheet" href="../site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<link rel="stylesheet" href="../site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />
<script type="text/javascript" src="https://ajax.googleapis.com/ajax/libs/jquery/1.9.1/jquery.min.js"></script>
<script type="text/javascript" src="https://ajax.googleapis.com/ajax/libs/jqueryui/1.9.1/jquery-ui.min.js"></script>
<script src="../site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="../site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="../site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>

<link rel="stylesheet"
      href="../site_libs/highlightjs/null.min.css"
      type="text/css" />

<script src="../site_libs/highlightjs/highlight.pack.js"></script>
<script>hljs.initHighlightingOnLoad();</script>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>

<script src="../js/doc_toc.js"></script>
<script src="../js/docs.js"></script>

<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.2/MathJax.js?config=TeX-MML-AM_CHTML"></script>
<script>
    MathJax.Hub.Config({
        extensions: ["tex2jax.js"],
        jax: ["input/TeX", "output/HTML-CSS"],
        tex2jax: {
        inlineMath: [ ['$','$'], ["\\(","\\)"] ],
        displayMath: [ ['$$','$$'], ["\\[","\\]"] ],
        processEscapes: true
        },
        "HTML-CSS": {
            preferredFont: "TeX",
            availableFonts: ["TeX"],
            styles: {
                scale: 110,
                ".MathJax_Display": {
                    "font-size": "110%",
                }
            }
        }
    });
</script>
<script>
function filterDataFrame(id) {
    var input = document.getElementById("search_" + id);
    var filter = input.value.toUpperCase();
    var table = document.getElementById("dataframe_" + id);
    var tr = table.getElementsByTagName("tr");
    // Loop through all table rows, and hide those who don't match the search query
    for (var i = 1; i < tr.length; i++) {
        for (var j = 0; j < tr[i].cells.length; ++j) {
            var matched = false;
            if (tr[i].cells[j].innerHTML.toUpperCase().indexOf(filter) != -1) {
                tr[i].style.display = "";
                matched = true
                break;
            }
            if (!matched)
                tr[i].style.display = "none";
        }
    }
}
function sortDataFrame(id, n, dtype) {
    var table = document.getElementById("dataframe_" + id);
    var tb = table.tBodies[0]; // use `<tbody>` to ignore `<thead>` and `<tfoot>` rows
    var tr = Array.prototype.slice.call(tb.rows, 0); // put rows into array
    if (dtype === 'numeric') {
        var fn = function(a, b) { 
            return parseFloat(a.cells[n].textContent) <= parseFloat(b.cells[n].textContent) ? -1 : 1;
        }
    } else {
        var fn = function(a, b) {
            var c = a.cells[n].textContent.trim().localeCompare(b.cells[n].textContent.trim()); 
            return c > 0 ? 1 : (c < 0 ? -1 : 0) }
    }
    var isSorted = function(array, fn) {
        if (array.length < 2)
            return 1;
        var direction = fn(array[0], array[1]); 
        for (var i = 1; i < array.length - 1; ++i) {
            var d = fn(array[i], array[i+1]);
            if (d == 0)
                continue;
            else if (direction == 0)
                direction = d;
            else if (direction != d)
                return 0;
            }
        return direction;
    }
    var sorted = isSorted(tr, fn);
    if (sorted == 1 || sorted == -1) {
        // if sorted already, reverse it
        for(var i = tr.length - 1; i >= 0; --i)
            tb.appendChild(tr[i]); // append each row in order
    } else {
        tr = tr.sort(fn);
        for(var i = 0; i < tr.length; ++i)
            tb.appendChild(tr[i]); // append each row in order
    }
}
</script>

<script>
$( document ).ready(function(){
            var cfg={'threshold':3,     // depth of toc (number of levels)
             'number_sections': false,
             'toc_cell': false,          // useless here
             'toc_window_display': true, // display the toc window
             "toc_section_display": "block", // display toc contents in the window
             'sideBar':true,       // sidebar or floating window
             'navigate_menu':false       // navigation menu (only in liveNotebook -- do not change)
            }
            var st={};                  // some variables used in the script
            st.rendering_toc_cell = false;
            st.config_loaded = false;
            st.extension_initialized=false;
            st.nbcontainer_marginleft = $('#notebook-container').css('margin-left')
            st.nbcontainer_marginright = $('#notebook-container').css('margin-right')
            st.nbcontainer_width = $('#notebook-container').css('width')
            st.oldTocHeight = undefined
            st.cell_toc = undefined;
            st.toc_index=0;
            // fire the main function with these parameters
            table_of_contents(cfg, st);
            var file=writeupDict[$("h1:first").attr("id")];
            $("#toc-level0 a").css("color","#126dce");
            $('a[href="#'+$("h1:first").attr("id")+'"]').hide()
            var docs=writeupArray;
            var docs_map=writeupArrayMap;
            var pos=writeupArray.indexOf(file);
            for (var a=pos;a>=0;a--){
                  $('<li><a href="'+docs[a]+'.html"><font color="#073642"><b>'+docs_map[docs[a]].replace(/_/g," ")+'</b></font></a></li>').insertBefore("#toc-level0 li:eq(0)");
            }
            $('a[href="'+file+'.html'+'"]').css("color","#126dce");
            for (var a=pos+1;a<docs.length;a++){
                  $(".toc #toc-level0").append('<li><a href="'+docs[a]+'.html"><font color="#073642"><b>'+docs_map[docs[a]].replace(/_/g," ")+'</b></font></a></li>');
            }
            // $("#toc-header").hide(); // comment out because it prevents search bar from displaying
    });
</script>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');
  // mark it active
  menuAnchor.parent().addClass('active');
  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>
<div class="container-fluid main-container">
<!-- tabsets -->
<script src="../site_libs/navigation-1.1/tabsets.js"></script>
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>



<title>Gene Mapping with CNV</title>

<style type = "text/css">
body {
  font-family: "Droid Sans";
  padding-top: 66px;
  padding-bottom: 40px;
}
</style>
</head>

<body>
<div tabindex="-1" id="notebook" class="border-box-sizing">
<div class="container" id="notebook-container">

<!-- code folding -->

<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="../index.html">Gene Mapping with CNV</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        
<li>
  <a href="../index.html">Overview</a>
</li>
        
<li>
  <a href="../analysis.html">Analysis</a>
</li>
        
<li>
  <a href="../prototype.html">Prototype</a>
</li>
        
<li>
  <a href="../dsc.html">Dsc</a>
</li>
        
<li>
  <a href="../setup/index.html">Setup</a>
</li>
        
<li>
  <a href="../writeup.html">Writeup</a>
</li>
        
      </ul>
        
<ul class="nav navbar-nav navbar-right">
<li>
   <a href="http://github.com/gaow/cnv-gene-mapping"> <span class="fa fa-github"></span> </a>
</li>
</ul>
        
      </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div>
<div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h1 id="CNV-Abstract">CNV Abstract<a class="anchor-link" href="#CNV-Abstract">&#182;</a></h1><p>For gene mapping</p>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div>
<div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Me">Me<a class="anchor-link" href="#Me">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div>
<div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p>A novel method to leverage copy number variations (CNVs) for gene mapping for psychiatric disorders</p>
<p>Through almost two decades, genome-wide association studies (GWAS) have been the main strategy to identify susceptible or causal variants or genes and understand underlying mechanisms for complex trait diseases. However, they gained limited achievement for early-onset disorders such as autism. Consequently, another strategy integrating CNVs has recently came into view for psychiatric disorders. CNVs are large genomic duplication or deletion events. Preliminary studies have proved that CNVs are an important source of genetic variation affecting many neurodevelopmental disorders. The challenge emerges that CNVs often span multiple genes, and the susceptible gene(s) is often unclear. We seek to develop a statistical framework that leverage genome-wide CNVs for mapping susceptibility genes of psychiatric diseases, and link these findings with those from single-nucleotide variants (SNVs). The advantage of our method is that its highly integrative characteristic, which can use enrichment results to re-rank genes and avoid a fixed cutoff.</p>
<p>We assume that a disease-related CNV must contain at least one risk gene. By using available CNV data from case-control autism studies, we will infer the posterior probabilities of genes affecting the disease risk. We first match disjoint CNV blocks (i.e. no common CNVs between blocks) with existing genes, especially genes in biological pathways such as calcium signaling pathway, then estimate the prior probabilities of genes by empirical Bayes approach (TORUS???) as well as integrating annotations from biological pathways and genomic functions, and obtain the enrichment of susceptibility genes. Next, we perform gene mapping by an iterative process to calculate the posterior (SuSiE???) of each gene given information from previous steps. Finally, we interpret and combine with SNV-based studies.</p>
<p>To achieve these, we first implement simulations to evaluate the Bayes factors of risk and non-risk genes and compare the power with related methods. All the sampling processes will based on real genomic and CNV data. Then we will apply the methods to real CNV datasets, and evaluate the current and predicted autism risk genes incorporating with SNVs shown signals of association with the disorder.</p>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div>
<div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p>1674W</p>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div>
<div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Gao">Gao<a class="anchor-link" href="#Gao">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div>
<div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p>TITLE: A novel method to leverage copy number variations for gene mapping for psychiatric disorders</p>
<p>AUTHORS: M. Qiao, G. Wang and X. He</p>
<p>For over a decade, genome-wide association studies (GWAS) have been the main strategy to uncover genetic architecture of complex diseases and traits. GWAS typically focus on studying single nucleotide polymorphisms (SNPs). Despite its success in a wide range of disease applications, SNPs thus discovered explain limited heritability in disorders of cognitive development such as Schizophrenia (SCZ). Copy number variations (CNV), on the other hand, are believed to play a critical role in complex disease etiology. CNVs are large genomic duplication or deletion events. Previous studies have suggested that CNVs are an important source of genetic variation affecting many neurodevelopmental disorders. However, since CNVs often span multiple genes, distinguish the susceptible gene(s) from other genes in the same CNV event is a difficult problem. Results of CNV associations are thus difficult to interpret. This hurdles the use of CNV data to unravel the genetics of complex diseases.</p>
<p>To address the challenge we developed a new approach that exploits large-scale genome-wide CNV data to map genes for psychiatric disorders. Unlike existing approaches that directly test for CNV associations, the new method seeks to identify in CNV events susceptibility genes in a rigorous statistical framework. Genome-wide CNV data are first clustered into disjoint analysis blocks, i.e. no CNV spans between any two blocks. For genes within a block we test for disease associations while accounting for correlations with other genes induced by CNV in the same block. We leverage knowledge of known biological pathways, e.g., calcium signaling pathway for SCZ, by applying tools developed for enrichment in GWAS (Wen 2016) to provide prior probability of association for CNV disrupted genes. We then develop dummy coding for genes in different CNV events (insertion, deletion, inversion, etc), and extended recently developed Bayesian variable selection method (Wang et al 2018) to model effect size for these events. Posterior probability of association as well as Bayes factor are obtained for genes, along with 95% credible sets each designed to capture one susceptibility gene. Using this new approach we perform gene-level association testing in several case-control CNV datasets such as SCZ. Our method identifies disease genes from CNV data, and has the potential to integrate with SNP data, elucidating the etiology of complex diseases from independent sources.</p>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div>
<div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Dr.-He">Dr. He<a class="anchor-link" href="#Dr.-He">&#182;</a></h2><p>2619 Characters</p>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div>
<div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p>TITLE: A novel statistical method to identify susceptibility genes from copy number variations</p>
<p>AUTHORS: M. Qiao, G. Wang and X. He</p>
<p>For over a decade, genome-wide association studies (GWAS) have been the main strategy to uncover genetic architecture of complex diseases and traits. GWAS typically focus on studying single nucleotide polymorphisms (SNPs). Despite its success in a wide range of disease applications, SNPs thus discovered explain limited heritability in disorders of cognitive development such as Schizophrenia (SCZ). Copy number variations (CNV), on the other hand, are believed to play a critical role in complex disease etiology. CNVs are large genomic duplication or deletion events. Previous studies have suggested that CNVs are an important source of genetic variation affecting many neurodevelopmental disorders. However, since CNVs often span multiple genes, distinguish the susceptible gene(s) from other genes in the same CNV event is a difficult problem. Results of CNV associations are thus difficult to interpret. This hurdles the use of CNV data to unravel the genetics of complex diseases.</p>
<p>To address the challenge we developed a new approach that exploits large-scale genome-wide CNV data in case-control studies to map genes for psychiatric disorders. Our approach is inspired by statistical fine-mapping of causal variants in linkage-disequilibrium blocks from GWAS. Unlike existing approaches that directly test for CNV associations with the traits, the new method seeks to identify in CNV events true susceptibility genes in a rigorous statistical framework. Genome-wide CNV data are first clustered into disjoint analysis blocks, i.e. no CNV spans between any two blocks. For genes within a block we test for disease associations while accounting for correlations among genes induced by CNV in the same block. We accomplish this by extension of a recently developed Bayesian variable selection method, SuSiE (Wang et al 2018). Our method thus selects a small number of putative risk genes among multiple correlated ones that best explain the CNV-phenotype data. Additionally, we leverage knowledge of known biological pathways to set prior probabilities of genes in CNV events to increase the power. Our model estimates posterior probabilities of all putative risk genes as well as 95% credible sets (i.e. the set of genes that cover all risk genes with high probability). Using this new approach we perform gene-level analysis in several case-control CNV datasets in SCZ. Since our method reports the statistical confidence of genes, it can be easily integrated with other gene-level datasets, e.g. results from exome-sequencing studies. This provides a powerful strategy to integrate data from independent sources.</p>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div>
<div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Final">Final<a class="anchor-link" href="#Final">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div>
<div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p>For over a decade, genome-wide association studies (GWAS) have been the main strategy to uncover genetic architecture of complex diseases. GWAS typically focus on single nucleotide polymorphisms (SNPs). Despite its success in a wide range of disease applications, SNPs thus discovered explain limited heritability in cognitive disorders such as Schizophrenia (SCZ). Copy number variations (CNVs), on the other hand, are believed to play a critical role in complex disease etiology. CNVs are large genomic duplication or deletion events. Previous studies have suggested that CNVs are an important source of genetic variation affecting many neurodevelopmental disorders. However, since CNVs often span multiple genes, distinguish susceptible gene(s) from other genes in the same CNV event is difficult. Results of CNV associations are thus difficult to interpret. This hurdles the use of CNV data to unravel the genetics of complex diseases.</p>
<p>To address the challenge we developed a new approach that exploits large-scale genome-wide CNV data in case-control studies to map genes. It is inspired by statistical fine-mapping of causal variants in linkage-disequilibrium blocks from GWAS. Unlike existing approaches that directly test for CNV associations, our method seeks to identify in CNV events true susceptible genes in a rigorous statistical framework. Genome-wide CNV data are first clustered into disjoint analysis blocks, i.e. no CNV spans between any two blocks. For genes within a block we test for disease associations while accounting for correlations among genes induced by CNV in the same block. We accomplish this by extension of a recently developed Bayesian variable selection method, SuSiE (Wang et al 2018). Our method thus selects a small number of putative risk genes among multiple correlated ones that best explain the CNV-phenotype data. Furthermore, we leverage knowledge of known biological pathways to set prior probabilities of genes in CNV events to increase power. Our model estimates posterior probabilities of all putative risk genes as well as 95% credible sets (i.e. the set of genes that cover all risk genes with high probability). Using this new approach we perform gene-level analysis in several case-control CNV datasets in SCZ. Since our method reports the statistical confidence of genes, it can be integrated with other gene-level datasets, e.g. results from exome-sequencing studies. This provides a powerful strategy to integrate data from independent sources.</p>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div>
<div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="RCC-MindBytes">RCC MindBytes<a class="anchor-link" href="#RCC-MindBytes">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div>
<div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p>Trainee Members Only
ASHG/Charles J. Epstein Trainee Awards for Excellence in Human Genetics Research</p>
<p>This is the only opportunity you will have to apply for the Epstein Trainee Research Awards.</p>
<p>Eligibility requirements:</p>
<ol>
<li>You MUST be an ASHG trainee member for the 2019 calendar year and be the first (presenting) author on this abstract. You can complete the membership application after you submit your abstract.</li>
<li>Your Nominator (see below) MUST be an ASHG member and complete the online nomination form verifying your primary role in the work described in this abstract by June 28, 2019.
If applying for an Epstein Trainee Research Award, select the type of award and provide the name and email address of your Nominator. All fields are required. Please inform your nominator about this submission and remind him/her to complete the form as soon as possible. Questions may be emailed to emantegani@ashg.org.</li>
</ol>
<p>Federal employees should check their policies to see if they are allowed to accept a monetary award prior to applying.</p>

</div>
</div>
</div>
<hr>
&copy 2016-2017 Min Qiao at <a href="http://xinhelab.org">Xin He lab</a>, The University of Chicago
<p><small>Exported from <a href="http://github.com/gaow/cnv-gene-mapping/blob/3a8885b238d6e24d2e25ff2d630f5047592d174c/writeup/20190603_ASHG_Abstract.ipynb"><code>writeup/20190603_ASHG_Abstract.ipynb</code></a> committed by minqiao on Thu Oct 3 23:31:12 2019 <a href="http://github.com/gaow/cnv-gene-mapping/commit/3a8885b238d6e24d2e25ff2d630f5047592d174c">revision 2, 3a8885b</a> <a href="https://stephenslab.github.io/ipynb-website/notes.html#Note-about-commit-ids"><span class="fa fa-question-circle"></span></a></small></p>
</div>
</div>
</body>
</html>
