<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="ipynb_website:version" content="0.9.7" />
<meta name="viewport" content="width=device-width, initial-scale=1" />

<link rel="stylesheet" type="text/css" href="../css/jt.css">

<link rel="stylesheet" type="text/css" href="../css/toc2.css">

<link href="../site_libs/jqueryui-1.11.4/jquery-ui.css">
<link rel="stylesheet" href="../site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<link rel="stylesheet" href="../site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />
<script type="text/javascript" src="https://ajax.googleapis.com/ajax/libs/jquery/1.9.1/jquery.min.js"></script>
<script type="text/javascript" src="https://ajax.googleapis.com/ajax/libs/jqueryui/1.9.1/jquery-ui.min.js"></script>
<script src="../site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="../site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="../site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>

<link rel="stylesheet"
      href="../site_libs/highlightjs/null.min.css"
      type="text/css" />

<script src="../site_libs/highlightjs/highlight.pack.js"></script>
<script>hljs.initHighlightingOnLoad();</script>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>

<script src="../js/doc_toc.js"></script>
<script src="../js/docs.js"></script>

<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.2/MathJax.js?config=TeX-MML-AM_CHTML"></script>
<script>
    MathJax.Hub.Config({
        extensions: ["tex2jax.js"],
        jax: ["input/TeX", "output/HTML-CSS"],
        tex2jax: {
        inlineMath: [ ['$','$'], ["\\(","\\)"] ],
        displayMath: [ ['$$','$$'], ["\\[","\\]"] ],
        processEscapes: true
        },
        "HTML-CSS": {
            preferredFont: "TeX",
            availableFonts: ["TeX"],
            styles: {
                scale: 110,
                ".MathJax_Display": {
                    "font-size": "110%",
                }
            }
        }
    });
</script>
<script>
function filterDataFrame(id) {
    var input = document.getElementById("search_" + id);
    var filter = input.value.toUpperCase();
    var table = document.getElementById("dataframe_" + id);
    var tr = table.getElementsByTagName("tr");
    // Loop through all table rows, and hide those who don't match the search query
    for (var i = 1; i < tr.length; i++) {
        for (var j = 0; j < tr[i].cells.length; ++j) {
            var matched = false;
            if (tr[i].cells[j].innerHTML.toUpperCase().indexOf(filter) != -1) {
                tr[i].style.display = "";
                matched = true
                break;
            }
            if (!matched)
                tr[i].style.display = "none";
        }
    }
}
function sortDataFrame(id, n, dtype) {
    var table = document.getElementById("dataframe_" + id);
    var tb = table.tBodies[0]; // use `<tbody>` to ignore `<thead>` and `<tfoot>` rows
    var tr = Array.prototype.slice.call(tb.rows, 0); // put rows into array
    if (dtype === 'numeric') {
        var fn = function(a, b) { 
            return parseFloat(a.cells[n].textContent) <= parseFloat(b.cells[n].textContent) ? -1 : 1;
        }
    } else {
        var fn = function(a, b) {
            var c = a.cells[n].textContent.trim().localeCompare(b.cells[n].textContent.trim()); 
            return c > 0 ? 1 : (c < 0 ? -1 : 0) }
    }
    var isSorted = function(array, fn) {
        if (array.length < 2)
            return 1;
        var direction = fn(array[0], array[1]); 
        for (var i = 1; i < array.length - 1; ++i) {
            var d = fn(array[i], array[i+1]);
            if (d == 0)
                continue;
            else if (direction == 0)
                direction = d;
            else if (direction != d)
                return 0;
            }
        return direction;
    }
    var sorted = isSorted(tr, fn);
    if (sorted == 1 || sorted == -1) {
        // if sorted already, reverse it
        for(var i = tr.length - 1; i >= 0; --i)
            tb.appendChild(tr[i]); // append each row in order
    } else {
        tr = tr.sort(fn);
        for(var i = 0; i < tr.length; ++i)
            tb.appendChild(tr[i]); // append each row in order
    }
}
</script>

<script>
$( document ).ready(function(){
            var cfg={'threshold':3,     // depth of toc (number of levels)
             'number_sections': false,
             'toc_cell': false,          // useless here
             'toc_window_display': true, // display the toc window
             "toc_section_display": "block", // display toc contents in the window
             'sideBar':true,       // sidebar or floating window
             'navigate_menu':false       // navigation menu (only in liveNotebook -- do not change)
            }
            var st={};                  // some variables used in the script
            st.rendering_toc_cell = false;
            st.config_loaded = false;
            st.extension_initialized=false;
            st.nbcontainer_marginleft = $('#notebook-container').css('margin-left')
            st.nbcontainer_marginright = $('#notebook-container').css('margin-right')
            st.nbcontainer_width = $('#notebook-container').css('width')
            st.oldTocHeight = undefined
            st.cell_toc = undefined;
            st.toc_index=0;
            // fire the main function with these parameters
            table_of_contents(cfg, st);
            var file=writeupDict[$("h1:first").attr("id")];
            $("#toc-level0 a").css("color","#126dce");
            $('a[href="#'+$("h1:first").attr("id")+'"]').hide()
            var docs=writeupArray;
            var docs_map=writeupArrayMap;
            var pos=writeupArray.indexOf(file);
            for (var a=pos;a>=0;a--){
                  $('<li><a href="'+docs[a]+'.html"><font color="#073642"><b>'+docs_map[docs[a]].replace(/_/g," ")+'</b></font></a></li>').insertBefore("#toc-level0 li:eq(0)");
            }
            $('a[href="'+file+'.html'+'"]').css("color","#126dce");
            for (var a=pos+1;a<docs.length;a++){
                  $(".toc #toc-level0").append('<li><a href="'+docs[a]+'.html"><font color="#073642"><b>'+docs_map[docs[a]].replace(/_/g," ")+'</b></font></a></li>');
            }
            // $("#toc-header").hide(); // comment out because it prevents search bar from displaying
    });
</script>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');
  // mark it active
  menuAnchor.parent().addClass('active');
  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>
<div class="container-fluid main-container">
<!-- tabsets -->
<script src="../site_libs/navigation-1.1/tabsets.js"></script>
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>



<title>Gene Mapping with CNV</title>

<style type = "text/css">
body {
  font-family: "Droid Sans";
  padding-top: 66px;
  padding-bottom: 40px;
}
</style>
</head>

<body>
<div tabindex="-1" id="notebook" class="border-box-sizing">
<div class="container" id="notebook-container">

<!-- code folding -->

<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="../index.html">Gene Mapping with CNV</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        
<li>
  <a href="../index.html">Overview</a>
</li>
        
<li>
  <a href="../analysis.html">Analysis</a>
</li>
        
<li>
  <a href="../prototype.html">Prototype</a>
</li>
        
<li>
  <a href="../dsc.html">Dsc</a>
</li>
        
<li>
  <a href="../setup/index.html">Setup</a>
</li>
        
<li>
  <a href="../writeup.html">Writeup</a>
</li>
        
      </ul>
        
<ul class="nav navbar-nav navbar-right">
<li>
   <a href="http://github.com/gaow/cnv-gene-mapping"> <span class="fa fa-github"></span> </a>
</li>
</ul>
        
      </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="CNV-meeting-notes-starting-from-June-2017">CNV meeting notes starting from June 2017<a class="anchor-link" href="#CNV-meeting-notes-starting-from-June-2017">&#182;</a></h3>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h4 id="Notes-for-meeting-06/02/2017">Notes for meeting 06/02/2017<a class="anchor-link" href="#Notes-for-meeting-06/02/2017">&#182;</a></h4><p>*1) Obtain $\alpha$ / $\alpha_0$ from ???</p>
<p>2) Add meaning of the last column in DAP output: log10 Bayes factor of the corresponding SNP in single-SNP testing</p>
<p>3) Add time consumed, begin time and end time</p>
<p>4) Constants added for all patterns disrupted by CNVs are the same, set to 1.0</p>
<p>5) Change the causal gene to the 2nd one or last one instead of 1st one</p>
<p>6) Make configuration data for ???</p>
<p>7) Add two more columns in pattern table: # of each pattern in cases and controls</p>
<p>8) Make toy example for 10 genes: at most one CNV for each sample, i.e. CNV can only disrupt adjacent gene(s)</p>
<p>9) For 4 genes toy example, ignore patterns which contain two CNVs in one sample</p>
<p>10) Set all constant to 0, and observe results generated from variational approximation and DAP</p>
<p>11) Make two or more blocks instead of only one in current toy example:</p>
<p>*** Estimation of $\alpha$: simulation procedure. (1) Many blocks per sample: sample genotypes; (2) Randomly choose a pathway; (3) Create causal gene list based on pathway. (4) Sample phenotypes: if a causal gene is disrupted, a larger $y$.</p>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h4 id="Notes-for-meeting-06/15/2017">Notes for meeting 06/15/2017<a class="anchor-link" href="#Notes-for-meeting-06/15/2017">&#182;</a></h4><p>1) First simulate CNVs: simu CNVs for different lengths: e.g. length (span 1~10 genes) follows geometric distribution; the longer CNV the less common/frequent it is; $0$ means no CNV in this region (wild type), most common in samples.</p>
<p>2) Sample fixed amount of CNVs, say 10, in this region. Binomial sampling these CNVs from 1st step???</p>
<p>3) Then simulate case-control: the number of CNV in cases are supposed to be more than in controls. Samples who have CNV overlapping with causal gene are more likely to be cases, say 80%; samples who have CNV not overlapping with causal gene are equally likely to be cases or controls, i.e. 50%.</p>
<p>4) Obtain $\alpha_0$: prior; Bayes factor: if sample size is small, pip would decrease, need to check Bayes factor</p>
<p>5) Check Swedish data, figure 3/table 1</p>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h4 id="Notes-for-meeting-07/21/2017">Notes for meeting 07/21/2017<a class="anchor-link" href="#Notes-for-meeting-07/21/2017">&#182;</a></h4><p>1) Data transformation: transform binary y {0,1} to {0,t}, for the aim of using linear model to perform logistic regression. $Z_\mu$ is the effect size.</p>
<p>$ P(X \geqslant t) = P(X=Z_\mu+X' \geqslant t) = P(X' \geqslant t-Z_\mu) = 1-\Phi(t-Z_\mu) $</p>
<p>Relative risk (RR):</p>
<p>$ RR = \frac{1-\Phi(t-Z_\mu)}{1-\Phi(t)} $</p>
<p>Solve $Z_\mu$ from a given RR:</p>
<p>$ \Phi(t-Z_\mu) = 1-RR*(1-\Phi(t)) $</p>
<p>2) TORUS prepare DAP input</p>
<p>(1) genesets: caucal/noncausal, one causal geneset and several noncausal genesets, each geneset contains tens to two hundreds genes</p>
<p>(2) check 20170710_dap_on_toy: .feather (we can write feather file) and .dap are needed; generate prior ($\alpha$) using TORUS; grid</p>
<p>3) DAP</p>
<p>4) Real data</p>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p>$f(k)=t, 1-\Phi(t)=k, y \in \{0,t\}$</p>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h4 id="Notes-for-meeting-08/22/2017">Notes for meeting 08/22/2017<a class="anchor-link" href="#Notes-for-meeting-08/22/2017">&#182;</a></h4><p>1) Use TORUS to estimate and simulate $\alpha$</p>
<p>Details: $\textbf{need to add details}$</p>
<p>2) Use "toy simulation of multiple genes and overlapping CNVs in a block (one causal gene)" to perform multi-causal-gene toy.</p>
<p>Perform Fisher test and DAP fine-mapping.</p>
<p>Expectation: DAP is supposed to differentiate the causal genes and their adjecent non-causal genes better than Fisher test, by having smaller PIP than Fisher's p-value; or comparing by amount; or using corresponding cutoff; or transfering p-value to FDR.</p>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h4 id="Notes-for-metting-09/19/2017">Notes for metting 09/19/2017<a class="anchor-link" href="#Notes-for-metting-09/19/2017">&#182;</a></h4><p>1) Dr. Carbonetto's individuall level data or individual TORUS</p>
<p>2) Variational paper, formula 2 in paper, use matlab</p>
<p><a href="https://github.com/pcarbo/bmapathway">github link</a></p>
<p><a href="http://journals.plos.org/plosgenetics/article/file?id=10.1371/journal.pgen.1003770&amp;type=printable">paper link</a></p>
<p>3) DAP</p>
<p>input $\alpha$: set to a fixed value, 0.01~0.1</p>
<p>4) Choose chromosome 16 to preprocess</p>
<p>DAP-g: record running time, and if blocks are long or short; if long, partition into short parts.</p>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h4 id="Notes-for-meeting-11/17/17">Notes for meeting 11/17/17<a class="anchor-link" href="#Notes-for-meeting-11/17/17">&#182;</a></h4><p>Ask Peter about 2, 3, 4</p>
<p>1) Summerize pathway data (format, annotation), test dataset; start from single pathway test.</p>
<p>2) Specify prior $\pi_j$ (from exome sequencing or GWASs).Is it is modularized? Or is it necessary to update $\theta_0$ and $\theta$, or take initial values?</p>
<p>3) Confirm whether $y$ could be binary in $\texttt{varbvs}$.</p>
<p>4) Confirm whether the model includes covariates.</p>
<p>5) We can fix prior $\bar{\beta}$ and $\sigma_{\beta}$; or find the parameter combination with greatest likelihood grid given a grid of them</p>
<p>6) Simulate real dataset: use real genotype and CNVs, simulate phenotype and risk parameters. Use Fisher's test to find signals.</p>
<p>7) How to account for deletion and duplication.</p>

</div>
</div>
</div>
<hr>
&copy 2016-2017 Min Qiao at <a href="http://xinhelab.org">Xin He lab</a>, The University of Chicago
<p><small>Exported from <a href="http://github.com/gaow/cnv-gene-mapping/blob/a7099265a1c18112619416807d8fa5fa85bdb8bd/writeup/CNV_meeting_2017.ipynb"><code>writeup/CNV_meeting_2017.ipynb</code></a> committed by minqiao on Tue Jun 25 13:14:55 2019 <a href="http://github.com/gaow/cnv-gene-mapping/commit/a7099265a1c18112619416807d8fa5fa85bdb8bd">revision 12, a709926</a> <a href="https://stephenslab.github.io/ipynb-website/notes.html#Note-about-commit-ids"><span class="fa fa-question-circle"></span></a></small></p>
</div>
</div>
</body>
</html>
