
<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="ipynb_website:version" content="0.9.4" />
<meta name="viewport" content="width=device-width, initial-scale=1" />

<link rel="stylesheet" type="text/css" href="../css/jt.css">

<link rel="stylesheet" type="text/css" href="../css/toc2.css">

<link href="../site_libs/jqueryui-1.11.4/jquery-ui.css">
<link rel="stylesheet" href="../site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<link rel="stylesheet" href="../site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />
<script type="text/javascript" src="https://ajax.googleapis.com/ajax/libs/jquery/1.9.1/jquery.min.js"></script>
<script type="text/javascript" src="https://ajax.googleapis.com/ajax/libs/jqueryui/1.9.1/jquery-ui.min.js"></script>
<script src="../site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="../site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="../site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>

<link rel="stylesheet"
      href="../site_libs/highlightjs-1.1/textmate.css"
      type="text/css" />

<script src="../site_libs/highlightjs-1.1/highlight.js"></script>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>

<script src="../js/doc_toc.js"></script>
<script src="../js/docs.js"></script>

<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.2/MathJax.js?config=TeX-MML-AM_CHTML"></script>
<script>
    MathJax.Hub.Config({
        extensions: ["tex2jax.js"],
        jax: ["input/TeX", "output/HTML-CSS"],
        tex2jax: {
        inlineMath: [ ['$','$'], ["\\(","\\)"] ],
        displayMath: [ ['$$','$$'], ["\\[","\\]"] ],
        processEscapes: true
        },
        "HTML-CSS": {
            preferredFont: "TeX",
            availableFonts: ["TeX"],
            styles: {
                scale: 110,
                ".MathJax_Display": {
                    "font-size": "110%",
                }
            }
        }
    });
</script>
<script>
function filterDataFrame(id) {
    var input = document.getElementById("search_" + id);
    var filter = input.value.toUpperCase();
    var table = document.getElementById("dataframe_" + id);
    var tr = table.getElementsByTagName("tr");
    // Loop through all table rows, and hide those who don't match the search query
    for (var i = 1; i < tr.length; i++) {
        for (var j = 0; j < tr[i].cells.length; ++j) {
            var matched = false;
            if (tr[i].cells[j].innerHTML.toUpperCase().indexOf(filter) != -1) {
                tr[i].style.display = "";
                matched = true
                break;
            }
            if (!matched)
                tr[i].style.display = "none";
        }
    }
}
function sortDataFrame(id, n, dtype) {
    var table = document.getElementById("dataframe_" + id);
    var tb = table.tBodies[0]; // use `<tbody>` to ignore `<thead>` and `<tfoot>` rows
    var tr = Array.prototype.slice.call(tb.rows, 0); // put rows into array
    if (dtype === 'numeric') {
        var fn = function(a, b) { 
            return parseFloat(a.cells[n].textContent) <= parseFloat(b.cells[n].textContent) ? -1 : 1;
        }
    } else {
        var fn = function(a, b) {
            var c = a.cells[n].textContent.trim().localeCompare(b.cells[n].textContent.trim()); 
            return c > 0 ? 1 : (c < 0 ? -1 : 0) }
    }
    var isSorted = function(array, fn) {
        if (array.length < 2)
            return 1;
        var direction = fn(array[0], array[1]); 
        for (var i = 1; i < array.length - 1; ++i) {
            var d = fn(array[i], array[i+1]);
            if (d == 0)
                continue;
            else if (direction == 0)
                direction = d;
            else if (direction != d)
                return 0;
            }
        return direction;
    }
    var sorted = isSorted(tr, fn);
    if (sorted == 1 || sorted == -1) {
        // if sorted already, reverse it
        for(var i = tr.length - 1; i >= 0; --i)
            tb.appendChild(tr[i]); // append each row in order
    } else {
        tr = tr.sort(fn);
        for(var i = 0; i < tr.length; ++i)
            tb.appendChild(tr[i]); // append each row in order
    }
}
</script>

<script>
$( document ).ready(function(){
            var cfg={'threshold':3,     // depth of toc (number of levels)
             'number_sections': false,
             'toc_cell': false,          // useless here
             'toc_window_display': true, // display the toc window
             "toc_section_display": "block", // display toc contents in the window
             'sideBar':true,       // sidebar or floating window
             'navigate_menu':false       // navigation menu (only in liveNotebook -- do not change)
            }
            var st={};                  // some variables used in the script
            st.rendering_toc_cell = false;
            st.config_loaded = false;
            st.extension_initialized=false;
            st.nbcontainer_marginleft = $('#notebook-container').css('margin-left')
            st.nbcontainer_marginright = $('#notebook-container').css('margin-right')
            st.nbcontainer_width = $('#notebook-container').css('width')
            st.oldTocHeight = undefined
            st.cell_toc = undefined;
            st.toc_index=0;
            // fire the main function with these parameters
            table_of_contents(cfg, st);
            var file=writeupDict[$("h1:first").attr("id")];
            $("#toc-level0 a").css("color","#126dce");
            $('a[href="#'+$("h1:first").attr("id")+'"]').hide()
            var docs=writeupArray;
            var docs_map=writeupArrayMap;
            var pos=writeupArray.indexOf(file);
            for (var a=pos;a>=0;a--){
                  $('<li><a href="'+docs[a]+'.html"><font color="#073642"><b>'+docs_map[docs[a]].replace(/_/g," ")+'</b></font></a></li>').insertBefore("#toc-level0 li:eq(0)");
            }
            $('a[href="'+file+'.html'+'"]').css("color","#126dce");
            for (var a=pos+1;a<docs.length;a++){
                  $(".toc #toc-level0").append('<li><a href="'+docs[a]+'.html"><font color="#073642"><b>'+docs_map[docs[a]].replace(/_/g," ")+'</b></font></a></li>');
            }
            // $("#toc-header").hide(); // comment out because it prevents search bar from displaying
    });
</script>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');
  // mark it active
  menuAnchor.parent().addClass('active');
  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>
<div class="container-fluid main-container">
<!-- tabsets -->
<script src="../site_libs/navigation-1.1/tabsets.js"></script>
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>



<title>Gene Mapping with CNV</title>

<style type = "text/css">
body {
  font-family: "Droid Sans";
  padding-top: 66px;
  padding-bottom: 40px;
}
</style>
</head>

<body>
<div tabindex="-1" id="notebook" class="border-box-sizing">
<div class="container" id="notebook-container">

<!-- code folding -->

<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="../index.html">Gene Mapping with CNV</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        
<li>
  <a href="../index.html">Overview</a>
</li>
        
<li>
  <a href="../analysis.html">Analysis</a>
</li>
        
<li>
  <a href="../prototype.html">Prototype</a>
</li>
        
<li>
  <a href="../utils.html">Utils</a>
</li>
        
<li>
  <a href="../setup.html">Setup</a>
</li>
        
<li>
  <a href="../writeup.html">Writeup</a>
</li>
        
      </ul>
        
<ul class="nav navbar-nav navbar-right">
<li>
   <a href="http://github.com/gaow/cnv-gene-mapping"> <span class="fa fa-github"></span> </a>
</li>
</ul>
        
      </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div>
<div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h1 id="Glossary">Glossary<a class="anchor-link" href="#Glossary">&#182;</a></h1>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div>
<div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Bioinformatics">Bioinformatics<a class="anchor-link" href="#Bioinformatics">&#182;</a></h2><h3 id="what-is-bioinformatics?">what is bioinformatics?<a class="anchor-link" href="#what-is-bioinformatics?">&#182;</a></h3><p>an interdisciplinary field that develops and applies computer and computational technologies to study biomedical questions.</p>
<p>a powerful technology to manage, query and analyze big data in life sciences, a top-down systems approach that generates new hypotheses, finds new patterns, and discovers new functional elements.</p>
<p>studies questions in biology and medicine, develops and applies methods in computer sciences, mathematics, statistics and physics, and overlaps with medical/clinical informatics, systems biology and synthetic biology.</p>
<p>BLAST: basic local alignment search tool, filtering, seeding, extension and evaluation.</p>
<p>BLAT: The BLAST-Like Alignment Tool</p>
<p>SRA: sequence read archive, a bioinformatics database that provides a public repository for DNA sequencing data, especially the "short reads" generated by High-throughput sequencing, typically less than 1,000 base pairs in length. over 1800 trillion nucleotide</p>
<p>NGS data analysis: more efficient and accurate, mapping of reads from resequencing, and calling of genetic variants</p>
<p>GATK: Genome Analysis Toolkit, a structured programming framework designed to enable the rapid development of efficient and robust analysis tools for next-generation DNA sequencers.</p>
<h3 id="alignment">alignment<a class="anchor-link" href="#alignment">&#182;</a></h3><p>global alignment</p>
<p>local alignment</p>
<h3 id="DBMS/database-management-system">DBMS/database management system<a class="anchor-link" href="#DBMS/database-management-system">&#182;</a></h3><p>underlying structure of a database is the data model, which is a collection of conceptual tools for describing data, data relationships, data semantics and constistency constraints.</p>
<p>The data model includes four categories: relational model (most widely used), entity-relationship model, object-based data model and semistructured data model.</p>
<p>In relational model, database is a set of named relations or tables; each relation has a set of named attributes, or columns; each tuple, or row, has a value for each attribute; each attribute has a type or domain.</p>
<h3 id="Bioinformatics-resource">Bioinformatics resource<a class="anchor-link" href="#Bioinformatics-resource">&#182;</a></h3><p>NCBI: national center for biotechnology information, NIH</p>
<p>EBI: European bioinformatics institute</p>
<p>UCSC genome bioinformatics, a good example of centralized genome resource</p>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div>
<div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="CNV-notes">CNV notes<a class="anchor-link" href="#CNV-notes">&#182;</a></h2><h3 id="Structural-variation-(SV)">Structural variation (SV)<a class="anchor-link" href="#Structural-variation-(SV)">&#182;</a></h3><p>Structural variation (also genomic structural variation) is the variation in structure of an organism's chromosome. It consists of many kinds of variation in the genome of one species, and usually includes microscopic and submicroscopic types, such as deletions, duplications, copy-number variants, insertions, inversions and translocations.</p>
<p><a href="https://en.wikipedia.org/wiki/Structural_variation">wiki</a></p>
<h3 id="copy-number-variation">copy number variation<a class="anchor-link" href="#copy-number-variation">&#182;</a></h3><p>Copy number variation (CNVs) is a relatively new field in genomics and it is defined as a phenomenon in which sections of the genome are repeated and the number of repeats in the genome varies between individuals in the human population.
CNVs &gt; 20kb are studied more extensively because they are easier to detect. Most common source of CNVs is NAHR (Non-allelic homologous recombination).</p>
<p><a href="https://en.wikipedia.org/wiki/Copy-number_variation">wiki</a></p>
<p>CNPs are Defined as CNV with frequency &gt; 1% in the population.</p>
<h3 id="De-novo-mutation">De novo mutation<a class="anchor-link" href="#De-novo-mutation">&#182;</a></h3><p>an alteration in a gene that is present for the first time in one family member as a result of a mutation in a germ cell (egg or sperm) of one of the parents or in the fertilized egg itself.</p>
<h3 id="SNV-with-coding-regions">SNV with coding regions<a class="anchor-link" href="#SNV-with-coding-regions">&#182;</a></h3><ol>
<li><p>stop codon gain/nonsense</p>
</li>
<li><p>non-synonymous/missense</p>
</li>
<li><p>synonymous/same sense/silent</p>
</li>
<li><p>affect splicing</p>
</li>
<li><p>stop codon loss</p>
</li>
</ol>
<p>Haploinsufficiency is the phenomenon where a diploid organism has only a single functional copy of a gene (with the other copy inactivated by mutation) and the single functional copy of the gene does not produce enough gene product (typically a protein) to bring about a wild-type condition, leading to an abnormal or diseased state.</p>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div>
<div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="NGS">NGS<a class="anchor-link" href="#NGS">&#182;</a></h2><h3 id="Sanger-sequencing">Sanger sequencing<a class="anchor-link" href="#Sanger-sequencing">&#182;</a></h3><p>To sequence a DNA, add mixture of normal nucleotides and low concentration of ddNTPs. The ddNTPs are fluorescently labeled: each type of ddNTP with a different color.</p>
<h3 id="Next-generation-sequencing/deep-sequencing">Next generation sequencing/deep sequencing<a class="anchor-link" href="#Next-generation-sequencing/deep-sequencing">&#182;</a></h3><p>workflow: alignment, variant calling, postprocessing (filtering/QC)</p>
<h3 id="NGS-Applications">NGS Applications<a class="anchor-link" href="#NGS-Applications">&#182;</a></h3><h4 id="Genomics">Genomics<a class="anchor-link" href="#Genomics">&#182;</a></h4><p>Whole genome sequencing</p>
<p>Whole exome sequencing and targeted sequencing</p>
<h4 id="Main-Applications-of-NGS-in-Genetics">Main Applications of NGS in Genetics<a class="anchor-link" href="#Main-Applications-of-NGS-in-Genetics">&#182;</a></h4><p>Discover rare variants of Mendelian Diseases</p>
<p>Rare variants in common complex diseases, often in case-control settings</p>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div>
<div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Structural-Variation">Structural Variation<a class="anchor-link" href="#Structural-Variation">&#182;</a></h2><p>Structural variation was originally defined as insertions, deletions and inversions greater than 1 kb in size.</p>
<p>SV discovery and genotyping requires accurate prediction of three features: copy, content and structure.</p>
<h3 id="Methods-for-discovery-and-then-for-genotyping">Methods for discovery and then for genotyping<a class="anchor-link" href="#Methods-for-discovery-and-then-for-genotyping">&#182;</a></h3><p>Microarrays have been the experimental workhorse of CNV discovery and genotyping.</p>
<p>These are represented primarily by array comparative genomic hybridization (array CGH) and SNP microarrays.</p>
<h3 id="Hybridization-based-microarray-approaches">Hybridization-based microarray approaches<a class="anchor-link" href="#Hybridization-based-microarray-approaches">&#182;</a></h3><h4 id="Array-CGH">Array CGH<a class="anchor-link" href="#Array-CGH">&#182;</a></h4><p>Array CGH platforms are based on the principle of comparative hybridization of two labeled samples (test and reference) to a set of hybridization targets.</p>
<h4 id="SNP-arrays">SNP arrays<a class="anchor-link" href="#SNP-arrays">&#182;</a></h4><p>a key advantage of SNP microarrays is the use of SNP allele-specific probes to increase CNV sensitivity</p>
<p>SNP arrays have proved popular in CNV-detection studies</p>

</div>
</div>
</div>
<hr>
&copy 2016-2017 Min Qiao at <a href="http://xinhelab.org">Xin He lab</a>, The University of Chicago
<p><small>Exported from <a href="http://github.com/gaow/cnv-gene-mapping/blob/2e5e756a767d6c7785fe8b1fc712677d08c1dd56/writeup/Glossary.ipynb"><code>writeup/Glossary.ipynb</code></a> committed by Min Qiao on Sat Aug 19 09:11:23 2017 <a href="http://github.com/gaow/cnv-gene-mapping/commit/2e5e756a767d6c7785fe8b1fc712677d08c1dd56">revision 1, 2e5e756</a> <a href="https://stephenslab.github.io/ipynb-website/notes.html#Note-about-commit-ids"><span class="fa fa-question-circle"></span></a></small></p>
</div>
</div>
</body>
</html>
