<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="ipynb_website:version" content="0.9.2" />
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link rel="stylesheet" type="text/css" href="../css/jt.css">
<link rel="stylesheet" type="text/css" href="../css/toc2.css">
<link href="../site_libs/jqueryui-1.11.4/jquery-ui.css">
<link rel="stylesheet" href="../site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<link rel="stylesheet" href="../site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />
<script type="text/javascript" src="https://ajax.googleapis.com/ajax/libs/jquery/1.9.1/jquery.min.js"></script>
<script type="text/javascript" src="https://ajax.googleapis.com/ajax/libs/jqueryui/1.9.1/jquery-ui.min.js"></script>
<script src="../site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="../site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="../site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<link rel="stylesheet"
      href="../site_libs/highlight/textmate.css"
      type="text/css" />
<script src="../site_libs/highlight/highlight.js"></script>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>
<script src="../js/toc2.js"></script>
<script src="../js/docs.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS_HTML"></script>
<script>
    MathJax.Hub.Config({
        extensions: ["tex2jax.js"],
        jax: ["input/TeX", "output/HTML-CSS"],
        tex2jax: {
        inlineMath: [ ['$','$'], ["\\(","\\)"] ],
        displayMath: [ ['$$','$$'], ["\\[","\\]"] ],
        processEscapes: true
        },
        "HTML-CSS": {
            preferredFont: "TeX",
            availableFonts: ["TeX"],
            styles: {
                scale: 110,
                ".MathJax_Display": {
                    "font-size": "110%",
                }
            }
        }
    });
</script>
<script>
$( document ).ready(function(){
            var cfg={'threshold':3,     // depth of toc (number of levels)
             'number_sections': false,
             'toc_cell': false,          // useless here
             'toc_window_display': true, // display the toc window
             "toc_section_display": "block", // display toc contents in the window
             'sideBar':true,       // sidebar or floating window
             'navigate_menu':false       // navigation menu (only in liveNotebook -- do not change)
            }
            var st={};                  // some variables used in the script
            st.rendering_toc_cell = false;
            st.config_loaded = false;
            st.extension_initialized=false;
            st.nbcontainer_marginleft = $('#notebook-container').css('margin-left')
            st.nbcontainer_marginright = $('#notebook-container').css('margin-right')
            st.nbcontainer_width = $('#notebook-container').css('width')
            st.oldTocHeight = undefined
            st.cell_toc = undefined;
            st.toc_index=0;
            // fire the main function with these parameters
            table_of_contents(cfg, st);
            var file=analysisDict[$("h1:first").attr("id")];
            $("#toc-level0 a").css("color","#126dce");
            $('a[href="#'+$("h1:first").attr("id")+'"]').hide()
            var docs=analysisArray;
            var pos=analysisArray.indexOf(file);
            for (var a=pos;a>=0;a--){
                  var name=docs[a]
                  $('<li><a href="'+name+'.html"><font color="#073642"><b>'+name.replace(/_/g," ")+'</b></font></a></li>').insertBefore("#toc-level0 li:eq(0)");
            }
            $('a[href="'+file+'.html'+'"]').css("color","#126dce");
            for (var a=pos+1;a<docs.length;a++){
                  var name=docs[a]
                  $(".toc #toc-level0").append('<li><a href="'+name+'.html"><font color="#073642"><b>'+name.replace(/_/g," ")+'</b></font></a></li>');
            }
            $("#toc-header").hide();
    });
</script>
<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');
  // mark it active
  menuAnchor.parent().addClass('active');
  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>
<div class="container-fluid main-container">
<!-- tabsets -->
<script src="../site_libs/navigation-1.1/tabsets.js"></script>
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>
<title>Gene Mapping with CNV</title>
<style type = "text/css">
body {
  font-family: "Droid Sans";
  padding-top: 66px;
  padding-bottom: 40px;
}
</style>
</head>
<body>
<div tabindex="-1" id="notebook" class="border-box-sizing">
<div class="container" id="notebook-container">
<!-- code folding -->
<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="../index.html">Gene Mapping with CNV</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">Overview</a>
</li>
<li>
  <a href="../analysis.html">Analysis</a>
</li>
<li>
  <a href="../prototype.html">Prototype</a>
</li>
<li>
  <a href="../utils.html">Utils</a>
</li>
<li>
  <a href="../setup.html">Setup</a>
</li>
<li>
  <a href="../writeup.html">Writeup</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
   <a href="http://github.com/gaow/cnv-gene-mapping"> <span class="fa fa-github"></span> </a>
</li>
</ul>
      </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->
<div class="cell border-box-sizing text_cell rendered">
<div class="prompt input_prompt">
</div>
<div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h1 id="Implement-TORUS-by-using-CNV-data-of-Schizophrenia-in-Sweden">Implement TORUS by using CNV data of Schizophrenia in Sweden<a class="anchor-link" href="#Implement-TORUS-by-using-CNV-data-of-Schizophrenia-in-Sweden">&#182;</a></h1><h3 id="Use-Sweden-Schizphrenia-individual-data-to-obtain-gene-and-CNV-overlap-configuration;-columns-represent-genes,-rows-represent-individuals">Use Sweden Schizphrenia individual data to obtain gene and CNV overlap configuration; columns represent genes, rows represent individuals<a class="anchor-link" href="#Use-Sweden-Schizphrenia-individual-data-to-obtain-gene-and-CNV-overlap-configuration;-columns-represent-genes,-rows-represent-individuals">&#182;</a></h3><h3 id="Use-the-transformed-file-to-perform-TOURS">Use the transformed file to perform TOURS<a class="anchor-link" href="#Use-the-transformed-file-to-perform-TOURS">&#182;</a></h3>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered">
<div class="prompt input_prompt">
</div>
<div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h4 id="Obtain-file-format-used-in-TORUS">Obtain file format used in TORUS<a class="anchor-link" href="#Obtain-file-format-used-in-TORUS">&#182;</a></h4><p>Use</p>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[1]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="kn">import</span> <span class="nn">pandas</span> <span class="k">as</span> <span class="nn">pd</span>
<span class="kn">from</span> <span class="nn">pandasql</span> <span class="k">import</span> <span class="n">sqldf</span>
<span class="kn">from</span> <span class="nn">utils</span> <span class="k">import</span> <span class="n">load_reference_gene</span><span class="p">,</span> <span class="n">load_data</span><span class="p">,</span> <span class="n">save_data</span><span class="p">,</span> <span class="n">get_analysis_blocks</span>
<span class="kn">import</span> <span class="nn">re</span>
<span class="kn">from</span> <span class="nn">collections</span> <span class="k">import</span> <span class="n">Counter</span>
<span class="kn">import</span> <span class="nn">numpy</span> <span class="k">as</span> <span class="nn">np</span>
<span class="kn">import</span> <span class="nn">fisher</span>
<span class="kn">from</span> <span class="nn">scipy</span> <span class="k">import</span> <span class="n">stats</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[2]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="c1"># load reference genes and Sweden CNV data</span>
<span class="n">ref_genes</span> <span class="o">=</span> <span class="n">load_reference_gene</span><span class="p">(</span><span class="s2">&quot;../data/refGene.txt.gz&quot;</span><span class="p">)</span>
<span class="n">ref_genes</span><span class="p">[</span><span class="s2">&quot;chrom&quot;</span><span class="p">]</span> <span class="o">=</span> <span class="n">ref_genes</span><span class="o">.</span><span class="n">apply</span><span class="p">(</span><span class="k">lambda</span> <span class="n">row</span><span class="p">:</span> <span class="n">row</span><span class="p">[</span><span class="s2">&quot;chrom&quot;</span><span class="p">][</span><span class="mi">3</span><span class="p">:</span><span class="mi">5</span><span class="p">]</span> 
                                     <span class="k">if</span> <span class="n">re</span><span class="o">.</span><span class="n">match</span><span class="p">(</span><span class="s2">&quot;^[c,h,r,0-9]*$&quot;</span><span class="p">,</span> <span class="n">row</span><span class="p">[</span><span class="s2">&quot;chrom&quot;</span><span class="p">][:</span><span class="mi">5</span><span class="p">])</span> <span class="k">else</span> <span class="n">row</span><span class="p">[</span><span class="s2">&quot;chrom&quot;</span><span class="p">][</span><span class="mi">3</span><span class="p">:</span><span class="mi">4</span><span class="p">],</span> <span class="n">axis</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span>
<span class="n">ref_genes</span><span class="p">[</span><span class="s2">&quot;chrom&quot;</span><span class="p">]</span> <span class="o">=</span> <span class="n">ref_genes</span><span class="o">.</span><span class="n">apply</span><span class="p">(</span><span class="k">lambda</span> <span class="n">row</span><span class="p">:</span> <span class="n">row</span><span class="p">[</span><span class="s2">&quot;chrom&quot;</span><span class="p">]</span> <span class="k">if</span> <span class="ow">not</span> <span class="n">row</span><span class="p">[</span><span class="s2">&quot;chrom&quot;</span><span class="p">]</span> <span class="ow">in</span> <span class="p">(</span><span class="s2">&quot;X&quot;</span><span class="p">,</span> <span class="s2">&quot;Y&quot;</span><span class="p">)</span> <span class="k">else</span> <span class="mi">23</span><span class="p">,</span> <span class="n">axis</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[3]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="n">chromosome</span> <span class="o">=</span> <span class="nb">list</span><span class="p">(</span><span class="nb">str</span><span class="p">(</span><span class="n">x</span><span class="p">)</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span><span class="mi">24</span><span class="p">))</span>
<span class="n">ref_genes</span><span class="p">[</span><span class="s2">&quot;chrom&quot;</span><span class="p">]</span> <span class="o">=</span> <span class="n">ref_genes</span><span class="o">.</span><span class="n">apply</span><span class="p">(</span><span class="k">lambda</span> <span class="n">row</span><span class="p">:</span> <span class="n">row</span><span class="p">[</span><span class="s2">&quot;chrom&quot;</span><span class="p">]</span> <span class="k">if</span> <span class="nb">str</span><span class="p">(</span><span class="n">row</span><span class="p">[</span><span class="s2">&quot;chrom&quot;</span><span class="p">])</span> <span class="ow">in</span> <span class="n">chromosome</span> <span class="k">else</span> <span class="mi">0</span><span class="p">,</span> <span class="n">axis</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span>
<span class="n">ref_genes</span> <span class="o">=</span> <span class="n">ref_genes</span><span class="p">[</span><span class="n">ref_genes</span><span class="o">.</span><span class="n">chrom</span> <span class="o">!=</span> <span class="mi">0</span><span class="p">]</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[4]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="n">sw_cnv</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_table</span><span class="p">(</span><span class="s2">&quot;../data/swcnv/swcnv.qc6.cnv&quot;</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">&quot;\s+&quot;</span><span class="p">)</span>
<span class="n">sw_indiv</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_table</span><span class="p">(</span><span class="s2">&quot;../data/swcnv/swcnv.qc6.cnv.indiv&quot;</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">&quot;\s+&quot;</span><span class="p">)</span> <span class="c1"># &quot;NSEG&quot;: number of CNV in each sample</span>
<span class="n">sw_cnv_indiv</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">merge</span><span class="p">(</span><span class="n">sw_cnv</span><span class="p">,</span> <span class="n">sw_indiv</span><span class="p">,</span> <span class="n">how</span><span class="o">=</span><span class="s2">&quot;inner&quot;</span><span class="p">,</span> <span class="n">on</span><span class="o">=</span><span class="p">[</span><span class="s2">&quot;FID&quot;</span><span class="p">,</span> <span class="s2">&quot;IID&quot;</span><span class="p">])</span>
<span class="n">sw_cnv_indiv</span><span class="p">[</span><span class="s2">&quot;PHE&quot;</span><span class="p">]</span> <span class="o">=</span> <span class="n">sw_cnv_indiv</span><span class="o">.</span><span class="n">apply</span><span class="p">(</span><span class="k">lambda</span> <span class="n">row</span><span class="p">:</span> <span class="mi">1</span> <span class="k">if</span> <span class="n">row</span><span class="p">[</span><span class="s2">&quot;PHE&quot;</span><span class="p">]</span><span class="o">==</span><span class="mi">2</span> <span class="k">else</span> <span class="mi">0</span><span class="p">,</span> <span class="n">axis</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[5]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="c1"># calcium pathway genes</span>
<span class="n">calpath_genes</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_table</span><span class="p">(</span><span class="s2">&quot;../data/calciumgeneset.txt&quot;</span><span class="p">,</span> <span class="n">skiprows</span> <span class="o">=</span> <span class="mi">2</span><span class="p">,</span> <span class="n">header</span> <span class="o">=</span> <span class="kc">None</span><span class="p">,</span> <span class="n">names</span> <span class="o">=</span> <span class="p">[</span><span class="s2">&quot;gene_name&quot;</span><span class="p">])</span>
<span class="n">calpath_genes_list</span> <span class="o">=</span> <span class="n">calpath_genes</span><span class="p">[</span><span class="s2">&quot;gene_name&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">tolist</span><span class="p">()</span>
<span class="n">calpath_genes_pos</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">merge</span><span class="p">(</span><span class="n">ref_genes</span><span class="p">,</span> <span class="n">calpath_genes</span><span class="p">,</span> <span class="n">how</span><span class="o">=</span><span class="s2">&quot;inner&quot;</span><span class="p">,</span> <span class="n">on</span><span class="o">=</span><span class="p">[</span><span class="s2">&quot;gene_name&quot;</span><span class="p">])</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[6]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="n">query</span> <span class="o">=</span> <span class="s1">&#39;&#39;&#39;</span>
<span class="s1">SELECT gene_name, chrom, min(tx_start), max(tx_end)</span>
<span class="s1">FROM calpath_genes_pos</span>
<span class="s1">GROUP BY gene_name</span>
<span class="s1">&#39;&#39;&#39;</span>
<span class="n">calpath_genes_pos</span> <span class="o">=</span> <span class="n">sqldf</span><span class="p">(</span><span class="n">query</span><span class="p">)</span>
<span class="n">calpath_genes_pos</span> <span class="o">=</span> <span class="n">calpath_genes_pos</span><span class="o">.</span><span class="n">rename</span><span class="p">(</span><span class="n">columns</span><span class="o">=</span><span class="p">{</span><span class="s2">&quot;min(tx_start)&quot;</span><span class="p">:</span> <span class="s2">&quot;gene_start&quot;</span><span class="p">,</span> <span class="s2">&quot;max(tx_end)&quot;</span><span class="p">:</span> <span class="s2">&quot;gene_end&quot;</span><span class="p">})</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[7]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="n">diff</span> <span class="o">=</span> <span class="nb">list</span><span class="p">(</span><span class="nb">set</span><span class="p">(</span><span class="n">calpath_genes_list</span><span class="p">)</span> <span class="o">-</span> <span class="nb">set</span><span class="p">(</span><span class="n">calpath_genes_pos</span><span class="p">[</span><span class="s2">&quot;gene_name&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">tolist</span><span class="p">()))</span>
<span class="n">calpath_genes_pos</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">calpath_genes_pos</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]]</span> <span class="o">=</span> <span class="p">[</span><span class="n">diff</span><span class="p">[</span><span class="mi">0</span><span class="p">],</span> <span class="s2">&quot;2&quot;</span><span class="p">,</span> <span class="mi">65205076</span><span class="p">,</span> <span class="mi">65206278</span><span class="p">]</span>
<span class="n">calpath_genes_pos</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">calpath_genes_pos</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]]</span> <span class="o">=</span> <span class="p">[</span><span class="n">diff</span><span class="p">[</span><span class="mi">1</span><span class="p">],</span> <span class="s2">&quot;15&quot;</span><span class="p">,</span> <span class="mi">41231149</span><span class="p">,</span> <span class="mi">41281887</span><span class="p">]</span>
<span class="n">calpath_genes_pos</span><span class="p">[</span><span class="s2">&quot;chr&quot;</span><span class="p">]</span> <span class="o">=</span> <span class="n">calpath_genes_pos</span><span class="o">.</span><span class="n">apply</span><span class="p">(</span><span class="k">lambda</span> <span class="n">row</span><span class="p">:</span> <span class="nb">int</span><span class="p">(</span><span class="n">row</span><span class="p">[</span><span class="s2">&quot;chrom&quot;</span><span class="p">])</span> 
                                                <span class="k">if</span> <span class="ow">not</span> <span class="n">row</span><span class="p">[</span><span class="s2">&quot;chrom&quot;</span><span class="p">]</span> <span class="ow">in</span> <span class="p">(</span><span class="s2">&quot;X&quot;</span><span class="p">,</span> <span class="s2">&quot;Y&quot;</span><span class="p">)</span> <span class="k">else</span> <span class="mi">23</span><span class="p">,</span> <span class="n">axis</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[8]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="n">query</span> <span class="o">=</span> <span class="s1">&#39;&#39;&#39;</span>
<span class="s1">SELECT sw.FID, sw.PHE, sw.CHR, sw.BP1, sw.BP2, sw.TYPE, cal.gene_name</span>
<span class="s1">FROM calpath_genes_pos cal LEFT JOIN sw_cnv_indiv sw</span>
<span class="s1">WHERE sw.CHR == cal.chr</span>
<span class="s1">AND (</span>
<span class="s1">(sw.BP1 &gt;= cal.gene_start AND sw.BP1 &lt;= cal.gene_end)</span>
<span class="s1">OR</span>
<span class="s1">(sw.BP2 &gt;= cal.gene_start AND sw.BP2 &lt;= cal.gene_end)</span>
<span class="s1">OR</span>
<span class="s1">(sw.BP1 &lt;= cal.gene_start AND sw.BP2 &gt;= cal.gene_end)</span>
<span class="s1">OR</span>
<span class="s1">(sw.BP1 &gt;= cal.gene_start AND sw.BP2 &lt;= cal.gene_end)</span>
<span class="s1">)</span>
<span class="s1">GROUP BY sw.PHE, sw.CHR, sw.FID, sw.BP1, sw.BP2, sw.TYPE, cal.gene_name</span>
<span class="s1">&#39;&#39;&#39;</span>
<span class="n">cnv_gene_overlap_calpath</span> <span class="o">=</span> <span class="n">sqldf</span><span class="p">(</span><span class="n">query</span><span class="p">)</span><span class="o">.</span><span class="n">drop_duplicates</span><span class="p">(</span><span class="n">subset</span><span class="o">=</span><span class="p">(</span><span class="s2">&quot;FID&quot;</span><span class="p">,</span> <span class="s2">&quot;PHE&quot;</span><span class="p">,</span> <span class="s2">&quot;CHR&quot;</span><span class="p">,</span> <span class="s2">&quot;BP1&quot;</span><span class="p">,</span> <span class="s2">&quot;BP2&quot;</span><span class="p">,</span> <span class="s2">&quot;TYPE&quot;</span><span class="p">,</span> <span class="s2">&quot;gene_name&quot;</span><span class="p">))</span>
<span class="n">cnv_gene_overlap_calpath</span><span class="p">[</span><span class="s2">&quot;tx_name&quot;</span><span class="p">]</span> <span class="o">=</span> <span class="s2">&quot;test_ABCDE&quot;</span>
<span class="nb">print</span> <span class="p">(</span><span class="n">cnv_gene_overlap_calpath</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="output_wrapper">
<div class="output">
<div class="output_area">
<div class="prompt"></div>
<div class="output_subarea output_stream output_stdout output_text">
<pre>         FID  PHE  CHR        BP1        BP2  TYPE gene_name     tx_name
0    PT-8TWI    0    1    1804302    2047584     3    CALML6  test_ABCDE
1    PT-BQT1    0    1  239678115  239800921     3     CHRM3  test_ABCDE
2    PT-BRNL    0    2  218749319  219480897     3     PLCD4  test_ABCDE
3    PT-L1G8    0    2   40674308   41559590     3    SLC8A1  test_ABCDE
4    PT-L31A    0    2   74969763   75323267     3     TACR1  test_ABCDE
5    PT-ESLA    0    3  123486169  123634548     1      MYLK  test_ABCDE
6    PT-FFX5    0    3    4317445    4747247     3     ITPR1  test_ABCDE
7    PT-8U8J    0    4    9688014    9794193     3      DRD5  test_ABCDE
8    PT-OPJ8    0    4  104179046  104766986     1     TACR3  test_ABCDE
9    PT-27QK    0    6  118701889  118900024     3       PLN  test_ABCDE
10   PT-29AJ    0    6  118853079  119033109     3       PLN  test_ABCDE
11   PT-2M7G    0    6  118842321  119039482     3       PLN  test_ABCDE
12   PT-8W9S    0    6  118835126  119032859     3       PLN  test_ABCDE
13   PT-8WJQ    0    6  118691108  119101336     3       PLN  test_ABCDE
14   PT-9YVZ    0    6  118835126  119033109     3       PLN  test_ABCDE
15   PT-BPIQ    0    6  118701889  118900024     3       PLN  test_ABCDE
16   PT-ESKZ    0    6  118786486  119048269     3       PLN  test_ABCDE
17   PT-ET7H    0    6  118868953  119048269     3       PLN  test_ABCDE
18   PT-FFKU    0    6  118786486  119056733     3       PLN  test_ABCDE
19   PT-FG7N    0    6  118719413  118891582     3       PLN  test_ABCDE
20   PT-FGHQ    0    6  118695049  118891582     3       PLN  test_ABCDE
21   PT-OO6J    0    6  118690822  119080379     3       PLN  test_ABCDE
22   PT-OPDH    0    6  118791804  119018502     3       PLN  test_ABCDE
23   PT-OPY7    0    6  118700137  118891582     3       PLN  test_ABCDE
24   PT-8VX9    0    7   32209359   32426450     3     PDE1C  test_ABCDE
25   PT-8VYD    0    7   32210926   32423399     3     PDE1C  test_ABCDE
26   PT-8WP8    0    7   32209359   32431274     3     PDE1C  test_ABCDE
27   PT-BQAQ    0    7   32210926   32423399     3     PDE1C  test_ABCDE
28   PT-BQH5    0    7   32210926   32412682     3     PDE1C  test_ABCDE
29   PT-FFVB    0    7   32209459   32423399     3     PDE1C  test_ABCDE
..       ...  ...  ...        ...        ...   ...       ...         ...
164  PT-3MHL    1   22   18876428   21465835     1     P2RX6  test_ABCDE
165  PT-8WLE    1   22   18876428   21465835     1     P2RX6  test_ABCDE
166  PT-8WQ7    1   22   18876428   21465835     1     P2RX6  test_ABCDE
167  PT-9ZCE    1   22   18876428   21542615     1     P2RX6  test_ABCDE
168  PT-BQ5V    1   22   22997811   25076598     3   ADORA2A  test_ABCDE
169  PT-BQA6    1   22   22997811   24997309     3   ADORA2A  test_ABCDE
170  PT-BRL3    1   22   23689983   25070067     3   ADORA2A  test_ABCDE
171  PT-BS88    1   22   23689983   24914451     3   ADORA2A  test_ABCDE
172  PT-ERRP    1   22   19634877   21463730     1     P2RX6  test_ABCDE
173  PT-ERRZ    1   22   18861748   21462353     1     P2RX6  test_ABCDE
174  PT-ISV8    1   22   20753615   21463730     1     P2RX6  test_ABCDE
175  PT-ITRD    1   22   23692738   25064809     3   ADORA2A  test_ABCDE
176  PT-OPSY    1   22   20789074   21463730     1     P2RX6  test_ABCDE
177  PT-OPXM    1   22   24447288   25032774     3   ADORA2A  test_ABCDE
178  PT-1RZS    1   23    2703391    4505504     3      PRKX  test_ABCDE
179  PT-1S19    1   23    2850801    3659193     3      PRKX  test_ABCDE
180  PT-1S1P    1   23    2916735    3659193     3      PRKX  test_ABCDE
181  PT-1S1Y    1   23    2850801    3637916     3      PRKX  test_ABCDE
182  PT-1S66    1   23    2703391    3659193     3      PRKX  test_ABCDE
183  PT-1S6I    1   23    2741339    3649902     3      PRKX  test_ABCDE
184  PT-1S7B    1   23    2741339    3649902     3      PRKX  test_ABCDE
185  PT-1S7C    1   23    2916735    3539248     3      PRKX  test_ABCDE
186  PT-1S7P    1   23    2979529    3659193     3      PRKX  test_ABCDE
187  PT-1S7Z    1   23    2741339    3649902     3      PRKX  test_ABCDE
188  PT-1S8F    1   23    2839942    3678164     3      PRKX  test_ABCDE
189  PT-1SBT    1   23    2811119    7355636     3      PRKX  test_ABCDE
190  PT-1SC1    1   23    2802568    4859798     3      PRKX  test_ABCDE
191  PT-1SC9    1   23    3070182    4736479     3      PRKX  test_ABCDE
192  PT-1SCR    1   23    2979529    3539248     3      PRKX  test_ABCDE
193  PT-FFLF    1   23    2712661    7044081     1      PRKX  test_ABCDE
[194 rows x 8 columns]
</pre>
</div>
</div>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[10]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="c1"># cnv_gene_overlap_calpath.reset_index(inplace=True)</span>
<span class="c1"># cnv_gene_overlap_calpath = cnv_gene_overlap_calpath.groupby((&quot;PHE&quot;, &quot;FID&quot;, &quot;CHR&quot;, &quot;BP1&quot;, &quot;BP2&quot;, &quot;TYPE&quot;))</span>
<span class="c1"># cnv_gene_overlap_calpath = cnv_gene_overlap_calpath[&quot;gene_name&quot;].unique()</span>
<span class="c1"># cnv_gene_overlap_calpath = cnv_gene_overlap_calpath.reset_index()</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[11]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="n">query</span> <span class="o">=</span> <span class="s1">&#39;&#39;&#39;</span>
<span class="s1">SELECT chrom, gene_name, tx_name, min(tx_start), max(tx_end)</span>
<span class="s1">FROM ref_genes</span>
<span class="s1">GROUP BY chrom, gene_name</span>
<span class="s1">&#39;&#39;&#39;</span>
<span class="n">ref_genes</span> <span class="o">=</span> <span class="n">sqldf</span><span class="p">(</span><span class="n">query</span><span class="p">)</span><span class="o">.</span><span class="n">drop_duplicates</span><span class="p">(</span><span class="n">subset</span> <span class="o">=</span> <span class="p">(</span><span class="s2">&quot;gene_name&quot;</span><span class="p">))</span>
<span class="n">ref_genes</span> <span class="o">=</span> <span class="n">ref_genes</span><span class="o">.</span><span class="n">rename</span><span class="p">(</span><span class="n">columns</span><span class="o">=</span><span class="p">{</span><span class="s2">&quot;min(tx_start)&quot;</span><span class="p">:</span> <span class="s2">&quot;gene_start&quot;</span><span class="p">,</span> <span class="s2">&quot;max(tx_end)&quot;</span><span class="p">:</span> <span class="s2">&quot;gene_end&quot;</span><span class="p">})</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered">
<div class="prompt input_prompt">
</div>
<div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h4 id="The-table-of-overlap-of-CNVs-(Sweden-data)-and-genes">The table of overlap of CNVs (Sweden data) and genes<a class="anchor-link" href="#The-table-of-overlap-of-CNVs-(Sweden-data)-and-genes">&#182;</a></h4>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[12]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="n">query</span> <span class="o">=</span> <span class="s1">&#39;&#39;&#39;</span>
<span class="s1">SELECT sw.FID, sw.CHR, sw.BP1, sw.BP2, ref.gene_name, ref.tx_name, sw.PHE, sw.TYPE</span>
<span class="s1">FROM ref_genes ref LEFT JOIN sw_cnv_indiv sw</span>
<span class="s1">WHERE sw.CHR == ref.chrom</span>
<span class="s1">AND (</span>
<span class="s1">(sw.BP1 &gt;= ref.gene_start AND sw.BP1 &lt;= ref.gene_end)</span>
<span class="s1">OR</span>
<span class="s1">(sw.BP2 &gt;= ref.gene_start AND sw.BP2 &lt;= ref.gene_end)</span>
<span class="s1">OR</span>
<span class="s1">(sw.BP1 &lt;= ref.gene_start AND sw.BP2 &gt;= ref.gene_end)</span>
<span class="s1">OR</span>
<span class="s1">(sw.BP1 &gt;= ref.gene_start AND sw.BP2 &lt;= ref.gene_end)</span>
<span class="s1">)</span>
<span class="s1">GROUP BY sw.PHE, sw.CHR, sw.FID, sw.BP1, sw.BP2, sw.TYPE, ref.gene_name</span>
<span class="s1">&#39;&#39;&#39;</span>
<span class="n">cnv_gene_overlap</span> <span class="o">=</span> <span class="n">sqldf</span><span class="p">(</span><span class="n">query</span><span class="p">)</span><span class="o">.</span><span class="n">drop_duplicates</span><span class="p">(</span><span class="n">subset</span><span class="o">=</span><span class="p">(</span><span class="s2">&quot;FID&quot;</span><span class="p">,</span> <span class="s2">&quot;PHE&quot;</span><span class="p">,</span> <span class="s2">&quot;CHR&quot;</span><span class="p">,</span> <span class="s2">&quot;BP1&quot;</span><span class="p">,</span> <span class="s2">&quot;BP2&quot;</span><span class="p">,</span> <span class="s2">&quot;TYPE&quot;</span><span class="p">,</span> <span class="s2">&quot;gene_name&quot;</span><span class="p">))</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[13]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="k">def</span> <span class="nf">get_block_matrix</span><span class="p">(</span><span class="n">input_data</span><span class="p">,</span> <span class="n">ref_genes</span><span class="p">,</span> <span class="n">deletion</span> <span class="o">=</span> <span class="kc">True</span><span class="p">,</span> <span class="n">make_block</span> <span class="o">=</span> <span class="kc">False</span><span class="p">,</span> <span class="n">dtype</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">uint8</span><span class="p">):</span>
    <span class="sd">&#39;&#39;&#39;input_data is a dataframe from a query of overlapping of CNVs and genes. Controls come first, cases follow.</span>
<span class="sd">       Therefore, controls (0) come first in &quot;phenotype&quot;, cases (1) follow, </span>
<span class="sd">       in order to coordinate the format of input data.</span>
<span class="sd">       The input_data may contain deletion and duplication. Separate them if needed.</span>
<span class="sd">       ref_genes is a dataframe from a query of reference genes</span>
<span class="sd">    &#39;&#39;&#39;</span>
    <span class="k">if</span> <span class="n">deletion</span><span class="p">:</span>
        <span class="n">dat</span> <span class="o">=</span> <span class="n">input_data</span><span class="p">[</span><span class="n">input_data</span><span class="p">[</span><span class="s2">&quot;TYPE&quot;</span><span class="p">]</span><span class="o">==</span><span class="mi">1</span><span class="p">]</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="n">dat</span> <span class="o">=</span> <span class="n">input_data</span><span class="p">[</span><span class="n">input_data</span><span class="p">[</span><span class="s2">&quot;TYPE&quot;</span><span class="p">]</span><span class="o">==</span><span class="mi">3</span><span class="p">]</span>
    <span class="n">n_case</span> <span class="o">=</span> <span class="n">dat</span><span class="p">[</span><span class="n">dat</span><span class="p">[</span><span class="s2">&quot;PHE&quot;</span><span class="p">]</span><span class="o">==</span><span class="mi">1</span><span class="p">]</span><span class="o">.</span><span class="n">drop_duplicates</span><span class="p">(</span><span class="n">subset</span><span class="o">=</span><span class="p">(</span><span class="s2">&quot;FID&quot;</span><span class="p">))</span><span class="o">.</span><span class="n">groupby</span><span class="p">(</span><span class="s2">&quot;PHE&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">size</span><span class="p">()[</span><span class="mi">1</span><span class="p">]</span>
    <span class="n">n_ctrl</span> <span class="o">=</span> <span class="n">dat</span><span class="p">[</span><span class="n">dat</span><span class="p">[</span><span class="s2">&quot;PHE&quot;</span><span class="p">]</span><span class="o">==</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">drop_duplicates</span><span class="p">(</span><span class="n">subset</span><span class="o">=</span><span class="p">(</span><span class="s2">&quot;FID&quot;</span><span class="p">))</span><span class="o">.</span><span class="n">groupby</span><span class="p">(</span><span class="s2">&quot;PHE&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">size</span><span class="p">()[</span><span class="mi">0</span><span class="p">]</span>
    <span class="nb">print</span> <span class="p">(</span><span class="n">n_case</span> <span class="o">+</span> <span class="n">n_ctrl</span><span class="p">)</span>
    <span class="n">dat_select</span> <span class="o">=</span> <span class="n">dat</span><span class="p">[[</span><span class="s2">&quot;CHR&quot;</span><span class="p">,</span> <span class="s2">&quot;BP1&quot;</span><span class="p">,</span> <span class="s2">&quot;BP2&quot;</span><span class="p">,</span> <span class="s2">&quot;tx_name&quot;</span><span class="p">,</span> <span class="s2">&quot;gene_name&quot;</span><span class="p">]]</span>
    <span class="n">dat_select</span> <span class="o">=</span> <span class="n">dat_select</span><span class="o">.</span><span class="n">rename</span><span class="p">(</span><span class="n">columns</span><span class="o">=</span><span class="p">{</span><span class="s2">&quot;CHR&quot;</span><span class="p">:</span> <span class="s2">&quot;chrom&quot;</span><span class="p">,</span> <span class="s2">&quot;BP1&quot;</span><span class="p">:</span> <span class="s2">&quot;cnv_start&quot;</span><span class="p">,</span> <span class="s2">&quot;BP2&quot;</span><span class="p">:</span> <span class="s2">&quot;cnv_terminate&quot;</span><span class="p">})</span>
    <span class="n">case</span> <span class="o">=</span> <span class="n">dat_select</span><span class="p">[</span><span class="n">dat</span><span class="p">[</span><span class="s2">&quot;PHE&quot;</span><span class="p">]</span><span class="o">==</span><span class="mi">1</span><span class="p">]</span>
    <span class="n">ctrl</span> <span class="o">=</span> <span class="n">dat_select</span><span class="p">[</span><span class="n">dat</span><span class="p">[</span><span class="s2">&quot;PHE&quot;</span><span class="p">]</span><span class="o">==</span><span class="mi">0</span><span class="p">]</span>
    <span class="n">genes</span> <span class="o">=</span> <span class="n">ref_genes</span><span class="p">[</span><span class="s2">&quot;gene_name&quot;</span><span class="p">]</span>
    <span class="nb">print</span> <span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">genes</span><span class="o">.</span><span class="n">tolist</span><span class="p">()))</span>
    <span class="n">regression_data</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">array</span><span class="p">([</span><span class="n">np</span><span class="o">.</span><span class="n">array</span><span class="p">(</span><span class="n">genes</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span><span class="n">dat</span><span class="p">[</span><span class="n">dat</span><span class="p">[</span><span class="s2">&quot;FID&quot;</span><span class="p">]</span><span class="o">==</span><span class="n">item</span><span class="p">][</span><span class="s2">&quot;gene_name&quot;</span><span class="p">]),</span> <span class="n">dtype</span><span class="o">=</span><span class="nb">float</span><span class="p">)</span> 
                                <span class="k">for</span> <span class="n">item</span> <span class="ow">in</span> <span class="n">dat</span><span class="o">.</span><span class="n">drop_duplicates</span><span class="p">(</span><span class="n">subset</span><span class="o">=</span><span class="p">(</span><span class="s2">&quot;FID&quot;</span><span class="p">))[</span><span class="s2">&quot;FID&quot;</span><span class="p">]])</span>
    <span class="n">phenotype</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">matrix</span><span class="p">([</span><span class="mi">0</span><span class="p">]</span><span class="o">*</span><span class="n">n_ctrl</span> <span class="o">+</span> <span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="o">*</span><span class="n">n_case</span><span class="p">)</span><span class="o">.</span><span class="n">T</span>
    <span class="n">regression_data</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">hstack</span><span class="p">((</span><span class="n">phenotype</span><span class="p">,</span> <span class="n">regression_data</span><span class="p">))</span>
    <span class="nb">print</span> <span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">shape</span><span class="p">(</span><span class="n">regression_data</span><span class="p">))</span>
    <span class="n">df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">(</span><span class="n">regression_data</span><span class="p">,</span> <span class="n">columns</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;phenotype&#39;</span><span class="p">]</span> <span class="o">+</span> <span class="n">genes</span><span class="o">.</span><span class="n">tolist</span><span class="p">())</span>
    <span class="n">newdf</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">()</span>
    <span class="k">for</span> <span class="n">col</span> <span class="ow">in</span> <span class="n">df</span><span class="p">:</span>
        <span class="k">if</span> <span class="nb">sum</span><span class="p">(</span><span class="n">df</span><span class="p">[</span><span class="n">col</span><span class="p">])</span> <span class="o">&gt;</span> <span class="mi">0</span><span class="p">:</span>
            <span class="n">newdf</span><span class="p">[</span><span class="n">col</span><span class="p">]</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="n">col</span><span class="p">]</span>
        <span class="k">else</span><span class="p">:</span> <span class="k">continue</span>
    <span class="k">if</span> <span class="ow">not</span> <span class="n">make_block</span><span class="p">:</span>
        <span class="n">res</span> <span class="o">=</span> <span class="n">newdf</span><span class="o">.</span><span class="n">astype</span><span class="p">(</span><span class="n">dtype</span><span class="p">,</span> <span class="n">copy</span> <span class="o">=</span> <span class="kc">True</span><span class="p">)</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="n">blocks</span> <span class="o">=</span> <span class="n">get_analysis_blocks</span><span class="p">(</span><span class="n">pd</span><span class="o">.</span><span class="n">concat</span><span class="p">([</span><span class="n">case</span><span class="p">,</span> <span class="n">ctrl</span><span class="p">]))</span>
        <span class="n">res</span> <span class="o">=</span> <span class="p">[</span><span class="n">newdf</span><span class="p">[[</span><span class="s1">&#39;phenotype&#39;</span><span class="p">]</span> <span class="o">+</span> <span class="n">item</span><span class="p">]</span><span class="o">.</span><span class="n">astype</span><span class="p">(</span><span class="n">dtype</span><span class="p">,</span> <span class="n">copy</span> <span class="o">=</span> <span class="kc">True</span><span class="p">)</span> <span class="k">for</span> <span class="n">item</span> <span class="ow">in</span> <span class="n">blocks</span><span class="p">]</span>
    <span class="n">save_data</span><span class="p">(</span><span class="n">res</span><span class="p">,</span> <span class="s2">&quot;data/calcium_pathway_Sweden_CNVs_</span><span class="si">{}</span><span class="s2">.data.blocks.pkl&quot;</span><span class="o">.</span><span class="n">format</span><span class="p">(</span><span class="s2">&quot;del&quot;</span> <span class="k">if</span> <span class="n">deletion</span> <span class="k">else</span> <span class="s2">&quot;dup&quot;</span><span class="p">))</span>
    <span class="k">return</span> <span class="p">{</span><span class="s2">&quot;data&quot;</span><span class="p">:</span> <span class="n">res</span><span class="p">}</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[14]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="n">res</span> <span class="o">=</span> <span class="n">get_block_matrix</span><span class="p">(</span><span class="n">cnv_gene_overlap_calpath</span><span class="p">,</span> <span class="n">ref_genes</span><span class="p">,</span> <span class="n">make_block</span> <span class="o">=</span> <span class="kc">True</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="output_wrapper">
<div class="output">
<div class="output_area">
<div class="prompt"></div>
<div class="output_subarea output_stream output_stdout output_text">
<pre>47
26910
(47, 26911)
</pre>
</div>
</div>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered">
<div class="prompt input_prompt">
</div>
<div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Run-TORUS">Run TORUS<a class="anchor-link" href="#Run-TORUS">&#182;</a></h3>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[15]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="c1"># from get_summary_for_torus import get_sum_stats_cal_simu as get_sum_Sweden_for_torus</span>
<span class="n">cal_pthwy_genes</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_table</span><span class="p">(</span><span class="s2">&quot;../data/calciumgeneset.txt&quot;</span><span class="p">,</span> <span class="n">skiprows</span> <span class="o">=</span> <span class="mi">2</span><span class="p">,</span> <span class="n">header</span> <span class="o">=</span> <span class="kc">None</span><span class="p">,</span> <span class="n">names</span> <span class="o">=</span> <span class="p">[</span><span class="s2">&quot;gene_name&quot;</span><span class="p">])</span>
<span class="n">causal_genes</span> <span class="o">=</span> <span class="n">cal_pthwy_genes</span><span class="p">[</span><span class="s2">&quot;gene_name&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">tolist</span><span class="p">()</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[16]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="n">col</span> <span class="o">=</span> <span class="p">[]</span>
<span class="k">for</span> <span class="n">item</span> <span class="ow">in</span> <span class="n">res</span><span class="p">[</span><span class="s2">&quot;data&quot;</span><span class="p">]:</span>
    <span class="n">col</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">shape</span><span class="p">(</span><span class="n">item</span><span class="p">)[</span><span class="mi">1</span><span class="p">])</span>
<span class="p">[</span><span class="n">i</span> <span class="k">for</span> <span class="n">i</span><span class="p">,</span> <span class="n">j</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">col</span><span class="p">)</span> <span class="k">if</span> <span class="n">j</span> <span class="o">==</span> <span class="nb">max</span><span class="p">(</span><span class="n">col</span><span class="p">)]</span>
</pre></div>
</div>
</div>
</div>
<div class="output_wrapper">
<div class="output">
<div class="output_area">
<div class="prompt output_prompt">Out[16]:</div>
<div class="output_text output_subarea output_execute_result">
<pre>[0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18]</pre>
</div>
</div>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[17]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="k">for</span> <span class="n">index</span><span class="p">,</span> <span class="n">item</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">res</span><span class="p">[</span><span class="s2">&quot;data&quot;</span><span class="p">]):</span>
    <span class="nb">print</span> <span class="p">(</span><span class="n">index</span><span class="p">,</span> <span class="p">[</span><span class="n">gene</span> <span class="k">for</span> <span class="n">gene</span> <span class="ow">in</span> <span class="n">item</span><span class="o">.</span><span class="n">columns</span><span class="o">.</span><span class="n">values</span><span class="p">[</span><span class="mi">1</span><span class="p">:]</span><span class="o">.</span><span class="n">tolist</span><span class="p">()</span> <span class="k">if</span> <span class="n">gene</span> <span class="ow">in</span> <span class="n">causal_genes</span><span class="p">])</span>
</pre></div>
</div>
</div>
</div>
<div class="output_wrapper">
<div class="output">
<div class="output_area">
<div class="prompt"></div>
<div class="output_subarea output_stream output_stdout output_text">
<pre>0 [&#39;ITPKB&#39;]
1 [&#39;RYR2&#39;]
2 [&#39;SLC8A1&#39;]
3 [&#39;PDE1A&#39;]
4 [&#39;MYLK&#39;]
5 [&#39;PPP3CA&#39;]
6 [&#39;TACR3&#39;]
7 [&#39;PDE1C&#39;]
8 [&#39;CACNA1B&#39;]
9 [&#39;P2RX2&#39;]
10 [&#39;ITPR2&#39;]
11 [&#39;BDKRB2&#39;]
12 [&#39;CHRNA7&#39;]
13 [&#39;ATP2A1&#39;]
14 [&#39;PHKB&#39;]
15 [&#39;PRKACA&#39;]
16 [&#39;PLCB1&#39;]
17 [&#39;P2RX6&#39;]
18 [&#39;PRKX&#39;]
</pre>
</div>
</div>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="n">max_block</span> <span class="o">=</span> <span class="n">res</span><span class="p">[</span><span class="s2">&quot;data&quot;</span><span class="p">][</span><span class="mi">25</span><span class="p">]</span>
<span class="n">block_genes</span> <span class="o">=</span> <span class="n">max_block</span><span class="o">.</span><span class="n">columns</span><span class="o">.</span><span class="n">values</span><span class="p">[</span><span class="mi">1</span><span class="p">:]</span><span class="o">.</span><span class="n">tolist</span><span class="p">()</span>
<span class="n">pheno</span> <span class="o">=</span> <span class="p">[]</span>
<span class="k">for</span> <span class="n">gene</span> <span class="ow">in</span> <span class="n">block_genes</span><span class="p">:</span>
    <span class="k">if</span> <span class="n">gene</span> <span class="ow">in</span> <span class="n">causal_genes</span><span class="p">:</span>
        <span class="n">pheno</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="mi">1</span><span class="p">)</span>
    <span class="k">else</span><span class="p">:</span> 
        <span class="n">pheno</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="mi">0</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="k">def</span> <span class="nf">get_sum_Sweden_for_torus</span><span class="p">(</span><span class="n">gene_df</span><span class="p">,</span> <span class="n">causal</span><span class="p">,</span> <span class="n">cat</span><span class="p">,</span> <span class="n">multi_cat</span> <span class="o">=</span> <span class="mi">0</span><span class="p">,</span> <span class="n">subst</span> <span class="o">=</span> <span class="o">-</span><span class="mi">10</span><span class="p">,</span> <span class="n">chrom</span> <span class="o">=</span> <span class="s2">&quot;chr6&quot;</span><span class="p">,</span> <span class="n">seed</span> <span class="o">=</span> <span class="mi">999</span><span class="p">):</span>
    <span class="sd">&#39;&#39;&#39;get summary statistics (e.g. t-statistic, p-value) to be used for TORUS, then transfer p-value to z-score</span>
<span class="sd">       Actually do NOT need SNP (gene) map and gene (chromosome) map</span>
<span class="sd">       If multi_cat!=0, then cat must be multiple categories</span>
<span class="sd">       If multi_cat==0, then cat can be any list (because it won&#39;t be used) but must be given</span>
<span class="sd">    &#39;&#39;&#39;</span>
    <span class="n">np</span><span class="o">.</span><span class="n">random</span><span class="o">.</span><span class="n">seed</span><span class="p">(</span><span class="n">seed</span><span class="p">)</span>
    <span class="n">cases</span> <span class="o">=</span> <span class="n">gene_df</span><span class="p">[</span><span class="n">gene_df</span><span class="p">[</span><span class="s2">&quot;phenotype&quot;</span><span class="p">]</span><span class="o">==</span><span class="mi">1</span><span class="p">]</span>
    <span class="n">ctrls</span> <span class="o">=</span> <span class="n">gene_df</span><span class="p">[</span><span class="n">gene_df</span><span class="p">[</span><span class="s2">&quot;phenotype&quot;</span><span class="p">]</span><span class="o">==</span><span class="mi">0</span><span class="p">]</span>
    <span class="n">df_cal_bf</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">()</span>
    <span class="n">df_cal_beta</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">()</span>
    <span class="n">df_cal_zscore</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">()</span>
    <span class="n">SNP_anno_cal</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">()</span>
    <span class="n">i</span> <span class="o">=</span> <span class="mi">1</span>
    <span class="k">for</span> <span class="n">gene</span> <span class="ow">in</span> <span class="nb">list</span><span class="p">(</span><span class="n">gene_df</span><span class="p">)[</span><span class="mi">1</span><span class="p">:]:</span>
        <span class="nb">print</span> <span class="p">(</span><span class="n">gene</span><span class="p">)</span>
        <span class="n">n_gene_case</span> <span class="o">=</span> <span class="nb">len</span><span class="p">(</span><span class="n">cases</span><span class="p">[</span><span class="n">cases</span><span class="p">[</span><span class="n">gene</span><span class="p">]</span><span class="o">==</span><span class="mi">1</span><span class="p">])</span>
        <span class="n">n_nogene_case</span> <span class="o">=</span> <span class="nb">len</span><span class="p">(</span><span class="n">cases</span><span class="p">[</span><span class="n">cases</span><span class="p">[</span><span class="n">gene</span><span class="p">]</span><span class="o">==</span><span class="mi">0</span><span class="p">])</span>
        <span class="n">n_gene_ctrl</span> <span class="o">=</span> <span class="nb">len</span><span class="p">(</span><span class="n">ctrls</span><span class="p">[</span><span class="n">ctrls</span><span class="p">[</span><span class="n">gene</span><span class="p">]</span><span class="o">==</span><span class="mi">1</span><span class="p">])</span>
        <span class="n">n_nogene_ctrl</span> <span class="o">=</span> <span class="nb">len</span><span class="p">(</span><span class="n">ctrls</span><span class="p">[</span><span class="n">ctrls</span><span class="p">[</span><span class="n">gene</span><span class="p">]</span><span class="o">==</span><span class="mi">0</span><span class="p">])</span>
        <span class="k">if</span> <span class="n">n_gene_case</span> <span class="o">==</span> <span class="mi">0</span> <span class="ow">or</span> <span class="n">n_gene_ctrl</span> <span class="o">==</span> <span class="mi">0</span><span class="p">:</span>
            <span class="n">odds_ratio</span> <span class="o">=</span> <span class="mi">0</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="n">odds_ratio</span> <span class="o">=</span> <span class="n">n_gene_case</span> <span class="o">*</span> <span class="n">n_nogene_ctrl</span> <span class="o">/</span> <span class="n">n_gene_ctrl</span> <span class="o">/</span> <span class="n">n_nogene_case</span>
        <span class="nb">print</span> <span class="p">(</span><span class="n">n_gene_case</span><span class="p">,</span> <span class="n">n_nogene_case</span><span class="p">,</span> <span class="n">n_gene_ctrl</span><span class="p">,</span> <span class="n">n_nogene_ctrl</span><span class="p">,</span> <span class="n">odds_ratio</span><span class="p">)</span>
        <span class="k">if</span> <span class="n">np</span><span class="o">.</span><span class="n">isinf</span><span class="p">(</span><span class="n">odds_ratio</span><span class="p">):</span>
            <span class="n">beta1</span> <span class="o">=</span> <span class="mi">10</span>
        <span class="k">elif</span> <span class="n">odds_ratio</span> <span class="o">==</span> <span class="mi">0</span><span class="p">:</span>
            <span class="n">beta1</span> <span class="o">=</span> <span class="o">-</span><span class="mi">10</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="n">beta1</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">log</span><span class="p">(</span><span class="n">odds_ratio</span><span class="p">)</span>
        <span class="n">fisher_stats</span> <span class="o">=</span> <span class="n">fisher</span><span class="o">.</span><span class="n">pvalue</span><span class="p">(</span><span class="n">n_gene_case</span><span class="p">,</span> <span class="n">n_gene_ctrl</span><span class="p">,</span> <span class="n">n_nogene_case</span><span class="p">,</span> <span class="n">n_nogene_ctrl</span><span class="p">)</span>
        <span class="n">p_value_two</span> <span class="o">=</span> <span class="n">fisher_stats</span><span class="o">.</span><span class="n">two_tail</span>
        <span class="n">p_value_left</span> <span class="o">=</span> <span class="n">fisher_stats</span><span class="o">.</span><span class="n">left_tail</span>
        <span class="n">p_value_right</span> <span class="o">=</span> <span class="n">fisher_stats</span><span class="o">.</span><span class="n">right_tail</span>
        <span class="k">if</span> <span class="n">p_value_right</span> <span class="o">&lt;</span> <span class="mf">0.5</span> <span class="ow">and</span> <span class="n">p_value_two</span> <span class="o">&lt;</span> <span class="mi">1</span><span class="p">:</span>
            <span class="n">z_score</span> <span class="o">=</span> <span class="n">stats</span><span class="o">.</span><span class="n">norm</span><span class="o">.</span><span class="n">ppf</span><span class="p">(</span><span class="mi">1</span><span class="o">-</span><span class="n">p_value_right</span><span class="p">)</span>
        <span class="k">elif</span> <span class="n">p_value_left</span> <span class="o">&lt;</span> <span class="mf">0.5</span> <span class="ow">and</span> <span class="n">p_value_two</span> <span class="o">&lt;</span> <span class="mi">1</span><span class="p">:</span>
            <span class="n">z_score</span> <span class="o">=</span> <span class="o">-</span><span class="n">stats</span><span class="o">.</span><span class="n">norm</span><span class="o">.</span><span class="n">ppf</span><span class="p">(</span><span class="mi">1</span><span class="o">-</span><span class="n">p_value_left</span><span class="p">)</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="n">z_score</span> <span class="o">=</span> <span class="n">subst</span>
        <span class="n">df_cal_bf</span> <span class="o">=</span> <span class="n">df_cal_bf</span><span class="o">.</span><span class="n">append</span><span class="p">([(</span><span class="s2">&quot;</span><span class="si">{}</span><span class="s2">.</span><span class="si">{}</span><span class="s2">&quot;</span><span class="o">.</span><span class="n">format</span><span class="p">(</span><span class="n">chrom</span><span class="p">,</span> <span class="n">gene</span><span class="p">),</span> <span class="n">chrom</span><span class="p">,</span> <span class="n">odds_ratio</span><span class="p">)])</span>
        <span class="n">df_cal_beta</span> <span class="o">=</span> <span class="n">df_cal_beta</span><span class="o">.</span><span class="n">append</span><span class="p">([(</span><span class="s2">&quot;</span><span class="si">{}</span><span class="s2">.</span><span class="si">{}</span><span class="s2">&quot;</span><span class="o">.</span><span class="n">format</span><span class="p">(</span><span class="n">chrom</span><span class="p">,</span> <span class="n">gene</span><span class="p">),</span> <span class="n">chrom</span><span class="p">,</span> <span class="n">beta1</span><span class="p">,</span> <span class="n">z_score</span><span class="p">,</span> <span class="n">p_value_two</span><span class="o">/</span><span class="mi">2</span><span class="p">)])</span>
        <span class="n">df_cal_zscore</span> <span class="o">=</span> <span class="n">df_cal_zscore</span><span class="o">.</span><span class="n">append</span><span class="p">([(</span><span class="s2">&quot;</span><span class="si">{}</span><span class="s2">.</span><span class="si">{}</span><span class="s2">&quot;</span><span class="o">.</span><span class="n">format</span><span class="p">(</span><span class="n">chrom</span><span class="p">,</span> <span class="n">gene</span><span class="p">),</span> <span class="n">chrom</span><span class="p">,</span> <span class="n">z_score</span><span class="p">)])</span>
        <span class="c1">### gene annotation needs to be determined</span>
        <span class="k">if</span> <span class="n">multi_cat</span> <span class="o">==</span> <span class="mi">0</span><span class="p">:</span>
            <span class="n">SNP_anno_cal</span> <span class="o">=</span> <span class="n">SNP_anno_cal</span><span class="o">.</span><span class="n">append</span><span class="p">([(</span><span class="s2">&quot;</span><span class="si">{}</span><span class="s2">.</span><span class="si">{}</span><span class="s2">&quot;</span><span class="o">.</span><span class="n">format</span><span class="p">(</span><span class="n">chrom</span><span class="p">,</span> <span class="n">gene</span><span class="p">),</span> <span class="mi">1</span> <span class="k">if</span> <span class="n">gene</span> <span class="ow">in</span> <span class="n">causal</span> <span class="k">else</span> <span class="mi">0</span><span class="p">)])</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="k">if</span> <span class="nb">len</span><span class="p">(</span><span class="n">cat</span><span class="p">)</span> <span class="o">!=</span> <span class="mi">0</span><span class="p">:</span>
                <span class="n">rand_cat</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">random</span><span class="o">.</span><span class="n">choice</span><span class="p">(</span><span class="n">cat</span><span class="p">)</span>
            <span class="n">SNP_anno_cal</span> <span class="o">=</span> <span class="n">SNP_anno_cal</span><span class="o">.</span><span class="n">append</span><span class="p">([(</span><span class="s2">&quot;</span><span class="si">{}</span><span class="s2">.</span><span class="si">{}</span><span class="s2">&quot;</span><span class="o">.</span><span class="n">format</span><span class="p">(</span><span class="n">chrom</span><span class="p">,</span> <span class="n">gene</span><span class="p">),</span> <span class="mi">1</span> <span class="k">if</span> <span class="n">gene</span> <span class="ow">in</span> <span class="n">causal</span> <span class="k">else</span> <span class="n">rand_cat</span><span class="p">)])</span>
            <span class="k">if</span> <span class="ow">not</span> <span class="n">gene</span> <span class="ow">in</span> <span class="n">causal</span><span class="p">:</span>
                <span class="n">cat</span><span class="o">.</span><span class="n">remove</span><span class="p">(</span><span class="n">rand_cat</span><span class="p">)</span>
        <span class="n">i</span> <span class="o">+=</span> <span class="mi">1</span>
    <span class="n">df_cal_bf</span><span class="o">.</span><span class="n">columns</span> <span class="o">=</span> <span class="p">[</span><span class="s2">&quot;SNP&quot;</span><span class="p">,</span> <span class="s2">&quot;gene&quot;</span><span class="p">,</span> <span class="s2">&quot;bf&quot;</span><span class="p">]</span>
    <span class="n">df_cal_beta</span><span class="o">.</span><span class="n">columns</span> <span class="o">=</span> <span class="p">[</span><span class="s2">&quot;SNP&quot;</span><span class="p">,</span> <span class="s2">&quot;gene&quot;</span><span class="p">,</span> <span class="s2">&quot;beta&quot;</span><span class="p">,</span> <span class="s2">&quot;t-stat&quot;</span><span class="p">,</span> <span class="s2">&quot;p-value&quot;</span><span class="p">]</span>
    <span class="n">df_cal_zscore</span><span class="o">.</span><span class="n">columns</span> <span class="o">=</span> <span class="p">[</span><span class="s2">&quot;SNP&quot;</span><span class="p">,</span> <span class="s2">&quot;gene&quot;</span><span class="p">,</span> <span class="s2">&quot;z-score&quot;</span><span class="p">]</span>
    <span class="n">SNP_anno_cal</span><span class="o">.</span><span class="n">columns</span> <span class="o">=</span> <span class="p">[</span><span class="s2">&quot;SNP&quot;</span><span class="p">,</span> <span class="s2">&quot;binding_d&quot;</span><span class="p">]</span>
    <span class="n">df_cal_bf</span><span class="o">.</span><span class="n">to_csv</span><span class="p">(</span><span class="s2">&quot;data/TORUS/calcium/bf_cal&quot;</span><span class="p">,</span> <span class="n">sep</span> <span class="o">=</span> <span class="s2">&quot;</span><span class="se">\t</span><span class="s2">&quot;</span><span class="p">,</span> <span class="n">header</span> <span class="o">=</span> <span class="kc">False</span><span class="p">,</span> <span class="n">index</span> <span class="o">=</span> <span class="kc">False</span><span class="p">)</span>
    <span class="n">df_cal_beta</span><span class="o">.</span><span class="n">to_csv</span><span class="p">(</span><span class="s2">&quot;data/TORUS/calcium/beta_cal&quot;</span><span class="p">,</span> <span class="n">sep</span> <span class="o">=</span> <span class="s2">&quot;</span><span class="se">\t</span><span class="s2">&quot;</span><span class="p">,</span> <span class="n">header</span> <span class="o">=</span> <span class="kc">True</span><span class="p">,</span> <span class="n">index</span> <span class="o">=</span> <span class="kc">False</span><span class="p">)</span>
    <span class="n">df_cal_zscore</span><span class="o">.</span><span class="n">to_csv</span><span class="p">(</span><span class="s2">&quot;data/TORUS/calcium/z_score_cal&quot;</span><span class="p">,</span> <span class="n">sep</span> <span class="o">=</span> <span class="s2">&quot;</span><span class="se">\t</span><span class="s2">&quot;</span><span class="p">,</span> <span class="n">header</span> <span class="o">=</span> <span class="kc">False</span><span class="p">,</span> <span class="n">index</span> <span class="o">=</span> <span class="kc">False</span><span class="p">)</span>
    <span class="n">SNP_anno_cal</span><span class="o">.</span><span class="n">to_csv</span><span class="p">(</span><span class="s2">&quot;data/TORUS/calcium/snp_anno_cal&quot;</span><span class="p">,</span> <span class="n">sep</span> <span class="o">=</span> <span class="s2">&quot;</span><span class="se">\t</span><span class="s2">&quot;</span><span class="p">,</span> <span class="n">header</span> <span class="o">=</span> <span class="kc">True</span><span class="p">,</span> <span class="n">index</span> <span class="o">=</span> <span class="kc">False</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="n">get_sum_Sweden_for_torus</span><span class="p">(</span><span class="n">gene_df</span> <span class="o">=</span> <span class="n">max_block</span><span class="p">,</span> <span class="n">causal</span> <span class="o">=</span> <span class="n">causal_genes</span><span class="p">,</span> <span class="n">cat</span> <span class="o">=</span> <span class="n">pheno</span><span class="p">,</span> <span class="n">multi_cat</span> <span class="o">=</span> <span class="mi">1</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="o">!</span>rm data/TORUS/calcium/*.gz
<span class="o">!</span>gzip data/TORUS/calcium/*_cal
<span class="o">!</span>torus.sh -d data/TORUS/calcium/beta_cal.gz -annot data/TORUS/calcium/snp_anno_cal.gz -est &gt; data/TORUS/calcium/cal_block_beta.est
</pre></div>
</div>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="nb">print</span> <span class="p">(</span><span class="n">block_genes</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span> 
</pre></div>
</div>
</div>
</div>
</div>
<hr>
&copy 2016-2017 Min Qiao at <a href="http://xinhelab.org">Xin He lab</a>, the University of Chicago
<p><small>Exported from <a href="http://github.com/gaow/cnv-gene-mapping/blob/a28974751957a328cc340b71662924dbeeb93d97/analysis/20171010_swcnv_TORUS.ipynb"><code>analysis/20171010_swcnv_TORUS.ipynb</code></a> committed by Min Qiao on Wed Oct 25 20:37:28 2017 <a href="http://github.com/gaow/cnv-gene-mapping/commit/a28974751957a328cc340b71662924dbeeb93d97">revision 369, a289747</a> <a href="https://stephenslab.github.io/ipynb-website/notes.html#Note-about-commit-ids"><span class="fa fa-question-circle"></span></a></small></p>
</div>
</div>
</body>
</html>
