<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="ipynb_website:version" content="0.9.2" />
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link rel="stylesheet" type="text/css" href="../css/jt.css">
<link rel="stylesheet" type="text/css" href="../css/toc2.css">
<link href="../site_libs/jqueryui-1.11.4/jquery-ui.css">
<link rel="stylesheet" href="../site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<link rel="stylesheet" href="../site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />
<script type="text/javascript" src="https://ajax.googleapis.com/ajax/libs/jquery/1.9.1/jquery.min.js"></script>
<script type="text/javascript" src="https://ajax.googleapis.com/ajax/libs/jqueryui/1.9.1/jquery-ui.min.js"></script>
<script src="../site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="../site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="../site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<link rel="stylesheet"
      href="../site_libs/highlight/textmate.css"
      type="text/css" />
<script src="../site_libs/highlight/highlight.js"></script>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>
<script src="../js/toc2.js"></script>
<script src="../js/docs.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS_HTML"></script>
<script>
    MathJax.Hub.Config({
        extensions: ["tex2jax.js"],
        jax: ["input/TeX", "output/HTML-CSS"],
        tex2jax: {
        inlineMath: [ ['$','$'], ["\\(","\\)"] ],
        displayMath: [ ['$$','$$'], ["\\[","\\]"] ],
        processEscapes: true
        },
        "HTML-CSS": {
            preferredFont: "TeX",
            availableFonts: ["TeX"],
            styles: {
                scale: 110,
                ".MathJax_Display": {
                    "font-size": "110%",
                }
            }
        }
    });
</script>
<script>
$( document ).ready(function(){
            var cfg={'threshold':3,     // depth of toc (number of levels)
             'number_sections': false,
             'toc_cell': false,          // useless here
             'toc_window_display': true, // display the toc window
             "toc_section_display": "block", // display toc contents in the window
             'sideBar':true,       // sidebar or floating window
             'navigate_menu':false       // navigation menu (only in liveNotebook -- do not change)
            }
            var st={};                  // some variables used in the script
            st.rendering_toc_cell = false;
            st.config_loaded = false;
            st.extension_initialized=false;
            st.nbcontainer_marginleft = $('#notebook-container').css('margin-left')
            st.nbcontainer_marginright = $('#notebook-container').css('margin-right')
            st.nbcontainer_width = $('#notebook-container').css('width')
            st.oldTocHeight = undefined
            st.cell_toc = undefined;
            st.toc_index=0;
            // fire the main function with these parameters
            table_of_contents(cfg, st);
            var file=analysisDict[$("h1:first").attr("id")];
            $("#toc-level0 a").css("color","#126dce");
            $('a[href="#'+$("h1:first").attr("id")+'"]').hide()
            var docs=analysisArray;
            var pos=analysisArray.indexOf(file);
            for (var a=pos;a>=0;a--){
                  var name=docs[a]
                  $('<li><a href="'+name+'.html"><font color="#073642"><b>'+name.replace(/_/g," ")+'</b></font></a></li>').insertBefore("#toc-level0 li:eq(0)");
            }
            $('a[href="'+file+'.html'+'"]').css("color","#126dce");
            for (var a=pos+1;a<docs.length;a++){
                  var name=docs[a]
                  $(".toc #toc-level0").append('<li><a href="'+name+'.html"><font color="#073642"><b>'+name.replace(/_/g," ")+'</b></font></a></li>');
            }
            $("#toc-header").hide();
    });
</script>
<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');
  // mark it active
  menuAnchor.parent().addClass('active');
  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>
<div class="container-fluid main-container">
<!-- tabsets -->
<script src="../site_libs/navigation-1.1/tabsets.js"></script>
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>
<title>Gene Mapping with CNV</title>
<style type = "text/css">
body {
  font-family: "Droid Sans";
  padding-top: 66px;
  padding-bottom: 40px;
}
</style>
</head>
<body>
<div tabindex="-1" id="notebook" class="border-box-sizing">
<div class="container" id="notebook-container">
<!-- code folding -->
<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="../index.html">Gene Mapping with CNV</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">Overview</a>
</li>
<li>
  <a href="../analysis.html">Analysis</a>
</li>
<li>
  <a href="../prototype.html">Prototype</a>
</li>
<li>
  <a href="../utils.html">Utils</a>
</li>
<li>
  <a href="../setup.html">Setup</a>
</li>
<li>
  <a href="../writeup.html">Writeup</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
   <a href="http://github.com/gaow/cnv-gene-mapping"> <span class="fa fa-github"></span> </a>
</li>
</ul>
      </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->
<div class="cell border-box-sizing text_cell rendered">
<div class="prompt input_prompt">
</div>
<div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h1 id="Toy-example-for-multi-genes-and-multi-CNVs-in-a-region">Toy example for multi-genes and multi-CNVs in a region<a class="anchor-link" href="#Toy-example-for-multi-genes-and-multi-CNVs-in-a-region">&#182;</a></h1><h2 id="One-or-more-causal-genes">One or more causal genes<a class="anchor-link" href="#One-or-more-causal-genes">&#182;</a></h2><h2 id="Obtain-summary-statistics-z_score-and-run-TORUS">Obtain summary statistics z_score and run TORUS<a class="anchor-link" href="#Obtain-summary-statistics-z_score-and-run-TORUS">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[1]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="kn">import</span> <span class="nn">numpy</span> <span class="k">as</span> <span class="nn">np</span>
<span class="kn">import</span> <span class="nn">pandas</span> <span class="k">as</span> <span class="nn">pd</span>
<span class="kn">import</span> <span class="nn">feather</span>
<span class="kn">from</span> <span class="nn">pandasql</span> <span class="k">import</span> <span class="n">sqldf</span>
<span class="kn">from</span> <span class="nn">scipy</span> <span class="k">import</span> <span class="n">stats</span>
<span class="kn">import</span> <span class="nn">sys</span>
<span class="kn">import</span> <span class="nn">fisher</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[2]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="k">def</span> <span class="nf">toy_multicnv_multicausal</span><span class="p">(</span><span class="n">p</span><span class="p">,</span> <span class="n">n_cnv</span><span class="p">,</span> <span class="n">causal</span><span class="p">,</span> <span class="n">n_geom</span><span class="o">=</span><span class="mi">10000</span><span class="p">,</span> <span class="n">n_gene</span><span class="o">=</span><span class="mi">10</span><span class="p">,</span> <span class="n">const1</span><span class="o">=</span><span class="mf">0.5</span><span class="p">,</span> <span class="n">const2</span><span class="o">=</span><span class="mf">0.05</span><span class="p">,</span> <span class="n">seed</span><span class="o">=</span><span class="mi">999</span><span class="p">):</span>
    <span class="sd">&#39;&#39;&#39;A region with multiple genes and CNVs. CNVs probably overlap. At least two causal genes.</span>
<span class="sd">    &quot;causal&quot; is a list of positions of causal genes.</span>
<span class="sd">    &quot;n_gene&quot; is the number of genes this region harbors.</span>
<span class="sd">    &quot;n_cnv&quot; is the number of CNV in this region. &quot;n_geom&quot; is the number of geometric variables generated.</span>
<span class="sd">    &quot;p&quot; is the probability of geometric dist, then randomly generate a certain number of geometric variables.</span>
<span class="sd">    I use these variables minus 1 as the length of CNVs, which means the number of genes that a CNV overlap with.</span>
<span class="sd">    The maximum length for CNV in this region is overlapping with 10 genes&#39;&#39;&#39;</span>
    <span class="n">np</span><span class="o">.</span><span class="n">random</span><span class="o">.</span><span class="n">seed</span><span class="p">(</span><span class="n">seed</span><span class="p">)</span>
    <span class="k">if</span> <span class="n">n_cnv</span> <span class="o">&gt;</span> <span class="n">n_geom</span><span class="p">:</span>
        <span class="nb">print</span> <span class="p">(</span><span class="s2">&quot;The number of CNV is larger than the number of geometric numbers&quot;</span><span class="p">)</span>
        <span class="n">sys</span><span class="o">.</span><span class="n">exit</span><span class="p">()</span>
        <span class="k">return</span> <span class="kc">None</span>
    <span class="n">geom_minus_1</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">random</span><span class="o">.</span><span class="n">geometric</span><span class="p">(</span><span class="n">p</span><span class="p">,</span> <span class="n">size</span><span class="o">=</span><span class="n">n_geom</span><span class="p">)</span> <span class="o">-</span> <span class="mi">1</span>
    <span class="k">if</span> <span class="n">n_cnv</span> <span class="o">&gt;</span> <span class="nb">len</span><span class="p">(</span><span class="n">geom_minus_1</span><span class="p">[</span><span class="n">geom_minus_1</span> <span class="o">&lt;=</span> <span class="n">n_gene</span><span class="p">]):</span>
        <span class="nb">print</span> <span class="p">(</span><span class="s2">&quot;Cannot take a larger sample than population when &#39;replace=False&#39;&quot;</span><span class="p">)</span>
        <span class="n">sys</span><span class="o">.</span><span class="n">exit</span><span class="p">()</span>
        <span class="k">return</span> <span class="kc">None</span>
    <span class="n">cnv_len</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">random</span><span class="o">.</span><span class="n">choice</span><span class="p">(</span><span class="n">geom_minus_1</span><span class="p">[</span><span class="n">geom_minus_1</span> <span class="o">&lt;=</span> <span class="n">n_gene</span><span class="p">],</span> <span class="n">n_cnv</span><span class="p">,</span> <span class="n">replace</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>
    <span class="n">cnv_start_pos</span> <span class="o">=</span> <span class="p">[</span><span class="n">np</span><span class="o">.</span><span class="n">random</span><span class="o">.</span><span class="n">choice</span><span class="p">(</span><span class="nb">range</span><span class="p">(</span><span class="n">n_gene</span><span class="o">+</span><span class="mi">1</span><span class="o">-</span><span class="n">i</span><span class="p">))</span> <span class="k">if</span> <span class="n">i</span><span class="o">!=</span><span class="mi">0</span> <span class="k">else</span> <span class="o">-</span><span class="mi">1</span> <span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="n">cnv_len</span><span class="p">]</span>
    <span class="n">ptn_ls</span> <span class="o">=</span> <span class="p">[]</span>
    <span class="k">for</span> <span class="n">j</span><span class="p">,</span><span class="n">i</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">cnv_start_pos</span><span class="p">):</span>
    <span class="c1">## j is the index of i in cnv start position list, so cnv_len[j] is the length of the corresponding cnv;</span>
    <span class="c1">## i is the corresponding cnv start position</span>
        <span class="k">if</span> <span class="n">i</span> <span class="o">==</span> <span class="o">-</span><span class="mi">1</span><span class="p">:</span>
            <span class="n">ptn</span> <span class="o">=</span> <span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">*</span><span class="n">n_gene</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="n">ptn</span> <span class="o">=</span> <span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">*</span><span class="n">i</span> <span class="o">+</span> <span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="o">*</span><span class="n">cnv_len</span><span class="p">[</span><span class="n">j</span><span class="p">]</span> <span class="o">+</span> <span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">*</span><span class="p">(</span><span class="n">n_gene</span> <span class="o">-</span> <span class="n">cnv_len</span><span class="p">[</span><span class="n">j</span><span class="p">]</span> <span class="o">-</span> <span class="n">i</span><span class="p">)</span>
        <span class="n">ptn_ls</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">ptn</span><span class="p">)</span>
    <span class="n">mat</span> <span class="o">=</span> <span class="p">[[</span><span class="n">np</span><span class="o">.</span><span class="n">random</span><span class="o">.</span><span class="n">normal</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="mi">1</span><span class="p">)]</span> <span class="o">+</span> <span class="n">line</span> <span class="k">for</span> <span class="n">line</span> <span class="ow">in</span> <span class="n">ptn_ls</span><span class="p">]</span>
    <span class="n">config</span> <span class="o">=</span> <span class="p">[]</span>
    <span class="k">for</span> <span class="n">line</span> <span class="ow">in</span> <span class="n">mat</span><span class="p">:</span>
        <span class="n">line</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span> <span class="o">=</span> <span class="n">line</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span> <span class="o">+</span> <span class="n">line</span><span class="p">[</span><span class="mi">1</span><span class="p">:]</span><span class="o">.</span><span class="n">count</span><span class="p">(</span><span class="mi">1</span><span class="p">)</span><span class="o">*</span><span class="n">const2</span>
        <span class="n">config</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">line</span><span class="p">[</span><span class="mi">0</span><span class="p">])</span>
<span class="c1">#         only one causal gene, and gene pos is no more than the number of gene covered in this region</span>
        <span class="k">if</span> <span class="nb">len</span><span class="p">(</span><span class="n">causal</span><span class="p">)</span> <span class="o">==</span> <span class="mi">1</span> <span class="ow">and</span> <span class="p">(</span><span class="n">causal</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span> <span class="o">&gt;</span> <span class="nb">len</span><span class="p">(</span><span class="n">line</span><span class="p">[</span><span class="mi">1</span><span class="p">:])</span> <span class="ow">or</span> <span class="n">causal</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span> <span class="o">&lt;=</span> <span class="mi">0</span><span class="p">):</span>
            <span class="nb">print</span> <span class="p">(</span><span class="s2">&quot;Single causal CNV index out of range&quot;</span><span class="p">)</span>
            <span class="n">sys</span><span class="o">.</span><span class="n">exit</span><span class="p">()</span>
        <span class="k">elif</span> <span class="nb">len</span><span class="p">(</span><span class="n">causal</span><span class="p">)</span> <span class="o">==</span> <span class="mi">1</span> <span class="ow">and</span> <span class="n">causal</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span> <span class="o">&lt;=</span> <span class="nb">len</span><span class="p">(</span><span class="n">line</span><span class="p">[</span><span class="mi">1</span><span class="p">:]):</span>
            <span class="n">line</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span> <span class="o">=</span> <span class="n">line</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span> <span class="o">+</span> <span class="n">line</span><span class="p">[</span><span class="n">causal</span><span class="p">[</span><span class="mi">0</span><span class="p">]]</span><span class="o">*</span><span class="n">const1</span>
            <span class="n">config</span><span class="p">[</span><span class="o">-</span><span class="mi">1</span><span class="p">]</span> <span class="o">=</span> <span class="n">line</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span>
        <span class="k">elif</span> <span class="nb">max</span><span class="p">(</span><span class="n">causal</span><span class="p">)</span> <span class="o">&gt;</span> <span class="nb">len</span><span class="p">(</span><span class="n">line</span><span class="p">[</span><span class="mi">1</span><span class="p">:])</span> <span class="ow">or</span> <span class="nb">min</span><span class="p">(</span><span class="n">causal</span><span class="p">)</span> <span class="o">&lt;=</span> <span class="mi">0</span><span class="p">:</span>
            <span class="nb">print</span> <span class="p">(</span><span class="s2">&quot;Causal CNV index out of range&quot;</span><span class="p">)</span>
            <span class="n">sys</span><span class="o">.</span><span class="n">exit</span><span class="p">()</span>
        <span class="k">elif</span> <span class="nb">any</span><span class="p">(</span><span class="n">line</span><span class="p">[</span><span class="n">x</span><span class="p">]</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">causal</span><span class="p">):</span>
            <span class="n">line</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span> <span class="o">=</span> <span class="n">line</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span> <span class="o">+</span> <span class="n">const1</span><span class="o">*</span><span class="p">[</span><span class="n">line</span><span class="p">[</span><span class="n">x</span><span class="p">]</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">causal</span><span class="p">]</span><span class="o">.</span><span class="n">count</span><span class="p">(</span><span class="mi">1</span><span class="p">)</span>
            <span class="n">config</span><span class="p">[</span><span class="o">-</span><span class="mi">1</span><span class="p">]</span> <span class="o">=</span> <span class="n">line</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="k">continue</span>
    <span class="k">for</span> <span class="n">line</span> <span class="ow">in</span> <span class="n">mat</span><span class="p">:</span>
        <span class="n">line</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span> <span class="o">=</span> <span class="mi">1</span> <span class="k">if</span> <span class="n">line</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span> <span class="o">&gt;=</span> <span class="n">np</span><span class="o">.</span><span class="n">median</span><span class="p">(</span><span class="n">config</span><span class="p">)</span> <span class="k">else</span> <span class="mi">0</span>
    <span class="n">mat</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">matrix</span><span class="p">(</span><span class="n">mat</span><span class="p">)</span>
    <span class="n">df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">(</span><span class="n">mat</span><span class="p">,</span> <span class="n">columns</span> <span class="o">=</span> <span class="p">[</span><span class="s2">&quot;phenotype&quot;</span><span class="p">]</span> <span class="o">+</span> <span class="p">[</span><span class="s2">&quot;gene</span><span class="si">{}</span><span class="s2">&quot;</span><span class="o">.</span><span class="n">format</span><span class="p">(</span><span class="n">i</span><span class="o">+</span><span class="mi">1</span><span class="p">)</span> <span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="n">mat</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="o">-</span><span class="mi">1</span><span class="p">)])</span>
    <span class="n">counts</span> <span class="o">=</span> <span class="n">df</span><span class="o">.</span><span class="n">groupby</span><span class="p">([</span><span class="s2">&quot;gene</span><span class="si">{}</span><span class="s2">&quot;</span><span class="o">.</span><span class="n">format</span><span class="p">(</span><span class="n">i</span><span class="o">+</span><span class="mi">1</span><span class="p">)</span> <span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="n">mat</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="o">-</span><span class="mi">1</span><span class="p">)]</span> <span class="o">+</span> <span class="p">[</span><span class="s2">&quot;phenotype&quot;</span><span class="p">])</span><span class="o">.</span><span class="n">size</span><span class="p">()</span>
    <span class="k">for</span> <span class="n">col</span> <span class="ow">in</span> <span class="n">df</span><span class="o">.</span><span class="n">columns</span><span class="p">:</span>
        <span class="n">df</span><span class="p">[</span><span class="n">col</span><span class="p">]</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="n">col</span><span class="p">]</span><span class="o">.</span><span class="n">astype</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">uint8</span><span class="p">)</span>
    <span class="n">output</span> <span class="o">=</span> <span class="n">feather</span><span class="o">.</span><span class="n">write_dataframe</span><span class="p">(</span><span class="n">df</span><span class="p">,</span> <span class="s2">&quot;data/toy_n</span><span class="si">{}</span><span class="s2">_p</span><span class="si">{}</span><span class="s2">_causal</span><span class="si">{}</span><span class="s2">_const</span><span class="si">{}</span><span class="s2">_</span><span class="si">{}</span><span class="s2">.feather&quot;</span>
                                     <span class="o">.</span><span class="n">format</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">cnv_len</span><span class="p">),</span> <span class="n">p</span><span class="p">,</span> <span class="n">causal</span><span class="p">,</span> <span class="n">const1</span><span class="p">,</span> <span class="n">const2</span><span class="p">))</span>
    <span class="k">return</span> <span class="n">df</span><span class="p">,</span> <span class="n">counts</span><span class="p">,</span> <span class="n">causal</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[3]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="n">df</span><span class="p">,</span> <span class="n">counts</span><span class="p">,</span> <span class="n">causal</span> <span class="o">=</span> <span class="n">toy_multicnv_multicausal</span><span class="p">(</span><span class="n">p</span><span class="o">=</span><span class="mf">0.5</span><span class="p">,</span> <span class="n">n_cnv</span><span class="o">=</span><span class="mi">5000</span><span class="p">,</span> <span class="n">causal</span><span class="o">=</span><span class="p">[</span><span class="mi">1</span><span class="p">,</span><span class="mi">15</span><span class="p">],</span> <span class="n">n_gene</span><span class="o">=</span><span class="mi">15</span><span class="p">,</span> <span class="n">const1</span><span class="o">=</span><span class="mf">0.45</span><span class="p">,</span> <span class="n">const2</span><span class="o">=</span><span class="mf">0.03</span><span class="p">)</span>
<span class="nb">print</span> <span class="p">(</span><span class="n">df</span><span class="p">[</span><span class="s2">&quot;phenotype&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">tolist</span><span class="p">()</span><span class="o">.</span><span class="n">count</span><span class="p">(</span><span class="mi">1</span><span class="p">))</span>
</pre></div>
</div>
</div>
</div>
<div class="output_wrapper">
<div class="output">
<div class="output_area">
<div class="prompt"></div>
<div class="output_subarea output_stream output_stdout output_text">
<pre>2500
</pre>
</div>
</div>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[4]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="kn">from</span> <span class="nn">utils</span> <span class="k">import</span> <span class="n">run_dap_lite</span>
<span class="n">fileout</span> <span class="o">=</span> <span class="s2">&quot;data/toy_multi_causal_10.dap&quot;</span>
<span class="n">run_dap_lite</span><span class="p">(</span><span class="n">df</span><span class="p">,</span> <span class="n">fileout</span><span class="p">,</span> <span class="n">grid</span><span class="o">=</span><span class="p">[(</span><span class="mi">0</span><span class="p">,</span> <span class="o">-</span><span class="mf">9.2</span><span class="p">)])</span>
</pre></div>
</div>
</div>
</div>
<div class="output_wrapper">
<div class="output">
<div class="output_area">
<div class="prompt"></div>
<div class="output_subarea output_stream output_stdout output_text">
<pre>2017-10-09 14:49:47.261231
2017-10-09 14:50:02.751030
</pre>
</div>
</div>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[4]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="k">def</span> <span class="nf">get_summary_stats</span><span class="p">(</span><span class="n">gene_df</span><span class="p">,</span> <span class="n">causal</span><span class="p">,</span> <span class="n">cat</span><span class="p">,</span> <span class="n">multi_cat</span> <span class="o">=</span> <span class="mi">0</span><span class="p">,</span> <span class="n">subst</span> <span class="o">=</span> <span class="o">-</span><span class="mi">10</span><span class="p">,</span> <span class="n">chrom</span> <span class="o">=</span> <span class="s2">&quot;chr6&quot;</span><span class="p">,</span> <span class="n">seed</span> <span class="o">=</span> <span class="mi">999</span><span class="p">):</span>
    <span class="sd">&#39;&#39;&#39;get summary statistics (e.g. t-statistic, p-value) to be used for TORUS, then transfer p-value to z-score</span>
<span class="sd">       Actually do NOT need SNP (gene) map and gene (chromosome) map</span>
<span class="sd">       If multi_cat!=0, then cat must be multiple categories</span>
<span class="sd">       If multi_cat==0, then cat can be any list (because it won&#39;t be used) but must be given</span>
<span class="sd">    &#39;&#39;&#39;</span>
    <span class="n">np</span><span class="o">.</span><span class="n">random</span><span class="o">.</span><span class="n">seed</span><span class="p">(</span><span class="n">seed</span><span class="p">)</span>
    <span class="n">cases</span> <span class="o">=</span> <span class="n">gene_df</span><span class="p">[</span><span class="n">gene_df</span><span class="p">[</span><span class="s2">&quot;phenotype&quot;</span><span class="p">]</span><span class="o">==</span><span class="mi">1</span><span class="p">]</span>
    <span class="n">ctrls</span> <span class="o">=</span> <span class="n">gene_df</span><span class="p">[</span><span class="n">gene_df</span><span class="p">[</span><span class="s2">&quot;phenotype&quot;</span><span class="p">]</span><span class="o">==</span><span class="mi">0</span><span class="p">]</span>
    <span class="n">df_bf</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">()</span>
    <span class="n">df_beta</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">()</span>
    <span class="n">df_zscore</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">()</span>
    <span class="n">SNP_anno</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">()</span>
    <span class="n">i</span> <span class="o">=</span> <span class="mi">1</span>
    <span class="k">for</span> <span class="n">gene</span> <span class="ow">in</span> <span class="nb">list</span><span class="p">(</span><span class="n">gene_df</span><span class="p">)[</span><span class="mi">1</span><span class="p">:]:</span>
        <span class="n">n_gene_case</span> <span class="o">=</span> <span class="nb">len</span><span class="p">(</span><span class="n">cases</span><span class="p">[</span><span class="n">cases</span><span class="p">[</span><span class="n">gene</span><span class="p">]</span><span class="o">==</span><span class="mi">1</span><span class="p">])</span>
        <span class="n">n_nogene_case</span> <span class="o">=</span> <span class="nb">len</span><span class="p">(</span><span class="n">cases</span><span class="p">[</span><span class="n">cases</span><span class="p">[</span><span class="n">gene</span><span class="p">]</span><span class="o">==</span><span class="mi">0</span><span class="p">])</span>
        <span class="n">n_gene_ctrl</span> <span class="o">=</span> <span class="nb">len</span><span class="p">(</span><span class="n">ctrls</span><span class="p">[</span><span class="n">ctrls</span><span class="p">[</span><span class="n">gene</span><span class="p">]</span><span class="o">==</span><span class="mi">1</span><span class="p">])</span>
        <span class="n">n_nogene_ctrl</span> <span class="o">=</span> <span class="nb">len</span><span class="p">(</span><span class="n">ctrls</span><span class="p">[</span><span class="n">ctrls</span><span class="p">[</span><span class="n">gene</span><span class="p">]</span><span class="o">==</span><span class="mi">0</span><span class="p">])</span>
        <span class="n">odds_ratio</span> <span class="o">=</span> <span class="n">stats</span><span class="o">.</span><span class="n">fisher_exact</span><span class="p">([[</span><span class="n">n_gene_case</span><span class="p">,</span> <span class="n">n_gene_ctrl</span><span class="p">],</span> <span class="p">[</span><span class="n">n_nogene_case</span><span class="p">,</span> <span class="n">n_nogene_ctrl</span><span class="p">]])[</span><span class="mi">0</span><span class="p">]</span>
<span class="c1">#         print (n_gene_case, n_gene_ctrl, odds_ratio)</span>
        <span class="n">beta1</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">log</span><span class="p">(</span><span class="n">odds_ratio</span><span class="p">)</span>
        <span class="n">fisher_stats</span> <span class="o">=</span> <span class="n">fisher</span><span class="o">.</span><span class="n">pvalue</span><span class="p">(</span><span class="n">n_gene_case</span><span class="p">,</span> <span class="n">n_gene_ctrl</span><span class="p">,</span> <span class="n">n_nogene_case</span><span class="p">,</span> <span class="n">n_nogene_ctrl</span><span class="p">)</span>
        <span class="n">p_value_two</span> <span class="o">=</span> <span class="n">fisher_stats</span><span class="o">.</span><span class="n">two_tail</span>
        <span class="n">p_value_left</span> <span class="o">=</span> <span class="n">fisher_stats</span><span class="o">.</span><span class="n">left_tail</span>
        <span class="n">p_value_right</span> <span class="o">=</span> <span class="n">fisher_stats</span><span class="o">.</span><span class="n">right_tail</span>
        <span class="k">if</span> <span class="n">p_value_right</span> <span class="o">&lt;</span> <span class="mf">0.5</span> <span class="ow">and</span> <span class="n">p_value_two</span> <span class="o">&lt;</span> <span class="mi">1</span><span class="p">:</span>
            <span class="n">z_score</span> <span class="o">=</span> <span class="n">stats</span><span class="o">.</span><span class="n">norm</span><span class="o">.</span><span class="n">ppf</span><span class="p">(</span><span class="mi">1</span><span class="o">-</span><span class="n">p_value_right</span><span class="p">)</span>
        <span class="k">elif</span> <span class="n">p_value_left</span> <span class="o">&lt;</span> <span class="mf">0.5</span> <span class="ow">and</span> <span class="n">p_value_two</span> <span class="o">&lt;</span> <span class="mi">1</span><span class="p">:</span>
            <span class="n">z_score</span> <span class="o">=</span> <span class="o">-</span><span class="n">stats</span><span class="o">.</span><span class="n">norm</span><span class="o">.</span><span class="n">ppf</span><span class="p">(</span><span class="mi">1</span><span class="o">-</span><span class="n">p_value_left</span><span class="p">)</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="n">z_score</span> <span class="o">=</span> <span class="n">subst</span>
        <span class="n">df_bf</span> <span class="o">=</span> <span class="n">df_bf</span><span class="o">.</span><span class="n">append</span><span class="p">([(</span><span class="s2">&quot;</span><span class="si">{}</span><span class="s2">.</span><span class="si">{}</span><span class="s2">&quot;</span><span class="o">.</span><span class="n">format</span><span class="p">(</span><span class="n">chrom</span><span class="p">,</span> <span class="n">gene</span><span class="p">),</span> <span class="n">chrom</span><span class="p">,</span> <span class="n">odds_ratio</span><span class="p">)])</span>
        <span class="n">df_beta</span> <span class="o">=</span> <span class="n">df_beta</span><span class="o">.</span><span class="n">append</span><span class="p">([(</span><span class="s2">&quot;</span><span class="si">{}</span><span class="s2">.</span><span class="si">{}</span><span class="s2">&quot;</span><span class="o">.</span><span class="n">format</span><span class="p">(</span><span class="n">chrom</span><span class="p">,</span> <span class="n">gene</span><span class="p">),</span> <span class="n">chrom</span><span class="p">,</span> <span class="n">beta1</span><span class="p">,</span> <span class="n">z_score</span><span class="p">,</span> <span class="n">p_value_two</span><span class="o">/</span><span class="mi">2</span><span class="p">)])</span>
        <span class="n">df_zscore</span> <span class="o">=</span> <span class="n">df_zscore</span><span class="o">.</span><span class="n">append</span><span class="p">([(</span><span class="s2">&quot;</span><span class="si">{}</span><span class="s2">.</span><span class="si">{}</span><span class="s2">&quot;</span><span class="o">.</span><span class="n">format</span><span class="p">(</span><span class="n">chrom</span><span class="p">,</span> <span class="n">gene</span><span class="p">),</span> <span class="n">chrom</span><span class="p">,</span> <span class="n">z_score</span><span class="p">)])</span>
        <span class="c1">### gene annotation needs to be determined</span>
        <span class="k">if</span> <span class="n">multi_cat</span> <span class="o">==</span> <span class="mi">0</span><span class="p">:</span>
            <span class="n">SNP_anno</span> <span class="o">=</span> <span class="n">SNP_anno</span><span class="o">.</span><span class="n">append</span><span class="p">([(</span><span class="s2">&quot;</span><span class="si">{}</span><span class="s2">.</span><span class="si">{}</span><span class="s2">&quot;</span><span class="o">.</span><span class="n">format</span><span class="p">(</span><span class="n">chrom</span><span class="p">,</span> <span class="n">gene</span><span class="p">),</span> <span class="mi">1</span> <span class="k">if</span> <span class="nb">int</span><span class="p">(</span><span class="n">gene</span><span class="p">[</span><span class="mi">4</span><span class="p">:])</span> <span class="ow">in</span> <span class="n">causal</span> <span class="k">else</span> <span class="mi">0</span><span class="p">)])</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="k">if</span> <span class="nb">len</span><span class="p">(</span><span class="n">cat</span><span class="p">)</span> <span class="o">!=</span> <span class="mi">0</span><span class="p">:</span>
                <span class="n">rand_cat</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">random</span><span class="o">.</span><span class="n">choice</span><span class="p">(</span><span class="n">cat</span><span class="p">)</span>
            <span class="n">SNP_anno</span> <span class="o">=</span> <span class="n">SNP_anno</span><span class="o">.</span><span class="n">append</span><span class="p">([(</span><span class="s2">&quot;</span><span class="si">{}</span><span class="s2">.</span><span class="si">{}</span><span class="s2">&quot;</span><span class="o">.</span><span class="n">format</span><span class="p">(</span><span class="n">chrom</span><span class="p">,</span> <span class="n">gene</span><span class="p">),</span> <span class="mi">1</span> <span class="k">if</span> <span class="nb">int</span><span class="p">(</span><span class="n">gene</span><span class="p">[</span><span class="mi">4</span><span class="p">:])</span> <span class="ow">in</span> <span class="n">causal</span> <span class="k">else</span> <span class="n">rand_cat</span><span class="p">)])</span>
            <span class="k">if</span> <span class="ow">not</span> <span class="nb">int</span><span class="p">(</span><span class="n">gene</span><span class="p">[</span><span class="mi">4</span><span class="p">:])</span> <span class="ow">in</span> <span class="n">causal</span><span class="p">:</span>
                <span class="n">cat</span><span class="o">.</span><span class="n">remove</span><span class="p">(</span><span class="n">rand_cat</span><span class="p">)</span>
        <span class="n">i</span> <span class="o">+=</span> <span class="mi">1</span>
    <span class="n">df_bf</span><span class="o">.</span><span class="n">columns</span> <span class="o">=</span> <span class="p">[</span><span class="s2">&quot;SNP&quot;</span><span class="p">,</span> <span class="s2">&quot;gene&quot;</span><span class="p">,</span> <span class="s2">&quot;bf&quot;</span><span class="p">]</span>
    <span class="n">df_beta</span><span class="o">.</span><span class="n">columns</span> <span class="o">=</span> <span class="p">[</span><span class="s2">&quot;SNP&quot;</span><span class="p">,</span> <span class="s2">&quot;gene&quot;</span><span class="p">,</span> <span class="s2">&quot;beta&quot;</span><span class="p">,</span> <span class="s2">&quot;t-stat&quot;</span><span class="p">,</span> <span class="s2">&quot;p-value&quot;</span><span class="p">]</span>
    <span class="n">df_zscore</span><span class="o">.</span><span class="n">columns</span> <span class="o">=</span> <span class="p">[</span><span class="s2">&quot;SNP&quot;</span><span class="p">,</span> <span class="s2">&quot;gene&quot;</span><span class="p">,</span> <span class="s2">&quot;z-score&quot;</span><span class="p">]</span>
    <span class="n">SNP_anno</span><span class="o">.</span><span class="n">columns</span> <span class="o">=</span> <span class="p">[</span><span class="s2">&quot;SNP&quot;</span><span class="p">,</span> <span class="s2">&quot;binding_d&quot;</span><span class="p">]</span>
    <span class="n">df_bf</span><span class="o">.</span><span class="n">to_csv</span><span class="p">(</span><span class="s2">&quot;data/TORUS/toy/bf&quot;</span><span class="p">,</span> <span class="n">sep</span> <span class="o">=</span> <span class="s2">&quot;</span><span class="se">\t</span><span class="s2">&quot;</span><span class="p">,</span> <span class="n">header</span> <span class="o">=</span> <span class="kc">False</span><span class="p">,</span> <span class="n">index</span> <span class="o">=</span> <span class="kc">False</span><span class="p">)</span>
    <span class="n">df_beta</span><span class="o">.</span><span class="n">to_csv</span><span class="p">(</span><span class="s2">&quot;data/TORUS/toy/beta&quot;</span><span class="p">,</span> <span class="n">sep</span> <span class="o">=</span> <span class="s2">&quot;</span><span class="se">\t</span><span class="s2">&quot;</span><span class="p">,</span> <span class="n">header</span> <span class="o">=</span> <span class="kc">True</span><span class="p">,</span> <span class="n">index</span> <span class="o">=</span> <span class="kc">False</span><span class="p">)</span>
    <span class="n">df_zscore</span><span class="o">.</span><span class="n">to_csv</span><span class="p">(</span><span class="s2">&quot;data/TORUS/toy/z_score&quot;</span><span class="p">,</span> <span class="n">sep</span> <span class="o">=</span> <span class="s2">&quot;</span><span class="se">\t</span><span class="s2">&quot;</span><span class="p">,</span> <span class="n">header</span> <span class="o">=</span> <span class="kc">False</span><span class="p">,</span> <span class="n">index</span> <span class="o">=</span> <span class="kc">False</span><span class="p">)</span>
    <span class="n">SNP_anno</span><span class="o">.</span><span class="n">to_csv</span><span class="p">(</span><span class="s2">&quot;data/TORUS/toy/snp_anno&quot;</span><span class="p">,</span> <span class="n">sep</span> <span class="o">=</span> <span class="s2">&quot;</span><span class="se">\t</span><span class="s2">&quot;</span><span class="p">,</span> <span class="n">header</span> <span class="o">=</span> <span class="kc">True</span><span class="p">,</span> <span class="n">index</span> <span class="o">=</span> <span class="kc">False</span><span class="p">)</span>
    <span class="k">return</span> <span class="n">df_bf</span><span class="p">,</span> <span class="n">df_beta</span><span class="p">,</span> <span class="n">df_zscore</span><span class="p">,</span> <span class="n">SNP_anno</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[6]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="o">!</span>python ../utils/extract_function.py --from 20170906_toy_multiCNVs_multigenes_TORUS.ipynb --to get_sum_stats_for_torus.py
</pre></div>
</div>
</div>
</div>
<div class="output_wrapper">
<div class="output">
<div class="output_area">
<div class="prompt"></div>
<div class="output_subarea output_stream output_stdout output_text">
<pre>Extracting from 20170906_toy_multiCNVs_multigenes_TORUS.ipynb
Done!
</pre>
</div>
</div>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[5]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="c1"># for 15 genes toy, there are 6 categories, 1 for causal, 0,2,3,4 and 5 for nonn-causal</span>
<span class="n">cat6</span> <span class="o">=</span> <span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">*</span><span class="mi">3</span> <span class="o">+</span> <span class="p">[</span><span class="mi">2</span><span class="p">]</span><span class="o">*</span><span class="mi">3</span> <span class="o">+</span> <span class="p">[</span><span class="mi">3</span><span class="p">]</span><span class="o">*</span><span class="mi">3</span> <span class="o">+</span> <span class="p">[</span><span class="mi">4</span><span class="p">]</span><span class="o">*</span><span class="mi">2</span> <span class="o">+</span> <span class="p">[</span><span class="mi">5</span><span class="p">]</span><span class="o">*</span><span class="mi">2</span>
<span class="n">df_bf</span><span class="p">,</span> <span class="n">df_beta</span><span class="p">,</span> <span class="n">df_zscore</span><span class="p">,</span> <span class="n">snp_anno</span> <span class="o">=</span> <span class="n">get_summary_stats</span><span class="p">(</span><span class="n">df</span><span class="p">,</span> <span class="n">causal</span><span class="p">,</span> <span class="n">cat</span> <span class="o">=</span> <span class="n">cat6</span><span class="p">,</span> <span class="n">multi_cat</span> <span class="o">=</span> <span class="mi">1</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered">
<div class="prompt input_prompt">
</div>
<div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="command">command<a class="anchor-link" href="#command">&#182;</a></h3><p>Do NOT need the optional annotation files - snp_map and gene_map, only need snp annotation.</p>
<p>gzip *</p>
<p>torus.sh -d beta.gz -annot snp_anno.gz -est &gt; toy_multi_beta.est</p>
<p>/# torus.sh -d beta.gz -smap snp_map.gz -gmap gene_map.gz -annot snp_anno.gz -est &gt; toy_multi_beta.est</p>
<p>/# torus.sh -d bf.gz --load_bf -smap snp_map.gz -gmap gene_map.gz -annot snp_anno.gz -qtl &gt; toy_multi_bf.est</p>
<p>/# torus.sh -d z_score.gz --load_zval -smap snp_map.gz -gmap gene_map.gz -annot snp_anno.gz -qtl &gt; toy_multi_zscore.est</p>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[22]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="n">df_bf</span><span class="p">,</span> <span class="n">df_beta</span><span class="p">,</span> <span class="n">df_zscore</span><span class="p">,</span> <span class="n">snp_anno</span> <span class="o">=</span> <span class="n">get_summary_stats</span><span class="p">(</span><span class="n">df</span><span class="p">,</span> <span class="n">causal</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="output_wrapper">
<div class="output">
<div class="output_area">
<div class="prompt"></div>
<div class="output_subarea output_stream output_stdout output_text">
<pre>134 63 2.19081163037
162 134 1.22343372956
195 143 1.39439952672
178 168 1.06408678889
187 175 1.07411524921
205 169 1.23204548091
204 180 1.14518002323
201 174 1.16873903196
183 180 1.01798302403
187 179 1.04830602835
182 170 1.0761305385
180 148 1.2329916123
173 119 1.48751774023
163 108 1.5447788396
132 63 2.15629021879
</pre>
</div>
</div>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[14]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="nb">print</span> <span class="p">(</span><span class="n">df_bf</span><span class="p">)</span>
<span class="nb">print</span> <span class="p">(</span><span class="n">df_beta</span><span class="p">)</span>
<span class="nb">print</span> <span class="p">(</span><span class="n">df_zscore</span><span class="p">)</span>
<span class="nb">print</span> <span class="p">(</span><span class="n">snp_anno</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="output_wrapper">
<div class="output">
<div class="output_area">
<div class="prompt"></div>
<div class="output_subarea output_stream output_stdout output_text">
<pre>           SNP  gene        bf
0   chr6.gene1  chr6  2.190812
0   chr6.gene2  chr6  1.223434
0   chr6.gene3  chr6  1.394400
0   chr6.gene4  chr6  1.064087
0   chr6.gene5  chr6  1.074115
0   chr6.gene6  chr6  1.232045
0   chr6.gene7  chr6  1.145180
0   chr6.gene8  chr6  1.168739
0   chr6.gene9  chr6  1.017983
0  chr6.gene10  chr6  1.048306
0  chr6.gene11  chr6  1.076131
0  chr6.gene12  chr6  1.232992
0  chr6.gene13  chr6  1.487518
0  chr6.gene14  chr6  1.544779
0  chr6.gene15  chr6  2.156290
           SNP  gene      beta    t-stat   p-value
0   chr6.gene1  chr6  0.784272  5.140413  0.000001
0   chr6.gene2  chr6  0.201661  1.618618  0.052765
0   chr6.gene3  chr6  0.332464  2.877339  0.002005
0   chr6.gene4  chr6  0.062117  0.501427  0.308035
0   chr6.gene5  chr6  0.071497  0.600209  0.274183
0   chr6.gene6  chr6  0.208676  1.882458  0.029887
0   chr6.gene7  chr6  0.135562  1.221669  0.110916
0   chr6.gene8  chr6  0.155925  1.396256  0.081319
0   chr6.gene9  chr6  0.017823  0.108983  0.456608
0  chr6.gene10  chr6  0.047176  0.380007  0.351970
0  chr6.gene11  chr6  0.073372  0.608019  0.271588
0  chr6.gene12  chr6  0.209443  1.771595  0.038231
0  chr6.gene13  chr6  0.397109  3.203872  0.000678
0  chr6.gene14  chr6  0.434881  3.382716  0.000359
0  chr6.gene15  chr6  0.768389  5.016009  0.000001
           SNP  gene   z-score
0   chr6.gene1  chr6  5.140413
0   chr6.gene2  chr6  1.618618
0   chr6.gene3  chr6  2.877339
0   chr6.gene4  chr6  0.501427
0   chr6.gene5  chr6  0.600209
0   chr6.gene6  chr6  1.882458
0   chr6.gene7  chr6  1.221669
0   chr6.gene8  chr6  1.396256
0   chr6.gene9  chr6  0.108983
0  chr6.gene10  chr6  0.380007
0  chr6.gene11  chr6  0.608019
0  chr6.gene12  chr6  1.771595
0  chr6.gene13  chr6  3.203872
0  chr6.gene14  chr6  3.382716
0  chr6.gene15  chr6  5.016009
           SNP  binding_d
0   chr6.gene1          1
0   chr6.gene2          0
0   chr6.gene3          0
0   chr6.gene4          0
0   chr6.gene5          0
0   chr6.gene6          0
0   chr6.gene7          0
0   chr6.gene8          0
0   chr6.gene9          0
0  chr6.gene10          0
0  chr6.gene11          0
0  chr6.gene12          0
0  chr6.gene13          0
0  chr6.gene14          0
0  chr6.gene15          1
</pre>
</div>
</div>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[9]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="k">for</span> <span class="n">idx</span><span class="p">,</span> <span class="n">item</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">df</span><span class="o">.</span><span class="n">columns</span><span class="o">.</span><span class="n">values</span><span class="p">):</span>
    <span class="nb">print</span> <span class="p">(</span><span class="n">idx</span><span class="p">,</span> <span class="n">item</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="output_wrapper">
<div class="output">
<div class="output_area">
<div class="prompt"></div>
<div class="output_subarea output_stream output_stdout output_text">
<pre>0 phenotype
1 gene1
2 gene2
3 gene3
4 gene4
5 gene5
6 gene6
7 gene7
8 gene8
9 gene9
10 gene10
11 gene11
12 gene12
13 gene13
14 gene14
15 gene15
</pre>
</div>
</div>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[41]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="n">stats</span><span class="o">.</span><span class="n">norm</span><span class="o">.</span><span class="n">ppf</span><span class="p">((</span><span class="mi">1</span><span class="o">+</span><span class="p">(</span><span class="mi">1</span><span class="o">-</span><span class="mf">0.9999999999985895</span><span class="p">))</span><span class="o">/</span><span class="mi">2</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="output_wrapper">
<div class="output">
<div class="output_area">
<div class="prompt output_prompt">Out[41]:</div>
<div class="output_text output_subarea output_execute_result">
<pre>1.7677085129515064e-12</pre>
</div>
</div>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[10]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="n">startp</span><span class="o">=</span><span class="p">[</span><span class="mi">0</span><span class="p">,</span><span class="mi">6</span><span class="p">,</span><span class="mi">8</span><span class="p">,</span><span class="mi">9</span><span class="p">,</span><span class="mi">1</span><span class="p">,</span><span class="mi">4</span><span class="p">,</span><span class="mi">3</span><span class="p">]</span>
<span class="k">for</span> <span class="n">j</span><span class="p">,</span> <span class="n">i</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">startp</span><span class="p">):</span>
    <span class="nb">print</span> <span class="p">(</span><span class="n">j</span><span class="p">,</span><span class="n">i</span><span class="p">,</span> <span class="n">startp</span><span class="p">[</span><span class="n">j</span><span class="p">])</span>
</pre></div>
</div>
</div>
</div>
<div class="output_wrapper">
<div class="output">
<div class="output_area">
<div class="prompt"></div>
<div class="output_subarea output_stream output_stdout output_text">
<pre>0 0 0
1 6 6
2 8 8
3 9 9
4 1 1
5 4 4
6 3 3
</pre>
</div>
</div>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[21]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="n">cat</span> <span class="o">=</span> <span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">*</span><span class="mi">3</span> <span class="o">+</span> <span class="p">[</span><span class="mi">2</span><span class="p">]</span><span class="o">*</span><span class="mi">3</span> <span class="o">+</span> <span class="p">[</span><span class="mi">3</span><span class="p">]</span><span class="o">*</span><span class="mi">3</span> <span class="o">+</span> <span class="p">[</span><span class="mi">4</span><span class="p">]</span><span class="o">*</span><span class="mi">2</span> <span class="o">+</span> <span class="p">[</span><span class="mi">5</span><span class="p">]</span><span class="o">*</span><span class="mi">2</span>
<span class="k">if</span> <span class="p">(</span><span class="mi">5</span><span class="p">)</span> <span class="ow">in</span> <span class="n">causal</span><span class="p">:</span>
    <span class="nb">print</span> <span class="p">(</span><span class="s2">&quot;balala&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span> 
</pre></div>
</div>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[121]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="c1"># # previous version, bug in it, fixed in function &quot;toy_multicnv_multicausal&quot;</span>
<span class="c1"># def toy_multi_gene_cnv(p, n_cnv, n_max=10, causal=5, const1=1, const2=0.1, seed=1):</span>
<span class="c1">#     &#39;&#39;&#39;A region with multiple genes and CNVs. CNVs may overlap or not overlap. Only one causal gene.&#39;&#39;&#39;</span>
<span class="c1">#     np.random.seed(seed)</span>
<span class="c1">#     cnv_len = np.random.choice((np.random.geometric(p, size=n_cnv) - 1), n_cnv, replace=False)</span>
<span class="c1">#     cnv_len = cnv_len[cnv_len &lt;= 10].tolist()</span>
<span class="c1">#     start_pos = [np.random.choice(range(n_max+1-i)) if i!=0 else -1 for i in cnv_len]</span>
<span class="c1">#     ptn_ls = []</span>
<span class="c1">#     for j,i in enumerate(start_pos):</span>
<span class="c1">#         if i == -1:</span>
<span class="c1">#             ptn = [0]*n_max</span>
<span class="c1">#         elif i == 0:</span>
<span class="c1">#             ptn = [0]*(i-1) + [1]*cnv_len[j] + [0]*(n_max-cnv_len[j]-i)</span>
<span class="c1">#         else:</span>
<span class="c1">#             ptn = [0]*(i-1) + [1]*cnv_len[j] + [0]*(n_max-cnv_len[j]-(i-1))</span>
<span class="c1">#         ptn_ls.append(ptn)</span>
<span class="c1">#     mat = [[np.random.normal(0,1)] + line for line in ptn_ls]</span>
<span class="c1">#     config = []</span>
<span class="c1">#     for line in mat:</span>
<span class="c1"># ##         config.append(line[0]+const1) if line[causal-1] == 1 else config.append(line[0])</span>
<span class="c1">#         if line[causal] == 1:</span>
<span class="c1">#             line[0] = line[0] + const1</span>
<span class="c1">#         config.append(line[0])</span>
<span class="c1"># #       the longer the CNV is, the less common it is, and larger OR</span>
<span class="c1">#         line[0] = line[0] + line[1:].count(1)*const2</span>
<span class="c1"># ##         line[0] = 1 if line[0] &gt; np.median(config) else 0</span>
<span class="c1">#         if line[0] &gt;= np.median(config):</span>
<span class="c1">#             line[0] = 1</span>
<span class="c1">#         else:</span>
<span class="c1">#             line[0] = 0</span>
<span class="c1">#     mat = np.matrix(mat)</span>
<span class="c1">#     df = pd.DataFrame(mat, columns = [&quot;phenotype&quot;] + [&quot;gene{}&quot;.format(i+1) for i in range(mat.shape[1]-1)])</span>
<span class="c1">#     counts = df.groupby([&quot;gene{}&quot;.format(i+1) for i in range(mat.shape[1]-1)] + [&quot;phenotype&quot;]).size()</span>
<span class="c1">#     for col in df.columns:</span>
<span class="c1">#         df[col] = df[col].astype(np.uint8)</span>
<span class="c1">#     output = feather.write_dataframe(df, &quot;data/toy_n{}_p{}_causal{}_const{}_{}.feather&quot;</span>
<span class="c1">#                                      .format(len(cnv_len), p, causal, const1, const2))</span>
<span class="c1">#     return df, counts</span>
</pre></div>
</div>
</div>
</div>
</div>
<hr>
&copy 2016-2017 Min Qiao at <a href="http://xinhelab.org">Xin He lab</a>, the University of Chicago
<p><small>Exported from <a href="http://github.com/gaow/cnv-gene-mapping/blob/dfd84ddd1e708bfd2c74f19995b02136239caff9/analysis/20170906_toy_multiCNVs_multigenes_TORUS.ipynb"><code>analysis/20170906_toy_multiCNVs_multigenes_TORUS.ipynb</code></a> committed by Min Qiao on Tue Oct 10 16:11:21 2017 <a href="http://github.com/gaow/cnv-gene-mapping/commit/dfd84ddd1e708bfd2c74f19995b02136239caff9">revision 350, dfd84dd</a> <a href="https://stephenslab.github.io/ipynb-website/notes.html#Note-about-commit-ids"><span class="fa fa-question-circle"></span></a></small></p>
</div>
</div>
</body>
</html>
