<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="ipynb_website:version" content="0.9.2" />
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link rel="stylesheet" type="text/css" href="../css/jt.css">
<link rel="stylesheet" type="text/css" href="../css/toc2.css">
<link href="../site_libs/jqueryui-1.11.4/jquery-ui.css">
<link rel="stylesheet" href="../site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<link rel="stylesheet" href="../site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />
<script type="text/javascript" src="https://ajax.googleapis.com/ajax/libs/jquery/1.9.1/jquery.min.js"></script>
<script type="text/javascript" src="https://ajax.googleapis.com/ajax/libs/jqueryui/1.9.1/jquery-ui.min.js"></script>
<script src="../site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="../site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="../site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<link rel="stylesheet"
      href="../site_libs/highlight/textmate.css"
      type="text/css" />
<script src="../site_libs/highlight/highlight.js"></script>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>
<script src="../js/toc2.js"></script>
<script src="../js/docs.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS_HTML"></script>
<script>
    MathJax.Hub.Config({
        extensions: ["tex2jax.js"],
        jax: ["input/TeX", "output/HTML-CSS"],
        tex2jax: {
        inlineMath: [ ['$','$'], ["\\(","\\)"] ],
        displayMath: [ ['$$','$$'], ["\\[","\\]"] ],
        processEscapes: true
        },
        "HTML-CSS": {
            preferredFont: "TeX",
            availableFonts: ["TeX"],
            styles: {
                scale: 110,
                ".MathJax_Display": {
                    "font-size": "110%",
                }
            }
        }
    });
</script>
<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');
  // mark it active
  menuAnchor.parent().addClass('active');
  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>
<div class="container-fluid main-container">
<!-- tabsets -->
<script src="../site_libs/navigation-1.1/tabsets.js"></script>
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>
<title>Gene Mapping with CNV</title>
<style type = "text/css">
body {
  font-family: "Droid Sans";
  padding-top: 66px;
  padding-bottom: 40px;
}
</style>
</head>
<body>
<div tabindex="-1" id="notebook" class="border-box-sizing">
<div class="container" id="notebook-container">
<!-- code folding -->
<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="../index.html">Gene Mapping with CNV</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">Overview</a>
</li>
<li>
  <a href="../analysis.html">Analysis</a>
</li>
<li>
  <a href="../setup.html">Setup</a>
</li>
<li>
  <a href="../writeup.html">Writeup</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
   <a href="http://github.com/gaow/cnv-gene-mapping"> <span class="fa fa-github"></span> </a>
</li>
</ul>
      </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->
<div class="cell border-box-sizing text_cell rendered">
<div class="prompt input_prompt">
</div>
<div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h1 id="Data-transform:-$y-\in-\{0,1\}$-to-$y-\in-\{0,t\}$">Data transform: $y \in \{0,1\}$ to $y \in \{0,t\}$<a class="anchor-link" href="#Data-transform:-$y-\in-\{0,1\}$-to-$y-\in-\{0,t\}$">&#182;</a></h1>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[1]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="kn">import</span> <span class="nn">numpy</span> <span class="k">as</span> <span class="nn">np</span>
<span class="kn">import</span> <span class="nn">pandas</span> <span class="k">as</span> <span class="nn">pd</span>
<span class="kn">from</span> <span class="nn">pprint</span> <span class="k">import</span> <span class="n">pprint</span>
<span class="kn">from</span> <span class="nn">pandasql</span> <span class="k">import</span> <span class="n">sqldf</span>
<span class="kn">from</span> <span class="nn">scipy</span> <span class="k">import</span> <span class="n">stats</span>
<span class="kn">import</span> <span class="nn">pickle</span>
<span class="kn">import</span> <span class="nn">feather</span>
<span class="kn">from</span> <span class="nn">collections</span> <span class="k">import</span> <span class="n">Counter</span>
<span class="kn">from</span> <span class="nn">datetime</span> <span class="k">import</span> <span class="n">datetime</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[2]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="c1"># the lifetime prevalence of schizophrenia is 4.0/1000, not 1%, but we first set it as 0.01</span>
<span class="c1"># according to https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1140960/</span>
<span class="n">prevalence</span> <span class="o">=</span> <span class="mf">0.01</span>
<span class="c1"># relative risk is normally between 2 and 10</span>
<span class="n">RR</span> <span class="o">=</span> <span class="mf">2.5</span>
<span class="n">t</span> <span class="o">=</span> <span class="n">stats</span><span class="o">.</span><span class="n">norm</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="mi">1</span><span class="p">)</span><span class="o">.</span><span class="n">ppf</span><span class="p">(</span><span class="mi">1</span><span class="o">-</span><span class="n">prevalence</span><span class="p">)</span>
<span class="c1"># effect size of beta is z_mu, prior</span>
<span class="n">z_mu</span> <span class="o">=</span> <span class="n">t</span> <span class="o">-</span> <span class="n">stats</span><span class="o">.</span><span class="n">norm</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="mi">1</span><span class="p">)</span><span class="o">.</span><span class="n">ppf</span><span class="p">(</span><span class="mi">1</span><span class="o">-</span><span class="n">RR</span><span class="o">*</span><span class="n">prevalence</span><span class="p">)</span>
<span class="nb">print</span> <span class="p">(</span><span class="n">t</span><span class="p">,</span> <span class="n">z_mu</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="output_wrapper">
<div class="output">
<div class="output_area">
<div class="prompt"></div>
<div class="output_subarea output_stream output_stdout output_text">
<pre>2.32634787404 0.366383889501
</pre>
</div>
</div>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[3]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="c1"># create prior file</span>
<span class="n">cal_pthwy_genes</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_table</span><span class="p">(</span><span class="s2">&quot;../data/calciumgeneset.txt&quot;</span><span class="p">,</span> <span class="n">skiprows</span> <span class="o">=</span> <span class="mi">2</span><span class="p">,</span> <span class="n">header</span> <span class="o">=</span> <span class="kc">None</span><span class="p">,</span> <span class="n">names</span> <span class="o">=</span> <span class="p">[</span><span class="s2">&quot;gene_name&quot;</span><span class="p">])</span>
<span class="n">ref_genes</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_table</span><span class="p">(</span><span class="s2">&quot;data/refGene.txt.gz&quot;</span><span class="p">,</span> <span class="n">compression</span><span class="o">=</span><span class="s2">&quot;gzip&quot;</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">&quot;</span><span class="se">\t</span><span class="s2">&quot;</span><span class="p">,</span> <span class="n">header</span> <span class="o">=</span> <span class="kc">None</span><span class="p">,</span> 
                          <span class="n">usecols</span><span class="o">=</span><span class="p">(</span><span class="mi">2</span><span class="p">,</span><span class="mi">12</span><span class="p">),</span> <span class="n">names</span><span class="o">=</span><span class="p">[</span><span class="s2">&quot;chrom&quot;</span><span class="p">,</span> <span class="s2">&quot;gene_name&quot;</span><span class="p">])</span><span class="o">.</span><span class="n">drop_duplicates</span><span class="p">(</span><span class="n">subset</span><span class="o">=</span><span class="p">(</span><span class="s2">&quot;gene_name&quot;</span><span class="p">))</span>
<span class="n">calpath_gene_pos</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">merge</span><span class="p">(</span><span class="n">ref_genes</span><span class="p">,</span> <span class="n">cal_pthwy_genes</span><span class="p">,</span> <span class="n">how</span><span class="o">=</span><span class="s2">&quot;inner&quot;</span><span class="p">,</span> <span class="n">on</span><span class="o">=</span><span class="p">[</span><span class="s2">&quot;gene_name&quot;</span><span class="p">])</span>
<span class="n">diff</span> <span class="o">=</span> <span class="nb">set</span><span class="p">(</span><span class="n">cal_pthwy_genes</span><span class="p">[</span><span class="s2">&quot;gene_name&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">tolist</span><span class="p">())</span> <span class="o">-</span> <span class="nb">set</span><span class="p">(</span><span class="n">calpath_gene_pos</span><span class="p">[</span><span class="s2">&quot;gene_name&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">tolist</span><span class="p">())</span>
<span class="c1"># add the two missing genes and their chromosome in the dataframe</span>
<span class="n">calpath_gene_pos</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">calpath_gene_pos</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]]</span> <span class="o">=</span> <span class="p">[</span><span class="s2">&quot;chr2&quot;</span><span class="p">,</span> <span class="nb">list</span><span class="p">(</span><span class="n">diff</span><span class="p">)[</span><span class="mi">0</span><span class="p">]]</span>
<span class="n">calpath_gene_pos</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">calpath_gene_pos</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]]</span> <span class="o">=</span> <span class="p">[</span><span class="s2">&quot;chr15&quot;</span><span class="p">,</span> <span class="nb">list</span><span class="p">(</span><span class="n">diff</span><span class="p">)[</span><span class="mi">1</span><span class="p">]]</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[4]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="n">calpath_gene_pos</span><span class="p">[</span><span class="s2">&quot;chr.gene&quot;</span><span class="p">]</span> <span class="o">=</span> <span class="n">calpath_gene_pos</span><span class="o">.</span><span class="n">apply</span><span class="p">(</span><span class="k">lambda</span> <span class="n">row</span><span class="p">:</span> <span class="s2">&quot;</span><span class="si">{}</span><span class="s2">.</span><span class="si">{}</span><span class="s2">&quot;</span><span class="o">.</span><span class="n">format</span><span class="p">(</span><span class="n">row</span><span class="p">[</span><span class="s2">&quot;chrom&quot;</span><span class="p">],</span> <span class="n">row</span><span class="p">[</span><span class="s2">&quot;gene_name&quot;</span><span class="p">]),</span> 
                                                      <span class="n">axis</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span>
<span class="n">calpath_gene_pos</span><span class="p">[</span><span class="s2">&quot;prior.pr&quot;</span><span class="p">]</span> <span class="o">=</span> <span class="n">calpath_gene_pos</span><span class="o">.</span><span class="n">apply</span><span class="p">(</span><span class="k">lambda</span> <span class="n">row</span><span class="p">:</span> <span class="n">np</span><span class="o">.</span><span class="n">random</span><span class="o">.</span><span class="n">uniform</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span> <span class="n">z_mu</span><span class="p">),</span> <span class="n">axis</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span>
<span class="n">calpath_gene_pos</span>
</pre></div>
</div>
</div>
</div>
<div class="output_wrapper">
<div class="output">
<div class="output_area">
<div class="prompt output_prompt">Out[4]:</div>
<div class="output_html rendered_html output_subarea output_execute_result">
<div>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>chrom</th>
      <th>gene_name</th>
      <th>chr.gene</th>
      <th>prior.pr</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>chr17</td>
      <td>CACNA1G</td>
      <td>chr17.CACNA1G</td>
      <td>0.266285</td>
    </tr>
    <tr>
      <th>1</th>
      <td>chr2</td>
      <td>PLCD4</td>
      <td>chr2.PLCD4</td>
      <td>0.107040</td>
    </tr>
    <tr>
      <th>2</th>
      <td>chr20</td>
      <td>PLCB4</td>
      <td>chr20.PLCB4</td>
      <td>0.362211</td>
    </tr>
    <tr>
      <th>3</th>
      <td>chr4</td>
      <td>CAMK2D</td>
      <td>chr4.CAMK2D</td>
      <td>0.140378</td>
    </tr>
    <tr>
      <th>4</th>
      <td>chr14</td>
      <td>SLC8A3</td>
      <td>chr14.SLC8A3</td>
      <td>0.169377</td>
    </tr>
    <tr>
      <th>5</th>
      <td>chr8</td>
      <td>PTK2B</td>
      <td>chr8.PTK2B</td>
      <td>0.210375</td>
    </tr>
    <tr>
      <th>6</th>
      <td>chr9</td>
      <td>GNA14</td>
      <td>chr9.GNA14</td>
      <td>0.059547</td>
    </tr>
    <tr>
      <th>7</th>
      <td>chr8</td>
      <td>VDAC3</td>
      <td>chr8.VDAC3</td>
      <td>0.157412</td>
    </tr>
    <tr>
      <th>8</th>
      <td>chr12</td>
      <td>PDE1B</td>
      <td>chr12.PDE1B</td>
      <td>0.075162</td>
    </tr>
    <tr>
      <th>9</th>
      <td>chr12</td>
      <td>CACNA1C</td>
      <td>chr12.CACNA1C</td>
      <td>0.256323</td>
    </tr>
    <tr>
      <th>10</th>
      <td>chr3</td>
      <td>ITPR1</td>
      <td>chr3.ITPR1</td>
      <td>0.271080</td>
    </tr>
    <tr>
      <th>11</th>
      <td>chr1</td>
      <td>PRKACB</td>
      <td>chr1.PRKACB</td>
      <td>0.294886</td>
    </tr>
    <tr>
      <th>12</th>
      <td>chr17</td>
      <td>ATP2A3</td>
      <td>chr17.ATP2A3</td>
      <td>0.347966</td>
    </tr>
    <tr>
      <th>13</th>
      <td>chr20</td>
      <td>ADRA1D</td>
      <td>chr20.ADRA1D</td>
      <td>0.220837</td>
    </tr>
    <tr>
      <th>14</th>
      <td>chr5</td>
      <td>HTR4</td>
      <td>chr5.HTR4</td>
      <td>0.296342</td>
    </tr>
    <tr>
      <th>15</th>
      <td>chr1</td>
      <td>PTAFR</td>
      <td>chr1.PTAFR</td>
      <td>0.304160</td>
    </tr>
    <tr>
      <th>16</th>
      <td>chr4</td>
      <td>CD38</td>
      <td>chr4.CD38</td>
      <td>0.004339</td>
    </tr>
    <tr>
      <th>17</th>
      <td>chrX</td>
      <td>SLC25A6</td>
      <td>chrX.SLC25A6</td>
      <td>0.333085</td>
    </tr>
    <tr>
      <th>18</th>
      <td>chr1</td>
      <td>PTGER3</td>
      <td>chr1.PTGER3</td>
      <td>0.075069</td>
    </tr>
    <tr>
      <th>19</th>
      <td>chr19</td>
      <td>SPHK2</td>
      <td>chr19.SPHK2</td>
      <td>0.173236</td>
    </tr>
    <tr>
      <th>20</th>
      <td>chr12</td>
      <td>ATP2A2</td>
      <td>chr12.ATP2A2</td>
      <td>0.208458</td>
    </tr>
    <tr>
      <th>21</th>
      <td>chr12</td>
      <td>P2RX2</td>
      <td>chr12.P2RX2</td>
      <td>0.173613</td>
    </tr>
    <tr>
      <th>22</th>
      <td>chr1</td>
      <td>CACNA1E</td>
      <td>chr1.CACNA1E</td>
      <td>0.323274</td>
    </tr>
    <tr>
      <th>23</th>
      <td>chr1</td>
      <td>ATP2B4</td>
      <td>chr1.ATP2B4</td>
      <td>0.093543</td>
    </tr>
    <tr>
      <th>24</th>
      <td>chr2</td>
      <td>ADCY3</td>
      <td>chr2.ADCY3</td>
      <td>0.310170</td>
    </tr>
    <tr>
      <th>25</th>
      <td>chr16</td>
      <td>PLCG2</td>
      <td>chr16.PLCG2</td>
      <td>0.139039</td>
    </tr>
    <tr>
      <th>26</th>
      <td>chr18</td>
      <td>GNAL</td>
      <td>chr18.GNAL</td>
      <td>0.226588</td>
    </tr>
    <tr>
      <th>27</th>
      <td>chr6</td>
      <td>ITPR3</td>
      <td>chr6.ITPR3</td>
      <td>0.100228</td>
    </tr>
    <tr>
      <th>28</th>
      <td>chr1</td>
      <td>RYR2</td>
      <td>chr1.RYR2</td>
      <td>0.267196</td>
    </tr>
    <tr>
      <th>29</th>
      <td>chr10</td>
      <td>CAMK2G</td>
      <td>chr10.CAMK2G</td>
      <td>0.027177</td>
    </tr>
    <tr>
      <th>...</th>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
    </tr>
    <tr>
      <th>148</th>
      <td>chr12</td>
      <td>P2RX7</td>
      <td>chr12.P2RX7</td>
      <td>0.302376</td>
    </tr>
    <tr>
      <th>149</th>
      <td>chr15</td>
      <td>ITPKA</td>
      <td>chr15.ITPKA</td>
      <td>0.320564</td>
    </tr>
    <tr>
      <th>150</th>
      <td>chr8</td>
      <td>TRHR</td>
      <td>chr8.TRHR</td>
      <td>0.134721</td>
    </tr>
    <tr>
      <th>151</th>
      <td>chr1</td>
      <td>CACNA1S</td>
      <td>chr1.CACNA1S</td>
      <td>0.259831</td>
    </tr>
    <tr>
      <th>152</th>
      <td>chr9</td>
      <td>GNAQ</td>
      <td>chr9.GNAQ</td>
      <td>0.275942</td>
    </tr>
    <tr>
      <th>153</th>
      <td>chr15</td>
      <td>PLCB2</td>
      <td>chr15.PLCB2</td>
      <td>0.121064</td>
    </tr>
    <tr>
      <th>154</th>
      <td>chr7</td>
      <td>ADCY1</td>
      <td>chr7.ADCY1</td>
      <td>0.343054</td>
    </tr>
    <tr>
      <th>155</th>
      <td>chr15</td>
      <td>RYR3</td>
      <td>chr15.RYR3</td>
      <td>0.265300</td>
    </tr>
    <tr>
      <th>156</th>
      <td>chr7</td>
      <td>PHKG1</td>
      <td>chr7.PHKG1</td>
      <td>0.157181</td>
    </tr>
    <tr>
      <th>157</th>
      <td>chrX</td>
      <td>CYSLTR1</td>
      <td>chrX.CYSLTR1</td>
      <td>0.068701</td>
    </tr>
    <tr>
      <th>158</th>
      <td>chr5</td>
      <td>PDGFRB</td>
      <td>chr5.PDGFRB</td>
      <td>0.293815</td>
    </tr>
    <tr>
      <th>159</th>
      <td>chr4</td>
      <td>SLC25A4</td>
      <td>chr4.SLC25A4</td>
      <td>0.070235</td>
    </tr>
    <tr>
      <th>160</th>
      <td>chr8</td>
      <td>PPP3CC</td>
      <td>chr8.PPP3CC</td>
      <td>0.356010</td>
    </tr>
    <tr>
      <th>161</th>
      <td>chr2</td>
      <td>LHCGR</td>
      <td>chr2.LHCGR</td>
      <td>0.094462</td>
    </tr>
    <tr>
      <th>162</th>
      <td>chr8</td>
      <td>ADRB3</td>
      <td>chr8.ADRB3</td>
      <td>0.330471</td>
    </tr>
    <tr>
      <th>163</th>
      <td>chr3</td>
      <td>OXTR</td>
      <td>chr3.OXTR</td>
      <td>0.327985</td>
    </tr>
    <tr>
      <th>164</th>
      <td>chr10</td>
      <td>CALML5</td>
      <td>chr10.CALML5</td>
      <td>0.363580</td>
    </tr>
    <tr>
      <th>165</th>
      <td>chr1</td>
      <td>HTR6</td>
      <td>chr1.HTR6</td>
      <td>0.337619</td>
    </tr>
    <tr>
      <th>166</th>
      <td>chr20</td>
      <td>PLCG1</td>
      <td>chr20.PLCG1</td>
      <td>0.244072</td>
    </tr>
    <tr>
      <th>167</th>
      <td>chr20</td>
      <td>MYLK2</td>
      <td>chr20.MYLK2</td>
      <td>0.164047</td>
    </tr>
    <tr>
      <th>168</th>
      <td>chr7</td>
      <td>CHRM2</td>
      <td>chr7.CHRM2</td>
      <td>0.268262</td>
    </tr>
    <tr>
      <th>169</th>
      <td>chr11</td>
      <td>P2RX3</td>
      <td>chr11.P2RX3</td>
      <td>0.051683</td>
    </tr>
    <tr>
      <th>170</th>
      <td>chrX</td>
      <td>CACNA1F</td>
      <td>chrX.CACNA1F</td>
      <td>0.190121</td>
    </tr>
    <tr>
      <th>171</th>
      <td>chrX</td>
      <td>SLC25A5</td>
      <td>chrX.SLC25A5</td>
      <td>0.180459</td>
    </tr>
    <tr>
      <th>172</th>
      <td>chrX</td>
      <td>ATP2B3</td>
      <td>chrX.ATP2B3</td>
      <td>0.180794</td>
    </tr>
    <tr>
      <th>173</th>
      <td>chr16</td>
      <td>PRKCB</td>
      <td>chr16.PRKCB</td>
      <td>0.272680</td>
    </tr>
    <tr>
      <th>174</th>
      <td>chr1</td>
      <td>CALML6</td>
      <td>chr1.CALML6</td>
      <td>0.147582</td>
    </tr>
    <tr>
      <th>175</th>
      <td>chr13</td>
      <td>CYSLTR2</td>
      <td>chr13.CYSLTR2</td>
      <td>0.222805</td>
    </tr>
    <tr>
      <th>176</th>
      <td>chr2</td>
      <td>CHP</td>
      <td>chr2.CHP</td>
      <td>0.351839</td>
    </tr>
    <tr>
      <th>177</th>
      <td>chr15</td>
      <td>LOC729317</td>
      <td>chr15.LOC729317</td>
      <td>0.328145</td>
    </tr>
  </tbody>
</table>
<p>178 rows × 4 columns</p>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[5]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="n">beta</span> <span class="o">=</span> <span class="n">z_mu</span>
<span class="n">y_tran</span> <span class="o">=</span> <span class="n">t</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[6]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="n">filename</span> <span class="o">=</span> <span class="s2">&quot;data/del_sim_scale3_shape1_N100_M100_999.feather&quot;</span>
<span class="n">fileout</span> <span class="o">=</span> <span class="s2">&quot;data/del_sim_scale3_shape1_N100_M100_999.dap&quot;</span>
<span class="n">input_file</span> <span class="o">=</span> <span class="n">feather</span><span class="o">.</span><span class="n">read_dataframe</span><span class="p">(</span><span class="n">filename</span><span class="p">)</span>
<span class="n">input_file</span><span class="p">[</span><span class="s2">&quot;phenotype&quot;</span><span class="p">]</span> <span class="o">=</span> <span class="n">input_file</span><span class="o">.</span><span class="n">apply</span><span class="p">(</span><span class="k">lambda</span> <span class="n">row</span><span class="p">:</span> <span class="n">y_tran</span> <span class="k">if</span> <span class="n">row</span><span class="p">[</span><span class="s2">&quot;phenotype&quot;</span><span class="p">]</span><span class="o">==</span><span class="mf">1.0</span> <span class="k">else</span> <span class="mi">0</span><span class="p">,</span> <span class="n">axis</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span>
<span class="k">def</span> <span class="nf">run_dap</span><span class="p">(</span><span class="n">df</span><span class="p">,</span> <span class="n">fileout</span><span class="p">,</span> <span class="n">multiplier</span> <span class="o">=</span> <span class="mi">2</span><span class="p">,</span> <span class="n">prefix</span> <span class="o">=</span> <span class="kc">None</span><span class="p">,</span> <span class="n">exec_path</span> <span class="o">=</span> <span class="kc">None</span><span class="p">):</span>
    <span class="sd">&#39;&#39;&#39;Convert pandas dataframe to dap input:</span>
<span class="sd">        - phenotype / genotype file</span>
<span class="sd">        - prior file</span>
<span class="sd">        - grid file (of effect size): omega^2 + phi^2 is what we care. Let&#39;s set it to</span>
<span class="sd">            *************** for now, as we only have one Y</span>
<span class="sd">    &#39;&#39;&#39;</span>
    <span class="nb">print</span> <span class="p">(</span><span class="nb">str</span><span class="p">(</span><span class="n">datetime</span><span class="o">.</span><span class="n">now</span><span class="p">()))</span>
    <span class="kn">import</span> <span class="nn">os</span>
    <span class="k">if</span> <span class="n">prefix</span> <span class="ow">is</span> <span class="kc">None</span><span class="p">:</span>
        <span class="kn">import</span> <span class="nn">time</span>
        <span class="n">prefix</span> <span class="o">=</span> <span class="s2">&quot;/tmp/F&quot;</span> <span class="o">+</span> <span class="nb">str</span><span class="p">(</span><span class="n">time</span><span class="o">.</span><span class="n">time</span><span class="p">())</span>
    <span class="k">if</span> <span class="n">exec_path</span> <span class="ow">is</span> <span class="kc">None</span><span class="p">:</span>
        <span class="n">exec_path</span> <span class="o">=</span> <span class="s1">&#39;dap/dap&#39;</span>
<span class="c1">#     chrom = [&quot;chr{}&quot;.format(i) for i in list(range(1,23))+[&quot;X&quot;]]</span>
<span class="c1">#     pos is all the gene names showed in .feather</span>
    <span class="n">pos</span> <span class="o">=</span> <span class="n">df</span><span class="o">.</span><span class="n">columns</span><span class="o">.</span><span class="n">values</span><span class="p">[</span><span class="mi">1</span><span class="p">:]</span><span class="o">.</span><span class="n">tolist</span><span class="p">()</span>
<span class="c1">#     first item in &quot;dat&quot; is phenotype for all samples (# of rows in .feather)</span>
    <span class="n">dat</span> <span class="o">=</span> <span class="p">[[</span><span class="s1">&#39;pheno&#39;</span><span class="p">,</span> <span class="s1">&#39;trait&#39;</span><span class="p">,</span> <span class="s1">&#39;chicago&#39;</span><span class="p">]</span> <span class="o">+</span> <span class="p">[</span><span class="nb">str</span><span class="p">(</span><span class="n">x</span><span class="p">)</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">df</span><span class="p">[</span><span class="s1">&#39;phenotype&#39;</span><span class="p">]]]</span>
    <span class="n">prior</span> <span class="o">=</span> <span class="p">[]</span>
    <span class="n">grid</span> <span class="o">=</span> <span class="p">[(</span><span class="mi">0</span><span class="p">,</span><span class="n">beta</span><span class="p">)]</span>
<span class="c1">#     item is gene name</span>
    <span class="k">for</span> <span class="n">idx</span><span class="p">,</span> <span class="n">item</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">df</span><span class="o">.</span><span class="n">columns</span><span class="o">.</span><span class="n">values</span><span class="p">):</span>
        <span class="k">if</span> <span class="n">item</span> <span class="o">==</span> <span class="s2">&quot;phenotype&quot;</span><span class="p">:</span>
            <span class="k">continue</span>
        <span class="n">dat</span><span class="o">.</span><span class="n">append</span><span class="p">([</span><span class="s1">&#39;geno&#39;</span><span class="p">,</span> <span class="s1">&#39;</span><span class="si">{}</span><span class="s1">.</span><span class="si">{}</span><span class="s1">&#39;</span><span class="o">.</span><span class="n">format</span><span class="p">(</span><span class="n">ref_genes</span><span class="p">[</span><span class="n">ref_genes</span><span class="p">[</span><span class="s2">&quot;gene_name&quot;</span><span class="p">]</span><span class="o">==</span><span class="n">item</span><span class="p">][</span><span class="s2">&quot;chrom&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">tolist</span><span class="p">()[</span><span class="mi">0</span><span class="p">],</span> <span class="n">item</span><span class="p">),</span> 
                    <span class="s1">&#39;chicago&#39;</span><span class="p">]</span> <span class="o">+</span> <span class="p">[</span><span class="nb">str</span><span class="p">(</span><span class="n">x</span><span class="p">)</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">df</span><span class="p">[</span><span class="n">item</span><span class="p">]])</span>
        <span class="n">n_overlap_gene</span> <span class="o">=</span> <span class="nb">len</span><span class="p">(</span><span class="nb">set</span><span class="p">(</span><span class="n">cal_pthwy_genes</span><span class="p">[</span><span class="s2">&quot;gene_name&quot;</span><span class="p">])</span> <span class="o">&amp;</span> <span class="nb">set</span><span class="p">(</span><span class="n">pos</span><span class="p">))</span>
        <span class="n">prior_causal</span> <span class="o">=</span> <span class="n">multiplier</span> <span class="o">/</span> <span class="nb">len</span><span class="p">(</span><span class="nb">set</span><span class="p">(</span><span class="n">pos</span><span class="p">))</span>
        <span class="n">prior_noncausal</span> <span class="o">=</span> <span class="p">(</span><span class="mi">1</span><span class="o">-</span><span class="n">prior_causal</span><span class="o">*</span><span class="n">n_overlap_gene</span><span class="p">)</span> <span class="o">/</span> <span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="nb">set</span><span class="p">(</span><span class="n">pos</span><span class="p">))</span><span class="o">-</span><span class="n">n_overlap_gene</span><span class="p">)</span>
        <span class="n">prior_pr</span> <span class="o">=</span> <span class="nb">str</span><span class="p">(</span><span class="n">prior_causal</span><span class="p">)</span> <span class="k">if</span> <span class="n">item</span> <span class="ow">in</span> <span class="n">cal_pthwy_genes</span><span class="p">[</span><span class="s2">&quot;gene_name&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">tolist</span><span class="p">()</span> <span class="k">else</span> <span class="nb">str</span><span class="p">(</span><span class="n">prior_noncausal</span><span class="p">)</span>
        <span class="n">prior</span><span class="o">.</span><span class="n">append</span><span class="p">([</span><span class="s1">&#39;</span><span class="si">{}</span><span class="s1">.</span><span class="si">{}</span><span class="s1">&#39;</span><span class="o">.</span><span class="n">format</span><span class="p">(</span><span class="n">ref_genes</span><span class="p">[</span><span class="n">ref_genes</span><span class="p">[</span><span class="s2">&quot;gene_name&quot;</span><span class="p">]</span><span class="o">==</span><span class="n">item</span><span class="p">][</span><span class="s2">&quot;chrom&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">tolist</span><span class="p">()[</span><span class="mi">0</span><span class="p">],</span> <span class="n">item</span><span class="p">),</span> <span class="n">prior_pr</span><span class="p">])</span>
<span class="c1">#     print (prior_causal, prior_noncausal)</span>
    <span class="k">with</span> <span class="nb">open</span><span class="p">(</span><span class="n">prefix</span> <span class="o">+</span> <span class="s1">&#39;.dat&#39;</span><span class="p">,</span> <span class="s1">&#39;w&#39;</span><span class="p">)</span> <span class="k">as</span> <span class="n">f</span><span class="p">:</span>
        <span class="n">f</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="s1">&#39;</span><span class="se">\n</span><span class="s1">&#39;</span><span class="o">.</span><span class="n">join</span><span class="p">([</span><span class="s1">&#39; &#39;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">x</span><span class="p">)</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">dat</span><span class="p">]))</span>
    <span class="k">with</span> <span class="nb">open</span><span class="p">(</span><span class="n">prefix</span> <span class="o">+</span> <span class="s1">&#39;.prior&#39;</span><span class="p">,</span> <span class="s1">&#39;w&#39;</span><span class="p">)</span> <span class="k">as</span> <span class="n">f</span><span class="p">:</span>
        <span class="n">f</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="s1">&#39;</span><span class="se">\n</span><span class="s1">&#39;</span><span class="o">.</span><span class="n">join</span><span class="p">([</span><span class="s1">&#39; &#39;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">x</span><span class="p">)</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">prior</span><span class="p">]))</span>
    <span class="k">with</span> <span class="nb">open</span><span class="p">(</span><span class="n">prefix</span> <span class="o">+</span> <span class="s1">&#39;.grid&#39;</span><span class="p">,</span> <span class="s1">&#39;w&#39;</span><span class="p">)</span> <span class="k">as</span> <span class="n">f</span><span class="p">:</span>
        <span class="n">f</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="s1">&#39;</span><span class="se">\n</span><span class="s1">&#39;</span><span class="o">.</span><span class="n">join</span><span class="p">([</span><span class="s1">&#39; &#39;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="nb">map</span><span class="p">(</span><span class="nb">str</span><span class="p">,</span> <span class="n">x</span><span class="p">))</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">grid</span><span class="p">]))</span>
    <span class="n">os</span><span class="o">.</span><span class="n">system</span><span class="p">(</span><span class="s2">&quot;</span><span class="si">{0}</span><span class="s2"> -d </span><span class="si">{1}</span><span class="s2">.dat -g </span><span class="si">{1}</span><span class="s2">.grid -t 8 -it 0.05 -prior </span><span class="si">{1}</span><span class="s2">.prior &gt; </span><span class="si">{2}</span><span class="s2">&quot;</span><span class="o">.</span><span class="n">format</span><span class="p">(</span><span class="n">exec_path</span><span class="p">,</span> <span class="n">prefix</span><span class="p">,</span> <span class="n">fileout</span><span class="p">))</span>
    <span class="nb">print</span> <span class="p">(</span><span class="nb">str</span><span class="p">(</span><span class="n">datetime</span><span class="o">.</span><span class="n">now</span><span class="p">()))</span>
<span class="c1">#     return prior</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[7]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="n">testp</span><span class="o">=</span><span class="n">run_dap</span><span class="p">(</span><span class="n">input_file</span><span class="p">,</span> <span class="n">fileout</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="output_wrapper">
<div class="output">
<div class="output_area">
<div class="prompt"></div>
<div class="output_subarea output_stream output_stdout output_text">
<pre>2017-08-04 20:22:54.007086
2017-08-04 20:23:02.148236
</pre>
</div>
</div>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[18]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="n">calpath_gene_pos</span><span class="o">.</span><span class="n">to_csv</span><span class="p">(</span><span class="s2">&quot;test.csv&quot;</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">&quot;</span><span class="se">\t</span><span class="s2">&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[10]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="c1"># prior_c = len(set(cal_pthwy_genes[&quot;gene_name&quot;]) &amp; set(df.columns.values[1:].tolist())) *2 / len(set(df.columns.values[1:].tolist())) / len(set(cal_pthwy_genes[&quot;gene_name&quot;]))</span>
<span class="n">prior_c</span> <span class="o">=</span> <span class="mi">2</span><span class="o">/</span><span class="nb">len</span><span class="p">(</span><span class="nb">set</span><span class="p">(</span><span class="n">df</span><span class="o">.</span><span class="n">columns</span><span class="o">.</span><span class="n">values</span><span class="p">[</span><span class="mi">1</span><span class="p">:]</span><span class="o">.</span><span class="n">tolist</span><span class="p">()))</span>
<span class="n">n_overlap_gene</span> <span class="o">=</span> <span class="nb">len</span><span class="p">(</span><span class="nb">set</span><span class="p">(</span><span class="n">cal_pthwy_genes</span><span class="p">[</span><span class="s2">&quot;gene_name&quot;</span><span class="p">])</span> <span class="o">&amp;</span> <span class="nb">set</span><span class="p">(</span><span class="n">df</span><span class="o">.</span><span class="n">columns</span><span class="o">.</span><span class="n">values</span><span class="p">[</span><span class="mi">1</span><span class="p">:]</span><span class="o">.</span><span class="n">tolist</span><span class="p">()))</span>
<span class="n">prior_nc</span> <span class="o">=</span> <span class="p">(</span><span class="mi">1</span><span class="o">-</span><span class="n">prior_c</span><span class="o">*</span><span class="n">n_overlap_gene</span><span class="p">)</span><span class="o">/</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="nb">set</span><span class="p">(</span><span class="n">df</span><span class="o">.</span><span class="n">columns</span><span class="o">.</span><span class="n">values</span><span class="p">[</span><span class="mi">1</span><span class="p">:]</span><span class="o">.</span><span class="n">tolist</span><span class="p">()))</span><span class="o">-</span><span class="n">n_overlap_gene</span><span class="p">)</span>
<span class="k">for</span> <span class="n">idx</span><span class="p">,</span> <span class="n">item</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">df</span><span class="o">.</span><span class="n">columns</span><span class="o">.</span><span class="n">values</span><span class="p">):</span>
    <span class="n">prior_t</span> <span class="o">=</span> <span class="nb">str</span><span class="p">(</span><span class="n">prior_c</span><span class="p">)</span> <span class="k">if</span> <span class="n">item</span> <span class="ow">in</span> <span class="n">cal_pthwy_genes</span><span class="p">[</span><span class="s2">&quot;gene_name&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">tolist</span><span class="p">()</span> <span class="k">else</span> <span class="nb">str</span><span class="p">(</span><span class="n">prior_nc</span><span class="p">)</span>
<span class="c1">#     print (prior_t)</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[11]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="c1"># chrom</span>
<span class="n">chrom</span> <span class="o">=</span> <span class="p">[</span><span class="s2">&quot;chr</span><span class="si">{}</span><span class="s2">&quot;</span><span class="o">.</span><span class="n">format</span><span class="p">(</span><span class="n">i</span><span class="p">)</span> <span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="nb">list</span><span class="p">(</span><span class="nb">range</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span><span class="mi">23</span><span class="p">))</span><span class="o">+</span><span class="p">[</span><span class="s2">&quot;X&quot;</span><span class="p">]]</span>
<span class="nb">print</span> <span class="p">((</span><span class="n">chrom</span><span class="p">))</span>
</pre></div>
</div>
</div>
</div>
<div class="output_wrapper">
<div class="output">
<div class="output_area">
<div class="prompt"></div>
<div class="output_subarea output_stream output_stdout output_text">
<pre>[&#39;chr1&#39;, &#39;chr2&#39;, &#39;chr3&#39;, &#39;chr4&#39;, &#39;chr5&#39;, &#39;chr6&#39;, &#39;chr7&#39;, &#39;chr8&#39;, &#39;chr9&#39;, &#39;chr10&#39;, &#39;chr11&#39;, &#39;chr12&#39;, &#39;chr13&#39;, &#39;chr14&#39;, &#39;chr15&#39;, &#39;chr16&#39;, &#39;chr17&#39;, &#39;chr18&#39;, &#39;chr19&#39;, &#39;chr20&#39;, &#39;chr21&#39;, &#39;chr22&#39;, &#39;chrX&#39;]
</pre>
</div>
</div>
</div>
</div>
</div>
<hr>
&copy 2016-2017 Min Qiao at <a href="http://xinhelab.org">Xin He lab</a>, the University of Chicago
<p><small>Exported from <a href="http://github.com/gaow/cnv-gene-mapping/blob/6a8f500b0d17d7ef7cf648a187826610bb4d8c92/analysis/20170726_DataTransform.ipynb"><code>analysis/20170726_DataTransform.ipynb</code></a> committed by Min Qiao on Fri Aug 4 20:36:41 2017 <a href="http://github.com/gaow/cnv-gene-mapping/commit/6a8f500b0d17d7ef7cf648a187826610bb4d8c92">revision 257, 6a8f500</a> <a href="https://stephenslab.github.io/ipynb-website/notes.html#Note-about-commit-ids"><span class="fa fa-question-circle"></span></a></small></p>
</div>
</div>
</body>
</html>
