<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="ipynb_website:version" content="0.9.7" />
<meta name="viewport" content="width=device-width, initial-scale=1" />

<link rel="stylesheet" type="text/css" href="../css/jt.css">

<link rel="stylesheet" type="text/css" href="../css/toc2.css">

<link href="../site_libs/jqueryui-1.11.4/jquery-ui.css">
<link rel="stylesheet" href="../site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<link rel="stylesheet" href="../site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />
<script type="text/javascript" src="https://ajax.googleapis.com/ajax/libs/jquery/1.9.1/jquery.min.js"></script>
<script type="text/javascript" src="https://ajax.googleapis.com/ajax/libs/jqueryui/1.9.1/jquery-ui.min.js"></script>
<script src="../site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="../site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="../site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>

<link rel="stylesheet"
      href="../site_libs/highlightjs/null.min.css"
      type="text/css" />

<script src="../site_libs/highlightjs/highlight.pack.js"></script>
<script>hljs.initHighlightingOnLoad();</script>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>

<script src="../js/doc_toc.js"></script>
<script src="../js/docs.js"></script>

<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.2/MathJax.js?config=TeX-MML-AM_CHTML"></script>
<script>
    MathJax.Hub.Config({
        extensions: ["tex2jax.js"],
        jax: ["input/TeX", "output/HTML-CSS"],
        tex2jax: {
        inlineMath: [ ['$','$'], ["\\(","\\)"] ],
        displayMath: [ ['$$','$$'], ["\\[","\\]"] ],
        processEscapes: true
        },
        "HTML-CSS": {
            preferredFont: "TeX",
            availableFonts: ["TeX"],
            styles: {
                scale: 110,
                ".MathJax_Display": {
                    "font-size": "110%",
                }
            }
        }
    });
</script>
<script>
function filterDataFrame(id) {
    var input = document.getElementById("search_" + id);
    var filter = input.value.toUpperCase();
    var table = document.getElementById("dataframe_" + id);
    var tr = table.getElementsByTagName("tr");
    // Loop through all table rows, and hide those who don't match the search query
    for (var i = 1; i < tr.length; i++) {
        for (var j = 0; j < tr[i].cells.length; ++j) {
            var matched = false;
            if (tr[i].cells[j].innerHTML.toUpperCase().indexOf(filter) != -1) {
                tr[i].style.display = "";
                matched = true
                break;
            }
            if (!matched)
                tr[i].style.display = "none";
        }
    }
}
function sortDataFrame(id, n, dtype) {
    var table = document.getElementById("dataframe_" + id);
    var tb = table.tBodies[0]; // use `<tbody>` to ignore `<thead>` and `<tfoot>` rows
    var tr = Array.prototype.slice.call(tb.rows, 0); // put rows into array
    if (dtype === 'numeric') {
        var fn = function(a, b) { 
            return parseFloat(a.cells[n].textContent) <= parseFloat(b.cells[n].textContent) ? -1 : 1;
        }
    } else {
        var fn = function(a, b) {
            var c = a.cells[n].textContent.trim().localeCompare(b.cells[n].textContent.trim()); 
            return c > 0 ? 1 : (c < 0 ? -1 : 0) }
    }
    var isSorted = function(array, fn) {
        if (array.length < 2)
            return 1;
        var direction = fn(array[0], array[1]); 
        for (var i = 1; i < array.length - 1; ++i) {
            var d = fn(array[i], array[i+1]);
            if (d == 0)
                continue;
            else if (direction == 0)
                direction = d;
            else if (direction != d)
                return 0;
            }
        return direction;
    }
    var sorted = isSorted(tr, fn);
    if (sorted == 1 || sorted == -1) {
        // if sorted already, reverse it
        for(var i = tr.length - 1; i >= 0; --i)
            tb.appendChild(tr[i]); // append each row in order
    } else {
        tr = tr.sort(fn);
        for(var i = 0; i < tr.length; ++i)
            tb.appendChild(tr[i]); // append each row in order
    }
}
</script>

<script>
$( document ).ready(function(){
            var cfg={'threshold':3,     // depth of toc (number of levels)
             'number_sections': false,
             'toc_cell': false,          // useless here
             'toc_window_display': true, // display the toc window
             "toc_section_display": "block", // display toc contents in the window
             'sideBar':true,       // sidebar or floating window
             'navigate_menu':false       // navigation menu (only in liveNotebook -- do not change)
            }
            var st={};                  // some variables used in the script
            st.rendering_toc_cell = false;
            st.config_loaded = false;
            st.extension_initialized=false;
            st.nbcontainer_marginleft = $('#notebook-container').css('margin-left')
            st.nbcontainer_marginright = $('#notebook-container').css('margin-right')
            st.nbcontainer_width = $('#notebook-container').css('width')
            st.oldTocHeight = undefined
            st.cell_toc = undefined;
            st.toc_index=0;
            // fire the main function with these parameters
            table_of_contents(cfg, st);
            var file=analysisDict[$("h1:first").attr("id")];
            $("#toc-level0 a").css("color","#126dce");
            $('a[href="#'+$("h1:first").attr("id")+'"]').hide()
            var docs=analysisArray;
            var docs_map=analysisArrayMap;
            var pos=analysisArray.indexOf(file);
            for (var a=pos;a>=0;a--){
                  $('<li><a href="'+docs[a]+'.html"><font color="#073642"><b>'+docs_map[docs[a]].replace(/_/g," ")+'</b></font></a></li>').insertBefore("#toc-level0 li:eq(0)");
            }
            $('a[href="'+file+'.html'+'"]').css("color","#126dce");
            for (var a=pos+1;a<docs.length;a++){
                  $(".toc #toc-level0").append('<li><a href="'+docs[a]+'.html"><font color="#073642"><b>'+docs_map[docs[a]].replace(/_/g," ")+'</b></font></a></li>');
            }
            // $("#toc-header").hide(); // comment out because it prevents search bar from displaying
    });
</script>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');
  // mark it active
  menuAnchor.parent().addClass('active');
  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>
<div class="container-fluid main-container">
<!-- tabsets -->
<script src="../site_libs/navigation-1.1/tabsets.js"></script>
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>



<title>Gene Mapping with CNV</title>

<style type = "text/css">
body {
  font-family: "Droid Sans";
  padding-top: 66px;
  padding-bottom: 40px;
}
</style>
</head>

<body>
<div tabindex="-1" id="notebook" class="border-box-sizing">
<div class="container" id="notebook-container">

<!-- code folding -->

<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="../index.html">Gene Mapping with CNV</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        
<li>
  <a href="../index.html">Overview</a>
</li>
        
<li>
  <a href="../analysis.html">Analysis</a>
</li>
        
<li>
  <a href="../prototype.html">Prototype</a>
</li>
        
<li>
  <a href="../workflow.html">Workflow</a>
</li>
        
<li>
  <a href="../setup/index.html">Setup</a>
</li>
        
<li>
  <a href="../writeup.html">Writeup</a>
</li>
        
      </ul>
        
<ul class="nav navbar-nav navbar-right">
<li>
   <a href="http://github.com/gaow/cnv-gene-mapping"> <span class="fa fa-github"></span> </a>
</li>
</ul>
        
      </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h1 id="Integrated-genome-wide-analysis-of-expression-quantitative-trait-loci-aids-interpretation-of-genomic-association-studies">Integrated genome-wide analysis of expression quantitative trait loci aids interpretation of genomic association studies<a class="anchor-link" href="#Integrated-genome-wide-analysis-of-expression-quantitative-trait-loci-aids-interpretation-of-genomic-association-studies">&#182;</a></h1><p>Roby Joehanes, Xiaoling Zhang, Tianxiao Huan, Chen Yao, Sai-xia Ying, Quang Tri Nguyen, Cumhur Yusuf Demirkale, Michael L. Feolo, Nataliya R. Sharopova, Anne Sturcke, Alejandro A. Schäffer, Nancy Heard-Costa, Han Chen, Po-ching Liu, Richard Wang, Kimberly A. Woodhouse, Kahraman Tanriverdi, Jane E. Freedman, Nalini Raghavachari, Josée Dupuis, Andrew D. Johnson, Christopher J. O’Donnell, Daniel Levy &amp; Peter J. Munson</p>
<p>Genome Biology volume 18, Article number: 16 (2017)</p>
<p>Online paper <a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-016-1142-6#article-info">link</a></p>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Background">Background<a class="anchor-link" href="#Background">&#182;</a></h2><ul>
<li>The vast majority of the SNPs from GWAS reside in non-coding regions - most disease-associated SNPs do not directly influence protein structure or function.</li>
<li>Identification of SNPs associated with gene expression levels, known as eQTLs, may improve understanding of the functional role of phenotype-associated SNPs in GWAS.</li>
<li>Previous eQTL studies: the small sample sizes limited their statistical power.</li>
</ul>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Methods">Methods<a class="anchor-link" href="#Methods">&#182;</a></h2><h3 id="Study-participants">Study participants<a class="anchor-link" href="#Study-participants">&#182;</a></h3><p>Whole blood eQTL from over 5000 samples in Framingham Heart Study (FHS) cohorts</p>
<ul>
<li>2770 individuals of eighth offspring cohort examination cycle (2005–2008)</li>
<li>3341 indicudials of second examination cycle (2006–2009)</li>
</ul>
<h3 id="Quality-control">Quality control<a class="anchor-link" href="#Quality-control">&#182;</a></h3><ul>
<li>It is detected that polymorphism-in-probe effects were generally minor</li>
</ul>
<h3 id="eQTL-Discovery">eQTL Discovery<a class="anchor-link" href="#eQTL-Discovery">&#182;</a></h3><ul>
<li>Use stepwise linear regression procedure to identify independent "lead-eQTLs" for each genetic region, and found over 19,000 independent, lead cis-eQTLs and almost 6,000 independent, lead trans-eQTLs</li>
</ul>
<h3 id="eQTL-Validation">eQTL Validation<a class="anchor-link" href="#eQTL-Validation">&#182;</a></h3><ul>
<li>Internal: 75% cis and 41% trans-eQTL are validated.</li>
<li>With previous studies: 50-70% previous cis-eQTL and 30-60% trans-eQTL are replicated</li>
<li>The replication in the other way is low due to lower power in previous work and different sequencing platforms, etc; but still 90% of cases the directions are consistent.</li>
</ul>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Statistical-analysis">Statistical analysis<a class="anchor-link" href="#Statistical-analysis">&#182;</a></h3><ul>
<li>Accounting for reported familial relationships, they removed the effects of sex, age, platelet count, white blood cell whole count, etc, from the expression data.</li>
<li>To infer hidden confounding factors, they use a Bayesian framework on the residualized gene expression data.</li>
</ul>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Enrichment-P-value-calculation">Enrichment P value calculation<a class="anchor-link" href="#Enrichment-P-value-calculation">&#182;</a></h3><ul>
<li>Enrichment: observed number divided by expected number, accounting for the LD structure of the available 8.5 million SNPs.</li>
<li>Expected numbers were obtained from the relevant 2 × 2 contingency tables.</li>
<li>The LD was computed between pairwise SNPs within the FHS dataset. </li>
</ul>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="cis-eQTL,-trans-eQTL,-primary-lead-eQTL-definition">cis-eQTL, trans-eQTL, primary lead eQTL definition<a class="anchor-link" href="#cis-eQTL,-trans-eQTL,-primary-lead-eQTL-definition">&#182;</a></h3><ul>
<li>cis: An SNP-transcript cluster pair is considered cis if the SNP resides within 1 Mb of the TSS on the same chromosome</li>
<li>trans: eQTLs that fall in blocks which did not contain the TSS for its target transcript cluster</li>
<li>primary lead eQTL:  the strongest eQTL, judged by P value for association, in its block</li>
</ul>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Results">Results<a class="anchor-link" href="#Results">&#182;</a></h2><h3 id="eQTL-discoveries">eQTL discoveries<a class="anchor-link" href="#eQTL-discoveries">&#182;</a></h3><ul>
<li>over 19,000 independent lead cis-eQTLs and over 6000 independent lead trans-eQTLs, with a false discovery rate (FDR) &lt; 5%.</li>
<li>48% are identified to be significant eQTLs</li>
</ul>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Enrichment-of-lead-eQTL-to-gene-structure">Enrichment of lead eQTL to gene structure<a class="anchor-link" href="#Enrichment-of-lead-eQTL-to-gene-structure">&#182;</a></h3><ul>
<li>Lead eQTLs are highly enriched in exonic locations (25-fold) within the transcribed region, more so than for intronic locations (12-fold)</li>
<li>Lead eQTLs are highly enriched in first exons and 5’ UTRs, especially, of transcribed regions (45 fold).</li>
<li>Other exons, the 3’-UTR, the first intron and subsequent introns showed less degrees of enrichment (21-fold, 20-fold, 11-fold, and 8-fold)</li>
<li>cis-eQTLs act preferentially through regulatory elements within the first exon, within the 5’-UTR or near the transcription start site (TSS).</li>
</ul>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Enrichment-of-lead-eQTLs-at-regulator-sites">Enrichment of lead eQTLs at regulator sites<a class="anchor-link" href="#Enrichment-of-lead-eQTLs-at-regulator-sites">&#182;</a></h3><ul>
<li>Regulator sites include DNAase hypersensitivity, transcription factor binding sites, and biochemically characterized regulatory promoter regions</li>
<li>It showed strong enrichment of regulatory evidence for primary and secondary lead cis-eQTLs (7-fold, P &lt; 1E-89).</li>
<li>The primary lead cis-eQTLs alone showed a stronger enrichment (8-fold, P &lt; 1E-69).</li>
</ul>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Clusters-of-trans-eQTL">Clusters of trans-eQTL<a class="anchor-link" href="#Clusters-of-trans-eQTL">&#182;</a></h3><ul>
<li>Some trans-eQTLs are associated with multiple distant transcripts and can be grouped into compact genomic blocks or clusters.</li>
<li>At gene level, we identified 59 distinct clusters of trans-eQTLs, each targeting a set of 6 to 141 distant transcripts.</li>
<li>The most prominent trans-eQTL clusters are on chromosomes 3 and 17, and are associated with expression of plateletspecific genes.</li>
<li>They found 13 platelet-related GWAS clusters, many of which also had target gene sets enriched with platelet specific genes.</li>
</ul>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="GWAS-analysis">GWAS analysis<a class="anchor-link" href="#GWAS-analysis">&#182;</a></h3><ul>
<li>Among the 58 GWAS SNPs for  coronary artery disease or myocardial infarction (CAD/MI), 21 loci or 36% are lead cis-eQTL.</li>
<li>The strongest eQTL (P &lt; 1E-455), rs1412445, is in the 3rd intron of LIPA transcript variant 1 and was a cis-eQTL for LIPA expression.</li>
<li>We also identified potentially novel, strong cis-eQTLs at UBE2Z locus and SH2B3 locus, where the CAD/MI GWAS risk SNP was in very strong LD with our lead eQTL.</li>
</ul>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Conclusion">Conclusion<a class="anchor-link" href="#Conclusion">&#182;</a></h2><h3 id="Advantages">Advantages<a class="anchor-link" href="#Advantages">&#182;</a></h3><ul>
<li>The study has large sample size, providing obvious benefits in terms of greater statistical power for discovery.</li>
<li>Expression measurement was carried out in a single laboratory with rigorous quality control.</li>
<li>The results provide an extensive resource of cis-eQTLs and trans-eQTLs at the gene and exon level, and this information may be useful for elucidating the biological underpinnings of many GWAS SNPs.</li>
</ul>
<h3 id="Limitations">Limitations<a class="anchor-link" href="#Limitations">&#182;</a></h3><ul>
<li>The homogeneity of the FHS population may limit the applicability of their results to populations of different ancestries.</li>
<li>Lack of population diversity might also increase the size of LD blocks and thereby limit the resolution.</li>
</ul>

</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span>
</pre></div>

    </div>
</div>
</div>

</div>
<hr>
&copy 2016-2017 Min Qiao at <a href="http://xinhelab.org">Xin He lab</a>, The University of Chicago
<p><small>Exported from <a href="http://github.com/gaow/cnv-gene-mapping/blob/345d6d20740960ff008d470162dca36910d6b8ab/analysis/20200702_eQTL_association.ipynb"><code>analysis/20200702_eQTL_association.ipynb</code></a> committed by minqiao on Wed Jul 8 15:22:52 2020 <a href="http://github.com/gaow/cnv-gene-mapping/commit/345d6d20740960ff008d470162dca36910d6b8ab">revision 1, 345d6d2</a> <a href="https://stephenslab.github.io/ipynb-website/notes.html#Note-about-commit-ids"><span class="fa fa-question-circle"></span></a></small></p>
</div>
</div>
</body>
</html>
