<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="ipynb_website:version" content="0.9.7" />
<meta name="viewport" content="width=device-width, initial-scale=1" />

<link rel="stylesheet" type="text/css" href="../css/jt.css">

<link rel="stylesheet" type="text/css" href="../css/toc2.css">

<link href="../site_libs/jqueryui-1.11.4/jquery-ui.css">
<link rel="stylesheet" href="../site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<link rel="stylesheet" href="../site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />
<script type="text/javascript" src="https://ajax.googleapis.com/ajax/libs/jquery/1.9.1/jquery.min.js"></script>
<script type="text/javascript" src="https://ajax.googleapis.com/ajax/libs/jqueryui/1.9.1/jquery-ui.min.js"></script>
<script src="../site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="../site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="../site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>

<link rel="stylesheet"
      href="../site_libs/highlightjs/null.min.css"
      type="text/css" />

<script src="../site_libs/highlightjs/highlight.pack.js"></script>
<script>hljs.initHighlightingOnLoad();</script>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>

<script src="../js/doc_toc.js"></script>
<script src="../js/docs.js"></script>

<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.2/MathJax.js?config=TeX-MML-AM_CHTML"></script>
<script>
    MathJax.Hub.Config({
        extensions: ["tex2jax.js"],
        jax: ["input/TeX", "output/HTML-CSS"],
        tex2jax: {
        inlineMath: [ ['$','$'], ["\\(","\\)"] ],
        displayMath: [ ['$$','$$'], ["\\[","\\]"] ],
        processEscapes: true
        },
        "HTML-CSS": {
            preferredFont: "TeX",
            availableFonts: ["TeX"],
            styles: {
                scale: 110,
                ".MathJax_Display": {
                    "font-size": "110%",
                }
            }
        }
    });
</script>
<script>
function filterDataFrame(id) {
    var input = document.getElementById("search_" + id);
    var filter = input.value.toUpperCase();
    var table = document.getElementById("dataframe_" + id);
    var tr = table.getElementsByTagName("tr");
    // Loop through all table rows, and hide those who don't match the search query
    for (var i = 1; i < tr.length; i++) {
        for (var j = 0; j < tr[i].cells.length; ++j) {
            var matched = false;
            if (tr[i].cells[j].innerHTML.toUpperCase().indexOf(filter) != -1) {
                tr[i].style.display = "";
                matched = true
                break;
            }
            if (!matched)
                tr[i].style.display = "none";
        }
    }
}
function sortDataFrame(id, n, dtype) {
    var table = document.getElementById("dataframe_" + id);
    var tb = table.tBodies[0]; // use `<tbody>` to ignore `<thead>` and `<tfoot>` rows
    var tr = Array.prototype.slice.call(tb.rows, 0); // put rows into array
    if (dtype === 'numeric') {
        var fn = function(a, b) { 
            return parseFloat(a.cells[n].textContent) <= parseFloat(b.cells[n].textContent) ? -1 : 1;
        }
    } else {
        var fn = function(a, b) {
            var c = a.cells[n].textContent.trim().localeCompare(b.cells[n].textContent.trim()); 
            return c > 0 ? 1 : (c < 0 ? -1 : 0) }
    }
    var isSorted = function(array, fn) {
        if (array.length < 2)
            return 1;
        var direction = fn(array[0], array[1]); 
        for (var i = 1; i < array.length - 1; ++i) {
            var d = fn(array[i], array[i+1]);
            if (d == 0)
                continue;
            else if (direction == 0)
                direction = d;
            else if (direction != d)
                return 0;
            }
        return direction;
    }
    var sorted = isSorted(tr, fn);
    if (sorted == 1 || sorted == -1) {
        // if sorted already, reverse it
        for(var i = tr.length - 1; i >= 0; --i)
            tb.appendChild(tr[i]); // append each row in order
    } else {
        tr = tr.sort(fn);
        for(var i = 0; i < tr.length; ++i)
            tb.appendChild(tr[i]); // append each row in order
    }
}
</script>

<script>
$( document ).ready(function(){
            var cfg={'threshold':3,     // depth of toc (number of levels)
             'number_sections': false,
             'toc_cell': false,          // useless here
             'toc_window_display': true, // display the toc window
             "toc_section_display": "block", // display toc contents in the window
             'sideBar':true,       // sidebar or floating window
             'navigate_menu':false       // navigation menu (only in liveNotebook -- do not change)
            }
            var st={};                  // some variables used in the script
            st.rendering_toc_cell = false;
            st.config_loaded = false;
            st.extension_initialized=false;
            st.nbcontainer_marginleft = $('#notebook-container').css('margin-left')
            st.nbcontainer_marginright = $('#notebook-container').css('margin-right')
            st.nbcontainer_width = $('#notebook-container').css('width')
            st.oldTocHeight = undefined
            st.cell_toc = undefined;
            st.toc_index=0;
            // fire the main function with these parameters
            table_of_contents(cfg, st);
            var file=dscDict[$("h1:first").attr("id")];
            $("#toc-level0 a").css("color","#126dce");
            $('a[href="#'+$("h1:first").attr("id")+'"]').hide()
            var docs=dscArray;
            var docs_map=dscArrayMap;
            var pos=dscArray.indexOf(file);
            for (var a=pos;a>=0;a--){
                  $('<li><a href="'+docs[a]+'.html"><font color="#073642"><b>'+docs_map[docs[a]].replace(/_/g," ")+'</b></font></a></li>').insertBefore("#toc-level0 li:eq(0)");
            }
            $('a[href="'+file+'.html'+'"]').css("color","#126dce");
            for (var a=pos+1;a<docs.length;a++){
                  $(".toc #toc-level0").append('<li><a href="'+docs[a]+'.html"><font color="#073642"><b>'+docs_map[docs[a]].replace(/_/g," ")+'</b></font></a></li>');
            }
            // $("#toc-header").hide(); // comment out because it prevents search bar from displaying
    });
</script>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');
  // mark it active
  menuAnchor.parent().addClass('active');
  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>
<div class="container-fluid main-container">
<!-- tabsets -->
<script src="../site_libs/navigation-1.1/tabsets.js"></script>
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>



<title>Gene Mapping with CNV</title>

<style type = "text/css">
body {
  font-family: "Droid Sans";
  padding-top: 66px;
  padding-bottom: 40px;
}
</style>
</head>

<body>
<div tabindex="-1" id="notebook" class="border-box-sizing">
<div class="container" id="notebook-container">

<!-- code folding -->

<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="../index.html">Gene Mapping with CNV</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        
<li>
  <a href="../index.html">Overview</a>
</li>
        
<li>
  <a href="../analysis.html">Analysis</a>
</li>
        
<li>
  <a href="../prototype.html">Prototype</a>
</li>
        
<li>
  <a href="../dsc.html">Dsc</a>
</li>
        
<li>
  <a href="../setup/index.html">Setup</a>
</li>
        
<li>
  <a href="../writeup.html">Writeup</a>
</li>
        
      </ul>
        
<ul class="nav navbar-nav navbar-right">
<li>
   <a href="http://github.com/gaow/cnv-gene-mapping"> <span class="fa fa-github"></span> </a>
</li>
</ul>
        
      </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Integrated-fine-mapping-of-non-coding-disease-variants-with-functional-genomics-data">Integrated fine-mapping of non-coding disease variants with functional genomics data<a class="anchor-link" href="#Integrated-fine-mapping-of-non-coding-disease-variants-with-functional-genomics-data">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Background-and-Public-Health-Significance">Background and Public Health Significance<a class="anchor-link" href="#Background-and-Public-Health-Significance">&#182;</a></h3><h4 id="Schizophrenia">Schizophrenia<a class="anchor-link" href="#Schizophrenia">&#182;</a></h4><p>Schizophrenia is a type of mental disorder, typically starting to develop in late adolescence or early adulthood. Schizophrenia can be severe due to its cognitive impairment, difficulty in social activities, and even potential to cause disability (9). Children may also develop schizophrenia (10). The prevalence of schizophrenia ranges between 0.25% and 0.64% in the United States (11).</p>
<p>A few hypothese for the mechanism of schizophrenia have been proposed, but it is not fully known yet. The risk factors for schizophrenia mainly include genetics and environment. The estimated heritability of schizophrenia is between 70% and 80%. Both rare variants with large effect and common variants with small effect can contribute to this disorder. Widely known environmental factors are living environment and drug use, etc.</p>
<p>Antipsychotic and  is the most popular medication for schizophrenia treatment, supported by social activity restoration, professional skill and job training and psychotherapy. However, involuntary hospitalization is necessary for severe schizophrenia cases or severe episodes, since merely medication is not sufficient to control the sumptoms.</p>
<p>heritability: <a href="https://www.sciencedirect.com/science/article/pii/S0065266016300529?via%3Dihub">https://www.sciencedirect.com/science/article/pii/S0065266016300529?via%3Dihub</a></p>
<h4 id="The-limitation-of-Genome-wide-Association-Studies-(GWAS)">The limitation of Genome-wide Association Studies (GWAS)<a class="anchor-link" href="#The-limitation-of-Genome-wide-Association-Studies-(GWAS)">&#182;</a></h4><p>Abundant common disease risk variants have been found in various GWAS over the past ten years. Most of these variants locate in non-coding regions of human genome, which are likely to be regulatory elements/sequences, such as enhancers, promoters and 5' untranslated region (5' UTR) (1-3). Functional interpretation for most of these non-coding variants remains unanswered, and it is still challenging to even identify the causal ones. GWAS for a particular trait often detects from millions to tens of millions variants, but most of them are common variants with small or modest effects, which limited their value for disease risk prediction.</p>
<h4 id="Open-chromatin-regions-and-allele-specific-open-chromatin-variants">Open chromatin regions and allele-specific open chromatin variants<a class="anchor-link" href="#Open-chromatin-regions-and-allele-specific-open-chromatin-variants">&#182;</a></h4><p>Chromatin's default tight coiling structure limits its accessibility. As a result, gene expression only happens when chromatin is in "opening" state. A new genomic technique, Assay for Transposase-Accessible Chromatin using sequencing (ATAC-seq), is widely used in recent years to access sequence on open chromatin, thus can map regions of transcription factor binding sites and nucleosome positions (4, 5). ATAC-seq takes advantage of next-generation sequencing (NGS) technology - the number of reads of a sequence is mainly determined by the extent of opening status of the chromatin region at a single nucleotide resolution.</p>
<p>The fact that open chromatin regions (OCRs) often overlap with regulatory sequence can help detect functional non-coding risk variants for some neuropsychiatric disorders, such as schizophrenia, in induced pluripotent stem cells (iPSC)-derived neurons from adult (6, 7). Therefore, it is a favorable approach to make use of ATAC-seq for chromatin accessibility profile in neuron cells to restrict variants detected by GWAS for neurodevelopmental disorders.</p>
<p>Another promising approach to overcome the challenge is to focus on allele-specific open chromatin (ASoC) variants in OCRs. ASoC variants are characterized by allelic imbalance in sequencing reads at heterozygous single nucleotide polymorphism (SNP) sites. They can be mapped by comparing the chromatin accessibilities of both alleles (SNPs) in the same individual. A binomial test is often used to test allelic imbalance of the sequencing reads. The main advantage of ASoC mapping compared to another commonly used approach, expression quantitative trait loci (eQTL), lies in its direct identification of putative functional disease variants (8). Since only a small fractional alleles in OCRs are actually functional, it will be more accurate to detect the functional SNPs and further limit GWAS variants.</p>
<h4 id="Public-Health-Significance">Public Health Significance<a class="anchor-link" href="#Public-Health-Significance">&#182;</a></h4><p>The significance of this study mainly lies in the potential detection of schizophrenia associated variants' regulatory activities by computational fine-mapping, endeavoring to provide a powerful statistical framework for the priorization of disease functional variants, as well as a brief landscape of neuron ASoC for schizophrenia. In addition, the regulatory variants from the fine-mapping conclusions are potential to provide more evidence for Clustered Regularly Interspaced Short Palindromic Repeats (CRISPR) editing, a fast-developing powerful gene editing technique, and may further assist to understand the mechanism of certain types of neurodevelopmental disorders.</p>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Aim">Aim<a class="anchor-link" href="#Aim">&#182;</a></h3><p>The goal of this proposal is to explore the enrichment of schizophrenia GWAS signals, as well as other neurodevelopmental disorders and common phenotypes for comparison, in OCRs and ASoC SNPs obtained in experiments, and to precisely identify causal variants among ASoC variants associated with schizophrenia, then investigate the mechanism of non-coding regulatory elements.</p>
<h4 id="Aim-1:-To-construct-statistical-model-for-GWAS-enrichment-analysis-for-genomic-features">Aim 1: To construct statistical model for GWAS enrichment analysis for genomic features<a class="anchor-link" href="#Aim-1:-To-construct-statistical-model-for-GWAS-enrichment-analysis-for-genomic-features">&#182;</a></h4><p>The statistical model to combine functional annotation with genetic fine-mapping is</p>
$$\beta_{j} = (1 - \pi_{j})\delta_0 + \pi_{j}g(\cdot) (1)$$$$\log\big[\frac{\pi_{j}}{1 - \pi_{j}} \big] = \alpha_0 + \sum_{k=1}^m \alpha_k d_{jk} (2)$$<p>where $\pi_{j}$ is prior inclusion probability of $\textit{j}$th SNP in a certain locus, thus $1-\pi_j$ denotes the probability of the coefficient to be zero; $\alpha_k$ is enrichment of annotation $\textit{k}$ in causal variants; $d_{jk}$ is the annotation $\textit{k}$ of $\textit{j}$th SNP, which is often binary.</p>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p>We assume GWAS effect size $\beta_j$ follows a spike-and-slab distribution. $\delta_0$ is 0, indicating a particular coefficient in the model to be 0, with probability $\pi_j$ for $\textit{j}$th SNP. $g(\cdot)$ is the prior distribution for the regression coefficient values, with probability $1-\pi_j$ for $\textit{j}$th SNP.</p>
<h4 id="Aim-2:-To-utilize-newly-developed-statistical-method-to-explore-the-enrichment-of-Schizophrenia-GWAS-signals-in-different-regulatory-regions">Aim 2: To utilize newly developed statistical method to explore the enrichment of Schizophrenia GWAS signals in different regulatory regions<a class="anchor-link" href="#Aim-2:-To-utilize-newly-developed-statistical-method-to-explore-the-enrichment-of-Schizophrenia-GWAS-signals-in-different-regulatory-regions">&#182;</a></h4><p>Aiming to explore whether Schizophrenia associated GWAS signals are enriched in certain types of functional genomic annotations, we propose to perform enrichment analysis of 15 types of genomic regions, 5 of which are from ATAC-seq OCR peaks of 5 types of neuron cell, 2 of which are from ASoC variants, while the rest are 8 types of commonly known generic genomic regions, including coding, promoter, conserved sequence, intron, 5'UTR, 3'UTR, DNase I hypersensitive (DHS) and repressed regions. A Bayesian hierarchical based tool, TORUS (12), will be applied for this purpose. We hypothesize that ASoC variants are more enriched for functional disease variants. We will evaluate enrichment by single-annotation (univariate) analysis for each single genomic region, and compare the effects and significance between OCR/ASoC and generic annotations. Furthermore, we will access joint enrichment effect using multiple-annotation (joint) analysis, where $m &gt; 1$ in (2), and different multiple-annotation combinations will be evaluated to obtain their respective prior inclusion probabilities ($\pi_{j}'s$) for fine-mapping process in Aim 3. In both univariate and joint analysis, we will obtain odds ratio (OR), $\alpha_k$ in (2), and credible intervals (CI) for each type of regulatory genomic region.</p>
<p>To illustrate the expected positive effects of OCRs/ASoC in neuron cells for Schizophrenia, which belongs to neurodevelopmental disorder, we will perform enrichment analysis for GWAS SNPs from non-neurodevelopmental traits, such as stroke, body fat percentage, body mass index, total cholesterol etc. We expect most enrichment effects for non-neurodevelopmental traits to be smaller and less significant than those in Schizophrenia.</p>
<h4 id="Aim-3:-To-perform-fine-mapping-of-putative-causal-SNPs-in-108-identified-loci">Aim 3: To perform fine-mapping of putative causal SNPs in 108 identified loci<a class="anchor-link" href="#Aim-3:-To-perform-fine-mapping-of-putative-causal-SNPs-in-108-identified-loci">&#182;</a></h4><p>Ripke et al identified 108 independent and significant genomic loci associated with Schizophrenia. We will employ SuSiE (14), a newly developed Bayesian variable selection and genetic fine-mapping software package, to determine each SNP of being causal based on posterior inclusion probability (PIP) over the 108 Schizophrenia significant loci, incorporated with external linkage disequilibrium (LD) from the 1000 Genomes project and prior inclusion probabilities as functions of genomic features for each SNP. Besides PIP, we will also obtain 95% credible sets of variants, indicating a minimum set of variants with 95% probability to capture one non-zero effect variant.</p>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Data">Data<a class="anchor-link" href="#Data">&#182;</a></h3><ol>
<li><p>A set of 20 iPSC lines used open chromatin mapping are reprogrammed by collaborators. The cell lines then differentiated into neural progenitor cells (NPC), subsequently to glutamatergic (iN-Glut), GABAergic (iN-GA), and dopaminergic (iN-DN) neurons. ATAC-seq is utilized to call OCR peaks (FDR&lt;0.05) for each cell type. The median length for all 5 types of neuron cells is 335 base pair (bp), and all OCR peaks cover about 4% of the whole human genome.</p>
</li>
<li><p>ASoC SNPs are obtained by testing heterozygous SNPs that showed allelic imbalance in ATAC-seq reads in iN-Glut and NPC neurons.</p>
</li>
<li><p>Schizophrenia GWAS risk variants from Psychiatric Genomics Consortium (PGC).</p>
</li>
</ol>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Reference">Reference<a class="anchor-link" href="#Reference">&#182;</a></h3><ol>
<li>M. J. Gandal et al., Transcriptome-wide isoform-level dysregulation in ASD, schizophrenia, and bipolar disorder. Science (2018).</li>
<li>P. Rajarajan et al., Neuron-specific signatures in the chromosomal connectome associated with schizophrenia risk. Science (2018).</li>
<li>M. Li et al., Integrative functional genomic analysis of human brain development and neuropsychiatric risks. Science (2018).</li>
<li>J. D. Buenrostro et al., Transposition of native chromatin for fast and sensitive epigenomic profiling of open chromatin, DNA-binding proteins and nucleosome position. Nature Methods (2013).</li>
<li>J. D. Buenrostro et al., ATAC-seq: A Method for Assaying Chromatin Accessibility Genome-Wide. Current Protocols in Molecular Biology (2015).</li>
<li>J. F. Degner et al., DNase I sensitivity QTLs are a major determinant of human expression variation. Nature 482, 390-394 (2012).</li>
<li>R. E. Thurman et al., The accessible chromatin landscape of the human genome. Nature 489, 75-82 (2012).</li>
<li>S. Zhang et al., Landscape of allele-specific open chromatin in human iPSC-differentiated neurons and its implication for mental disorders. European Neuropsychopharmacology 29 (2019).</li>
<li>J. van Os et al., Schizophrenia. Lancet 374 (2009).</li>
<li>American Psychiatric Association. Schizophrenia. 295.90 (F20.9). American Psychiatric Publishing (2013).</li>
<li>E. Q. Wu et al., Annual prevalence of diagnosed schizophrenia in the USA: a claims data analysis approach. Psychol Med (2006).</li>
<li>X. Wen et al., Molecular QTL discovery incorporating genomic annotations using Bayesian false discovery rate control. Ann Appl Stat 10,  (2016).</li>
<li>C. Schizophrenia Working Group of the Psychiatric Genomics, Biological insights from 108 schizophrenia-associated genetic loci. Nature 511, (2014).</li>
<li>G. Wang et al., A simple new approach to variable selection in regression, with application to genetic fine-mapping. biorxiv 2018</li>
</ol>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Alzheimer's-Disease">Alzheimer's Disease<a class="anchor-link" href="#Alzheimer's-Disease">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Background">Background<a class="anchor-link" href="#Background">&#182;</a></h3><p>Alzheimer's disease (AD) is a type of neurodegenerative disease featured by its chronicity and irreversibility. Patients who develop AD have difficulty in remembering recent events at the beginning. Later symptoms include problems with language, loss of cognitive functioning, extreme and rapid changes in mood, disability in self-care, motivation loss and other behavioral issues. As the disease progresses, patients' body functions are gradually lost, until death. It is also the main cause of dementia.</p>
<p>The usual onset for AD is over 65 years old, which is late-onset. On the other hand, there are still a small subgroup, less than 10%, of early-onset cases occur from a person's 30s to mid-60s. The life expectancy is about 3 to 9 years. AD is currently ranked as the 6th leading cause of death in the United States (US). Worldwide, the number of deaths caused by AD for year of 2015 is about 1.9 million, and the prevalence of the disease in 2015 is roughly 29.8 million. There is no effective cure for AD, and caregiving and family supports is important and essential for patients, placing heavy burden to caregivers and families and even the whole society.</p>
<p>Alzheimer's disease is a severe threat to people's physical and mental health, a heavy burden to family members and caregivers, as well as a huge consumption of social and community resources. The cost for caring US AD and other dementia patients in 2017 is estimated as high as 277 billion dollars. By the year of 2050, the total number of AD patients all over the world is expected to increase to 13.8 million with the acceleration of population aging process, and the caring cost will rapidly increase to over 1 trillion dollars.</p>
<p>Although the brain imaging shows that, for AD patients, plaques and tangles gradually spread throughout their cerebral cortex and the brain tissue begins to shrink, the biological mechanism of developing AD is not yet fully understood. Some researches indicate that deteriorate vascular and metabolic conditions, such as heart disease, high blood pressure, diabetes and obesity, are associated with decline in cognition, and more social engagement and physical activities are potential to help old people stay healthy. Meanwhile, geneticists find that the disease is highly heritable. For familial early-onset patients, rare variants in a few genes with strong effect are often detected playing the main role. The most recent large-scale AD genome-wide association study (GWAS) have identified 94 lead single nucleotide polymorphism (SNP) sites located in 29 risk genomic loci.</p>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p>277 billion: <a href="https://www.webmd.com/alzheimers/news/20180320/the-high-costs-of-alzheimers#1">https://www.webmd.com/alzheimers/news/20180320/the-high-costs-of-alzheimers#1</a></p>
<p>13.8 million: <a href="https://www.alzinfo.org/articles/alzheimers-cases-triple-2050/">https://www.alzinfo.org/articles/alzheimers-cases-triple-2050/</a></p>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Aims-and-strategies-(methods)">Aims and strategies (methods)<a class="anchor-link" href="#Aims-and-strategies-(methods)">&#182;</a></h3><p>The overall goal of this proposal is to interpret the function of the variants at AD GWAS identified loci, most of which are enriched in non-coding regions and activated transcription regions, and to identify potential causal variants in those associated with AD. Two steps will be conducted to achieve the goal. First, utilize new statistical method in chapter 1 to explore the enrichment of Alzheimer's GWAS signals in OCRs and ASoC SNPs obtained using in ATAC-seq method, as well as other well-known functional regions. Second, apply Bayesian fine-mapping approach in chapter 1 for putative causal SNPs in 29 loci identified in AD GWAS, using prior inclusion probabilities obtained in aim 1.</p>
<h4 id="Aim-1:-To-explore-whether-AD-GWAS-signals-are-enriched-in-different-regulatory-regions">Aim 1: To explore whether AD GWAS signals are enriched in different regulatory regions<a class="anchor-link" href="#Aim-1:-To-explore-whether-AD-GWAS-signals-are-enriched-in-different-regulatory-regions">&#182;</a></h4><p>We will utilize statistical method in chapter 1 by both univariate analysis for each single genomic region and joint analysis. For joint analysis, we will test 3 multiple annotation combinations: 1) 5 OCRs in neuron cells, 2) 5 OCRs and 2 ASoC in neuron cells, 3) coding, promoter and conserved sequence regions. Then, we will obtain prior inclusion probabilities for fine-mapping in aim 2. Next, we will compare the enrichment odds ratio with those in schizophrenia.</p>
<h4 id="Aim-2:-To-identify-putative-causal-variants-in-29-loci-identified-in-AD-GWAS">Aim 2: To identify putative causal variants in 29 loci identified in AD GWAS<a class="anchor-link" href="#Aim-2:-To-identify-putative-causal-variants-in-29-loci-identified-in-AD-GWAS">&#182;</a></h4><p>We propose to apply Bayesian fine-mapping approach in chapter 1 to identify genetic variant(s) responsible for AD, assuming there is at least one causal variant exists. Meanwhile, I will apply second method, Deterministic Approximation of Posteriors (DAP), for this purpose. Then, we will check the consistency, i.e. whether the same variant(s) are identified in a certain locus, and compare PIPs for the two methods.</p>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="reference">reference<a class="anchor-link" href="#reference">&#182;</a></h3>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="A-statistical-method-to-identify-causal-genes-in-copy-number-variation-data">A statistical method to identify causal genes in copy number variation data<a class="anchor-link" href="#A-statistical-method-to-identify-causal-genes-in-copy-number-variation-data">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Background-and-Significance">Background and Significance<a class="anchor-link" href="#Background-and-Significance">&#182;</a></h3><h4 id="Background">Background<a class="anchor-link" href="#Background">&#182;</a></h4><p>For over a decade, genome-wide association studies (GWAS) have been the main strategy to uncover genetic architecture of complex diseases and traits and detected tens of thousands of loci associated with some psychiatric disorders (1). GWAS typically focus on studying single nucleotide polymorphisms (SNPs). Despite their success in a wide range of disease applications, GWAS have major limitations: 1) Risk alleles are usually at low frequencies thus difficult to detect, especially for in early-onset cognitive development disorders such as Schizophrenia and autism (2); 2) Most loci have small effect sizes, not likely to be deleterious mutations (3); 3) Furthermore, causal variants or genes in GWAS loci are often unclear. Therefore, SNPs only explain limited heritability.</p>
<p>Copy number variations (CNVs), on the other hand, are believed to play a critical role in complex disease etiology (4). CNVs are large genomic insertion or deletion events, which are a type of structural variation of an organism's chromosome. The length of CNVs vary enormously, often spread from 50 base pairs to kilo- or even mega-bases. Computational methods to call CNVs have been developed from sequencing data, making great opportunities for CNV analysis after DNA sequencing (5).</p>
<p>Existing methods for analyzing CNV data are usually focused on either the detection of association of CNVs and disease risk, or the identification of gene sets with CNV burden (6, 7), for example, CNV deletions that affecting a gene set are more common in cases than controls.</p>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Aim-and-Public-Health-Significance">Aim and Public Health Significance<a class="anchor-link" href="#Aim-and-Public-Health-Significance">&#182;</a></h3><h4 id="Aim">Aim<a class="anchor-link" href="#Aim">&#182;</a></h4><p>Previous studies have suggested that CNVs are an important source of genetic variation affecting neurodevelopmental disorders. It is estimated that 10% of individuals with developmental delay carry at least one CNV larger than 500 kilo-bases (6). However, since the challenge is that CNVs often span multiple genes, distinguish the susceptible or causal gene(s) from other genes in the same CNV event is a difficult problem. Results of CNV associations are thus difficult to interpret. This hurdles the use of CNV data to unravel the genetics of complex diseases. The goal of this proposal is to develop a statistical framework that leverage genome-wide CNVs for mapping susceptibility genes of psychiatric diseases, and link these findings with those from single-nucleotide variant (SNV) -based studies. The advantage of our method is its highly integrative characteristic, which can use enrichment results to re-rank genes and avoid a fixed cutoff.</p>
<h4 id="Public-Health-Significance">Public Health Significance<a class="anchor-link" href="#Public-Health-Significance">&#182;</a></h4><p>CNV is less studied comparing to SNV, and CNV and SNV studies are usually conducted separately. CNVs often collect numerous SNVs and account for much larger genetic variation because of their lengths, thus they play a significant role in neurodevelopmental disorders. This proposed study provides a rigorous statistical framework to extract gene-level knowledge from CNVs for the first time, and to link the discoveries with those from SNVs for more accurate identification of disease genes.</p>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Methods">Methods<a class="anchor-link" href="#Methods">&#182;</a></h3><p>To address the challenge we will develop a new approach that exploits large-scale genome-wide CNV data in case-control studies to map genes. It is inspired by statistical fine-mapping of causal variants in linkage-disequilibrium blocks from GWAS. Unlike existing approaches that directly test for CNV associations, our method seeks to identify in CNV events true susceptible genes in a rigorous statistical framework. Genome-wide CNV data are first clustered into disjoint analysis blocks, i.e. no CNV spans between any two blocks. For genes within a block we test for disease associations while accounting for correlations among genes induced by CNV in the same block. We accomplish this by extension of a recently developed Bayesian variable selection method, SuSiE (7). Our method thus selects a small number of putative risk genes among multiple correlated ones that best explain the CNV-phenotype data. Furthermore, we leverage knowledge of known biological pathways to set prior probabilities of genes in CNV events to increase power. Our model estimates posterior probabilities of all putative risk genes as well as 95% credible sets (i.e. the set of genes that cover all risk genes with high probability). Using this new approach we perform gene-level analysis in several case-control CNV datasets in Schizophrenia. Since our method reports the statistical confidence of genes, it can be integrated with other gene-level datasets, e.g. results from exome-sequencing studies. This provides a powerful strategy to integrate data from independent sources.</p>
<p>To infer CNV configuration $B(Z)$ from case-control data, we leverage the statistical machinery of Bayesian regression. Specifically, let $\beta_j$ be the effect size of the $j$-th gene.</p>
$$\beta_j|Z_j = 0\sim0, \beta_j|Z_j = 1\sim N(\mu,\sigma^2), \text{logit P}(y_i = 1) = \beta_0+\rho w_i + \sum_j \beta_j x_{ij}$$<p>where $w_i$ denotes possible covariates (such as the total number of CNVs in subject $i$) and $\rho$ their effects. Inference on this model leads to the support of each CNV configuration $B(Z)$.</p>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Integrated-association-analysis-of-CNV-and-SNP-data-for-causal-genes-mapping-in-psychiatric-disorders">Integrated association analysis of CNV and SNP data for causal genes mapping in psychiatric disorders<a class="anchor-link" href="#Integrated-association-analysis-of-CNV-and-SNP-data-for-causal-genes-mapping-in-psychiatric-disorders">&#182;</a></h3><p>We will apply the method to individual-level schizophrenia CNV data and non-individual level case-control CNV data.</p>
<ol>
<li>Swedish Schizophrenia Population-Based Case-control Exome Sequencing CNV data from dbGAP.</li>
<li>Schizophrenia case-control CNV data International Schizophrenia Consortium (ISC) study.</li>
</ol>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Reference">Reference<a class="anchor-link" href="#Reference">&#182;</a></h3><ol>
<li>J. Gratten et al., Large-scale genomics unveils the genetic architecture of psychiatric disorders. Nature Neuroscience 17 (2014).</li>
<li>B. Devlin et al., Genetic architecture in autism spectrum disorder. Current Opinion in Genetics &amp; Development 22 (2012).</li>
<li>J. Hardy et al., Genomewide association studies and human disease. The New England Journal of Medicine 360 (2009).</li>
<li>C. Lowther et al., Genomic Disorders in Psychiatry-What Does the Clinician Need to Know? Current Psychiatry Reports 19 (2017).</li>
<li>C. Alkan et al., Genome structural variation discovery and genotyping. Nature Reviews Genetics 12 (2011).</li>
<li>S. Girirajan et al., Human copy number variation and complex genetic disease. Annual Review of Genetics 45 (2011).</li>
<li>G. Wang et al., A simple new approach to variable selection in regression, with application to genetic fine-mapping. biorxiv 2018</li>
</ol>

</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span>
</pre></div>

    </div>
</div>
</div>

</div>
<hr>
&copy 2016-2017 Min Qiao at <a href="http://xinhelab.org">Xin He lab</a>, The University of Chicago
<p><small>Exported from <a href="http://github.com/gaow/cnv-gene-mapping/blob/06b7431656558076017f93d48f8d163668b0319a/dsc/20190908_proposal_manuscript.ipynb"><code>dsc/20190908_proposal_manuscript.ipynb</code></a> committed by minqiao on Fri Jan 10 15:40:33 2020 <a href="http://github.com/gaow/cnv-gene-mapping/commit/06b7431656558076017f93d48f8d163668b0319a">revision 6, 06b7431</a> <a href="https://stephenslab.github.io/ipynb-website/notes.html#Note-about-commit-ids"><span class="fa fa-question-circle"></span></a></small></p>
<p><small>Exported from <a href="http://github.com/gaow/cnv-gene-mapping/blob/06b7431656558076017f93d48f8d163668b0319a/dsc/20190908_proposal_manuscript.ipynb"><code>dsc/20190908_proposal_manuscript.ipynb</code></a> committed by minqiao on Fri Jan 10 15:40:33 2020 <a href="http://github.com/gaow/cnv-gene-mapping/commit/06b7431656558076017f93d48f8d163668b0319a">revision 6, 06b7431</a> <a href="https://stephenslab.github.io/ipynb-website/notes.html#Note-about-commit-ids"><span class="fa fa-question-circle"></span></a></small></p>
<p><small>Exported from <a href="http://github.com/gaow/cnv-gene-mapping/blob/06b7431656558076017f93d48f8d163668b0319a/dsc/20190908_proposal_manuscript.ipynb"><code>dsc/20190908_proposal_manuscript.ipynb</code></a> committed by minqiao on Fri Jan 10 14:40:33 2020 <a href="http://github.com/gaow/cnv-gene-mapping/commit/06b7431656558076017f93d48f8d163668b0319a">revision 6, 06b7431</a> <a href="https://stephenslab.github.io/ipynb-website/notes.html#Note-about-commit-ids"><span class="fa fa-question-circle"></span></a></small></p>
</div>
</div>
</body>
</html>
