<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="ipynb_website:version" content="0.9.7" />
<meta name="viewport" content="width=device-width, initial-scale=1" />

<link rel="stylesheet" type="text/css" href="../css/jt.css">

<link rel="stylesheet" type="text/css" href="../css/toc2.css">

<link href="../site_libs/jqueryui-1.11.4/jquery-ui.css">
<link rel="stylesheet" href="../site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<link rel="stylesheet" href="../site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />
<script type="text/javascript" src="https://ajax.googleapis.com/ajax/libs/jquery/1.9.1/jquery.min.js"></script>
<script type="text/javascript" src="https://ajax.googleapis.com/ajax/libs/jqueryui/1.9.1/jquery-ui.min.js"></script>
<script src="../site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="../site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="../site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>

<link rel="stylesheet"
      href="../site_libs/highlightjs/null.min.css"
      type="text/css" />

<script src="../site_libs/highlightjs/highlight.pack.js"></script>
<script>hljs.initHighlightingOnLoad();</script>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>

<script src="../js/doc_toc.js"></script>
<script src="../js/docs.js"></script>

<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.2/MathJax.js?config=TeX-MML-AM_CHTML"></script>
<script>
    MathJax.Hub.Config({
        extensions: ["tex2jax.js"],
        jax: ["input/TeX", "output/HTML-CSS"],
        tex2jax: {
        inlineMath: [ ['$','$'], ["\\(","\\)"] ],
        displayMath: [ ['$$','$$'], ["\\[","\\]"] ],
        processEscapes: true
        },
        "HTML-CSS": {
            preferredFont: "TeX",
            availableFonts: ["TeX"],
            styles: {
                scale: 110,
                ".MathJax_Display": {
                    "font-size": "110%",
                }
            }
        }
    });
</script>
<script>
function filterDataFrame(id) {
    var input = document.getElementById("search_" + id);
    var filter = input.value.toUpperCase();
    var table = document.getElementById("dataframe_" + id);
    var tr = table.getElementsByTagName("tr");
    // Loop through all table rows, and hide those who don't match the search query
    for (var i = 1; i < tr.length; i++) {
        for (var j = 0; j < tr[i].cells.length; ++j) {
            var matched = false;
            if (tr[i].cells[j].innerHTML.toUpperCase().indexOf(filter) != -1) {
                tr[i].style.display = "";
                matched = true
                break;
            }
            if (!matched)
                tr[i].style.display = "none";
        }
    }
}
function sortDataFrame(id, n, dtype) {
    var table = document.getElementById("dataframe_" + id);
    var tb = table.tBodies[0]; // use `<tbody>` to ignore `<thead>` and `<tfoot>` rows
    var tr = Array.prototype.slice.call(tb.rows, 0); // put rows into array
    if (dtype === 'numeric') {
        var fn = function(a, b) { 
            return parseFloat(a.cells[n].textContent) <= parseFloat(b.cells[n].textContent) ? -1 : 1;
        }
    } else {
        var fn = function(a, b) {
            var c = a.cells[n].textContent.trim().localeCompare(b.cells[n].textContent.trim()); 
            return c > 0 ? 1 : (c < 0 ? -1 : 0) }
    }
    var isSorted = function(array, fn) {
        if (array.length < 2)
            return 1;
        var direction = fn(array[0], array[1]); 
        for (var i = 1; i < array.length - 1; ++i) {
            var d = fn(array[i], array[i+1]);
            if (d == 0)
                continue;
            else if (direction == 0)
                direction = d;
            else if (direction != d)
                return 0;
            }
        return direction;
    }
    var sorted = isSorted(tr, fn);
    if (sorted == 1 || sorted == -1) {
        // if sorted already, reverse it
        for(var i = tr.length - 1; i >= 0; --i)
            tb.appendChild(tr[i]); // append each row in order
    } else {
        tr = tr.sort(fn);
        for(var i = 0; i < tr.length; ++i)
            tb.appendChild(tr[i]); // append each row in order
    }
}
</script>

<script>
$( document ).ready(function(){
            var cfg={'threshold':3,     // depth of toc (number of levels)
             'number_sections': false,
             'toc_cell': false,          // useless here
             'toc_window_display': true, // display the toc window
             "toc_section_display": "block", // display toc contents in the window
             'sideBar':true,       // sidebar or floating window
             'navigate_menu':false       // navigation menu (only in liveNotebook -- do not change)
            }
            var st={};                  // some variables used in the script
            st.rendering_toc_cell = false;
            st.config_loaded = false;
            st.extension_initialized=false;
            st.nbcontainer_marginleft = $('#notebook-container').css('margin-left')
            st.nbcontainer_marginright = $('#notebook-container').css('margin-right')
            st.nbcontainer_width = $('#notebook-container').css('width')
            st.oldTocHeight = undefined
            st.cell_toc = undefined;
            st.toc_index=0;
            // fire the main function with these parameters
            table_of_contents(cfg, st);
            var file=dscDict[$("h1:first").attr("id")];
            $("#toc-level0 a").css("color","#126dce");
            $('a[href="#'+$("h1:first").attr("id")+'"]').hide()
            var docs=dscArray;
            var docs_map=dscArrayMap;
            var pos=dscArray.indexOf(file);
            for (var a=pos;a>=0;a--){
                  $('<li><a href="'+docs[a]+'.html"><font color="#073642"><b>'+docs_map[docs[a]].replace(/_/g," ")+'</b></font></a></li>').insertBefore("#toc-level0 li:eq(0)");
            }
            $('a[href="'+file+'.html'+'"]').css("color","#126dce");
            for (var a=pos+1;a<docs.length;a++){
                  $(".toc #toc-level0").append('<li><a href="'+docs[a]+'.html"><font color="#073642"><b>'+docs_map[docs[a]].replace(/_/g," ")+'</b></font></a></li>');
            }
            // $("#toc-header").hide(); // comment out because it prevents search bar from displaying
    });
</script>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');
  // mark it active
  menuAnchor.parent().addClass('active');
  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>
<div class="container-fluid main-container">
<!-- tabsets -->
<script src="../site_libs/navigation-1.1/tabsets.js"></script>
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>



<title>Gene Mapping with CNV</title>

<style type = "text/css">
body {
  font-family: "Droid Sans";
  padding-top: 66px;
  padding-bottom: 40px;
}
</style>
</head>

<body>
<div tabindex="-1" id="notebook" class="border-box-sizing">
<div class="container" id="notebook-container">

<!-- code folding -->

<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="../index.html">Gene Mapping with CNV</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        
<li>
  <a href="../index.html">Overview</a>
</li>
        
<li>
  <a href="../analysis.html">Analysis</a>
</li>
        
<li>
  <a href="../prototype.html">Prototype</a>
</li>
        
<li>
  <a href="../dsc.html">Dsc</a>
</li>
        
<li>
  <a href="../setup/index.html">Setup</a>
</li>
        
<li>
  <a href="../writeup.html">Writeup</a>
</li>
        
      </ul>
        
<ul class="nav navbar-nav navbar-right">
<li>
   <a href="http://github.com/gaow/cnv-gene-mapping"> <span class="fa fa-github"></span> </a>
</li>
</ul>
        
      </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Introduction-(Use-dot-for-numbering)">Introduction (Use dot for numbering)<a class="anchor-link" href="#Introduction-(Use-dot-for-numbering)">&#182;</a></h2><ol>
<li>Copy number variations (CNVs): large genomic insertion or deletion events;</li>
<li>Length of CNVs: spread from 50 base pairs to kilo- or even mega-bases;</li>
<li>Previous CNV research: (1) detection of CNVs associated with disease risks, (2) identification of gene sets with CNV burden;</li>
<li>Challenge: CNVs often span multiple genes (1-30 or more), unclear which one has genetic susceptibility in the same CNV event.</li>
</ol>
<p>Figure 5</p>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Motivation-and-Aim">Motivation and Aim<a class="anchor-link" href="#Motivation-and-Aim">&#182;</a></h2><ol>
<li>Genome-wide association studies (GWAS) have limitations: (1) Risk alleles are usually at low frequencies and difficult to detect, (2) most loci have small effect sizes, not likely to be deleterious mutations, (3) causal variants or genes in GWAS loci are often unclear;</li>
<li>CNVs are believed to play a critical role in psychiatric disorders, such as schizophrenia (SCZ) [1];</li>
<li>The prevalence of SCZ ranges between 0.25% and 0.64% in the United States [2]; symptoms of SCZ include: (1) early-onset, usually late adolescence and early adulthood, (2) cognitive impairment/dysfunction, a core feature of SCZ, (3) difficulty in social activities, (4) potential to cause disability;</li>
<li>Inspired by statistical fine-mapping of causal variants in linkage-disequilibrium blocks from GWAS, we aim to develop a new approach that exploits large-scale genome-wide CNV data in case-control studies to map genes for psychiatric disorders;</li>
<li>It can be integrated with other gene-level datasets, e.g. results from exome-sequencing studies.</li>
</ol>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Data-available">Data available<a class="anchor-link" href="#Data-available">&#182;</a></h2><ol>
<li>Swedish schizophrenia population-based case-control exome sequencing CNV data from dbGAP;</li>
<li>Schizophrenia case-control CNV data from International Schizophrenia Consortium (ISC) study;</li>
<li>hg19 refGene from UCSC genome annotation database.</li>
</ol>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Approach-and-algorighm">Approach and algorighm<a class="anchor-link" href="#Approach-and-algorighm">&#182;</a></h2><p>Suppose the genome is divided into disjoint blocks and no CNVs in common between blocks (i.e. allow overlapping CNVs). Within a block $R$, we may have multiple, possibly overlapping, CNV events, and assume there is at least one causal gene in $R$. To infer CNV-gene configuration from case-control data, we leverage the statistical machinery of Bayesian regression.</p>
<p>Assume a mixture prior with spike-and-slab for $β_j$ and logistic regression model for the phenotype:
$$\beta_{j} = (1 - \pi_{j})\delta_0 + \pi_{j}g(\cdot)$$
where $$g(\cdot) \sim N(\mu,\sigma^2)$$
$$\text{logit P}(y_i = 1) =\log\big[\frac{\pi_{j}}{1 - \pi_{j}} \big] = \alpha_0 + \sum_{j=1}^m \alpha_j d_{{ij}}$$</p>
<p>$\pi_{j}$: prior inclusion probability of $\textit{j}$-th gene in a CNV-gene block</p>
<p>$y_i$: the phenotypic status of sample $i$</p>
<p>$\alpha_j$: effect size of $\textit{j}$-th gene</p>
<p>$d_{ij}$: the overlapping status with CNV event of $j$-th gene in sample $i$</p>
<p>$\mu, \sigma$: prior mean and standard error for spike, where $\mu \neq 0$</p>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Simulation-processes">Simulation processes<a class="anchor-link" href="#Simulation-processes">&#182;</a></h2><ol>
<li>Obtain CNV-gene blocks: simulate CNV-gene blocks containing at least a certain number of genes/exons (no overlapping CNVs between blocks). </li>
<li>Obtain genome-wide CNV-gene pattern (X matrix): randomly sample each block and merge them together as a simulated individual. Repeat this process $100,000$ times.</li>
<li>Obtain spike-and-slab prior: set penetrance/prevalence ($p$) as $0.05$, then $p \approx \frac{e^{\beta_0}}{1-e^{\beta_0}}$ and $\beta_0 \approx \log \frac{p}{1-p}$. Set $\pi_j = 0.05$, then $95\%$ $\beta_j's$ are adjusted to 0. Odds ratio (OR) for $j$-th gene is $\text{exp}(\beta_j)$.</li>
<li>Obtain phenotype y: first calculate $y=\frac{e^{x\boldsymbol{\beta}+\beta_0}}{1+e^{x\boldsymbol{\beta}+\beta_0}}$, then use Bernoulli($y_i$) to categorize each $y_i$ to either cases ($1's$) or controls ($0's$). Select all cases (about $5\%$) and randomly select equal number of controls.</li>
</ol>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Results">Results<a class="anchor-link" href="#Results">&#182;</a></h2><ol>
<li>Use R package <code>varbvs</code> to obtain prior parameters for MCMC method, $\pi = 0.0438$, $\mu = 0.777$ and $\sigma = 0.844$.</li>
<li>Map the genes for susceptibility in one CNV-gene block using R software packages, <code>SuSiE</code> and <code>varbvs</code>, to obtain posterior inclusion probabilities (PIP) and potentially credible set (CS);</li>
<li>Perform Bayesian Logistic Regression using python package <code>PyMC3</code>.</li>
</ol>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<ul>
<li><p>Block example 1 (5 genes, 1 positive effect gene):</p>

<pre><code>  sample_ID  phenotype    740 741 742 743 744
  655 1   0   0   1   0   0
  774 1   0   0   1   0   0
  826 1   0   0   1   0   0
  950 1   0   0   1   0   0
  1006    1   0   0   1   0   0
  1011    1   0   0   1   0   0
  1454    1   0   0   1   0   0
  1678    1   1   1   1   1   1
  2079    1   0   0   0   1   0
  2276    1   0   0   1   0   0
  2691    1   0   0   1   0   0
  3187    1   1   1   1   1   1
  3837    1   0   0   1   0   0
  4900    1   0   0   1   0   0
  5713    1   0   0   1   0   0
  6080    1   1   1   1   1   1
  6278    1   0   0   1   0   0
  6368    1   0   0   0   1   0
  7037    0   0   0   0   1   0
  7891    0   1   1   1   1   1
  8321    0   0   0   1   0   0
  11199   0   0   0   1   0   0
  12311   0   1   1   1   1   1

  gene index  d_c d_nc    nd_c    nd_nc   p
  gene_742    16   6690   4   6702    0.011757
  gene_743    5   6701    3   6703    0.726481
  gene_744    3   6703    2   6704    1.000000
  gene_741    3   6703    2   6704    1.000000
  gene_740    3   6703    2   6704    1.000000</code></pre>
</li>
</ul>
<table>
<thead><tr>
<th style="text-align:center">gene index</th>
<th style="text-align:center">simulated effect</th>
<th style="text-align:center">SuSiE</th>
<th style="text-align:center">varbvs</th>
<th style="text-align:center">PyMC3</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:center">740</td>
<td style="text-align:center">0</td>
<td style="text-align:center">0.0442</td>
<td style="text-align:center">0.0913</td>
<td style="text-align:center">0.0290</td>
</tr>
<tr>
<td style="text-align:center">741</td>
<td style="text-align:center">0</td>
<td style="text-align:center">0.0442</td>
<td style="text-align:center">0.0913</td>
<td style="text-align:center">0.0370</td>
</tr>
<tr>
<td style="text-align:center">742</td>
<td style="text-align:center">0.9806</td>
<td style="text-align:center"><strong>0.8173</strong></td>
<td style="text-align:center"><strong>0.6076</strong></td>
<td style="text-align:center"><strong>0.4680</strong></td>
</tr>
<tr>
<td style="text-align:center">743</td>
<td style="text-align:center">0</td>
<td style="text-align:center">0.0501</td>
<td style="text-align:center">0.0804</td>
<td style="text-align:center">0.0255</td>
</tr>
<tr>
<td style="text-align:center">744</td>
<td style="text-align:center">0</td>
<td style="text-align:center">0.0442</td>
<td style="text-align:center">0.0913</td>
<td style="text-align:center">0.0310</td>
</tr>
</tbody>
</table>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<ul>
<li><p>Block example 2 (14 genes, 2 positive effect genes)</p>

<pre><code>  sample_ID   phenotype   1019    1020    1021    1022    1023    1024    1025    1026    1027    1028    1029    1030    1031    1032
  789  1  1   1   1   1   1   1   1   1   1   1   1   1   1   1
  1198    1   1   1   1   1   1   1   1   1   1   1   1   1   1   1
  1291    1   1   1   1   1   1   1   1   1   1   1   1   1   1   1
  1748    1   0   0   1   0   0   0   0   0   0   0   0   0   0   0
  2306    1   1   1   1   1   1   1   1   1   1   1   1   1   1   1
  2377    1   0   0   1   0   0   0   0   0   0   0   0   0   0   0
  2401    1   0   0   1   0   0   0   0   0   0   0   0   0   0   0
  3285    1   1   1   0   0   0   0   0   0   0   0   0   0   0   0
  3477    1   1   1   1   1   1   1   1   1   1   1   1   1   1   1
  3550    1   1   1   0   0   0   0   0   0   0   0   0   0   0   0
  4497    1   1   1   1   1   1   1   1   1   1   1   1   1   1   1
  4555    1   0   0   1   0   0   0   0   0   0   0   0   0   0   0
  5394    1   0   0   1   0   0   0   0   0   0   0   0   0   0   0
  5531    1   0   0   1   0   0   0   0   0   0   0   0   0   0   0
  6230    1   0   0   1   0   0   0   0   0   0   0   0   0   0   0
  6390    1   1   1   1   1   1   1   1   1   1   1   1   1   1   1
  9210    0   1   1   0   0   0   0   0   0   0   0   0   0   0   0
  9488    0   1   1   1   1   1   1   1   1   1   1   1   1   1   1
  11470   0   1   1   1   1   1   1   1   1   1   1   1   1   1   1
  11809   0   1   1   0   0   0   0   0   0   0   0   0   0   0   0
  12698   0   0   0   1   0   0   0   0   0   0   0   0   0   0   0

  gene index  d_c  d_nc   nd_c  nd_nc p
  gene_1021   14   6692   3   6703    0.012672
  gene_1023   7   6699    2   6704    0.179541
  gene_1024   7   6699    2   6704    0.179541
  gene_1022   7   6699    2   6704    0.179541
  gene_1025   7   6699    2   6704    0.179541
  gene_1027   7   6699    2   6704    0.179541
  gene_1030   7   6699    2   6704    0.179541
  gene_1031   7   6699    2   6704    0.179541
  gene_1032   7   6699    2   6704    0.179541
  gene_1029   7   6699    2   6704    0.179541
  gene_1026   7   6699    2   6704    0.179541
  gene_1028   7   6699    2   6704    0.179541
  gene_1020   9   6697    4   6702    0.266611
  gene_1019   9   6697    4   6702    0.266611</code></pre>
</li>
</ul>
<table>
<thead><tr>
<th style="text-align:center">gene index</th>
<th style="text-align:center">simulated effect</th>
<th style="text-align:center">SuSiE</th>
<th style="text-align:center">varbvs</th>
<th style="text-align:center">PyMC3</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:center">1019</td>
<td style="text-align:center">0</td>
<td style="text-align:center">0.0414</td>
<td style="text-align:center">0.0500</td>
<td style="text-align:center">0.0385</td>
</tr>
<tr>
<td style="text-align:center">1020</td>
<td style="text-align:center">0</td>
<td style="text-align:center">0.0414</td>
<td style="text-align:center">0.0500</td>
<td style="text-align:center">0.0365</td>
</tr>
<tr>
<td style="text-align:center">1021</td>
<td style="text-align:center">0.60</td>
<td style="text-align:center">0.2900</td>
<td style="text-align:center">0.2861</td>
<td style="text-align:center">0.3040</td>
</tr>
<tr>
<td style="text-align:center">1022</td>
<td style="text-align:center">0</td>
<td style="text-align:center">0.0570</td>
<td style="text-align:center">0.0617</td>
<td style="text-align:center">0.0580</td>
</tr>
<tr>
<td style="text-align:center">1023</td>
<td style="text-align:center">0</td>
<td style="text-align:center">0.0570</td>
<td style="text-align:center">0.0617</td>
<td style="text-align:center">0.0505</td>
</tr>
<tr>
<td style="text-align:center">1024</td>
<td style="text-align:center">0</td>
<td style="text-align:center">0.0570</td>
<td style="text-align:center">0.0617</td>
<td style="text-align:center">0.0650</td>
</tr>
<tr>
<td style="text-align:center">1025</td>
<td style="text-align:center">0</td>
<td style="text-align:center">0.0570</td>
<td style="text-align:center">0.0617</td>
<td style="text-align:center">0.0495</td>
</tr>
<tr>
<td style="text-align:center">1026</td>
<td style="text-align:center">0</td>
<td style="text-align:center">0.0570</td>
<td style="text-align:center">0.0617</td>
<td style="text-align:center">0.0475</td>
</tr>
<tr>
<td style="text-align:center">1027</td>
<td style="text-align:center">0</td>
<td style="text-align:center">0.0570</td>
<td style="text-align:center">0.0617</td>
<td style="text-align:center">0.0640</td>
</tr>
<tr>
<td style="text-align:center">1028</td>
<td style="text-align:center">0</td>
<td style="text-align:center">0.0570</td>
<td style="text-align:center">0.0617</td>
<td style="text-align:center">0.0635</td>
</tr>
<tr>
<td style="text-align:center">1029</td>
<td style="text-align:center">0</td>
<td style="text-align:center">0.0570</td>
<td style="text-align:center">0.0617</td>
<td style="text-align:center">0.0585</td>
</tr>
<tr>
<td style="text-align:center">1030</td>
<td style="text-align:center">0</td>
<td style="text-align:center">0.0570</td>
<td style="text-align:center">0.0617</td>
<td style="text-align:center">0.0485</td>
</tr>
<tr>
<td style="text-align:center">1031</td>
<td style="text-align:center">0</td>
<td style="text-align:center">0.0570</td>
<td style="text-align:center">0.0617</td>
<td style="text-align:center">0.0580</td>
</tr>
<tr>
<td style="text-align:center">1032</td>
<td style="text-align:center">1.07</td>
<td style="text-align:center">0.0570</td>
<td style="text-align:center">0.0617</td>
<td style="text-align:center">0.0580</td>
</tr>
</tbody>
</table>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<ul>
<li><p>Block example 3 (12 genes, 3 positive effect genes):</p>

<pre><code>  gene index  d_c d_nc   nd_c  nd_nc   p
  gene_671    99  6607    9   6697    1.956916e-20
  gene_672    99  6607    9   6697    1.956916e-20
  gene_676    20  6686    0   6706    1.880480e-06
  gene_678    20  6686    0   6706    1.880480e-06
  gene_677    20  6686    0   6706    1.880480e-06
  gene_673    23  6683    5   6701    9.016016e-04
  gene_674    23  6683    5   6701    9.016016e-04
  gene_675    23  6683    5   6701    9.016016e-04
  gene_667    12  6694    2   6704    1.289476e-02
  gene_668    17  6689    6   6700    3.453722e-02
  gene_670    17  6689    6   6700    3.453722e-02
  gene_669    17  6689    6   6700    3.453722e-02</code></pre>
</li>
</ul>
<table>
<thead><tr>
<th style="text-align:center">gene index</th>
<th style="text-align:center">simulated effect</th>
<th style="text-align:center">SuSiE</th>
<th style="text-align:center">varbvs</th>
<th style="text-align:center">PyMC3</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:center">667</td>
<td style="text-align:center">0</td>
<td style="text-align:center">8.8818e-16</td>
<td style="text-align:center">0.0435</td>
<td style="text-align:center">0.0240</td>
</tr>
<tr>
<td style="text-align:center">668</td>
<td style="text-align:center">0</td>
<td style="text-align:center">3.3307e-16</td>
<td style="text-align:center">0.0330</td>
<td style="text-align:center">0.0195</td>
</tr>
<tr>
<td style="text-align:center">669</td>
<td style="text-align:center">0</td>
<td style="text-align:center">3.3307e-16</td>
<td style="text-align:center">0.0330</td>
<td style="text-align:center">0.0110</td>
</tr>
<tr>
<td style="text-align:center">670</td>
<td style="text-align:center">0</td>
<td style="text-align:center">3.3307e-16</td>
<td style="text-align:center">0.0330</td>
<td style="text-align:center">0.0150</td>
</tr>
<tr>
<td style="text-align:center">671</td>
<td style="text-align:center">1.2865</td>
<td style="text-align:center"><strong>0.4999</strong></td>
<td style="text-align:center">1</td>
<td style="text-align:center">0.5460</td>
</tr>
<tr>
<td style="text-align:center">672</td>
<td style="text-align:center">0.5374</td>
<td style="text-align:center"><strong>0.4999</strong></td>
<td style="text-align:center">0.0192</td>
<td style="text-align:center">0.5185</td>
</tr>
<tr>
<td style="text-align:center">673</td>
<td style="text-align:center">0</td>
<td style="text-align:center">7.9936e-15</td>
<td style="text-align:center">0.0338</td>
<td style="text-align:center">0.0185</td>
</tr>
<tr>
<td style="text-align:center">674</td>
<td style="text-align:center">0</td>
<td style="text-align:center">7.9936e-15</td>
<td style="text-align:center">0.0338</td>
<td style="text-align:center">0.0230</td>
</tr>
<tr>
<td style="text-align:center">675</td>
<td style="text-align:center">0</td>
<td style="text-align:center">7.9936e-15</td>
<td style="text-align:center">0.0338</td>
<td style="text-align:center">0.0300</td>
</tr>
<tr>
<td style="text-align:center">676</td>
<td style="text-align:center">0</td>
<td style="text-align:center">5.2757e-13</td>
<td style="text-align:center">0.0614</td>
<td style="text-align:center">0.1270</td>
</tr>
<tr>
<td style="text-align:center">677</td>
<td style="text-align:center">0</td>
<td style="text-align:center">5.2757e-13</td>
<td style="text-align:center">0.0614</td>
<td style="text-align:center">0.1325</td>
</tr>
<tr>
<td style="text-align:center">678</td>
<td style="text-align:center">0.9833</td>
<td style="text-align:center">5.2757e-13</td>
<td style="text-align:center">0.0614</td>
<td style="text-align:center">0.1080</td>
</tr>
</tbody>
</table>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Reference">Reference<a class="anchor-link" href="#Reference">&#182;</a></h2><ol>
<li>C. Lowther et al., Genomic Disorders in Psychiatry-What Does the Clinician Need to Know? Current Psychiatry Reports 19 (2017).</li>
<li>E. Q. Wu et al., Annual prevalence of diagnosed schizophrenia in the USA: a claims data analysis approach. Psychol Med (2006).</li>
</ol>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="ASHG-talks">ASHG talks<a class="anchor-link" href="#ASHG-talks">&#182;</a></h2><p>Session 81</p>
<p>Poison exons in neurodevelopmental disorders: from development and disease to therapeutic target</p>
<ol>
<li>What is poison exons: small exonic regions that lead to premature truncation of a protein when spliced into an RNA transcript (contain stop codon).</li>
<li>When does it occur: Inclusion of poison exons occurs during specific times in neurodevelopment, while splicing occurs in a cell-specific manner.</li>
<li>Many genes involved in neurodevelopmental disorders harbor these poison exons, epilepsy, autism, and malformations of cortical development, including ion channels, epigenetic regulators.</li>
<li>How to identify poison exons in neuronal development, splicing mechanisms, variants that disrupt splicing</li>
<li><p>The identification of patient-specific variants</p>

<pre><code> Dr. Mefford: describe the current genetic landscape of neurodevelopmental disorders and strategies to identify the molecular etiology in undiagnosed cases;
 Dr. Zhang: discuss the studies that identified poison exons in neuronal development, the splicing mechanisms that govern their use, and variants detected in patients with malformations of cortical development that disrupt their splicing;
 Dr. Carvill: 
 Dr. Isom: </code></pre>
</li>
</ol>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p>This session will focus on the role of poison exons in epilepsy. Poison exons are small exonic regions that, when spliced into an RNA transcript, lead to premature truncation of a protein. Inclusion of poison exons occurs during specific times in neurodevelopment, and splicing occurs in a cell-specific manner. Many of the genes implicated in neurodevelopmental disorders, including epilepsy, autism, and malformations of cortical development, harbor these poison exons. These include ion channels (SCN1A/2A/8A), epigenetic regulators (CHD2, MBD5), and cytoskeletal proteins (FLNA). In this session, Dr. Mefford will open by describing the current genetic landscape of neurodevelopmental disorders and strategies to identify the molecular etiology in undiagnosed cases. Dr. Mefford will also introduce the concept of poison exons and their function. Dr. Zhang will discuss the studies that identified poison exons in neuronal development, the splicing mechanisms that govern their use, and variants detected in patients with malformations of cortical development that disrupt their splicing. Dr. Carvill will discuss the identification of patient-specific variants that lead to aberrant inclusion of poison exons in genes implicated in epilepsy, including SCN1A, and functional studies in patient-derived induced neurons. Dr. Isom will describe an antisense oligonucleotide that targets an SCN1A poison exon, preventing its inclusion and thus restoring full-length protein and preventing SCN1A-related mortality in an animal model of epilepsy.</p>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="ASHG-posters">ASHG posters<a class="anchor-link" href="#ASHG-posters">&#182;</a></h2><p>Approximately LD independent regions for biobank scale genetic analyses</p>
<ol>
<li>For genetic fine-mapping methods to produce valid estimates: loci are assumed to be independent</li>
<li>Introduce a new method for identifying approximately LD independent regions that minimizes different functins of the full "off-diagonal" LD band at region boundaries and apply it to the UK Biobank. (minimize off-diagonal LD scores)</li>
<li>Use $\sum_{ij}1_{\{r_{ij}&gt;0.60\}}$ instead of a naive odLDSC $\sum_{ij}r_{ij}^2$</li>
<li>affect fine-mapping</li>
</ol>

</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span>
</pre></div>

    </div>
</div>
</div>

</div>
<hr>
&copy 2016-2017 Min Qiao at <a href="http://xinhelab.org">Xin He lab</a>, The University of Chicago
<p><small>Exported from <a href="http://github.com/gaow/cnv-gene-mapping/blob/a38c6d13dd72bf20e4371c654f5b0e91ba1f63d9/dsc/20191012_ASHG_poster.ipynb"><code>dsc/20191012_ASHG_poster.ipynb</code></a> committed by minqiao on Tue Oct 22 14:02:10 2019 <a href="http://github.com/gaow/cnv-gene-mapping/commit/a38c6d13dd72bf20e4371c654f5b0e91ba1f63d9">revision 3, a38c6d1</a> <a href="https://stephenslab.github.io/ipynb-website/notes.html#Note-about-commit-ids"><span class="fa fa-question-circle"></span></a></small></p>
</div>
</div>
</body>
</html>
