<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="ipynb_website:version" content="0.9.2" />
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link rel="stylesheet" type="text/css" href="../css/jt.css">
<link rel="stylesheet" type="text/css" href="../css/toc2.css">
<link href="../site_libs/jqueryui-1.11.4/jquery-ui.css">
<link rel="stylesheet" href="../site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<link rel="stylesheet" href="../site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />
<script type="text/javascript" src="https://ajax.googleapis.com/ajax/libs/jquery/1.9.1/jquery.min.js"></script>
<script type="text/javascript" src="https://ajax.googleapis.com/ajax/libs/jqueryui/1.9.1/jquery-ui.min.js"></script>
<script src="../site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="../site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="../site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<link rel="stylesheet"
      href="../site_libs/highlight/textmate.css"
      type="text/css" />
<script src="../site_libs/highlight/highlight.js"></script>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>
<script src="../js/toc2.js"></script>
<script src="../js/docs.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS_HTML"></script>
<script>
    MathJax.Hub.Config({
        extensions: ["tex2jax.js"],
        jax: ["input/TeX", "output/HTML-CSS"],
        tex2jax: {
        inlineMath: [ ['$','$'], ["\\(","\\)"] ],
        displayMath: [ ['$$','$$'], ["\\[","\\]"] ],
        processEscapes: true
        },
        "HTML-CSS": {
            preferredFont: "TeX",
            availableFonts: ["TeX"],
            styles: {
                scale: 110,
                ".MathJax_Display": {
                    "font-size": "110%",
                }
            }
        }
    });
</script>
<script>
$( document ).ready(function(){
            var cfg={'threshold':3,     // depth of toc (number of levels)
             'number_sections': false,
             'toc_cell': false,          // useless here
             'toc_window_display': true, // display the toc window
             "toc_section_display": "block", // display toc contents in the window
             'sideBar':true,       // sidebar or floating window
             'navigate_menu':false       // navigation menu (only in liveNotebook -- do not change)
            }
            var st={};                  // some variables used in the script
            st.rendering_toc_cell = false;
            st.config_loaded = false;
            st.extension_initialized=false;
            st.nbcontainer_marginleft = $('#notebook-container').css('margin-left')
            st.nbcontainer_marginright = $('#notebook-container').css('margin-right')
            st.nbcontainer_width = $('#notebook-container').css('width')
            st.oldTocHeight = undefined
            st.cell_toc = undefined;
            st.toc_index=0;
            // fire the main function with these parameters
            table_of_contents(cfg, st);
            var file=utilsDict[$("h1:first").attr("id")];
            $("#toc-level0 a").css("color","#126dce");
            $('a[href="#'+$("h1:first").attr("id")+'"]').hide()
            var docs=utilsArray;
            var pos=utilsArray.indexOf(file);
            for (var a=pos;a>=0;a--){
                  var name=docs[a]
                  $('<li><a href="'+name+'.html"><font color="#073642"><b>'+name.replace(/_/g," ")+'</b></font></a></li>').insertBefore("#toc-level0 li:eq(0)");
            }
            $('a[href="'+file+'.html'+'"]').css("color","#126dce");
            for (var a=pos+1;a<docs.length;a++){
                  var name=docs[a]
                  $(".toc #toc-level0").append('<li><a href="'+name+'.html"><font color="#073642"><b>'+name.replace(/_/g," ")+'</b></font></a></li>');
            }
            $("#toc-header").hide();
    });
</script>
<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');
  // mark it active
  menuAnchor.parent().addClass('active');
  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>
<div class="container-fluid main-container">
<!-- tabsets -->
<script src="../site_libs/navigation-1.1/tabsets.js"></script>
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>
<title>Gene Mapping with CNV</title>
<style type = "text/css">
body {
  font-family: "Droid Sans";
  padding-top: 66px;
  padding-bottom: 40px;
}
</style>
</head>
<body>
<div tabindex="-1" id="notebook" class="border-box-sizing">
<div class="container" id="notebook-container">
<!-- code folding -->
<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="../index.html">Gene Mapping with CNV</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">Overview</a>
</li>
<li>
  <a href="../analysis.html">Analysis</a>
</li>
<li>
  <a href="../prototype.html">Prototype</a>
</li>
<li>
  <a href="../utils.html">Utils</a>
</li>
<li>
  <a href="../setup.html">Setup</a>
</li>
<li>
  <a href="../writeup.html">Writeup</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
   <a href="http://github.com/gaow/cnv-gene-mapping"> <span class="fa fa-github"></span> </a>
</li>
</ul>
      </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->
<div class="cell border-box-sizing text_cell rendered">
<div class="prompt input_prompt">
</div>
<div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h1 id="Python-utilities">Python utilities<a class="anchor-link" href="#Python-utilities">&#182;</a></h1>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered">
<div class="prompt input_prompt">
</div>
<div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p>This is a centralized script for all Python functions / classes ever used repeatedly, exported as Python module <code>utils.py</code>. <strong>This notebook should not / cannot be executed as is!</strong>.</p>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered">
<div class="prompt input_prompt">
</div>
<div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Load-modules">Load modules<a class="anchor-link" href="#Load-modules">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[10]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="kn">import</span> <span class="nn">os</span><span class="o">,</span> <span class="nn">sys</span>
<span class="kn">import</span> <span class="nn">random</span>
<span class="kn">from</span> <span class="nn">random</span> <span class="kn">import</span> <span class="n">randint</span>
<span class="kn">import</span> <span class="nn">pandas</span> <span class="kn">as</span> <span class="nn">pd</span>
<span class="kn">from</span> <span class="nn">pandasql</span> <span class="kn">import</span> <span class="n">sqldf</span>
<span class="kn">import</span> <span class="nn">numpy</span> <span class="kn">as</span> <span class="nn">np</span>
<span class="kn">import</span> <span class="nn">matplotlib.pyplot</span> <span class="kn">as</span> <span class="nn">plt</span>
<span class="kn">import</span> <span class="nn">seaborn</span> <span class="kn">as</span> <span class="nn">sns</span>
<span class="kn">import</span> <span class="nn">scipy</span> <span class="kn">as</span> <span class="nn">sp</span>
<span class="kn">from</span> <span class="nn">scipy</span> <span class="kn">import</span> <span class="n">stats</span>
<span class="kn">import</span> <span class="nn">glob</span>
<span class="kn">import</span> <span class="nn">math</span>
<span class="kn">from</span> <span class="nn">sklearn.mixture</span> <span class="kn">import</span> <span class="n">GaussianMixture</span>
<span class="kn">from</span> <span class="nn">datetime</span> <span class="kn">import</span> <span class="n">datetime</span>
<span class="kn">import</span> <span class="nn">pickle</span>
<span class="kn">from</span> <span class="nn">fisher</span> <span class="kn">import</span> <span class="n">pvalue</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered">
<div class="prompt input_prompt">
</div>
<div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="File-I/O-functions">File I/O functions<a class="anchor-link" href="#File-I/O-functions">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[18]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="k">def</span> <span class="nf">load_reference_gene</span><span class="p">(</span><span class="n">filename</span><span class="p">):</span>
    <span class="sd">&#39;&#39;&#39;Load reference gene database&#39;&#39;&#39;</span>
    <span class="n">ref_gene</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_table</span><span class="p">(</span><span class="n">filename</span><span class="p">,</span> <span class="n">compression</span><span class="o">=</span><span class="s2">&quot;gzip&quot;</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">&quot;</span><span class="se">\t</span><span class="s2">&quot;</span><span class="p">,</span> 
                         <span class="n">header</span> <span class="o">=</span> <span class="bp">None</span><span class="p">,</span> <span class="n">usecols</span><span class="o">=</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span><span class="mi">2</span><span class="p">,</span><span class="mi">4</span><span class="p">,</span><span class="mi">5</span><span class="p">,</span><span class="mi">12</span><span class="p">),</span> 
                         <span class="n">names</span> <span class="o">=</span> <span class="p">[</span><span class="s2">&quot;tx_name&quot;</span><span class="p">,</span> <span class="s2">&quot;chrom&quot;</span><span class="p">,</span> <span class="s2">&quot;tx_start&quot;</span><span class="p">,</span> <span class="s2">&quot;tx_end&quot;</span><span class="p">,</span> <span class="s2">&quot;gene_name&quot;</span><span class="p">])</span>
    <span class="k">return</span> <span class="n">ref_gene</span><span class="o">.</span><span class="n">drop_duplicates</span><span class="p">(</span><span class="n">subset</span><span class="o">=</span><span class="p">(</span><span class="s2">&quot;chrom&quot;</span><span class="p">,</span> <span class="s2">&quot;tx_start&quot;</span><span class="p">,</span> <span class="s2">&quot;tx_end&quot;</span><span class="p">))</span>
<span class="k">def</span> <span class="nf">load_pthwy_gene</span><span class="p">(</span><span class="n">filename</span><span class="p">,</span> <span class="n">n_skiprow</span> <span class="o">=</span> <span class="mi">2</span><span class="p">):</span>
    <span class="sd">&#39;&#39;&#39;Load pathway genes. For example, calcium pathway genes.</span>
<span class="sd">       The input file must contain a column named &quot;gene_name&quot;.</span>
<span class="sd">    &#39;&#39;&#39;</span>
    <span class="n">pthwy_gene</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_table</span><span class="p">(</span><span class="n">filename</span><span class="p">,</span> <span class="n">skiprows</span> <span class="o">=</span> <span class="n">n_skiprow</span><span class="p">,</span> <span class="n">header</span> <span class="o">=</span> <span class="bp">None</span><span class="p">,</span> <span class="n">names</span> <span class="o">=</span> <span class="p">[</span><span class="s2">&quot;gene_name&quot;</span><span class="p">])</span>
    <span class="k">return</span> <span class="n">pthwy_gene</span>
<span class="k">def</span> <span class="nf">load_cnv_data</span><span class="p">(</span><span class="n">filename</span><span class="p">):</span>
    <span class="sd">&#39;&#39;&#39;load cnv data&#39;&#39;&#39;</span>
    <span class="n">cnvbed</span> <span class="o">=</span> <span class="p">{}</span>
    <span class="n">dataset</span> <span class="o">=</span> <span class="bp">None</span>
    <span class="k">for</span> <span class="n">line</span> <span class="ow">in</span> <span class="nb">open</span><span class="p">(</span><span class="n">filename</span><span class="p">)</span><span class="o">.</span><span class="n">readlines</span><span class="p">():</span>
        <span class="k">if</span> <span class="ow">not</span> <span class="n">line</span><span class="o">.</span><span class="n">startswith</span><span class="p">(</span><span class="s2">&quot;chr&quot;</span><span class="p">):</span>
            <span class="n">dataset</span> <span class="o">=</span> <span class="n">line</span><span class="o">.</span><span class="n">strip</span><span class="p">()</span><span class="o">.</span><span class="n">split</span><span class="p">()[</span><span class="mi">1</span><span class="p">]</span><span class="o">.</span><span class="n">lstrip</span><span class="p">(</span><span class="s2">&quot;name=&quot;</span><span class="p">)</span>
            <span class="n">cnvbed</span><span class="p">[</span><span class="n">dataset</span><span class="p">]</span> <span class="o">=</span> <span class="p">{}</span>
            <span class="k">continue</span>
        <span class="n">line</span> <span class="o">=</span> <span class="n">line</span><span class="o">.</span><span class="n">strip</span><span class="p">()</span><span class="o">.</span><span class="n">split</span><span class="p">()</span>
        <span class="k">if</span> <span class="ow">not</span> <span class="n">line</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span> <span class="ow">in</span> <span class="n">cnvbed</span><span class="p">[</span><span class="n">dataset</span><span class="p">]:</span>
            <span class="n">cnvbed</span><span class="p">[</span><span class="n">dataset</span><span class="p">][</span><span class="n">line</span><span class="p">[</span><span class="mi">0</span><span class="p">]]</span> <span class="o">=</span> <span class="p">[]</span>
        <span class="n">cnvbed</span><span class="p">[</span><span class="n">dataset</span><span class="p">][</span><span class="n">line</span><span class="p">[</span><span class="mi">0</span><span class="p">]]</span><span class="o">.</span><span class="n">append</span><span class="p">((</span><span class="nb">int</span><span class="p">(</span><span class="n">line</span><span class="p">[</span><span class="mi">1</span><span class="p">]),</span><span class="nb">int</span><span class="p">(</span><span class="n">line</span><span class="p">[</span><span class="mi">2</span><span class="p">])))</span>
    <span class="k">for</span> <span class="n">dataset</span> <span class="ow">in</span> <span class="n">cnvbed</span><span class="o">.</span><span class="n">keys</span><span class="p">():</span>
        <span class="k">for</span> <span class="n">chrom</span> <span class="ow">in</span> <span class="n">cnvbed</span><span class="p">[</span><span class="n">dataset</span><span class="p">]:</span>
            <span class="n">cnvbed</span><span class="p">[</span><span class="n">dataset</span><span class="p">][</span><span class="n">chrom</span><span class="p">]</span><span class="o">.</span><span class="n">sort</span><span class="p">()</span>
    <span class="n">cnvbed_df</span> <span class="o">=</span> <span class="p">{}</span>
    <span class="k">for</span> <span class="n">dataset</span> <span class="ow">in</span> <span class="n">cnvbed</span><span class="o">.</span><span class="n">keys</span><span class="p">():</span>
        <span class="n">cnvbed_df</span><span class="p">[</span><span class="n">dataset</span><span class="p">]</span> <span class="o">=</span> <span class="p">{</span><span class="s2">&quot;chrom&quot;</span><span class="p">:[],</span> <span class="s2">&quot;cnv_start&quot;</span><span class="p">:[],</span> <span class="s2">&quot;cnv_end&quot;</span><span class="p">:[]}</span>
        <span class="k">for</span> <span class="n">chrom</span> <span class="ow">in</span> <span class="n">cnvbed</span><span class="p">[</span><span class="n">dataset</span><span class="p">]:</span>
            <span class="n">start</span><span class="p">,</span> <span class="n">end</span> <span class="o">=</span> <span class="nb">tuple</span><span class="p">(</span><span class="nb">zip</span><span class="p">(</span><span class="o">*</span><span class="n">cnvbed</span><span class="p">[</span><span class="n">dataset</span><span class="p">][</span><span class="n">chrom</span><span class="p">]))</span> 
            <span class="n">cnvbed_df</span><span class="p">[</span><span class="n">dataset</span><span class="p">][</span><span class="s2">&quot;chrom&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">extend</span><span class="p">([</span><span class="n">chrom</span><span class="p">]</span> <span class="o">*</span> <span class="nb">len</span><span class="p">(</span><span class="n">start</span><span class="p">))</span>
            <span class="n">cnvbed_df</span><span class="p">[</span><span class="n">dataset</span><span class="p">][</span><span class="s2">&quot;cnv_start&quot;</span><span class="p">]</span> <span class="o">+=</span> <span class="nb">list</span><span class="p">(</span><span class="n">start</span><span class="p">)</span>
            <span class="n">cnvbed_df</span><span class="p">[</span><span class="n">dataset</span><span class="p">][</span><span class="s2">&quot;cnv_end&quot;</span><span class="p">]</span> <span class="o">+=</span> <span class="nb">list</span><span class="p">(</span><span class="n">end</span><span class="p">)</span>
        <span class="n">cnvbed_df</span><span class="p">[</span><span class="n">dataset</span><span class="p">]</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="o">.</span><span class="n">from_dict</span><span class="p">(</span><span class="n">cnvbed_df</span><span class="p">[</span><span class="n">dataset</span><span class="p">])</span><span class="o">.</span><span class="n">drop_duplicates</span><span class="p">(</span>
                                                <span class="n">subset</span><span class="o">=</span><span class="p">(</span><span class="s2">&quot;chrom&quot;</span><span class="p">,</span> <span class="s2">&quot;cnv_start&quot;</span><span class="p">,</span> <span class="s2">&quot;cnv_end&quot;</span><span class="p">))</span>
    <span class="k">return</span> <span class="n">cnvbed_df</span>
<span class="k">def</span> <span class="nf">save_data</span><span class="p">(</span><span class="n">data</span><span class="p">,</span> <span class="n">filename</span><span class="p">):</span>
    <span class="n">pickle</span><span class="o">.</span><span class="n">dump</span><span class="p">(</span><span class="n">data</span><span class="p">,</span> <span class="nb">open</span><span class="p">(</span><span class="n">filename</span><span class="p">,</span> <span class="s2">&quot;wb&quot;</span><span class="p">))</span>
<span class="k">def</span> <span class="nf">load_data</span><span class="p">(</span><span class="n">filename</span><span class="p">):</span>
    <span class="k">return</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_pickle</span><span class="p">(</span><span class="n">filename</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered">
<div class="prompt input_prompt">
</div>
<div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Global-variables">Global variables<a class="anchor-link" href="#Global-variables">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[12]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="k">class</span> <span class="nc">Environment</span><span class="p">(</span><span class="nb">dict</span><span class="p">):</span>
    <span class="k">def</span> <span class="fm">__init__</span><span class="p">(</span><span class="bp">self</span><span class="p">):</span>
        <span class="n">parameters</span> <span class="o">=</span> <span class="p">{</span><span class="s1">&#39;block_size&#39;</span><span class="p">:</span> <span class="mi">100000</span><span class="p">,</span>
                      <span class="s1">&#39;avg_cnv_per_individual&#39;</span><span class="p">:</span> <span class="mi">5</span><span class="p">,</span>
                      <span class="s1">&#39;n_case&#39;</span><span class="p">:</span> <span class="mi">5</span><span class="p">,</span>
                      <span class="s1">&#39;n_ctrl&#39;</span><span class="p">:</span> <span class="mi">5</span><span class="p">,</span>
                       <span class="c1"># set Gamma shape to be 3 instead of 5</span>
                       <span class="c1"># &#39;odds_ratio_params&#39; : None # for H_0</span>
                      <span class="s1">&#39;odds_ratio_params&#39;</span><span class="p">:</span> <span class="p">{</span><span class="s1">&#39;shape&#39;</span><span class="p">:</span> <span class="mi">25</span><span class="p">,</span> <span class="s1">&#39;scale&#39;</span><span class="p">:</span> <span class="mi">1</span><span class="p">},</span>
                      <span class="s1">&#39;prevalence&#39;</span><span class="p">:</span> <span class="mf">0.005</span><span class="p">,</span>
                      <span class="s1">&#39;n_causal_gene&#39;</span><span class="p">:</span> <span class="mi">200</span><span class="p">,</span>
                      <span class="s1">&#39;refgene_file&#39;</span><span class="p">:</span> <span class="s1">&#39;../data/refGene.txt.gz&#39;</span><span class="p">,</span>
                      <span class="s1">&#39;pthwy_gene_file&#39;</span><span class="p">:</span> <span class="s1">&#39;../data/calciumgeneset.txt&#39;</span><span class="p">,</span>
                      <span class="s1">&#39;cnv_file&#39;</span><span class="p">:</span> <span class="s1">&#39;../data/ISC-r1.CNV.bed&#39;</span><span class="p">,</span>
                      <span class="s1">&#39;case_dataset&#39;</span><span class="p">:</span> <span class="s1">&#39;delCases&#39;</span><span class="p">,</span>
                      <span class="s1">&#39;ctrl_dataset&#39;</span><span class="p">:</span> <span class="s1">&#39;delControls&#39;</span><span class="p">,</span>
                      <span class="s1">&#39;output&#39;</span><span class="p">:</span> <span class="s1">&#39;del_sample&#39;</span><span class="p">,</span>
                      <span class="s1">&#39;ncpu&#39;</span><span class="p">:</span> <span class="mi">6</span>
                     <span class="p">}</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">update</span><span class="p">(</span><span class="n">parameters</span><span class="p">)</span>
        <span class="c1">## select causal genes randomly, instead of the first 100 in enrichment analysis</span>
<span class="c1">#         self.ref_gene_name = load_reference_gene(parameters[&quot;refgene_file&quot;])[&quot;gene_name&quot;].tolist()</span>
<span class="c1">#         self.causal_genes = random.sample(self.ref_gene_name, parameters[&#39;n_causal_gene&#39;])</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">causal_genes</span> <span class="o">=</span> <span class="n">load_pthwy_gene</span><span class="p">(</span><span class="n">parameters</span><span class="p">[</span><span class="s2">&quot;pthwy_gene_file&quot;</span><span class="p">])[</span><span class="s2">&quot;gene_name&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">tolist</span><span class="p">()</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">seed</span> <span class="o">=</span> <span class="mi">12</span>
<span class="n">args</span> <span class="o">=</span> <span class="n">Environment</span><span class="p">()</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered">
<div class="prompt input_prompt">
</div>
<div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="CNV-data-processing">CNV data processing<a class="anchor-link" href="#CNV-data-processing">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[13]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="k">def</span> <span class="nf">count_cnv_by_block</span><span class="p">(</span><span class="n">df</span><span class="p">,</span> <span class="n">block_size</span><span class="p">):</span>
    <span class="c1"># check how many CNVs start in block_size = 100K blocks genome</span>
    <span class="c1"># this cell produces `block_counts`: ((chrom, block_position), count)</span>
    <span class="k">def</span> <span class="nf">count_by_blocks</span><span class="p">(</span><span class="n">chrom</span><span class="p">):</span>
        <span class="n">data</span> <span class="o">=</span> <span class="n">df</span><span class="o">.</span><span class="n">query</span><span class="p">(</span><span class="s1">&#39;chrom == &quot;{}&quot;&#39;</span><span class="o">.</span><span class="n">format</span><span class="p">(</span><span class="n">chrom</span><span class="p">))[</span><span class="s1">&#39;cnv_start&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">tolist</span><span class="p">()</span>
        <span class="n">start_pos</span> <span class="o">=</span> <span class="nb">min</span><span class="p">(</span><span class="n">data</span><span class="p">)</span>
        <span class="n">end_pos</span> <span class="o">=</span> <span class="nb">max</span><span class="p">(</span><span class="n">data</span><span class="p">)</span>
        <span class="n">counts</span><span class="p">,</span> <span class="n">bins</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">histogram</span><span class="p">(</span><span class="n">data</span><span class="p">,</span> <span class="n">bins</span> <span class="o">=</span> <span class="nb">int</span><span class="p">((</span><span class="n">end_pos</span> <span class="o">-</span> <span class="n">start_pos</span><span class="p">)</span> <span class="o">/</span> <span class="n">block_size</span><span class="p">)</span> <span class="o">+</span> <span class="mi">1</span><span class="p">,</span> 
                                    <span class="nb">range</span> <span class="o">=</span> <span class="p">(</span><span class="n">start_pos</span><span class="p">,</span> <span class="n">end_pos</span><span class="p">))</span>
        <span class="k">return</span> <span class="n">counts</span><span class="p">,</span> <span class="p">[</span><span class="nb">int</span><span class="p">(</span><span class="n">x</span><span class="p">)</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">bins</span><span class="p">]</span>
    <span class="n">block_counts</span> <span class="o">=</span> <span class="p">[]</span>
    <span class="k">for</span> <span class="n">chrom</span> <span class="ow">in</span> <span class="nb">set</span><span class="p">(</span><span class="n">df</span><span class="p">[</span><span class="s1">&#39;chrom&#39;</span><span class="p">]):</span>
        <span class="n">counts</span><span class="p">,</span> <span class="n">bins</span> <span class="o">=</span> <span class="n">count_by_blocks</span><span class="p">(</span><span class="n">chrom</span><span class="p">)</span>
        <span class="c1"># most blocks contain 0 CNV start. Add 0.5 to each block, so that CNV could start within any block.</span>
        <span class="n">block_counts</span><span class="o">.</span><span class="n">extend</span><span class="p">([((</span><span class="n">chrom</span><span class="p">,</span> <span class="n">x</span><span class="p">),</span> <span class="n">y</span><span class="o">+</span><span class="mf">0.5</span><span class="p">)</span> <span class="k">for</span> <span class="n">x</span><span class="p">,</span> <span class="n">y</span> <span class="ow">in</span> <span class="nb">zip</span><span class="p">(</span><span class="n">bins</span><span class="p">,</span> <span class="n">counts</span><span class="p">)])</span>
    <span class="k">return</span> <span class="n">block_counts</span>
<span class="k">def</span> <span class="nf">fit_truncated_gaussian_mix</span><span class="p">(</span><span class="n">x</span><span class="p">,</span> <span class="n">k</span> <span class="o">=</span> <span class="mi">10</span><span class="p">):</span>
    <span class="n">x</span> <span class="o">=</span> <span class="n">x</span><span class="o">.</span><span class="n">extend</span><span class="p">([</span><span class="o">-</span><span class="n">i</span> <span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="n">x</span><span class="p">])</span>
    <span class="n">clf</span> <span class="o">=</span> <span class="n">GaussianMixture</span><span class="p">(</span><span class="n">n_components</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span> <span class="n">covariance_type</span><span class="o">=</span><span class="s1">&#39;full&#39;</span><span class="p">)</span>
    <span class="n">clf</span><span class="o">.</span><span class="n">fit</span><span class="p">(</span><span class="n">x</span><span class="p">)</span>
    <span class="k">return</span> <span class="n">clf</span>
<span class="k">def</span> <span class="nf">sample_cnv_length</span><span class="p">(</span><span class="n">data</span><span class="p">,</span> <span class="n">mean_num_cnv</span><span class="p">):</span>
    <span class="k">return</span> <span class="n">np</span><span class="o">.</span><span class="n">random</span><span class="o">.</span><span class="n">choice</span><span class="p">(</span><span class="n">data</span><span class="p">,</span> <span class="n">np</span><span class="o">.</span><span class="n">random</span><span class="o">.</span><span class="n">poisson</span><span class="p">(</span><span class="n">mean_num_cnv</span><span class="p">))</span>
<span class="k">def</span> <span class="nf">get_sample_blocks</span><span class="p">(</span><span class="n">block_counts</span><span class="p">,</span> <span class="n">num_cnv</span><span class="p">):</span>
    <span class="sd">&#39;&#39;&#39;sample blocks from blocks across genome&#39;&#39;&#39;</span>
    <span class="n">probability_distribution</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">array</span><span class="p">([</span><span class="n">x</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">block_counts</span><span class="p">])</span>
    <span class="n">sample_idx</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">random</span><span class="o">.</span><span class="n">choice</span><span class="p">(</span><span class="nb">range</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">block_counts</span><span class="p">)),</span> <span class="n">num_cnv</span><span class="p">,</span> 
                                  <span class="n">p</span> <span class="o">=</span> <span class="n">probability_distribution</span> <span class="o">/</span> <span class="nb">sum</span><span class="p">(</span><span class="n">probability_distribution</span><span class="p">))</span>
    <span class="k">return</span> <span class="nb">sorted</span><span class="p">([</span><span class="n">block_counts</span><span class="p">[</span><span class="n">idx</span><span class="p">][</span><span class="mi">0</span><span class="p">]</span> <span class="k">for</span> <span class="n">idx</span> <span class="ow">in</span> <span class="n">sample_idx</span><span class="p">])</span>
<span class="k">def</span> <span class="nf">assign_cnv_to_sample</span><span class="p">(</span><span class="n">sample_blocks</span><span class="p">,</span> <span class="n">sample_len</span><span class="p">,</span> <span class="n">block_size</span><span class="p">):</span>
    <span class="n">samples</span> <span class="o">=</span> <span class="p">{</span><span class="s1">&#39;chrom&#39;</span><span class="p">:</span> <span class="p">[],</span> <span class="s1">&#39;cnv_start&#39;</span><span class="p">:</span> <span class="p">[],</span> <span class="s1">&#39;cnv_terminate&#39;</span><span class="p">:</span> <span class="p">[]}</span>
    <span class="k">for</span> <span class="n">x</span><span class="p">,</span> <span class="n">y</span> <span class="ow">in</span> <span class="nb">zip</span><span class="p">(</span><span class="n">sample_blocks</span><span class="p">,</span> <span class="n">sample_len</span><span class="p">):</span>
        <span class="n">start_pos</span> <span class="o">=</span> <span class="n">randint</span><span class="p">(</span><span class="n">x</span><span class="p">[</span><span class="mi">1</span><span class="p">],</span> <span class="n">x</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span> <span class="o">+</span> <span class="n">block_size</span><span class="p">)</span>
        <span class="n">samples</span><span class="p">[</span><span class="s1">&#39;cnv_start&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">start_pos</span><span class="p">)</span>
        <span class="n">samples</span><span class="p">[</span><span class="s1">&#39;cnv_terminate&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">start_pos</span> <span class="o">+</span> <span class="nb">int</span><span class="p">(</span><span class="n">y</span><span class="p">))</span>
        <span class="n">samples</span><span class="p">[</span><span class="s1">&#39;chrom&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">x</span><span class="p">[</span><span class="mi">0</span><span class="p">])</span>
    <span class="k">return</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">(</span><span class="n">samples</span><span class="p">)</span>
<span class="k">def</span> <span class="nf">annotate_samples</span><span class="p">(</span><span class="n">samples</span><span class="p">,</span> <span class="n">gene_df</span><span class="p">):</span>
    <span class="n">query</span> <span class="o">=</span> <span class="s2">&quot;&quot;&quot;</span>
<span class="s2">        SELECT cnv.chrom, cnv.cnv_start, cnv.cnv_terminate, gene.tx_name, gene.gene_name</span>
<span class="s2">        FROM samples cnv LEFT JOIN gene_df gene</span>
<span class="s2">        WHERE cnv.chrom == gene.chrom </span>
<span class="s2">        AND (</span>
<span class="s2">        (cnv.cnv_start &gt;= gene.tx_start AND cnv.cnv_start &lt;= gene.tx_end)</span>
<span class="s2">        OR</span>
<span class="s2">        (cnv.cnv_terminate &gt;= gene.tx_start AND cnv.cnv_terminate &lt;= gene.tx_end)</span>
<span class="s2">        OR</span>
<span class="s2">        (cnv.cnv_start &lt;= gene.tx_start AND cnv.cnv_terminate &gt;= gene.tx_end)</span>
<span class="s2">        )</span>
<span class="s2">        &quot;&quot;&quot;</span>
        <span class="c1"># drop_duplicates(): make sure the case that CNV spread multiple txs but each gene to be counted only once</span>
    <span class="k">return</span> <span class="n">sqldf</span><span class="p">(</span><span class="n">query</span><span class="p">)</span><span class="o">.</span><span class="n">drop_duplicates</span><span class="p">(</span><span class="n">subset</span><span class="o">=</span><span class="p">(</span><span class="s2">&quot;chrom&quot;</span><span class="p">,</span> <span class="s2">&quot;cnv_start&quot;</span><span class="p">,</span> <span class="s2">&quot;cnv_terminate&quot;</span><span class="p">,</span> <span class="s2">&quot;gene_name&quot;</span><span class="p">))</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered">
<div class="prompt input_prompt">
</div>
<div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Simulation-related-codes">Simulation related codes<a class="anchor-link" href="#Simulation-related-codes">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[14]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="k">def</span> <span class="nf">get_causal_genes</span><span class="p">(</span><span class="n">causal_genes</span><span class="p">,</span> <span class="n">sample_genes</span><span class="p">):</span>
    <span class="sd">&#39;&#39;&#39;get causal genes for each simulated sample&#39;&#39;&#39;</span>
    <span class="k">return</span> <span class="p">[</span><span class="n">x</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">causal_genes</span> <span class="k">if</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">sample_genes</span><span class="p">]</span>
<span class="k">def</span> <span class="nf">get_ccnv</span><span class="p">():</span>
    <span class="sd">&#39;&#39;&#39;get causal cnvs&#39;&#39;&#39;</span>
    <span class="k">return</span> <span class="bp">None</span>
<span class="k">def</span> <span class="nf">p_case</span><span class="p">(</span><span class="n">p</span><span class="p">,</span> <span class="n">num_causal_genes_in_sample</span><span class="p">,</span> <span class="n">shape</span><span class="p">,</span> <span class="n">scale</span><span class="p">):</span>
    <span class="k">if</span> <span class="n">num_causal_genes_in_sample</span> <span class="o">==</span> <span class="mi">0</span><span class="p">:</span>
        <span class="k">return</span> <span class="n">p</span>
    <span class="n">baseline_odds</span> <span class="o">=</span> <span class="n">p</span> <span class="o">/</span> <span class="p">(</span><span class="mi">1</span> <span class="o">-</span> <span class="n">p</span><span class="p">)</span>
    <span class="k">if</span> <span class="n">shape</span> <span class="ow">is</span> <span class="bp">None</span> <span class="ow">or</span> <span class="n">scale</span> <span class="ow">is</span> <span class="bp">None</span><span class="p">:</span>
        <span class="n">odds_ratio</span> <span class="o">=</span> <span class="mi">1</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="n">odds_ratio</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">prod</span><span class="p">([</span><span class="n">np</span><span class="o">.</span><span class="n">random</span><span class="o">.</span><span class="n">gamma</span><span class="p">(</span><span class="n">shape</span><span class="p">,</span> <span class="n">scale</span><span class="p">)</span> 
                              <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="n">num_causal_genes_in_sample</span><span class="p">)])</span>
    <span class="c1"># obtain the power of fisher test by setting odds ratio to 1</span>
    <span class="c1"># odds_ratio = 1</span>
    <span class="n">odds</span> <span class="o">=</span> <span class="n">baseline_odds</span> <span class="o">*</span> <span class="n">odds_ratio</span>
    <span class="k">return</span> <span class="n">odds</span> <span class="o">/</span> <span class="p">(</span><span class="mi">1</span> <span class="o">+</span> <span class="n">odds</span><span class="p">)</span>
<span class="k">def</span> <span class="nf">simulate_core</span><span class="p">(</span><span class="n">case_data</span><span class="p">,</span> <span class="n">ctrl_data</span><span class="p">,</span> <span class="n">debug</span><span class="p">,</span> <span class="n">args</span><span class="p">,</span> <span class="n">cnv_length</span><span class="p">,</span> <span class="n">block_counts</span><span class="p">,</span> <span class="n">refgene</span><span class="p">,</span> <span class="n">N1</span><span class="p">,</span> <span class="n">N2</span><span class="p">):</span>
    <span class="n">cnt1</span> <span class="o">=</span> <span class="mi">0</span>
    <span class="n">cnt2</span> <span class="o">=</span> <span class="mi">0</span>
    <span class="n">status</span> <span class="o">=</span> <span class="mi">1</span>
    <span class="k">while</span><span class="p">(</span><span class="n">status</span><span class="p">):</span>
        <span class="n">sample_len</span> <span class="o">=</span> <span class="n">sample_cnv_length</span><span class="p">(</span><span class="n">cnv_length</span><span class="p">,</span> <span class="n">args</span><span class="p">[</span><span class="s2">&quot;avg_cnv_per_individual&quot;</span><span class="p">])</span>
        <span class="n">sample_blocks</span> <span class="o">=</span> <span class="n">get_sample_blocks</span><span class="p">(</span><span class="n">block_counts</span><span class="p">,</span> <span class="nb">len</span><span class="p">(</span><span class="n">sample_len</span><span class="p">))</span>
        <span class="n">samples</span> <span class="o">=</span> <span class="n">assign_cnv_to_sample</span><span class="p">(</span><span class="n">sample_blocks</span><span class="p">,</span> <span class="n">sample_len</span><span class="p">,</span> <span class="n">args</span><span class="p">[</span><span class="s2">&quot;block_size&quot;</span><span class="p">])</span>
        <span class="n">samples</span> <span class="o">=</span> <span class="n">annotate_samples</span><span class="p">(</span><span class="n">samples</span><span class="p">,</span> <span class="n">refgene</span><span class="p">)</span>
        <span class="n">causal_genes_in_sample</span> <span class="o">=</span> <span class="n">get_causal_genes</span><span class="p">(</span><span class="n">args</span><span class="o">.</span><span class="n">causal_genes</span><span class="p">,</span> <span class="n">samples</span><span class="p">[</span><span class="s1">&#39;gene_name&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">tolist</span><span class="p">())</span>
        <span class="n">p</span> <span class="o">=</span> <span class="n">p_case</span><span class="p">(</span><span class="n">args</span><span class="p">[</span><span class="s2">&quot;prevalence&quot;</span><span class="p">],</span> <span class="nb">len</span><span class="p">(</span><span class="n">causal_genes_in_sample</span><span class="p">),</span> 
                   <span class="n">args</span><span class="p">[</span><span class="s2">&quot;odds_ratio_params&quot;</span><span class="p">][</span><span class="s2">&quot;shape&quot;</span><span class="p">],</span> <span class="n">args</span><span class="p">[</span><span class="s2">&quot;odds_ratio_params&quot;</span><span class="p">][</span><span class="s2">&quot;scale&quot;</span><span class="p">])</span>
        <span class="k">if</span> <span class="n">random</span><span class="o">.</span><span class="n">random</span><span class="p">()</span> <span class="o">&lt;</span> <span class="n">p</span> <span class="ow">and</span> <span class="n">cnt1</span> <span class="o">&lt;</span> <span class="n">N1</span><span class="p">:</span>
            <span class="c1"># sample data is a case</span>
            <span class="n">case_data</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">samples</span><span class="p">)</span>
            <span class="n">debug_data</span> <span class="o">=</span> <span class="nb">dict</span><span class="p">(</span><span class="n">debug</span><span class="p">)</span>
            <span class="n">debug_data</span><span class="p">[</span><span class="s1">&#39;p in case&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">p</span><span class="p">)</span>
            <span class="n">debug_data</span><span class="p">[</span><span class="s1">&#39;number of causal genes in case&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">causal_genes_in_sample</span><span class="p">))</span>
            <span class="n">debug_data</span><span class="p">[</span><span class="s1">&#39;simulated CNV length in case&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">extend</span><span class="p">(</span><span class="n">sample_len</span><span class="p">)</span>
            <span class="n">debug_data</span><span class="p">[</span><span class="s1">&#39;number of genes overlap CNV in case&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="nb">set</span><span class="p">(</span><span class="n">samples</span><span class="p">[</span><span class="s1">&#39;gene_name&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">tolist</span><span class="p">())))</span>
            <span class="n">debug</span><span class="o">.</span><span class="n">update</span><span class="p">(</span><span class="n">debug_data</span><span class="p">)</span>
            <span class="n">cnt1</span> <span class="o">+=</span> <span class="mi">1</span>
        <span class="k">if</span> <span class="n">random</span><span class="o">.</span><span class="n">random</span><span class="p">()</span> <span class="o">&gt;</span> <span class="n">p</span> <span class="ow">and</span> <span class="n">cnt2</span> <span class="o">&lt;</span> <span class="n">N2</span><span class="p">:</span>
            <span class="c1"># sample data is a control</span>
            <span class="n">ctrl_data</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">samples</span><span class="p">)</span>
            <span class="n">debug_data</span> <span class="o">=</span> <span class="nb">dict</span><span class="p">(</span><span class="n">debug</span><span class="p">)</span>
            <span class="n">debug_data</span><span class="p">[</span><span class="s1">&#39;p in ctrl&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">p</span><span class="p">)</span>
            <span class="n">debug_data</span><span class="p">[</span><span class="s1">&#39;number of causal genes in ctrl&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">causal_genes_in_sample</span><span class="p">))</span>
            <span class="n">debug_data</span><span class="p">[</span><span class="s1">&#39;simulated CNV length in ctrl&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">extend</span><span class="p">(</span><span class="n">sample_len</span><span class="p">)</span>
            <span class="n">debug_data</span><span class="p">[</span><span class="s1">&#39;number of genes overlap CNV in ctrl&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="nb">set</span><span class="p">(</span><span class="n">samples</span><span class="p">[</span><span class="s1">&#39;gene_name&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">tolist</span><span class="p">())))</span>
            <span class="n">debug</span><span class="o">.</span><span class="n">update</span><span class="p">(</span><span class="n">debug_data</span><span class="p">)</span>
            <span class="n">cnt2</span> <span class="o">+=</span> <span class="mi">1</span>
        <span class="k">if</span> <span class="n">cnt1</span> <span class="o">==</span> <span class="n">N1</span> <span class="ow">and</span> <span class="n">cnt2</span> <span class="o">==</span> <span class="n">N2</span><span class="p">:</span>
            <span class="n">status</span> <span class="o">=</span> <span class="mi">0</span>
    <span class="k">return</span> <span class="n">status</span>
<span class="k">def</span> <span class="nf">simulate</span><span class="p">(</span><span class="n">refgene</span><span class="p">,</span> <span class="n">cnv_data</span><span class="p">,</span> <span class="n">args</span><span class="p">):</span>
    <span class="kn">from</span> <span class="nn">multiprocessing</span> <span class="kn">import</span> <span class="n">Pool</span><span class="p">,</span> <span class="n">Manager</span>
    <span class="n">manager</span> <span class="o">=</span> <span class="n">Manager</span><span class="p">()</span>
    <span class="n">np</span><span class="o">.</span><span class="n">random</span><span class="o">.</span><span class="n">seed</span><span class="p">(</span><span class="n">args</span><span class="o">.</span><span class="n">seed</span><span class="p">)</span>
    <span class="n">df</span> <span class="o">=</span> <span class="n">cnv_data</span><span class="o">.</span><span class="n">drop_duplicates</span><span class="p">(</span><span class="n">subset</span><span class="o">=</span><span class="p">(</span><span class="s2">&quot;chrom&quot;</span><span class="p">,</span> <span class="s2">&quot;cnv_start&quot;</span><span class="p">,</span> <span class="s2">&quot;cnv_end&quot;</span><span class="p">))</span>
    <span class="n">block_counts</span> <span class="o">=</span> <span class="n">count_cnv_by_block</span><span class="p">(</span><span class="n">df</span><span class="p">,</span> <span class="n">args</span><span class="p">[</span><span class="s1">&#39;block_size&#39;</span><span class="p">])</span>
    <span class="n">cnv_length</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="s1">&#39;cnv_end&#39;</span><span class="p">]</span> <span class="o">-</span> <span class="n">df</span><span class="p">[</span><span class="s1">&#39;cnv_start&#39;</span><span class="p">]</span>
    <span class="n">debug</span> <span class="o">=</span> <span class="p">{</span><span class="s1">&#39;p in case&#39;</span><span class="p">:</span> <span class="p">[],</span> <span class="s1">&#39;p in ctrl&#39;</span><span class="p">:</span> <span class="p">[],</span> <span class="s1">&#39;args&#39;</span><span class="p">:</span> <span class="nb">dict</span><span class="p">(</span><span class="n">args</span><span class="p">),</span> <span class="s1">&#39;time&#39;</span><span class="p">:</span> <span class="p">[</span><span class="nb">str</span><span class="p">(</span><span class="n">datetime</span><span class="o">.</span><span class="n">now</span><span class="p">()),</span> <span class="bp">None</span><span class="p">],</span>
             <span class="s1">&#39;causal genes&#39;</span><span class="p">:</span> <span class="n">args</span><span class="o">.</span><span class="n">causal_genes</span><span class="p">,</span> <span class="s1">&#39;number of genes overlap CNV in case&#39;</span><span class="p">:</span> <span class="p">[],</span> 
             <span class="s1">&#39;number of genes overlap CNV in ctrl&#39;</span><span class="p">:</span> <span class="p">[],</span>
             <span class="s1">&#39;number of causal genes in case&#39;</span><span class="p">:</span> <span class="p">[],</span> <span class="s1">&#39;number of causal genes in ctrl&#39;</span><span class="p">:</span> <span class="p">[],</span> 
             <span class="s1">&#39;simulated CNV length in case&#39;</span><span class="p">:</span> <span class="p">[],</span> <span class="s1">&#39;simulated CNV length in ctrl&#39;</span><span class="p">:</span> <span class="p">[],</span> <span class="s1">&#39;seed&#39;</span><span class="p">:</span> <span class="n">args</span><span class="o">.</span><span class="n">seed</span><span class="p">}</span>
    <span class="n">manager</span> <span class="o">=</span> <span class="n">Manager</span><span class="p">()</span>
    <span class="n">case_data</span> <span class="o">=</span> <span class="n">manager</span><span class="o">.</span><span class="n">list</span><span class="p">()</span>
    <span class="n">ctrl_data</span> <span class="o">=</span> <span class="n">manager</span><span class="o">.</span><span class="n">list</span><span class="p">()</span>
    <span class="n">debug</span> <span class="o">=</span> <span class="n">manager</span><span class="o">.</span><span class="n">dict</span><span class="p">(</span><span class="n">debug</span><span class="p">)</span>
    <span class="n">pool</span> <span class="o">=</span> <span class="n">Pool</span><span class="p">(</span><span class="n">args</span><span class="p">[</span><span class="s1">&#39;ncpu&#39;</span><span class="p">])</span>
    <span class="n">N1</span> <span class="o">=</span> <span class="p">[</span><span class="mi">1</span><span class="p">]</span> <span class="o">*</span> <span class="n">args</span><span class="p">[</span><span class="s1">&#39;n_case&#39;</span><span class="p">]</span> <span class="o">+</span> <span class="p">[</span><span class="mi">0</span><span class="p">]</span> <span class="o">*</span> <span class="n">args</span><span class="p">[</span><span class="s1">&#39;n_ctrl&#39;</span><span class="p">]</span>
    <span class="n">N2</span> <span class="o">=</span> <span class="p">[</span><span class="mi">0</span><span class="p">]</span> <span class="o">*</span> <span class="n">args</span><span class="p">[</span><span class="s1">&#39;n_case&#39;</span><span class="p">]</span> <span class="o">+</span> <span class="p">[</span><span class="mi">1</span><span class="p">]</span> <span class="o">*</span> <span class="n">args</span><span class="p">[</span><span class="s1">&#39;n_ctrl&#39;</span><span class="p">]</span>
    <span class="c1"># elog saves exceptions</span>
    <span class="n">elog</span> <span class="o">=</span> <span class="p">[</span><span class="n">pool</span><span class="o">.</span><span class="n">apply_async</span><span class="p">(</span><span class="n">simulate_core</span><span class="p">,</span> <span class="n">args</span> <span class="o">=</span> <span class="p">(</span><span class="n">case_data</span><span class="p">,</span> <span class="n">ctrl_data</span><span class="p">,</span> <span class="n">debug</span><span class="p">,</span> <span class="n">args</span><span class="p">,</span>
                                                    <span class="n">cnv_length</span><span class="p">,</span> <span class="n">block_counts</span><span class="p">,</span> <span class="n">refgene</span><span class="p">,</span> <span class="n">x</span><span class="p">,</span> <span class="n">y</span><span class="p">))</span> 
            <span class="k">for</span> <span class="n">x</span><span class="p">,</span> <span class="n">y</span> <span class="ow">in</span> <span class="nb">zip</span><span class="p">(</span><span class="n">N1</span><span class="p">,</span> <span class="n">N2</span><span class="p">)]</span>
    <span class="n">pool</span><span class="o">.</span><span class="n">close</span><span class="p">()</span>
    <span class="n">pool</span><span class="o">.</span><span class="n">join</span><span class="p">()</span>
    <span class="k">for</span> <span class="n">item</span> <span class="ow">in</span> <span class="n">elog</span><span class="p">:</span>
        <span class="n">item</span><span class="o">.</span><span class="n">get</span><span class="p">()</span>
    <span class="n">debug</span> <span class="o">=</span> <span class="nb">dict</span><span class="p">(</span><span class="n">debug</span><span class="p">)</span>
    <span class="n">debug</span><span class="p">[</span><span class="s1">&#39;time&#39;</span><span class="p">][</span><span class="mi">1</span><span class="p">]</span> <span class="o">=</span> <span class="nb">str</span><span class="p">(</span><span class="n">datetime</span><span class="o">.</span><span class="n">now</span><span class="p">())</span>
    <span class="k">return</span> <span class="p">{</span><span class="s1">&#39;case&#39;</span><span class="p">:</span> <span class="nb">list</span><span class="p">(</span><span class="n">case_data</span><span class="p">),</span> <span class="s1">&#39;ctrl&#39;</span><span class="p">:</span> <span class="nb">list</span><span class="p">(</span><span class="n">ctrl_data</span><span class="p">),</span> <span class="s1">&#39;debug&#39;</span><span class="p">:</span> <span class="n">debug</span><span class="p">}</span>
<span class="k">def</span> <span class="nf">run_simulation</span><span class="p">(</span><span class="n">args</span><span class="p">,</span> <span class="n">simulation_id</span> <span class="o">=</span> <span class="bp">None</span><span class="p">):</span>
    <span class="n">ref_gene</span> <span class="o">=</span> <span class="n">load_reference_gene</span><span class="p">(</span><span class="n">args</span><span class="p">[</span><span class="s1">&#39;refgene_file&#39;</span><span class="p">])</span>
    <span class="n">cnv_data</span> <span class="o">=</span> <span class="n">load_cnv_data</span><span class="p">(</span><span class="n">args</span><span class="p">[</span><span class="s1">&#39;cnv_file&#39;</span><span class="p">])</span>
    <span class="n">sample_data</span> <span class="o">=</span> <span class="n">simulate</span><span class="p">(</span><span class="n">ref_gene</span><span class="p">,</span> <span class="n">pd</span><span class="o">.</span><span class="n">concat</span><span class="p">([</span><span class="n">cnv_data</span><span class="p">[</span><span class="n">args</span><span class="p">[</span><span class="s1">&#39;case_dataset&#39;</span><span class="p">]],</span> <span class="n">cnv_data</span><span class="p">[</span><span class="n">args</span><span class="p">[</span><span class="s1">&#39;ctrl_dataset&#39;</span><span class="p">]]]),</span>
                           <span class="n">args</span><span class="p">)</span>
    <span class="n">save_data</span><span class="p">(</span><span class="n">sample_data</span><span class="p">,</span> <span class="s1">&#39;{}{}.data.pkl&#39;</span><span class="o">.</span>\
              <span class="n">format</span><span class="p">(</span><span class="n">args</span><span class="p">[</span><span class="s1">&#39;output&#39;</span><span class="p">],</span> <span class="s1">&#39;_{}&#39;</span><span class="o">.</span><span class="n">format</span><span class="p">(</span><span class="n">simulation_id</span><span class="p">)</span> <span class="k">if</span> <span class="n">simulation_id</span> <span class="ow">is</span> <span class="ow">not</span> <span class="bp">None</span> <span class="k">else</span> <span class="s1">&#39;&#39;</span><span class="p">))</span>
    <span class="k">return</span> <span class="n">sample_data</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered">
<div class="prompt input_prompt">
</div>
<div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Codes-for-data-analysis">Codes for data analysis<a class="anchor-link" href="#Codes-for-data-analysis">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[15]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="k">def</span> <span class="nf">get_analysis_blocks</span><span class="p">(</span><span class="n">data</span><span class="p">):</span>
    <span class="sd">&#39;&#39;&#39;Determine from `data` table (pd.DataFrame) independent genomic blocks.</span>
<span class="sd">       For simulated samples use `pd.concat([pd.concat(samples[&#39;case&#39;]), pd.concat(samples[&#39;ctrl&#39;])])`</span>
<span class="sd">       as input&#39;&#39;&#39;</span>
    <span class="kn">from</span> <span class="nn">scipy.sparse.csgraph</span> <span class="kn">import</span> <span class="n">connected_components</span>
    <span class="c1">#</span>
    <span class="k">def</span> <span class="nf">reductionFunction</span><span class="p">(</span><span class="n">df</span><span class="p">):</span>
        <span class="c1"># create a 2D graph of connectivity between date ranges</span>
        <span class="n">start</span> <span class="o">=</span> <span class="n">df</span><span class="o">.</span><span class="n">cnv_start</span><span class="o">.</span><span class="n">values</span>
        <span class="n">end</span> <span class="o">=</span> <span class="n">df</span><span class="o">.</span><span class="n">cnv_terminate</span><span class="o">.</span><span class="n">values</span>
        <span class="n">graph</span> <span class="o">=</span> <span class="p">(</span><span class="n">start</span> <span class="o">&lt;=</span> <span class="n">end</span><span class="p">[:,</span> <span class="bp">None</span><span class="p">])</span> <span class="o">&amp;</span> <span class="p">(</span><span class="n">end</span> <span class="o">&gt;=</span> <span class="n">start</span><span class="p">[:,</span> <span class="bp">None</span><span class="p">])</span>
        <span class="n">n_components</span><span class="p">,</span> <span class="n">indices</span> <span class="o">=</span> <span class="n">connected_components</span><span class="p">(</span><span class="n">graph</span><span class="p">)</span>
        <span class="k">return</span> <span class="n">df</span><span class="o">.</span><span class="n">groupby</span><span class="p">(</span><span class="n">indices</span><span class="p">)</span><span class="o">.</span><span class="n">aggregate</span><span class="p">({</span><span class="s1">&#39;cnv_start&#39;</span><span class="p">:</span> <span class="s1">&#39;min&#39;</span><span class="p">,</span>
                                              <span class="s1">&#39;cnv_terminate&#39;</span><span class="p">:</span> <span class="s1">&#39;max&#39;</span><span class="p">,</span>
                                              <span class="s1">&#39;tx_name&#39;</span><span class="p">:</span> <span class="k">lambda</span> <span class="n">x</span><span class="p">:</span> <span class="s1">&#39;,&#39;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="nb">sorted</span><span class="p">(</span><span class="nb">set</span><span class="p">(</span><span class="n">x</span><span class="p">))),</span>
                                              <span class="s1">&#39;gene_name&#39;</span><span class="p">:</span> <span class="k">lambda</span> <span class="n">x</span><span class="p">:</span> <span class="s1">&#39;,&#39;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="nb">sorted</span><span class="p">(</span><span class="nb">set</span><span class="p">(</span><span class="n">x</span><span class="p">)))})</span>
    <span class="c1">#</span>
    <span class="k">return</span> <span class="p">[</span><span class="n">x</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s1">&#39;,&#39;</span><span class="p">)</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">data</span><span class="o">.</span><span class="n">groupby</span><span class="p">([</span><span class="s1">&#39;chrom&#39;</span><span class="p">])</span><span class="o">.</span><span class="n">apply</span><span class="p">(</span><span class="n">reductionFunction</span><span class="p">)</span><span class="o">.</span><span class="n">reset_index</span><span class="p">(</span><span class="s1">&#39;chrom&#39;</span><span class="p">)[</span><span class="s1">&#39;gene_name&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">tolist</span><span class="p">()]</span>
<span class="k">def</span> <span class="nf">get_gene_table</span><span class="p">(</span><span class="n">gene_df</span><span class="p">):</span>
    <span class="n">gene_table</span> <span class="o">=</span> <span class="p">{}</span>
    <span class="k">for</span> <span class="n">item</span> <span class="ow">in</span> <span class="p">[</span><span class="s2">&quot;case&quot;</span><span class="p">,</span> <span class="s2">&quot;ctrl&quot;</span><span class="p">]:</span>
        <span class="n">gene</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">concat</span><span class="p">(</span><span class="n">gene_df</span><span class="p">[</span><span class="n">item</span><span class="p">])</span>
        <span class="n">query</span> <span class="o">=</span> <span class="s1">&#39;&#39;&#39;</span>
<span class="s1">        SELECT chrom, gene_name, count(gene_name)</span>
<span class="s1">        FROM gene</span>
<span class="s1">        GROUP BY chrom, gene_name</span>
<span class="s1">        ORDER BY count(gene_name) DESC</span>
<span class="s1">        &#39;&#39;&#39;</span>
        <span class="n">gene_table</span><span class="p">[</span><span class="n">item</span><span class="p">]</span> <span class="o">=</span> <span class="n">sqldf</span><span class="p">(</span><span class="n">query</span><span class="p">)</span>
    <span class="n">gene_table</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">merge</span><span class="p">(</span><span class="n">gene_table</span><span class="p">[</span><span class="s2">&quot;case&quot;</span><span class="p">],</span> <span class="n">gene_table</span><span class="p">[</span><span class="s2">&quot;ctrl&quot;</span><span class="p">],</span> <span class="n">how</span> <span class="o">=</span> <span class="s2">&quot;outer&quot;</span><span class="p">,</span> <span class="n">on</span> <span class="o">=</span> <span class="p">[</span><span class="s2">&quot;chrom&quot;</span><span class="p">,</span> <span class="s2">&quot;gene_name&quot;</span><span class="p">])</span>
    <span class="n">gene_table</span><span class="p">[</span><span class="s2">&quot;count(gene_name)_x&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">fillna</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span> <span class="n">inplace</span><span class="o">=</span><span class="bp">True</span><span class="p">)</span>
    <span class="n">gene_table</span><span class="p">[</span><span class="s2">&quot;count(gene_name)_y&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">fillna</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span> <span class="n">inplace</span><span class="o">=</span><span class="bp">True</span><span class="p">)</span>
    <span class="n">gene_table</span> <span class="o">=</span> <span class="n">gene_table</span><span class="o">.</span><span class="n">rename</span><span class="p">(</span><span class="n">columns</span><span class="o">=</span><span class="p">{</span><span class="s2">&quot;count(gene_name)_x&quot;</span><span class="p">:</span><span class="s2">&quot;n_case_gene&quot;</span><span class="p">,</span> <span class="s2">&quot;count(gene_name)_y&quot;</span><span class="p">:</span><span class="s2">&quot;n_ctrl_gene&quot;</span><span class="p">})</span>
    <span class="n">n_gene_case</span> <span class="o">=</span> <span class="nb">sum</span><span class="p">(</span><span class="n">gene_table</span><span class="p">[</span><span class="s2">&quot;n_case_gene&quot;</span><span class="p">])</span>
    <span class="n">n_gene_ctrl</span> <span class="o">=</span> <span class="nb">sum</span><span class="p">(</span><span class="n">gene_table</span><span class="p">[</span><span class="s2">&quot;n_ctrl_gene&quot;</span><span class="p">])</span>
    <span class="n">gene_table</span><span class="p">[</span><span class="s2">&quot;n_case_nogene&quot;</span><span class="p">]</span> <span class="o">=</span> <span class="n">n_gene_case</span> <span class="o">-</span> <span class="n">gene_table</span><span class="p">[</span><span class="s2">&quot;n_case_gene&quot;</span><span class="p">]</span>
    <span class="n">gene_table</span><span class="p">[</span><span class="s2">&quot;n_ctrl_nogene&quot;</span><span class="p">]</span> <span class="o">=</span> <span class="n">n_gene_ctrl</span> <span class="o">-</span> <span class="n">gene_table</span><span class="p">[</span><span class="s2">&quot;n_ctrl_gene&quot;</span><span class="p">]</span>
    <span class="n">gene_table</span> <span class="o">=</span> <span class="n">gene_table</span><span class="p">[[</span><span class="s2">&quot;gene_name&quot;</span><span class="p">,</span> <span class="s2">&quot;n_case_gene&quot;</span><span class="p">,</span> <span class="s2">&quot;n_ctrl_gene&quot;</span><span class="p">,</span> <span class="s2">&quot;n_case_nogene&quot;</span><span class="p">,</span> <span class="s2">&quot;n_ctrl_nogene&quot;</span><span class="p">]]</span>
    <span class="k">return</span> <span class="n">gene_table</span>
<span class="k">def</span> <span class="nf">get_stats</span><span class="p">(</span><span class="n">gene_table</span><span class="p">,</span> <span class="n">sort</span> <span class="o">=</span> <span class="mi">0</span><span class="p">):</span>
    <span class="c1"># from website https://pypi.python.org/pypi/fisher/</span>
    <span class="n">stats_table</span> <span class="o">=</span> <span class="p">[(</span><span class="n">pvalue</span><span class="p">(</span><span class="n">row</span><span class="p">[</span><span class="s2">&quot;n_case_gene&quot;</span><span class="p">],</span> <span class="n">row</span><span class="p">[</span><span class="s2">&quot;n_ctrl_gene&quot;</span><span class="p">],</span> <span class="n">row</span><span class="p">[</span><span class="s2">&quot;n_case_nogene&quot;</span><span class="p">],</span> <span class="n">row</span><span class="p">[</span><span class="s2">&quot;n_ctrl_nogene&quot;</span><span class="p">]),</span> <span class="n">row</span><span class="p">[</span><span class="s2">&quot;gene_name&quot;</span><span class="p">])</span> 
                   <span class="k">for</span> <span class="n">idx</span><span class="p">,</span> <span class="n">row</span> <span class="ow">in</span> <span class="n">gene_table</span><span class="o">.</span><span class="n">iterrows</span><span class="p">()]</span>
    <span class="n">p_value</span> <span class="o">=</span> <span class="p">[</span><span class="n">x</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">two_tail</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">stats_table</span><span class="p">]</span>
    <span class="n">oddsratio_table</span> <span class="o">=</span> <span class="p">[(</span><span class="n">stats</span><span class="o">.</span><span class="n">fisher_exact</span><span class="p">([[</span><span class="n">row</span><span class="p">[</span><span class="s2">&quot;n_case_gene&quot;</span><span class="p">],</span> <span class="n">row</span><span class="p">[</span><span class="s2">&quot;n_ctrl_gene&quot;</span><span class="p">]],</span> <span class="p">[</span><span class="n">row</span><span class="p">[</span><span class="s2">&quot;n_case_nogene&quot;</span><span class="p">],</span> <span class="n">row</span><span class="p">[</span><span class="s2">&quot;n_ctrl_nogene&quot;</span><span class="p">]]])[</span><span class="mi">0</span><span class="p">],</span> <span class="n">row</span><span class="p">[</span><span class="s2">&quot;gene_name&quot;</span><span class="p">])</span> 
                       <span class="k">for</span> <span class="n">idx</span><span class="p">,</span> <span class="n">row</span> <span class="ow">in</span> <span class="n">gene_table</span><span class="o">.</span><span class="n">iterrows</span><span class="p">()]</span>
    <span class="k">if</span> <span class="ow">not</span> <span class="n">sort</span> <span class="o">==</span> <span class="mi">0</span><span class="p">:</span>
        <span class="n">stats_table</span> <span class="o">=</span> <span class="nb">sorted</span><span class="p">(</span><span class="n">stats_table</span><span class="p">,</span> <span class="n">reverse</span><span class="o">=</span><span class="bp">True</span><span class="p">,</span> <span class="n">key</span> <span class="o">=</span> <span class="k">lambda</span> <span class="n">x</span><span class="p">:</span> <span class="o">-</span><span class="n">np</span><span class="o">.</span><span class="n">log10</span><span class="p">(</span><span class="n">x</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">two_tail</span><span class="p">))</span>
        <span class="n">oddsratio_table</span> <span class="o">=</span> <span class="nb">sorted</span><span class="p">(</span><span class="n">oddsratio_table</span><span class="p">,</span> <span class="n">reverse</span><span class="o">=</span><span class="bp">True</span><span class="p">,</span> <span class="n">key</span><span class="o">=</span><span class="k">lambda</span> <span class="n">x</span><span class="p">:</span> <span class="n">x</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span> <span class="k">if</span> <span class="n">np</span><span class="o">.</span><span class="n">isfinite</span><span class="p">(</span><span class="n">x</span><span class="p">[</span><span class="mi">0</span><span class="p">])</span> <span class="k">else</span> <span class="o">-</span><span class="n">x</span><span class="p">[</span><span class="mi">0</span><span class="p">])</span>
    <span class="n">logp_2side</span> <span class="o">=</span> <span class="p">[</span><span class="o">-</span><span class="n">np</span><span class="o">.</span><span class="n">log10</span><span class="p">(</span><span class="n">x</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">two_tail</span><span class="p">)</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">stats_table</span><span class="p">]</span>
    <span class="n">logp_gene</span> <span class="o">=</span> <span class="p">[</span><span class="n">x</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">stats_table</span><span class="p">]</span>
    <span class="n">OR_2side</span> <span class="o">=</span> <span class="p">[</span><span class="n">x</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">oddsratio_table</span><span class="p">]</span>
    <span class="n">OR_gene</span> <span class="o">=</span> <span class="p">[</span><span class="n">x</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">oddsratio_table</span><span class="p">]</span>
    <span class="n">stats_table</span> <span class="o">=</span> <span class="p">{</span><span class="s2">&quot;p_value&quot;</span><span class="p">:</span> <span class="n">p_value</span><span class="p">,</span> <span class="s2">&quot;logp_2side&quot;</span><span class="p">:</span> <span class="n">logp_2side</span><span class="p">,</span> <span class="s2">&quot;logp_gene&quot;</span><span class="p">:</span> <span class="n">logp_gene</span><span class="p">,</span> 
                   <span class="s2">&quot;OR_2side&quot;</span><span class="p">:</span> <span class="n">OR_2side</span><span class="p">,</span> <span class="s2">&quot;OR_gene&quot;</span><span class="p">:</span> <span class="n">OR_gene</span><span class="p">}</span>
    <span class="k">return</span> <span class="n">stats_table</span>
<span class="k">def</span> <span class="nf">get_stats_from_input</span><span class="p">(</span><span class="n">input_data</span><span class="p">,</span> <span class="n">sort_data</span> <span class="o">=</span> <span class="mi">0</span><span class="p">):</span>
    <span class="sd">&#39;&#39;&#39;input data saved from run_simulate step: sample_dup and sample_del separately&#39;&#39;&#39;</span>
    <span class="n">input_data</span> <span class="o">=</span> <span class="n">load_data</span><span class="p">(</span><span class="n">input_data</span><span class="p">)</span>
    <span class="n">sample_gene_table</span> <span class="o">=</span> <span class="n">get_gene_table</span><span class="p">(</span><span class="n">input_data</span><span class="p">)</span>
    <span class="n">sample_stats_table</span> <span class="o">=</span> <span class="n">get_stats</span><span class="p">(</span><span class="n">del_sample_gene</span><span class="p">,</span> <span class="n">sort</span> <span class="o">=</span> <span class="n">sort_data</span><span class="p">)</span>
    <span class="k">return</span> <span class="n">sample_stats_table</span>
<span class="k">def</span> <span class="nf">run_stats</span><span class="p">(</span><span class="n">input_data</span><span class="p">,</span> <span class="n">output_data</span><span class="p">):</span>
    <span class="n">stats_table</span> <span class="o">=</span> <span class="n">get_stats_from_input</span><span class="p">(</span><span class="n">input_data</span><span class="p">,</span> <span class="n">sort_data</span><span class="o">=</span><span class="mi">0</span><span class="p">)</span>
    <span class="n">stats_table</span><span class="p">[</span><span class="s1">&#39;debug&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="p">{</span><span class="s1">&#39;simulation_args&#39;</span><span class="p">:</span> <span class="n">input_data</span><span class="p">[</span><span class="s1">&#39;args&#39;</span><span class="p">]}</span>
    <span class="n">save_data</span><span class="p">(</span><span class="n">stats_table</span><span class="p">,</span> <span class="n">output_data</span><span class="p">)</span>
    <span class="k">return</span> <span class="n">stats_table</span>
<span class="k">def</span> <span class="nf">pkl_to_matrix</span><span class="p">(</span><span class="n">input_data</span><span class="p">,</span> <span class="n">make_block</span> <span class="o">=</span> <span class="bp">False</span><span class="p">,</span> <span class="n">dtype</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">uint8</span><span class="p">):</span>
    <span class="n">dat</span> <span class="o">=</span> <span class="n">load_data</span><span class="p">(</span><span class="n">input_data</span><span class="p">)</span>
    <span class="n">ref</span> <span class="o">=</span> <span class="n">load_reference_gene</span><span class="p">(</span><span class="n">dat</span><span class="p">[</span><span class="s1">&#39;debug&#39;</span><span class="p">][</span><span class="s1">&#39;args&#39;</span><span class="p">][</span><span class="s2">&quot;refgene_file&quot;</span><span class="p">])</span>
    <span class="n">genes</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">Series</span><span class="p">(</span><span class="nb">list</span><span class="p">(</span><span class="nb">set</span><span class="p">(</span><span class="n">ref</span><span class="p">[</span><span class="s1">&#39;gene_name&#39;</span><span class="p">])))</span>
    <span class="n">regression_data</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">array</span><span class="p">([</span><span class="n">np</span><span class="o">.</span><span class="n">array</span><span class="p">(</span><span class="n">genes</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span><span class="n">item</span><span class="p">[</span><span class="s2">&quot;gene_name&quot;</span><span class="p">]),</span> <span class="n">dtype</span> <span class="o">=</span> <span class="nb">float</span><span class="p">)</span> 
                                <span class="k">for</span> <span class="n">item</span> <span class="ow">in</span> <span class="n">dat</span><span class="p">[</span><span class="s1">&#39;case&#39;</span><span class="p">]</span> <span class="o">+</span> <span class="n">dat</span><span class="p">[</span><span class="s1">&#39;ctrl&#39;</span><span class="p">]])</span>
    <span class="n">phenotype</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">matrix</span><span class="p">([</span><span class="mi">1</span><span class="p">]</span><span class="o">*</span><span class="nb">len</span><span class="p">(</span><span class="n">dat</span><span class="p">[</span><span class="s1">&#39;case&#39;</span><span class="p">])</span> <span class="o">+</span> <span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">*</span><span class="nb">len</span><span class="p">(</span><span class="n">dat</span><span class="p">[</span><span class="s1">&#39;ctrl&#39;</span><span class="p">]))</span><span class="o">.</span><span class="n">T</span>
    <span class="n">regression_data</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">hstack</span><span class="p">((</span><span class="n">phenotype</span><span class="p">,</span> <span class="n">regression_data</span><span class="p">))</span>
<span class="c1">#     mask = np.ravel((regression_data==0).all(0))</span>
<span class="c1">#     regression_data = np.hstack((phenotype, regression_data[:, mask]))</span>
    <span class="n">df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">(</span><span class="n">regression_data</span><span class="p">,</span> <span class="n">columns</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;phenotype&#39;</span><span class="p">]</span> <span class="o">+</span> <span class="n">genes</span><span class="o">.</span><span class="n">tolist</span><span class="p">())</span>
    <span class="n">newdf</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">()</span>
    <span class="k">for</span> <span class="n">col</span> <span class="ow">in</span> <span class="n">df</span><span class="p">:</span>
        <span class="k">if</span> <span class="nb">sum</span><span class="p">(</span><span class="n">df</span><span class="p">[</span><span class="n">col</span><span class="p">])</span> <span class="o">&gt;</span> <span class="mi">0</span><span class="p">:</span>
            <span class="n">newdf</span><span class="p">[</span><span class="n">col</span><span class="p">]</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="n">col</span><span class="p">]</span>
        <span class="k">else</span><span class="p">:</span> <span class="k">continue</span>
    <span class="k">if</span> <span class="ow">not</span> <span class="n">make_block</span><span class="p">:</span>
        <span class="n">res</span> <span class="o">=</span> <span class="n">newdf</span><span class="o">.</span><span class="n">astype</span><span class="p">(</span><span class="n">dtype</span><span class="p">,</span> <span class="n">copy</span> <span class="o">=</span> <span class="bp">True</span><span class="p">)</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="n">blocks</span> <span class="o">=</span> <span class="n">get_analysis_blocks</span><span class="p">(</span><span class="n">pd</span><span class="o">.</span><span class="n">concat</span><span class="p">([</span><span class="n">pd</span><span class="o">.</span><span class="n">concat</span><span class="p">(</span><span class="n">dat</span><span class="p">[</span><span class="s1">&#39;case&#39;</span><span class="p">]),</span> <span class="n">pd</span><span class="o">.</span><span class="n">concat</span><span class="p">(</span><span class="n">dat</span><span class="p">[</span><span class="s1">&#39;ctrl&#39;</span><span class="p">])]))</span>
        <span class="n">res</span> <span class="o">=</span> <span class="p">[</span><span class="n">newdf</span><span class="p">[[</span><span class="s1">&#39;phenotype&#39;</span><span class="p">]</span> <span class="o">+</span> <span class="n">item</span><span class="p">]</span><span class="o">.</span><span class="n">astype</span><span class="p">(</span><span class="n">dtype</span><span class="p">,</span> <span class="n">copy</span> <span class="o">=</span> <span class="bp">True</span><span class="p">)</span> <span class="k">for</span> <span class="n">item</span> <span class="ow">in</span> <span class="n">blocks</span><span class="p">]</span>
    <span class="k">return</span> <span class="p">{</span><span class="s2">&quot;data&quot;</span><span class="p">:</span> <span class="n">res</span><span class="p">,</span> <span class="s2">&quot;debug&quot;</span><span class="p">:</span> <span class="n">dat</span><span class="p">[</span><span class="s1">&#39;debug&#39;</span><span class="p">]}</span>
<span class="k">def</span> <span class="nf">run_dap_lite</span><span class="p">(</span><span class="n">df</span><span class="p">,</span> <span class="n">fout</span><span class="p">,</span> <span class="n">prefix</span> <span class="o">=</span> <span class="bp">None</span><span class="p">,</span> <span class="n">exec_path</span> <span class="o">=</span> <span class="bp">None</span><span class="p">):</span>
    <span class="sd">&#39;&#39;&#39;Convert pandas dataframe to dap input:</span>
<span class="sd">        - phenotype / genotype file</span>
<span class="sd">        - prior file</span>
<span class="sd">        - grid file (of effect size): omega^2 + phi^2 is what we care. Let&#39;s set it to</span>
<span class="sd">            1 1; 2 2; 3 3 and 4 4 for now, as we only have one Y</span>
<span class="sd">    &#39;&#39;&#39;</span>
    <span class="k">print</span> <span class="p">(</span><span class="nb">str</span><span class="p">(</span><span class="n">datetime</span><span class="o">.</span><span class="n">now</span><span class="p">()))</span>
    <span class="kn">import</span> <span class="nn">os</span>
    <span class="k">if</span> <span class="n">prefix</span> <span class="ow">is</span> <span class="bp">None</span><span class="p">:</span>
        <span class="kn">import</span> <span class="nn">time</span>
        <span class="n">prefix</span> <span class="o">=</span> <span class="s2">&quot;/tmp/F&quot;</span> <span class="o">+</span> <span class="nb">str</span><span class="p">(</span><span class="n">time</span><span class="o">.</span><span class="n">time</span><span class="p">())</span>
    <span class="k">if</span> <span class="n">exec_path</span> <span class="ow">is</span> <span class="bp">None</span><span class="p">:</span>
        <span class="n">exec_path</span> <span class="o">=</span> <span class="s1">&#39;dap/dap&#39;</span>
    <span class="n">chrom</span> <span class="o">=</span> <span class="s1">&#39;chr6&#39;</span>
    <span class="n">pos</span> <span class="o">=</span> <span class="mi">100000</span>
    <span class="n">dat</span> <span class="o">=</span> <span class="p">[[</span><span class="s1">&#39;pheno&#39;</span><span class="p">,</span> <span class="s1">&#39;trait&#39;</span><span class="p">,</span> <span class="s1">&#39;chicago&#39;</span><span class="p">]</span> <span class="o">+</span> <span class="p">[</span><span class="nb">str</span><span class="p">(</span><span class="n">x</span><span class="p">)</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">df</span><span class="p">[</span><span class="s1">&#39;phenotype&#39;</span><span class="p">]]]</span>
    <span class="n">prior</span> <span class="o">=</span> <span class="p">[]</span>
    <span class="n">grid</span> <span class="o">=</span> <span class="p">[(</span><span class="mi">1</span><span class="p">,</span><span class="mi">1</span><span class="p">),(</span><span class="mi">2</span><span class="p">,</span><span class="mi">2</span><span class="p">),(</span><span class="mi">3</span><span class="p">,</span><span class="mi">3</span><span class="p">),(</span><span class="mi">4</span><span class="p">,</span><span class="mi">4</span><span class="p">)]</span>
    <span class="k">for</span> <span class="n">idx</span><span class="p">,</span> <span class="n">item</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">df</span><span class="o">.</span><span class="n">columns</span><span class="o">.</span><span class="n">values</span><span class="p">):</span>
        <span class="k">if</span> <span class="n">item</span> <span class="o">==</span> <span class="s1">&#39;phenotype&#39;</span><span class="p">:</span>
            <span class="k">continue</span>
        <span class="n">dat</span><span class="o">.</span><span class="n">append</span><span class="p">([</span><span class="s1">&#39;geno&#39;</span><span class="p">,</span> <span class="s1">&#39;{}.{}&#39;</span><span class="o">.</span><span class="n">format</span><span class="p">(</span><span class="n">chrom</span><span class="p">,</span> <span class="n">pos</span> <span class="o">+</span> <span class="n">idx</span><span class="p">),</span> <span class="s1">&#39;chicago&#39;</span><span class="p">]</span> <span class="o">+</span> <span class="p">[</span><span class="nb">str</span><span class="p">(</span><span class="n">x</span><span class="p">)</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">df</span><span class="p">[</span><span class="n">item</span><span class="p">]])</span>
        <span class="n">prior</span><span class="o">.</span><span class="n">append</span><span class="p">([</span><span class="s1">&#39;{}.{}&#39;</span><span class="o">.</span><span class="n">format</span><span class="p">(</span><span class="n">chrom</span><span class="p">,</span> <span class="n">pos</span> <span class="o">+</span> <span class="n">idx</span><span class="p">),</span> <span class="nb">str</span><span class="p">(</span><span class="mi">1</span><span class="o">/</span><span class="p">(</span><span class="n">df</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span> <span class="o">-</span> <span class="mi">1</span><span class="p">))])</span>
    <span class="k">with</span> <span class="nb">open</span><span class="p">(</span><span class="n">prefix</span> <span class="o">+</span> <span class="s1">&#39;.dat&#39;</span><span class="p">,</span> <span class="s1">&#39;w&#39;</span><span class="p">)</span> <span class="k">as</span> <span class="n">f</span><span class="p">:</span>
        <span class="n">f</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="s1">&#39;</span><span class="se">\n</span><span class="s1">&#39;</span><span class="o">.</span><span class="n">join</span><span class="p">([</span><span class="s1">&#39; &#39;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">x</span><span class="p">)</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">dat</span><span class="p">]))</span>
    <span class="k">with</span> <span class="nb">open</span><span class="p">(</span><span class="n">prefix</span> <span class="o">+</span> <span class="s1">&#39;.prior&#39;</span><span class="p">,</span> <span class="s1">&#39;w&#39;</span><span class="p">)</span> <span class="k">as</span> <span class="n">f</span><span class="p">:</span>
        <span class="n">f</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="s1">&#39;</span><span class="se">\n</span><span class="s1">&#39;</span><span class="o">.</span><span class="n">join</span><span class="p">([</span><span class="s1">&#39; &#39;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">x</span><span class="p">)</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">prior</span><span class="p">]))</span>
    <span class="k">with</span> <span class="nb">open</span><span class="p">(</span><span class="n">prefix</span> <span class="o">+</span> <span class="s1">&#39;.grid&#39;</span><span class="p">,</span> <span class="s1">&#39;w&#39;</span><span class="p">)</span> <span class="k">as</span> <span class="n">f</span><span class="p">:</span>
        <span class="n">f</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="s1">&#39;</span><span class="se">\n</span><span class="s1">&#39;</span><span class="o">.</span><span class="n">join</span><span class="p">([</span><span class="s1">&#39; &#39;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="nb">map</span><span class="p">(</span><span class="nb">str</span><span class="p">,</span> <span class="n">x</span><span class="p">))</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">grid</span><span class="p">]))</span>
    <span class="n">os</span><span class="o">.</span><span class="n">system</span><span class="p">(</span><span class="s2">&quot;{0} -d {1}.dat -g {1}.grid -t 8 -it 0.05 -prior {1}.prior &gt; {2}&quot;</span><span class="o">.</span><span class="n">format</span><span class="p">(</span><span class="n">exec_path</span><span class="p">,</span> <span class="n">prefix</span><span class="p">,</span> <span class="n">fout</span><span class="p">))</span>
    <span class="k">print</span> <span class="p">(</span><span class="nb">str</span><span class="p">(</span><span class="n">datetime</span><span class="o">.</span><span class="n">now</span><span class="p">()))</span>
<span class="c1">#     return dat</span>
<span class="k">def</span> <span class="nf">run_dap</span><span class="p">(</span><span class="n">df</span><span class="p">,</span> <span class="n">fileout</span><span class="p">,</span> <span class="n">pthwy_genes</span><span class="p">,</span> <span class="n">ref_genes</span><span class="p">,</span> <span class="n">dap_method</span> <span class="o">=</span> <span class="s2">&quot;dap&quot;</span><span class="p">,</span> <span class="n">grid</span> <span class="o">=</span> <span class="p">[(</span><span class="mi">0</span><span class="p">,</span><span class="mf">0.1</span><span class="p">)],</span> <span class="n">multiplier</span> <span class="o">=</span> <span class="mi">5</span><span class="p">,</span>  
            <span class="n">exec_path</span> <span class="o">=</span> <span class="s1">&#39;dap&#39;</span><span class="p">,</span> <span class="n">prefix</span> <span class="o">=</span> <span class="bp">None</span><span class="p">,</span> <span class="n">dry_run</span> <span class="o">=</span> <span class="bp">False</span><span class="p">,</span> <span class="n">ncpu</span> <span class="o">=</span> <span class="mi">6</span><span class="p">):</span>
    <span class="sd">&#39;&#39;&#39;Convert pandas dataframe to dap input:</span>
<span class="sd">        - phenotype / genotype file</span>
<span class="sd">        - prior file</span>
<span class="sd">        - grid file (of effect size): omega^2 + phi^2 is what we care. Let&#39;s set it to</span>
<span class="sd">          (0, beta) for now, as we only have one Y</span>
<span class="sd">        dap_method: default to `dap`, other choices are `dap1`, `dap-g` (dap greedy)</span>
<span class="sd">        You can experiment with multiple dap implementation using `dap_method` and `dry_run`</span>
<span class="sd">        For example you set `dry_run = True`, ncpu = 3 and `dap_method = &quot;dap&quot;` </span>
<span class="sd">        you&#39;ll get a line of command printed on screen that will run `dap` if you execute it in terminal.</span>
<span class="sd">        Then you can change `dap_method = &quot;dap1&quot;`, run again to generate the command for `dap1` and execute</span>
<span class="sd">        it in another terminal window.</span>
<span class="sd">    &#39;&#39;&#39;</span>
    <span class="kn">import</span> <span class="nn">time</span>
    <span class="n">times</span> <span class="o">=</span> <span class="p">[</span><span class="n">time</span><span class="o">.</span><span class="n">time</span><span class="p">(),</span> <span class="bp">None</span><span class="p">,</span> <span class="bp">None</span><span class="p">]</span>
    <span class="k">if</span> <span class="n">prefix</span> <span class="ow">is</span> <span class="bp">None</span><span class="p">:</span>
        <span class="n">prefix</span> <span class="o">=</span> <span class="s2">&quot;/tmp/F&quot;</span> <span class="o">+</span> <span class="nb">str</span><span class="p">(</span><span class="n">time</span><span class="o">.</span><span class="n">time</span><span class="p">())</span>
    <span class="n">exec_path</span> <span class="o">=</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">exec_path</span> <span class="ow">or</span> <span class="s1">&#39;&#39;</span><span class="p">,</span> <span class="n">dap_method</span><span class="p">)</span>
<span class="c1">#     chrom = [&quot;chr{}&quot;.format(i) for i in list(range(1,23))+[&quot;X&quot;]]</span>
<span class="c1">#     pos is all the gene names showed in .feather</span>
    <span class="n">pos</span> <span class="o">=</span> <span class="n">df</span><span class="o">.</span><span class="n">columns</span><span class="o">.</span><span class="n">values</span><span class="p">[</span><span class="mi">1</span><span class="p">:]</span><span class="o">.</span><span class="n">tolist</span><span class="p">()</span>
<span class="c1">#     first item in &quot;dat&quot; is phenotype for all samples (# of rows in .feather)</span>
    <span class="n">dat</span> <span class="o">=</span> <span class="p">[[</span><span class="s1">&#39;pheno&#39;</span><span class="p">,</span> <span class="s1">&#39;trait&#39;</span><span class="p">,</span> <span class="s1">&#39;chicago&#39;</span><span class="p">]</span> <span class="o">+</span> <span class="p">[</span><span class="nb">str</span><span class="p">(</span><span class="n">x</span><span class="p">)</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">df</span><span class="p">[</span><span class="s1">&#39;phenotype&#39;</span><span class="p">]]]</span>
    <span class="n">prior</span> <span class="o">=</span> <span class="p">[]</span>
<span class="c1">#     item is gene namePerformance Cake PanPerformance Cake PanPerformance Cake Pan</span>
    <span class="k">for</span> <span class="n">idx</span><span class="p">,</span> <span class="n">item</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">df</span><span class="o">.</span><span class="n">columns</span><span class="o">.</span><span class="n">values</span><span class="p">):</span>
        <span class="k">if</span> <span class="n">item</span> <span class="o">==</span> <span class="s2">&quot;phenotype&quot;</span><span class="p">:</span>
            <span class="k">continue</span>
        <span class="n">dat</span><span class="o">.</span><span class="n">append</span><span class="p">([</span><span class="s1">&#39;geno&#39;</span><span class="p">,</span> <span class="s1">&#39;{}.{}&#39;</span><span class="o">.</span><span class="n">format</span><span class="p">(</span><span class="n">ref_genes</span><span class="p">[</span><span class="n">ref_genes</span><span class="p">[</span><span class="s2">&quot;gene_name&quot;</span><span class="p">]</span><span class="o">==</span><span class="n">item</span><span class="p">][</span><span class="s2">&quot;chrom&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">tolist</span><span class="p">()[</span><span class="mi">0</span><span class="p">],</span> <span class="n">item</span><span class="p">),</span> 
                    <span class="s1">&#39;chicago&#39;</span><span class="p">]</span> <span class="o">+</span> <span class="p">[</span><span class="nb">str</span><span class="p">(</span><span class="n">x</span><span class="p">)</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">df</span><span class="p">[</span><span class="n">item</span><span class="p">]])</span>
        <span class="n">n_overlap_gene</span> <span class="o">=</span> <span class="nb">len</span><span class="p">(</span><span class="nb">set</span><span class="p">(</span><span class="n">pthwy_genes</span><span class="p">[</span><span class="s2">&quot;gene_name&quot;</span><span class="p">])</span> <span class="o">&amp;</span> <span class="nb">set</span><span class="p">(</span><span class="n">pos</span><span class="p">))</span>
        <span class="n">prior_causal</span> <span class="o">=</span> <span class="n">multiplier</span> <span class="o">/</span> <span class="nb">len</span><span class="p">(</span><span class="nb">set</span><span class="p">(</span><span class="n">pos</span><span class="p">))</span>
        <span class="n">div</span> <span class="o">=</span> <span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="nb">set</span><span class="p">(</span><span class="n">pos</span><span class="p">))</span><span class="o">-</span><span class="n">n_overlap_gene</span><span class="p">)</span> <span class="ow">or</span> <span class="mi">1</span>
        <span class="n">prior_noncausal</span> <span class="o">=</span> <span class="p">(</span><span class="mi">1</span><span class="o">-</span><span class="n">prior_causal</span><span class="o">*</span><span class="n">n_overlap_gene</span><span class="p">)</span> <span class="o">/</span> <span class="n">div</span>
        <span class="k">if</span> <span class="n">multiplier</span> <span class="o">&lt;=</span> <span class="mf">1.0</span><span class="p">:</span>
            <span class="n">prior_pr</span> <span class="o">=</span> <span class="nb">str</span><span class="p">(</span><span class="mi">1</span> <span class="o">/</span> <span class="nb">len</span><span class="p">(</span><span class="nb">set</span><span class="p">(</span><span class="n">pos</span><span class="p">)))</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="n">prior_pr</span> <span class="o">=</span> <span class="nb">str</span><span class="p">(</span><span class="n">prior_causal</span><span class="p">)</span> <span class="k">if</span> <span class="n">item</span> <span class="ow">in</span> <span class="n">pthwy_genes</span><span class="p">[</span><span class="s2">&quot;gene_name&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">tolist</span><span class="p">()</span> <span class="k">else</span> <span class="nb">str</span><span class="p">(</span><span class="n">prior_noncausal</span><span class="p">)</span>
        <span class="n">prior</span><span class="o">.</span><span class="n">append</span><span class="p">([</span><span class="s1">&#39;{}.{}&#39;</span><span class="o">.</span><span class="n">format</span><span class="p">(</span><span class="n">ref_genes</span><span class="p">[</span><span class="n">ref_genes</span><span class="p">[</span><span class="s2">&quot;gene_name&quot;</span><span class="p">]</span><span class="o">==</span><span class="n">item</span><span class="p">][</span><span class="s2">&quot;chrom&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">tolist</span><span class="p">()[</span><span class="mi">0</span><span class="p">],</span> <span class="n">item</span><span class="p">),</span> <span class="n">prior_pr</span><span class="p">])</span>
<span class="c1">#     print (prior_causal, prior_noncausal)</span>
    <span class="k">with</span> <span class="nb">open</span><span class="p">(</span><span class="n">prefix</span> <span class="o">+</span> <span class="s1">&#39;.dat&#39;</span><span class="p">,</span> <span class="s1">&#39;w&#39;</span><span class="p">)</span> <span class="k">as</span> <span class="n">f</span><span class="p">:</span>
        <span class="n">f</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="s1">&#39;</span><span class="se">\n</span><span class="s1">&#39;</span><span class="o">.</span><span class="n">join</span><span class="p">([</span><span class="s1">&#39; &#39;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">x</span><span class="p">)</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">dat</span><span class="p">]))</span>
    <span class="k">with</span> <span class="nb">open</span><span class="p">(</span><span class="n">prefix</span> <span class="o">+</span> <span class="s1">&#39;.prior&#39;</span><span class="p">,</span> <span class="s1">&#39;w&#39;</span><span class="p">)</span> <span class="k">as</span> <span class="n">f</span><span class="p">:</span>
        <span class="n">f</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="s1">&#39;</span><span class="se">\n</span><span class="s1">&#39;</span><span class="o">.</span><span class="n">join</span><span class="p">([</span><span class="s1">&#39; &#39;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">x</span><span class="p">)</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">prior</span><span class="p">]))</span>
    <span class="k">with</span> <span class="nb">open</span><span class="p">(</span><span class="n">prefix</span> <span class="o">+</span> <span class="s1">&#39;.grid&#39;</span><span class="p">,</span> <span class="s1">&#39;w&#39;</span><span class="p">)</span> <span class="k">as</span> <span class="n">f</span><span class="p">:</span>
        <span class="n">f</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="s1">&#39;</span><span class="se">\n</span><span class="s1">&#39;</span><span class="o">.</span><span class="n">join</span><span class="p">([</span><span class="s1">&#39; &#39;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="nb">map</span><span class="p">(</span><span class="nb">str</span><span class="p">,</span> <span class="n">x</span><span class="p">))</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">grid</span><span class="p">]))</span>
    <span class="n">times</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span> <span class="o">=</span> <span class="n">time</span><span class="o">.</span><span class="n">time</span><span class="p">()</span>
    <span class="k">if</span> <span class="n">dap_method</span> <span class="o">==</span> <span class="s1">&#39;dap&#39;</span><span class="p">:</span>
        <span class="n">cmd</span> <span class="o">=</span> <span class="s2">&quot;{0} -d {1}.dat -g {1}.grid -it 0.05 -prior {1}.prior -t {2} &gt;&gt; {3} &quot;</span><span class="o">.</span>\
            <span class="n">format</span><span class="p">(</span><span class="n">exec_path</span><span class="p">,</span> <span class="n">prefix</span><span class="p">,</span> <span class="n">ncpu</span><span class="p">,</span> <span class="n">fileout</span><span class="p">)</span>
    <span class="k">elif</span> <span class="n">dap_method</span> <span class="o">==</span> <span class="s1">&#39;dap-g&#39;</span><span class="p">:</span>
        <span class="n">cmd</span> <span class="o">=</span>  <span class="s2">&quot;{0} -d {1}.dat -g {1}.grid -prior {1}.prior -t {2} &gt;&gt; {3} &quot;</span><span class="o">.</span>\
            <span class="n">format</span><span class="p">(</span><span class="n">exec_path</span><span class="p">,</span> <span class="n">prefix</span><span class="p">,</span> <span class="n">ncpu</span><span class="p">,</span> <span class="n">fileout</span><span class="p">)</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="n">cmd</span> <span class="o">=</span> <span class="s2">&quot;{0} -d {1}.dat -prior {1}.prior &gt;&gt; {2} &quot;</span><span class="o">.</span>\
            <span class="n">format</span><span class="p">(</span><span class="n">exec_path</span><span class="p">,</span> <span class="n">prefix</span><span class="p">,</span> <span class="n">fileout</span><span class="p">)</span>
    <span class="k">if</span> <span class="ow">not</span> <span class="n">dry_run</span><span class="p">:</span>
        <span class="c1"># print(cmd)</span>
        <span class="n">os</span><span class="o">.</span><span class="n">system</span><span class="p">(</span><span class="n">cmd</span><span class="p">)</span>
        <span class="n">times</span><span class="p">[</span><span class="mi">2</span><span class="p">]</span> <span class="o">=</span> <span class="n">time</span><span class="o">.</span><span class="n">time</span><span class="p">()</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="c1"># generate cmd, then run under the folder of &quot;analysis&quot;</span>
        <span class="k">print</span><span class="p">(</span><span class="n">cmd</span><span class="p">)</span>
    <span class="k">return</span> <span class="n">times</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered">
<div class="prompt input_prompt">
</div>
<div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Power-and-type-I-error">Power and type I error<a class="anchor-link" href="#Power-and-type-I-error">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[16]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="k">def</span> <span class="nf">get_power</span><span class="p">(</span><span class="n">stats_table</span><span class="p">,</span> <span class="n">causal_genes</span><span class="p">,</span> <span class="n">p</span> <span class="o">=</span> <span class="mf">0.05</span><span class="p">):</span>
    <span class="sd">&#39;&#39;&#39;get power for each simulated dataset.</span>
<span class="sd">    First get overlapped genes from stats table and causal genes, and find corresponding p value</span>
<span class="sd">    Then power = #(p_value &lt; 0.05) / #(p_value)&#39;&#39;&#39;</span>
    <span class="n">overlap_causal_genes</span> <span class="o">=</span> <span class="p">[</span><span class="n">gene</span> <span class="k">for</span> <span class="n">gene</span> <span class="ow">in</span> <span class="n">causal_genes</span> <span class="k">if</span> <span class="n">gene</span> <span class="ow">in</span> <span class="n">stats_table</span><span class="p">[</span><span class="s2">&quot;genes&quot;</span><span class="p">]]</span>
    <span class="n">overlap_p_value</span> <span class="o">=</span> <span class="p">[</span><span class="n">stats_table</span><span class="p">[</span><span class="s2">&quot;p_value&quot;</span><span class="p">][</span><span class="n">stats_table</span><span class="p">[</span><span class="s2">&quot;genes&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">index</span><span class="p">(</span><span class="n">gene</span><span class="p">)]</span> 
                       <span class="k">for</span> <span class="n">gene</span> <span class="ow">in</span> <span class="n">overlap_causal_genes</span><span class="p">]</span>
    <span class="n">pvalue_less_than_p</span> <span class="o">=</span> <span class="p">[</span><span class="n">x</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">overlap_p_value</span> <span class="k">if</span> <span class="n">x</span> <span class="o">&lt;</span> <span class="n">p</span><span class="p">]</span>
    <span class="k">return</span> <span class="nb">len</span><span class="p">(</span><span class="n">pvalue_less_than_p</span><span class="p">)</span><span class="o">/</span><span class="nb">len</span><span class="p">(</span><span class="n">overlap_causal_genes</span><span class="p">)</span>
<span class="k">def</span> <span class="nf">get_typeIerror</span><span class="p">(</span><span class="n">stats_table</span><span class="p">,</span> <span class="n">causal_genes</span><span class="p">,</span> <span class="n">p</span> <span class="o">=</span> <span class="mf">0.05</span><span class="p">):</span>
    <span class="sd">&#39;&#39;&#39;get type I error for each simulated dataset.</span>
<span class="sd">    First get the overlapped genes from noncausal genes and genes in stats table, find corresponding p value</span>
<span class="sd">    Then type I error = #(p_value &lt; 0.05) / #(p_value)</span>
<span class="sd">    &#39;&#39;&#39;</span>
    <span class="n">noncausal_overlap_genes</span> <span class="o">=</span> <span class="p">[</span><span class="n">gene</span> <span class="k">for</span> <span class="n">gene</span> <span class="ow">in</span> <span class="n">stats_table</span><span class="p">[</span><span class="s2">&quot;genes&quot;</span><span class="p">]</span> <span class="k">if</span> <span class="n">gene</span> <span class="ow">not</span> <span class="ow">in</span> <span class="n">causal_genes</span><span class="p">]</span>
    <span class="n">noncausal_overlap_genes_pvalue</span> <span class="o">=</span> <span class="p">[</span><span class="n">stats_table</span><span class="p">[</span><span class="s2">&quot;p_value&quot;</span><span class="p">][</span><span class="n">stats_table</span><span class="p">[</span><span class="s2">&quot;genes&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">index</span><span class="p">(</span><span class="n">gene</span><span class="p">)]</span>
                                      <span class="k">for</span> <span class="n">gene</span> <span class="ow">in</span> <span class="n">noncausal_overlap_genes</span><span class="p">]</span>
    <span class="n">pvalue_less_than_p</span> <span class="o">=</span> <span class="p">[</span><span class="n">x</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">noncausal_overlap_genes_pvalue</span> <span class="k">if</span> <span class="n">x</span> <span class="o">&lt;</span> <span class="n">p</span><span class="p">]</span>
    <span class="k">return</span> <span class="nb">len</span><span class="p">(</span><span class="n">pvalue_less_than_p</span><span class="p">)</span><span class="o">/</span><span class="nb">len</span><span class="p">(</span><span class="n">noncausal_overlap_genes_pvalue</span><span class="p">)</span>
<span class="k">def</span> <span class="nf">test_contingency_table</span><span class="p">(</span><span class="n">gene_table</span><span class="p">,</span> <span class="n">method</span> <span class="o">=</span> <span class="s2">&quot;Fisher&quot;</span><span class="p">,</span> <span class="n">option</span> <span class="o">=</span> <span class="bp">False</span><span class="p">):</span> 
    <span class="k">if</span> <span class="p">(</span><span class="n">method</span> <span class="o">==</span> <span class="s2">&quot;Fisher&quot;</span><span class="p">):</span>
        <span class="n">stats_table</span> <span class="o">=</span> <span class="p">[(</span><span class="n">pvalue</span><span class="p">(</span><span class="n">row</span><span class="p">[</span><span class="s2">&quot;n_case_gene&quot;</span><span class="p">],</span> <span class="n">row</span><span class="p">[</span><span class="s2">&quot;n_ctrl_gene&quot;</span><span class="p">],</span> <span class="n">row</span><span class="p">[</span><span class="s2">&quot;n_case_nogene&quot;</span><span class="p">],</span> <span class="n">row</span><span class="p">[</span><span class="s2">&quot;n_ctrl_nogene&quot;</span><span class="p">]),</span> 
                        <span class="n">row</span><span class="p">[</span><span class="s2">&quot;gene_name&quot;</span><span class="p">])</span> <span class="k">for</span> <span class="n">idx</span><span class="p">,</span> <span class="n">row</span> <span class="ow">in</span> <span class="n">gene_table</span><span class="o">.</span><span class="n">iterrows</span><span class="p">()]</span>
        <span class="n">p_value</span> <span class="o">=</span> <span class="p">[</span><span class="n">x</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">two_tail</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">stats_table</span><span class="p">]</span>
        <span class="n">genes</span> <span class="o">=</span> <span class="p">[</span><span class="n">x</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">stats_table</span><span class="p">]</span>
        <span class="n">stats_table</span> <span class="o">=</span> <span class="p">{</span><span class="s2">&quot;genes&quot;</span><span class="p">:</span> <span class="n">genes</span><span class="p">,</span> <span class="s2">&quot;p_value&quot;</span><span class="p">:</span> <span class="n">p_value</span><span class="p">}</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="n">table</span> <span class="o">=</span> <span class="p">[(</span> <span class="n">stats</span><span class="o">.</span><span class="n">chi2_contingency</span><span class="p">([[</span><span class="n">row</span><span class="p">[</span><span class="s2">&quot;n_case_gene&quot;</span><span class="p">],</span> <span class="n">row</span><span class="p">[</span><span class="s2">&quot;n_ctrl_gene&quot;</span><span class="p">]],</span> 
                                           <span class="p">[</span><span class="n">row</span><span class="p">[</span><span class="s2">&quot;n_case_nogene&quot;</span><span class="p">],</span> <span class="n">row</span><span class="p">[</span><span class="s2">&quot;n_ctrl_nogene&quot;</span><span class="p">]]],</span> 
                                          <span class="n">correction</span> <span class="o">=</span> <span class="n">option</span><span class="p">),</span> <span class="n">row</span><span class="p">[</span><span class="s2">&quot;gene_name&quot;</span><span class="p">]</span> <span class="p">)</span> 
                 <span class="k">for</span> <span class="n">idx</span><span class="p">,</span> <span class="n">row</span> <span class="ow">in</span> <span class="n">gene_table</span><span class="o">.</span><span class="n">iterrows</span><span class="p">()]</span>
        <span class="n">p_value</span> <span class="o">=</span> <span class="p">[</span><span class="n">x</span><span class="p">[</span><span class="mi">0</span><span class="p">][</span><span class="mi">1</span><span class="p">]</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">table</span><span class="p">]</span>
        <span class="n">gene</span> <span class="o">=</span> <span class="p">[</span><span class="n">x</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">table</span><span class="p">]</span>
        <span class="n">stats_table</span> <span class="o">=</span> <span class="p">{</span><span class="s2">&quot;p_value&quot;</span><span class="p">:</span> <span class="n">p_value</span><span class="p">,</span> <span class="s2">&quot;genes&quot;</span><span class="p">:</span> <span class="n">gene</span><span class="p">}</span>
    <span class="k">return</span> <span class="n">stats_table</span>
<span class="k">def</span> <span class="nf">get_power_and_typeIerror</span><span class="p">(</span><span class="n">input_data</span><span class="p">,</span> <span class="n">method_option</span> <span class="o">=</span> <span class="s2">&quot;chi2&quot;</span><span class="p">,</span> <span class="n">correction_option</span> <span class="o">=</span> <span class="bp">False</span><span class="p">,</span> <span class="n">p_option</span> <span class="o">=</span> <span class="mf">0.05</span><span class="p">):</span>
    <span class="sd">&#39;&#39;&#39;use function &quot;load_data&quot; and &quot;get_gene_table&quot; from simulation.py, use simulated dataset as input data,</span>
<span class="sd">    and get stats table by using Fisher or chisquare test.</span>
<span class="sd">    Then get power and type I error by using functions above, input is stats table and causal genes&#39;&#39;&#39;</span>
    <span class="n">sample_table</span> <span class="o">=</span> <span class="n">load_data</span><span class="p">(</span><span class="n">input_data</span><span class="p">)</span>
    <span class="n">causal_genes</span> <span class="o">=</span> <span class="n">sample_table</span><span class="p">[</span><span class="s2">&quot;debug&quot;</span><span class="p">][</span><span class="s2">&quot;causal genes&quot;</span><span class="p">]</span>
    <span class="n">gene_table</span> <span class="o">=</span> <span class="n">get_gene_table</span><span class="p">(</span><span class="n">sample_table</span><span class="p">)</span>
    <span class="n">stats_table</span> <span class="o">=</span> <span class="n">test_contingency_table</span><span class="p">(</span><span class="n">gene_table</span><span class="p">,</span> <span class="n">method</span> <span class="o">=</span> <span class="n">method_option</span><span class="p">,</span> <span class="n">option</span> <span class="o">=</span> <span class="n">correction_option</span><span class="p">)</span>
    <span class="n">power</span> <span class="o">=</span> <span class="n">get_power</span><span class="p">(</span><span class="n">stats_table</span><span class="p">,</span> <span class="n">causal_genes</span><span class="p">,</span> <span class="n">p</span> <span class="o">=</span> <span class="n">p_option</span><span class="p">)</span>
    <span class="n">typeI_error</span> <span class="o">=</span> <span class="n">get_typeIerror</span><span class="p">(</span><span class="n">stats_table</span><span class="p">,</span> <span class="n">causal_genes</span><span class="p">,</span> <span class="n">p</span> <span class="o">=</span> <span class="n">p_option</span><span class="p">)</span>
    <span class="k">return</span> <span class="p">{</span><span class="s2">&quot;power&quot;</span><span class="p">:</span> <span class="n">power</span><span class="p">,</span> <span class="s2">&quot;typeI_error&quot;</span><span class="p">:</span> <span class="n">typeI_error</span><span class="p">,</span> <span class="s2">&quot;debug&quot;</span><span class="p">:</span> <span class="n">sample_table</span><span class="p">[</span><span class="s2">&quot;debug&quot;</span><span class="p">][</span><span class="s2">&quot;args&quot;</span><span class="p">]}</span>
<span class="k">def</span> <span class="nf">run_power_typeIerror</span><span class="p">(</span><span class="n">datasets</span><span class="p">):</span>
    <span class="sd">&#39;&#39;&#39;input data must be a list of simulated datasets&#39;&#39;&#39;</span>
    <span class="n">res</span> <span class="o">=</span> <span class="p">{}</span>
    <span class="n">i</span> <span class="o">=</span> <span class="mi">0</span>
    <span class="k">for</span> <span class="n">data</span> <span class="ow">in</span> <span class="n">datasets</span><span class="p">:</span>
        <span class="n">res_data</span> <span class="o">=</span> <span class="n">get_power_and_typeIerror</span><span class="p">(</span><span class="n">data</span><span class="p">)</span>
        <span class="n">res</span><span class="p">[</span><span class="s2">&quot;dataset_{}&quot;</span><span class="o">.</span><span class="n">format</span><span class="p">(</span><span class="n">i</span><span class="p">)]</span> <span class="o">=</span> <span class="n">res_data</span>
        <span class="n">i</span> <span class="o">+=</span> <span class="mi">1</span>
    <span class="k">return</span> <span class="n">res</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered">
<div class="prompt input_prompt">
</div>
<div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Obsolete-codes">Obsolete codes<a class="anchor-link" href="#Obsolete-codes">&#182;</a></h2><p>Yet may still be useful</p>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered">
<div class="prompt input_prompt">
</div>
<div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Simulation-codes">Simulation codes<a class="anchor-link" href="#Simulation-codes">&#182;</a></h3><p>Here is the version using randomly sampled causal genes, and single process computation (use <code>ctrl</code> + <code>/</code> on PC or <code>cmd</code> + <code>/</code> on Mac to comment and uncomment)</p>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="c1"># def p_case(p, num_causal_genes_in_sample, sim_args):</span>
<span class="c1">#     if num_causal_genes_in_sample == 0:</span>
<span class="c1">#         return p</span>
<span class="c1">#     baseline_odds = p / (1 - p)</span>
<span class="c1">#     if sim_args[&quot;odds_ratio_params&quot;][&quot;shape&quot;] is None or sim_args[&quot;odds_ratio_params&quot;][&quot;scale&quot;] is None:</span>
<span class="c1">#         odds_ratio = 1</span>
<span class="c1">#     else:</span>
<span class="c1">#         odds_ratio = np.prod([np.random.gamma(sim_args[&quot;odds_ratio_params&quot;][&#39;shape&#39;], </span>
<span class="c1">#                                               sim_args[&quot;odds_ratio_params&quot;][&#39;scale&#39;]) </span>
<span class="c1">#                               for x in range(num_causal_genes_in_sample)])</span>
<span class="c1">#     # obtain the power of fisher test by setting odds ratio to 1</span>
<span class="c1">#     # odds_ratio = 1</span>
<span class="c1">#     odds = baseline_odds * odds_ratio</span>
<span class="c1">#     return odds / (1 + odds)</span>
<span class="c1"># class Environment(dict):</span>
<span class="c1">#     def __init__(self):</span>
<span class="c1">#         parameters = {&#39;block_size&#39;: 100000,</span>
<span class="c1">#                       &#39;avg_cnv_per_individual&#39;: 5,</span>
<span class="c1">#                       &#39;n_case&#39;: 2000,</span>
<span class="c1">#                       &#39;n_ctrl&#39;: 2000,</span>
<span class="c1">#                        # set Gamma shape to be 3 instead of 5</span>
<span class="c1">#                        # &#39;odds_ratio_params&#39; : None # for H_0</span>
<span class="c1">#                       &#39;odds_ratio_params&#39;: {&#39;shape&#39;: 5, &#39;scale&#39;: 1},</span>
<span class="c1">#                       &#39;prevalence&#39;: 0.005,</span>
<span class="c1">#                       &#39;n_causal_gene&#39;: 200,</span>
<span class="c1">#                       &#39;refgene_file&#39;: &#39;data/refGene.txt.gz&#39;,</span>
<span class="c1">#                       &#39;cnv_file&#39;: &#39;data/ISC-r1.CNV.bed&#39;,</span>
<span class="c1">#                       &#39;case_dataset&#39;: &#39;delCases&#39;,</span>
<span class="c1">#                       &#39;ctrl_dataset&#39;: &#39;delControls&#39;,</span>
<span class="c1">#                       &#39;output&#39;: &#39;del_sample&#39; </span>
<span class="c1">#                      }</span>
<span class="c1">#         self.update(parameters)</span>
<span class="c1">#         ## select causal genes randomly, instead of the first 100 in enrichment analysis</span>
<span class="c1">#         self.ref_gene_name = load_reference_gene(parameters[&quot;refgene_file&quot;])[&quot;gene_name&quot;].tolist()</span>
<span class="c1">#         self.causal_genes = random.sample(self.ref_gene_name, parameters[&#39;n_causal_gene&#39;])</span>
<span class="c1">#         self.seed = 999</span>
<span class="c1"># def simulate(refgene, cnv_data, args, causal_genes):</span>
<span class="c1">#     df = cnv_data.drop_duplicates(subset=(&quot;chrom&quot;, &quot;cnv_start&quot;, &quot;cnv_end&quot;))</span>
<span class="c1">#     block_counts = count_cnv_by_block(df, args[&#39;block_size&#39;])</span>
<span class="c1">#     cnv_length = cnv_data[&#39;cnv_end&#39;] - cnv_data[&#39;cnv_start&#39;]</span>
<span class="c1">#     status = 1</span>
<span class="c1">#     case_data = []</span>
<span class="c1">#     ctrl_data = []</span>
<span class="c1">#     debug = {&#39;p&#39;: [], &#39;niter&#39;: 0, &#39;time&#39;: [str(datetime.now()), None], &#39;args&#39;: dict(args), </span>
<span class="c1">#              &#39;causal genes&#39;: causal_genes, &#39;number of causal genes&#39;: [], &#39;number of genes overlap CNV&#39;: [],</span>
<span class="c1">#              &#39;simulated CNV length in case&#39;: [], &#39;simulated CNV length in ctrl&#39;: []}</span>
<span class="c1">#     while(status):</span>
<span class="c1">#         sample_len = sample_cnv_length(cnv_length, args[&#39;avg_cnv_per_individual&#39;])</span>
<span class="c1">#         sample_blocks = get_sample_blocks(block_counts, len(sample_len))</span>
<span class="c1">#         samples = assign_cnv_to_sample(sample_blocks, sample_len, args[&#39;block_size&#39;])</span>
<span class="c1">#         samples = annotate_samples(samples, refgene)</span>
<span class="c1">#         causal_genes_in_sample = get_causal_genes(causal_genes, samples[&#39;gene_name&#39;].tolist())</span>
<span class="c1">#         p = p_case(args[&#39;prevalence&#39;], len(causal_genes_in_sample), args)</span>
<span class="c1">#         # add the number of causal genes overlapped with simulated CNVs for each simulated sample</span>
<span class="c1">#         debug[&#39;number of causal genes&#39;].append(len(causal_genes_in_sample))</span>
<span class="c1">#         # add the number of genes overlapped with simulated CNVs, both causal and non-causal genes</span>
<span class="c1">#         debug[&#39;number of genes overlap CNV&#39;].append(len( set(samples[&#39;gene_name&#39;].tolist()) ))</span>
<span class="c1">#         #debug[&#39;p&#39;].append(p)</span>
<span class="c1">#         if random.random() &lt; p and len(case_data) &lt; args[&#39;n_case&#39;]:</span>
<span class="c1">#             # sample data is a case</span>
<span class="c1">#             case_data.append(samples)</span>
<span class="c1">#             debug[&#39;p&#39;].append(p)</span>
<span class="c1">#             debug[&#39;simulated CNV length in case&#39;].extend(sample_len)</span>
<span class="c1">#         if random.random() &gt; p and len(ctrl_data) &lt; args[&#39;n_ctrl&#39;]:</span>
<span class="c1">#             # sample data is a control</span>
<span class="c1">#             ctrl_data.append(samples)</span>
<span class="c1">#             debug[&#39;p&#39;].append(p)</span>
<span class="c1">#             debug[&#39;simulated CNV length in ctrl&#39;].extend(sample_len)</span>
<span class="c1">#         if len(case_data) == args[&#39;n_case&#39;] and len(ctrl_data) == args[&#39;n_ctrl&#39;]:</span>
<span class="c1">#             status = 0</span>
<span class="c1">#         debug[&#39;niter&#39;] += 1</span>
<span class="c1">#     debug[&#39;time&#39;][1] = str(datetime.now())</span>
<span class="c1">#     return {&#39;case&#39;: case_data, &#39;ctrl&#39;: ctrl_data, &#39;debug&#39;: debug}</span>
<span class="c1"># def save_data(data, filename):</span>
<span class="c1">#     pickle.dump(data, open(filename, &quot;wb&quot;))</span>
<span class="c1"># def load_data(filename):</span>
<span class="c1">#     return pickle.load(open(filename, &quot;rb&quot;))</span>
<span class="c1"># def run_simulation(args, simulation_id = 0):</span>
<span class="c1">#     np.random.seed(args.seed)</span>
<span class="c1">#     ref_gene = load_reference_gene(args[&#39;refgene_file&#39;])</span>
<span class="c1">#     cnv_data = load_cnv_data(args[&#39;cnv_file&#39;])</span>
<span class="c1">#     sample_data = simulate(ref_gene, pd.concat([cnv_data[args[&#39;case_dataset&#39;]], cnv_data[args[&#39;ctrl_dataset&#39;]]]),</span>
<span class="c1">#                           args, args.causal_genes)</span>
<span class="c1">#     save_data(sample_data, &#39;{}_{}.data.pkl&#39;.format(args[&#39;output&#39;], simulation_id))</span>
<span class="c1">#     return sample_data</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered">
<div class="prompt input_prompt">
</div>
<div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Export-all-to-module-utils.py">Export all to module <code>utils.py</code><a class="anchor-link" href="#Export-all-to-module-utils.py">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[8]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="err">!</span><span class="n">python</span> <span class="n">extract_function</span><span class="o">.</span><span class="n">py</span> <span class="o">--</span><span class="kn">from</span> <span class="nn">Python_Utils.ipynb</span> <span class="o">--</span><span class="n">to</span> <span class="o">../</span><span class="n">analysis</span><span class="o">/</span><span class="n">utils</span><span class="o">.</span><span class="n">py</span> <span class="o">../</span><span class="n">prototype</span><span class="o">/</span><span class="n">utils</span><span class="o">.</span><span class="n">py</span> <span class="n">utils</span><span class="o">.</span><span class="n">py</span>
</pre></div>
</div>
</div>
</div>
<div class="output_wrapper">
<div class="output">
<div class="output_area">
<div class="prompt"></div>
<div class="output_subarea output_stream output_stdout output_text">
<pre>Extracting from Python_Utils.ipynb
Done!
</pre>
</div>
</div>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[20]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="c1"># check the reason that there is no result</span>
<span class="kn">from</span> <span class="nn">utils</span> <span class="kn">import</span> <span class="n">load_data</span>
<span class="n">gene_df</span> <span class="o">=</span> <span class="n">load_data</span><span class="p">(</span><span class="s2">&quot;../analysis/data/calcium_pathway_4000.data.pkl&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[22]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="kn">from</span> <span class="nn">utils</span> <span class="kn">import</span> <span class="n">get_gene_table</span>
<span class="n">gene_table</span> <span class="o">=</span> <span class="n">get_gene_table</span><span class="p">(</span><span class="n">gene_df</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[26]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="kn">from</span> <span class="nn">utils</span> <span class="kn">import</span> <span class="n">get_stats</span>
<span class="n">stats_tble</span> <span class="o">=</span> <span class="n">get_stats</span><span class="p">(</span><span class="n">gene_table</span><span class="p">,</span> <span class="n">sort</span> <span class="o">=</span> <span class="mi">0</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[27]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="n">stats_tble</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span>
</pre></div>
</div>
</div>
</div>
</div>
<hr>
&copy 2016-2017 Min Qiao at <a href="http://xinhelab.org">Xin He lab</a>, the University of Chicago
<p><small>Exported from <a href="http://github.com/gaow/cnv-gene-mapping/blob/ce29b677ae2ce9767b85f83fb4d4104ba8549e78/utils/Python_Utils.ipynb"><code>utils/Python_Utils.ipynb</code></a> committed by Min Qiao on Wed Aug 23 19:52:01 2017 <a href="http://github.com/gaow/cnv-gene-mapping/commit/ce29b677ae2ce9767b85f83fb4d4104ba8549e78">revision 298, ce29b67</a> <a href="https://stephenslab.github.io/ipynb-website/notes.html#Note-about-commit-ids"><span class="fa fa-question-circle"></span></a></small></p>
</div>
</div>
</body>
</html>
