<!DOCTYPE html>
<html>
<head><meta charset="utf-8" />
<title>dbGaP_Proposal</title>

<script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.1.10/require.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/2.0.3/jquery.min.js"></script>








<!-- Custom stylesheet, it must be in the same directory as the html file -->
<link rel="stylesheet" href="custom.css">

<!-- Loading mathjax macro -->
<!-- Load mathjax -->
    <script src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS_HTML"></script>
    <!-- MathJax configuration -->
    <script type="text/x-mathjax-config">
    MathJax.Hub.Config({
        tex2jax: {
            inlineMath: [ ['$','$'], ["\\(","\\)"] ],
            displayMath: [ ['$$','$$'], ["\\[","\\]"] ],
            processEscapes: true,
            processEnvironments: true
        },
        // Center justify equations in code and markdown cells. Elsewhere
        // we use CSS to left justify single line equations in code cells.
        displayAlign: 'center',
        "HTML-CSS": {
            styles: {'.MathJax_Display': {"margin": 0}},
            linebreaks: { automatic: true }
        }
    });
    </script>
    <!-- End of mathjax configuration --></head>

 <link rel="stylesheet" href="https://code.jquery.com/ui/1.11.4/themes/smoothness/jquery-ui.css">

<script type="text/javascript" src="https://ajax.googleapis.com/ajax/libs/jquery/1.9.1/jquery.min.js"></script>
<script type="text/javascript" src="https://ajax.googleapis.com/ajax/libs/jqueryui/1.9.1/jquery-ui.min.js"></script>

<style>  /* defined here in case the main.css below cannot be loaded */
.lev1 {margin-left: 80px}
.lev2 {margin-left: 100px}
.lev3 {margin-left: 120px}
.lev4 {margin-left: 140px}
.lev5 {margin-left: 160px}
.lev6 {margin-left: 180px}
</style>


<link rel="stylesheet" type="text/css" href="../../css/jt.css">
<link rel="stylesheet" type="text/css" href="../../css/toc2.css">

<script src="../../js/toc2.js"></script>
<script src="../../js/docs.js"></script>

<script>
    MathJax.Hub.Config({
        "HTML-CSS": {
            preferredFont: "TeX",
            availableFonts: ["STIX","TeX"],
            styles: {
                scale: 110,
                ".MathJax_Display": {
                    "font-size": "110%",
                }
            }
        }
    });
</script>

<script>
$( document ).ready(function(){

            var cfg={'threshold':3,     // depth of toc (number of levels)
             'number_sections': false,  // sections numbering
             'toc_cell': false,          // useless here
             'toc_window_display': true, // display the toc window
             "toc_section_display": "block", // display toc contents in the window
             'sideBar':true,       // sidebar or floating window
             'navigate_menu':false       // navigation menu (only in liveNotebook -- do not change)
            }

            var st={};                  // some variables used in the script
            st.rendering_toc_cell = false;
            st.config_loaded = false;
            st.extension_initialized=false;
            st.nbcontainer_marginleft = $('#notebook-container').css('margin-left')
            st.nbcontainer_marginright = $('#notebook-container').css('margin-right')
            st.nbcontainer_width = $('#notebook-container').css('width')
            st.oldTocHeight = undefined
            st.cell_toc = undefined;
            st.toc_index=0;

            // fire the main function with these parameters


            table_of_contents(cfg,st);


            var file=writeupDict[$("h1:first").attr("id")];
            $("#toc-level0 a").css("color","#126dce");
            $('a[href="#'+$("h1:first").attr("id")+'"]').hide()
            
            var tuts=writeup;
            var pos=writeup.indexOf(file);
        
            for (var a=pos;a>=0;a--){
                  var name=tuts[a]
                  $('<li><a href="'+name+'.html">'+name.replace(/_/g," ")+'</a></li>').insertBefore("#toc-level0 li:eq(0)");
            }
            $('a[href="'+file+'.html'+'"]').css("color","#126dce");

            $('<li id="indexHome"><a href="../../writeup.html"><b>Tutorials Home<b></a></li>').insertBefore("#toc-level0 li:eq(0)");
            for (var a=pos+1;a<tuts.length;a++){
                  var name=tuts[a]
                  $(".toc #toc-level0").append('<li><a href="'+name+'.html">'+name.replace(/_/g," ")+'</a></li>');
            }
            $("#toc-header").hide();


    });
</script>
<body>
  <div tabindex="-1" id="notebook" class="border-box-sizing">
    <div class="container" id="notebook-container">

<div class="cell border-box-sizing text_cell rendered">
<div class="prompt input_prompt">
</div>
<div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p>The application below follows from templates on <a href="https://github.com/aaronwolen/gibbs-brain-project/wiki/Other-dbGap-applications">this site</a>.</p>
<h2 id="Integration-of-CNV-and-SNV-for-gene-mapping-of-psychiatric-disorders:-methods-and-data-analysis">Integration of CNV and SNV for gene mapping of psychiatric disorders: methods and data analysis<a class="anchor-link" href="#Integration-of-CNV-and-SNV-for-gene-mapping-of-psychiatric-disorders:-methods-and-data-analysis">&#182;</a></h2><p>OMB control number: 0925-0670 Expiration date: 03/31/2019</p>
<h2 id="Non-technical-summary">Non-technical summary<a class="anchor-link" href="#Non-technical-summary">&#182;</a></h2><p>We seek to develop methods that leverage genome-wide copy number variations (CNVs) for gene mapping of psychiatric diseases. CNVs are large genomic insertion or deletion events, an important source of genetic variation affecting neuropsychiatric disorders such as autism and intellectual disability. Recent evidence has implicated that CNVs and single-nucleotide variants (SNVs) converge on the same set of genes in etiology of many psychiatric traits. However, current methods for gene mapping typically analyze one type of data at a time -- either focusing on SNV association mapping, or CNV based gene-set analysis. The fact that CNVs often spread multiple genes raises analytic challenge of identifying the causal genes among all genes involved. We will develop a principled statistical framework to integrate SNV and CNV data at the gene-level, accommodating both trio-sequencing and case-control studies, to increase power of psychiatric gene discovery.</p>
<h2 id="Research-Use-Statement-(RUS)">Research Use Statement (RUS)<a class="anchor-link" href="#Research-Use-Statement-(RUS)">&#182;</a></h2><p>Recent studies have indicated that SNVs and CNVs of psychiatric disorders converge on the same set of target genes, making joint SNV-CNV analysis an extremely attractive strategy. We recently developed a method for integrating autism CNV and SNV data (Sanders &amp; He et al, Neuron, 2015). The method extracts gene level evidence from CNV data, and then combine with gene-level test from SNV data. The method, however, suffers from several limitations: for instance, it assumes one causal gene per CNV; and it assumes all CNVs are independent (this would be violated for overlapping CNVs). We will extend this work and develop a more rigorous statistical model to infer causal genes from CNV data. Our idea is that whether a CNV is associated with autism depends on whether it overlaps with autism genes. Thus we can use the Bayes rule to infer the most probable risk genes from observed CNVs. This model allows multiple causal genes in one CNV and can easily incorporate gene-level information from SNV data. To develop and test this method, we need access to individual-level genotype and phenotype datasets specific for autism (accession number: phs000267.v4) containing the CNV data from a recent paper by Pinto et al, in AJHG, 2014 (PMID: 24768552). Our method will generate summary statistics (e.g. odds ratio estimates, standard errors) from the CNV dataset and combine with summary statistics from SNV data. By using summary statistics the data integration project will not create additional risks to participants. We acknowledge the use restrictions of each dataset and will not use any of these datasets to seek to reveal the identity of individuals.</p>

</div>
</div>
</div>
    </div>
  </div>
</body>
</html>
