{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Exploratory of a toy CNV dataset in VCF format"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "## install package saasCNV, first install DNAcopy\n",
    "## source(\"https://bioconductor.org/biocLite.R\")\n",
    "## biocLite(\"DNAcopy\")\n",
    "## install.packages(\"saasCNV\")\n",
    "library(saasCNV)\n",
    "## first covert VCF file to a data frame\n",
    "vcf_table <- vcf2txt(vcf.file=\"WES_example.vcf\", normal.col = 10, tumor.col = 11)\n",
    "\n",
    "## not used\n",
    "## read.delim is used to read .txt.gz, not vcf\n",
    "## vcf_table <- read.delim(file=\"WES_example.vcf\", as.is=TRUE)\n",
    "## use the data generated by vcf2txt, whose input is vcf\n",
    "\n",
    "## construct data frame for CNV inference with NGS data\n",
    "seq.data <- cnv.data(vcf=vcf_table, min.chr.probe=100, verbose=FALSE)\n",
    "## see how seq.data looks like\n",
    "head(seq.data)\n",
    "data(seq.data)\n",
    "\n",
    "## ??? contained in library saasCNV ???\n",
    "data(seq.segs.merge)\n",
    "## CNV Calling from Sequencing Data\n",
    "seq.cnv <- cnv.call(data=seq.data, sample.id=\"PT116\", segs.stat=seq.segs.merge, maxL=2000, N=1000, pvalue.cutoff=0.05)\n",
    "\n",
    "## the last column is CNV\n",
    "seq.cnv[, ncol(seq.cnv)]\n",
    "table(seq.cnv[, ncol(seq.cnv)])\n",
    "## gain       LOH      loss    normal undecided \n",
    "##   14        14        35         4         7 \n",
    "head(seq.cnv)\n",
    "data(seq.cnv)\n",
    "## visualize genome-wide SCNA profile in 2D cluster plot\n",
    "diagnosis.cluster.plot(segs=seq.cnv, chrs=sub(\"^chr\",\"\",unique(seq.cnv$chr)), min.snps=10, max.cex=3, ref.num.probe=1000)\n",
    "\n",
    "## visualize genome-wide SCNA profile\n",
    "genome.wide.plot(data=seq.data, segs=seq.cnv, sample.id=\"PT116\", chrs=sub(\"^chr\",\"\",unique(seq.cnv$chr)), cex=0.3)\n",
    "\n",
    "## Joint Segmentation on log2ratio and log2mBAF Dimensions, for chr 1-22\n",
    "seq.segs <- joint.segmentation(data=seq.data, min.snps=10, global.pval.cutoff=1e-4, max.chpts=30, verbose=TRUE)\n",
    "\n",
    "## Merge Adjacent Segments\n",
    "seq.segs.merge <- merging.segments(data=seq.data, segs.stat=seq.segs, use.null.data=TRUE, N=1000, maxL=2000, merge.pvalue.cutoff=0.05, verbose=TRUE)\n",
    "head(seq.segs.merge)\n",
    "data(seq.segs.merge)\n",
    "\n",
    "## CNV Analysis Pipeline for WGS and WES Data\n",
    "sample.id <- \"WES_0116\"\n",
    "output.dir <- file.path(getwd(), \"test_saasCNV\")\n",
    "NGS.CNV(vcf=vcf_table, output.dir=output.dir, sample.id=sample.id, min.chr.probe=100, min.snps=10, joint.segmentation.pvalue.cutoff=1e-4, max.chpts=30, do.merge=TRUE, use.null.data=TRUE, num.perm=1000, maxL=2000, merge.pvalue.cutoff=0.05, do.cnvcall.on.merge=TRUE, cnvcall.pvalue.cutoff=0.05, do.plot=TRUE, cex=0.3, ref.num.probe=1000, do.gene.anno=TRUE, gene.anno.file=\"refGene_hg19.txt.gz\", seed=123456789, verbose=TRUE)\n",
    "\n",
    "## Gene Annotation\n",
    "gene.anno <- read.delim(file=\"refGene_hg19.txt.gz\", as.is=TRUE, comment.char=\"\")\n",
    "seq.cnv.anno <- reannotate.CNV.res(res=seq.cnv, gene=gene.anno, only.CNV=TRUE)\n",
    "\n",
    "## CNV Analysis Pipeline for SNP array Data\n",
    "## download snp_table.txt.gz\n",
    "## url <- \"https://zhangz05.u.hpc.mssm.edu/saasCNV/data/snp_table.txt.gz\"\n",
    "snp_table <- read.delim(file=\"snp_table.txt.gz\", as.is=TRUE)\n",
    "SNP.CNV(snp=snp_table, output.dir=output.dir, sample.id=sample.id, min.chr.probe=100, min.snps=10, joint.segmentation.pvalue.cutoff=1e-4, max.chpts=30, do.merge=TRUE, use.null.data=TRUE, num.perm=1000, maxL=5000, merge.pvalue.cutoff=0.05, do.cnvcall.on.merge=TRUE, cnvcall.pvalue.cutoff=0.05, do.boundary.refine=TRUE, do.plot=TRUE, cex=0.3, ref.num.probe=5000, do.gene.anno=TRUE, gene.anno.file=\"refGene_hg19.txt.gz\", seed=123456789, verbose=TRUE)\n",
    "\n",
    "## Construct Data Frame for CNV Inference with SNP Array Data\n",
    "snp.data <- snp.cnv.data(snp=snp_table, min.chr.probe=100, verbose=TRUE)\n",
    "head(snp.data)\n",
    "\n",
    "## Refine Segment Boundaries\n",
    "data(snp.cnv)\n",
    "snp.cnv.refine <- snp.refine.boundary(data=snp.data, segs.stat=snp.cnv)\n",
    "head(snp.cnv.refine)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "3.4.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
