var tipuesearch = {"pages": [
{"mainTitle": "Analysis", "title": "Analysis", "text": "Analysis ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis.html#Analysis"},
{"mainTitle": "Analysis", "title": "Notebooks", "text": "Notebooks ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis.html#Notebooks"},
{"mainTitle": "Analysis", "title": "2017/10", "text": "2017 10 ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis.html#2017/10"},
{"mainTitle": "Analysis", "title": "2017/09", "text": "2017 09 ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis.html#2017/09"},
{"mainTitle": "Analysis", "title": "2017/08", "text": "2017 08 ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis.html#2017/08"},
{"mainTitle": "Analysis", "title": "2017/07", "text": "2017 07 ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis.html#2017/07"},
{"mainTitle": "Analysis", "title": "2017/06", "text": "2017 06 ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis.html#2017/06"},
{"mainTitle": "Analysis", "title": "2017/05", "text": "2017 05 ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis.html#2017/05"},
{"mainTitle": "Analysis", "title": "2017/04", "text": "2017 04 ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis.html#2017/04"},
{"mainTitle": "Analysis", "title": "2017/02", "text": "2017 02 ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis.html#2017/02"},
{"mainTitle": "Analysis", "title": "Pipelines", "text": "Pipelines ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis.html#Pipelines"},
{"mainTitle": "Prototype", "title": "Prototype", "text": "Prototype ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/prototype.html", "url": "http://gaow.github.io/cnv-gene-mapping/prototype.html#Prototype"},
{"mainTitle": "Prototype", "title": "2017/08", "text": "2017 08 ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/prototype.html", "url": "http://gaow.github.io/cnv-gene-mapping/prototype.html#2017/08"},
{"mainTitle": "Prototype", "title": "2017/07", "text": "2017 07 ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/prototype.html", "url": "http://gaow.github.io/cnv-gene-mapping/prototype.html#2017/07"},
{"mainTitle": "Prototype", "title": "2017/06", "text": "2017 06 ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/prototype.html", "url": "http://gaow.github.io/cnv-gene-mapping/prototype.html#2017/06"},
{"mainTitle": "Prototype", "title": "2017/05", "text": "2017 05 ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/prototype.html", "url": "http://gaow.github.io/cnv-gene-mapping/prototype.html#2017/05"},
{"mainTitle": "Prototype", "title": "2017/01", "text": "2017 01 ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/prototype.html", "url": "http://gaow.github.io/cnv-gene-mapping/prototype.html#2017/01"},
{"mainTitle": "Prototype", "title": "2016/08", "text": "2016 08 ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/prototype.html", "url": "http://gaow.github.io/cnv-gene-mapping/prototype.html#2016/08"},
{"mainTitle": "Dsc", "title": "Dsc", "text": "Dsc ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc.html#Dsc"},
{"mainTitle": "Dsc", "title": "2019/06", "text": "2019 06 ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc.html#2019/06"},
{"mainTitle": "Dsc", "title": "2019/07", "text": "2019 07 ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc.html#2019/07"},
{"mainTitle": "Dsc", "title": "2019/08", "text": "2019 08 ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc.html#2019/08"},
{"mainTitle": "Dsc", "title": "2019/09", "text": "2019 09 ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc.html#2019/09"},
{"mainTitle": "Dsc", "title": "2019/10", "text": "2019 10 ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc.html#2019/10"},
{"mainTitle": "Dsc", "title": "2020/01", "text": "2020 01 ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc.html#2020/01"},
{"mainTitle": "Dsc", "title": "2020/02", "text": "2020 02 ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc.html#2020/02"},
{"mainTitle": "Setup", "title": "Setup", "text": "Setup ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/setup.html", "url": "http://gaow.github.io/cnv-gene-mapping/setup.html#Setup"},
{"mainTitle": "Writeup", "title": "Writeup", "text": "Writeup ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/writeup.html", "url": "http://gaow.github.io/cnv-gene-mapping/writeup.html#Writeup"},
{"mainTitle": "Writeup", "title": "2019/06", "text": "2019 06 ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/writeup.html", "url": "http://gaow.github.io/cnv-gene-mapping/writeup.html#2019/06"},
{"mainTitle": "CNV gene mapping", "title": "CNV gene mapping", "text": "CNV gene mapping ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/index.html", "url": "http://gaow.github.io/cnv-gene-mapping/index.html#CNV-gene-mapping"},
{"mainTitle": "CNV gene mapping", "title": "CNV", "text": "CNV ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/index.html", "url": "http://gaow.github.io/cnv-gene-mapping/index.html#CNV"},
{"mainTitle": "CNV gene mapping", "title": "Challenge and aim", "text": "Challenge and aim ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/index.html", "url": "http://gaow.github.io/cnv-gene-mapping/index.html#Challenge-and-aim"},
{"mainTitle": "CNV gene mapping", "title": "Solution", "text": "Solution ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/index.html", "url": "http://gaow.github.io/cnv-gene-mapping/index.html#Solution"},
{"mainTitle": "Toy example of multi-gene configuration overlapped with multi-CNV", "title": "Toy example of multi-gene configuration overlapped with multi-CNV", "text": "Toy example of multi-gene configuration overlapped with multi-CNV ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170705_toy_example_multi_outline.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170705_toy_example_multi_outline.html#Toy-example-of-multi-gene-configuration-overlapped-with-multi-CNV"},
{"mainTitle": "Toy example of multi-gene configuration overlapped with multi-CNV", "title": "Simulate CNVs", "text": "Simulate CNVs ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170705_toy_example_multi_outline.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170705_toy_example_multi_outline.html#Simulate-CNVs"},
{"mainTitle": "Toy example of multi-gene configuration overlapped with multi-CNV", "title": "Assign CNVs to cases and controls", "text": "Assign CNVs to cases and controls ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170705_toy_example_multi_outline.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170705_toy_example_multi_outline.html#Assign-CNVs-to-cases-and-controls"},
{"mainTitle": "Toy example of multi-gene configuration overlapped with multi-CNV", "title": "Results", "text": "Results ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170705_toy_example_multi_outline.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170705_toy_example_multi_outline.html#Results"},
{"mainTitle": "Toy example of multi-gene configuration overlapped with multi-CNV", "title": "CNV maximum length overlap with 10 genes; sample size 2000; 5th gene is causal", "text": "CNV maximum length overlap with 10 genes sample size 2000 5th gene is causal ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170705_toy_example_multi_outline.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170705_toy_example_multi_outline.html#CNV-maximum-length-overlap-with-10-genes;-sample-size-2000;-5th-gene-is-causal"},
{"mainTitle": "Toy example of multi-gene configuration overlapped with multi-CNV", "title": "Constant $0$", "text": "Constant 0 ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170705_toy_example_multi_outline.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170705_toy_example_multi_outline.html#Constant-$0$"},
{"mainTitle": "Toy example of multi-gene configuration overlapped with multi-CNV", "title": "Constant $0.1$", "text": "Constant 0.1 ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170705_toy_example_multi_outline.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170705_toy_example_multi_outline.html#Constant-$0.1$"},
{"mainTitle": "Toy example of multi-gene configuration overlapped with multi-CNV", "title": "Constant $0.5$", "text": "Constant 0.5 ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170705_toy_example_multi_outline.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170705_toy_example_multi_outline.html#Constant-$0.5$"},
{"mainTitle": "Toy example of multi-gene configuration overlapped with multi-CNV", "title": "Constant $1.0$", "text": "Constant 1.0 ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170705_toy_example_multi_outline.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170705_toy_example_multi_outline.html#Constant-$1.0$"},
{"mainTitle": "Toy example of multi-gene configuration overlapped with multi-CNV", "title": "DAP results annotation", "text": "DAP results annotation ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170705_toy_example_multi_outline.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170705_toy_example_multi_outline.html#DAP-results-annotation"},
{"mainTitle": "R: use varbvs for all datasets from simulation and save as .RDS", "title": "R: use varbvs for all datasets from simulation and save as .RDS", "text": "R use varbvs for all datasets from simulation and save as .RDS ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170525_R_varbvs.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170525_R_varbvs.html#R:-use-varbvs-for-all-datasets-from-simulation-and-save-as-.RDS"},
{"mainTitle": "Integration of $\beta$", "title": "Integration of $\beta$", "text": "Integration of \beta ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20171030_Obtain_beta.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20171030_Obtain_beta.html#Integration-of-$\beta$"},
{"mainTitle": "DAP on simulation results", "title": "DAP on simulation results", "text": "DAP on simulation results ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170806_Overview_of_dap_on_simulation.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170806_Overview_of_dap_on_simulation.html#DAP-on-simulation-results"},
{"mainTitle": "DAP on simulation results", "title": "Data transformation: transform $y \in \{0,1\}$ to $y \in \{0,t\}$", "text": "Data transformation transform y \in \ 0,1\ to y \in \ 0,t\ ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170806_Overview_of_dap_on_simulation.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170806_Overview_of_dap_on_simulation.html#Data-transformation:-transform-$y-\in-\{0,1\}$-to-$y-\in-\{0,t\}$"},
{"mainTitle": "DAP on simulation results", "title": "Prepare DAP input", "text": "Prepare DAP input ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170806_Overview_of_dap_on_simulation.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170806_Overview_of_dap_on_simulation.html#Prepare-DAP-input"},
{"mainTitle": "DAP on simulation results", "title": "Effect Size Grid file: prior effect size specifications for causal variants.", "text": "Effect Size Grid file prior effect size specifications for causal variants. ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170806_Overview_of_dap_on_simulation.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170806_Overview_of_dap_on_simulation.html#Effect-Size-Grid-file:-prior-effect-size-specifications-for-causal-variants."},
{"mainTitle": "DAP on simulation results", "title": "Prior file: specifies SNP-level/gene-level prior for each candidate SNP/gene.", "text": "Prior file specifies SNP-level gene-level prior for each candidate SNP gene. ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170806_Overview_of_dap_on_simulation.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170806_Overview_of_dap_on_simulation.html#Prior-file:-specifies-SNP-level/gene-level-prior-for-each-candidate-SNP/gene."},
{"mainTitle": "DAP on simulation results", "title": "Run DAP", "text": "Run DAP ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170806_Overview_of_dap_on_simulation.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170806_Overview_of_dap_on_simulation.html#Run-DAP"},
{"mainTitle": "Obtain .feather file from .pkl to analyze in R", "title": "Obtain .feather file from .pkl to analyze in R", "text": "Obtain .feather file from .pkl to analyze in R ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170425_feather.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170425_feather.html#Obtain-.feather-file-from-.pkl-to-analyze-in-R"},
{"mainTitle": "Toy example of gene configuration overlapped with CNV", "title": "Toy example of gene configuration overlapped with CNV", "text": "Toy example of gene configuration overlapped with CNV ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170623_toy_example_outline.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170623_toy_example_outline.html#Toy-example-of-gene-configuration-overlapped-with-CNV"},
{"mainTitle": "Toy example of gene configuration overlapped with CNV", "title": "Convergence of genes and CNV", "text": "Convergence of genes and CNV ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170623_toy_example_outline.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170623_toy_example_outline.html#Convergence-of-genes-and-CNV"},
{"mainTitle": "Toy example of gene configuration overlapped with CNV", "title": "Cases and Controls assignment", "text": "Cases and Controls assignment ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170623_toy_example_outline.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170623_toy_example_outline.html#Cases-and-Controls-assignment"},
{"mainTitle": "Toy example of gene configuration overlapped with CNV", "title": "3 adjacent genes", "text": "3 adjacent genes ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170623_toy_example_outline.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170623_toy_example_outline.html#3-adjacent-genes"},
{"mainTitle": "Toy example of gene configuration overlapped with CNV", "title": "4 adjacent genes", "text": "4 adjacent genes ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170623_toy_example_outline.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170623_toy_example_outline.html#4-adjacent-genes"},
{"mainTitle": "Toy example of gene configuration overlapped with CNV", "title": "15 adjacent genes. CNVs overlap with maximumly 10 genes.", "text": "15 adjacent genes. CNVs overlap with maximumly 10 genes. ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170623_toy_example_outline.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170623_toy_example_outline.html#15-adjacent-genes.-CNVs-overlap-with-maximumly-10-genes."},
{"mainTitle": "Toy example of gene configuration overlapped with CNV", "title": "Methods", "text": "Methods ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170623_toy_example_outline.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170623_toy_example_outline.html#Methods"},
{"mainTitle": "Toy example of gene configuration overlapped with CNV", "title": "Results", "text": "Results ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170623_toy_example_outline.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170623_toy_example_outline.html#Results"},
{"mainTitle": "Toy example of gene configuration overlapped with CNV", "title": "4 genes, extreme case, sample size 2000, constant 1.0.", "text": "4 genes, extreme case, sample size 2000, constant 1.0. ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170623_toy_example_outline.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170623_toy_example_outline.html#4-genes,-extreme-case,-sample-size-2000,-constant-1.0."},
{"mainTitle": "Toy example of gene configuration overlapped with CNV", "title": "4 genes, sample size 1320, constant 1.0", "text": "4 genes, sample size 1320, constant 1.0 ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170623_toy_example_outline.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170623_toy_example_outline.html#4-genes,-sample-size-1320,-constant-1.0"},
{"mainTitle": "Toy example of gene configuration overlapped with CNV", "title": "4 genes, sample size 1320, constant 0, constant 1.0", "text": "4 genes, sample size 1320, constant 0, constant 1.0 ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170623_toy_example_outline.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170623_toy_example_outline.html#4-genes,-sample-size-1320,-constant-0,-constant-1.0"},
{"mainTitle": "Toy example of gene configuration overlapped with CNV", "title": "3 genes, sample size 1210, constant 1.0", "text": "3 genes, sample size 1210, constant 1.0 ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170623_toy_example_outline.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170623_toy_example_outline.html#3-genes,-sample-size-1210,-constant-1.0"},
{"mainTitle": "Toy example of gene configuration overlapped with CNV", "title": "10 genes, sample size 1250, constant 0 (randomly assign to cases and controls)", "text": "10 genes, sample size 1250, constant 0 randomly assign to cases and controls ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170623_toy_example_outline.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170623_toy_example_outline.html#10-genes,-sample-size-1250,-constant-0-(randomly-assign-to-cases-and-controls)"},
{"mainTitle": "Toy example of gene configuration overlapped with CNV", "title": "10 genes, sample size 1250, constant 1.0. The 8th gene is causal.", "text": "10 genes, sample size 1250, constant 1.0. The 8th gene is causal. ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170623_toy_example_outline.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170623_toy_example_outline.html#10-genes,-sample-size-1250,-constant-1.0.-The-8th-gene-is-causal."},
{"mainTitle": "Toy example of gene configuration overlapped with CNV", "title": "PIP for different constants", "text": "PIP for different constants ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170623_toy_example_outline.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170623_toy_example_outline.html#PIP-for-different-constants"},
{"mainTitle": "Toy example of gene configuration overlapped with CNV", "title": "Issue", "text": "Issue ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170623_toy_example_outline.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170623_toy_example_outline.html#Issue"},
{"mainTitle": "Toy example of gene configuration overlapped with CNV", "title": "Constant added might be too large", "text": "Constant added might be too large ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170623_toy_example_outline.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170623_toy_example_outline.html#Constant-added-might-be-too-large"},
{"mainTitle": "Toy example of gene configuration overlapped with CNV", "title": "Size for each pattern might not accurate", "text": "Size for each pattern might not accurate ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170623_toy_example_outline.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170623_toy_example_outline.html#Size-for-each-pattern-might-not-accurate"},
{"mainTitle": "Implement TORUS by using CNV data of Schizophrenia in Sweden", "title": "Implement TORUS by using CNV data of Schizophrenia in Sweden", "text": "Implement TORUS by using CNV data of Schizophrenia in Sweden ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20171010_swcnv_TORUS.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20171010_swcnv_TORUS.html#Implement-TORUS-by-using-CNV-data-of-Schizophrenia-in-Sweden"},
{"mainTitle": "Implement TORUS by using CNV data of Schizophrenia in Sweden", "title": "1. Use Sweden Schizphrenia individual data to obtain gene and CNV overlap configuration; columns represent genes, rows represent individuals", "text": "1. Use Sweden Schizphrenia individual data to obtain gene and CNV overlap configuration columns represent genes, rows represent individuals ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20171010_swcnv_TORUS.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20171010_swcnv_TORUS.html#1.-Use-Sweden-Schizphrenia-individual-data-to-obtain-gene-and-CNV-overlap-configuration;-columns-represent-genes,-rows-represent-individuals"},
{"mainTitle": "Implement TORUS by using CNV data of Schizophrenia in Sweden", "title": "2. Export data as pkl and feather format to use in varbvs", "text": "2. Export data as pkl and feather format to use in varbvs ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20171010_swcnv_TORUS.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20171010_swcnv_TORUS.html#2.-Export-data-as-pkl-and-feather-format-to-use-in-varbvs"},
{"mainTitle": "Implement TORUS by using CNV data of Schizophrenia in Sweden", "title": "3. Obtain CNV blocks and use it to perform TORUS", "text": "3. Obtain CNV blocks and use it to perform TORUS ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20171010_swcnv_TORUS.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20171010_swcnv_TORUS.html#3.-Obtain-CNV-blocks-and-use-it-to-perform-TORUS"},
{"mainTitle": "Implement TORUS by using CNV data of Schizophrenia in Sweden", "title": "The table of overlap of CNVs (Sweden data) and genes", "text": "The table of overlap of CNVs Sweden data and genes ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20171010_swcnv_TORUS.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20171010_swcnv_TORUS.html#The-table-of-overlap-of-CNVs-(Sweden-data)-and-genes"},
{"mainTitle": "Implement TORUS by using CNV data of Schizophrenia in Sweden", "title": "1. Get blocks of CNVs, and genes", "text": "1. Get blocks of CNVs, and genes ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20171010_swcnv_TORUS.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20171010_swcnv_TORUS.html#1.-Get-blocks-of-CNVs,-and-genes"},
{"mainTitle": "Implement TORUS by using CNV data of Schizophrenia in Sweden", "title": "2. Export overlaps of CNVs and genes as .feather format", "text": "2. Export overlaps of CNVs and genes as .feather format ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20171010_swcnv_TORUS.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20171010_swcnv_TORUS.html#2.-Export-overlaps-of-CNVs-and-genes-as-.feather-format"},
{"mainTitle": "Implement TORUS by using CNV data of Schizophrenia in Sweden", "title": "Run TORUS", "text": "Run TORUS ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20171010_swcnv_TORUS.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20171010_swcnv_TORUS.html#Run-TORUS"},
{"mainTitle": "Plots of p-value and LFDR for all genes in simulated dataset", "title": "Plots of p-value and LFDR for all genes in simulated dataset", "text": "Plots of p-value and LFDR for all genes in simulated dataset ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170504_lfdr_pvalue_dist.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170504_lfdr_pvalue_dist.html#Plots-of-p-value-and-LFDR-for-all-genes-in-simulated-dataset"},
{"mainTitle": "Use TORUS to estimate α for toy of multi-CNV-causal-gene example", "title": "Use TORUS to estimate α for toy of multi-CNV-causal-gene example", "text": "Use TORUS to estimate for toy of multi-CNV-causal-gene example ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170906_Overview_TORUS_Estimate_Alpha.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170906_Overview_TORUS_Estimate_Alpha.html#Use-TORUS-to-estimate-α-for-toy-of-multi-CNV-causal-gene-example"},
{"mainTitle": "Use TORUS to estimate α for toy of multi-CNV-causal-gene example", "title": "Methods", "text": "Methods ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170906_Overview_TORUS_Estimate_Alpha.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170906_Overview_TORUS_Estimate_Alpha.html#Methods"},
{"mainTitle": "Use TORUS to estimate α for toy of multi-CNV-causal-gene example", "title": "Results for toy example", "text": "Results for toy example ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170906_Overview_TORUS_Estimate_Alpha.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170906_Overview_TORUS_Estimate_Alpha.html#Results-for-toy-example"},
{"mainTitle": "Use TORUS to estimate α for toy of multi-CNV-causal-gene example", "title": "Gene annotation has two categories, 0 for non-causal genes and 1 for causal genes.", "text": "Gene annotation has two categories, 0 for non-causal genes and 1 for causal genes. ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170906_Overview_TORUS_Estimate_Alpha.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170906_Overview_TORUS_Estimate_Alpha.html#Gene-annotation-has-two-categories,-0-for-non-causal-genes-and-1-for-causal-genes."},
{"mainTitle": "Use TORUS to estimate α for toy of multi-CNV-causal-gene example", "title": "Gene annotation has five categories, 1 for causal genes, 0,2,3,4,5 randomly for noncausal genes.", "text": "Gene annotation has five categories, 1 for causal genes, 0,2,3,4,5 randomly for noncausal genes. ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170906_Overview_TORUS_Estimate_Alpha.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170906_Overview_TORUS_Estimate_Alpha.html#Gene-annotation-has-five-categories,-1-for-causal-genes,-0,2,3,4,5-randomly-for-noncausal-genes."},
{"mainTitle": "Use TORUS to estimate α for toy of multi-CNV-causal-gene example", "title": "Another selection of category for noncausal: define gene 2 and 14 as category 2, and gene 3 to 13 are randomly assigned to 0,3,4,5", "text": "Another selection of category for noncausal define gene 2 and 14 as category 2, and gene 3 to 13 are randomly assigned to 0,3,4,5 ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170906_Overview_TORUS_Estimate_Alpha.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170906_Overview_TORUS_Estimate_Alpha.html#Another-selection-of-category-for-noncausal:-define-gene-2-and-14-as-category-2,-and-gene-3-to-13-are-randomly-assigned-to-0,3,4,5"},
{"mainTitle": "Use TORUS to estimate α for toy of multi-CNV-causal-gene example", "title": "Results for simulation of calcium pathway", "text": "Results for simulation of calcium pathway ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170906_Overview_TORUS_Estimate_Alpha.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170906_Overview_TORUS_Estimate_Alpha.html#Results-for-simulation-of-calcium-pathway"},
{"mainTitle": "Use TORUS to estimate α for toy of multi-CNV-causal-gene example", "title": "Use Sweden Schizophrenia data - calcium pathway, to perform TORUS", "text": "Use Sweden Schizophrenia data - calcium pathway, to perform TORUS ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170906_Overview_TORUS_Estimate_Alpha.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170906_Overview_TORUS_Estimate_Alpha.html#Use-Sweden-Schizophrenia-data---calcium-pathway,-to-perform-TORUS"},
{"mainTitle": "Implement of table 1 and Figure 3 in CNV in schizophrenia in Sweden", "title": "Implement of table 1 and Figure 3 in CNV in schizophrenia in Sweden", "text": "Implement of table 1 and Figure 3 in CNV in schizophrenia in Sweden ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170721_swcnv.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170721_swcnv.html#Implement-of-table-1-and-Figure-3-in-CNV-in-schizophrenia-in-Sweden"},
{"mainTitle": "Implement of table 1 and Figure 3 in CNV in schizophrenia in Sweden", "title": "Table 1", "text": "Table 1 ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170721_swcnv.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170721_swcnv.html#Table-1"},
{"mainTitle": "Implement of table 1 and Figure 3 in CNV in schizophrenia in Sweden", "title": "Figure 3", "text": "Figure 3 ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170721_swcnv.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170721_swcnv.html#Figure-3"},
{"mainTitle": "Gene Mapping with CNV", "title": "Use data from the paper of "Contribution of CNV to Schizophrenia" for analysis", "text": "Use data from the paper of 'Contribution of CNV to Schizophrenia' for analysis ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20171005_SCZ_CNV.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20171005_SCZ_CNV.html#Use-data-from-the-paper-of-"Contribution-of-CNV-to-Schizophrenia"-for-analysis"},
{"mainTitle": "Obtain power and type I error for each simulated dataset (.pkl)", "title": "Obtain power and type I error for each simulated dataset (.pkl)", "text": "Obtain power and type I error for each simulated dataset .pkl ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170503_Power_typeIerror.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170503_Power_typeIerror.html#Obtain-power-and-type-I-error-for-each-simulated-dataset-(.pkl)"},
{"mainTitle": "Toy example of multiple gene overlapped with CNVs", "title": "Toy example of multiple gene overlapped with CNVs", "text": "Toy example of multiple gene overlapped with CNVs ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170627_toy_multigenes.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170627_toy_multigenes.html#Toy-example-of-multiple-gene-overlapped-with-CNVs"},
{"mainTitle": "Overview of DAP and TORUS", "title": "Overview of DAP and TORUS", "text": "Overview of DAP and TORUS ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170801_Overview_TORUS_DAP.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170801_Overview_TORUS_DAP.html#Overview-of-DAP-and-TORUS"},
{"mainTitle": "Overview of DAP and TORUS", "title": "TORUS: QTL discovery utilizing genomic annotations; Implement a computational procedure for discovering molecular QTLs incorporating genomic annotations.", "text": "TORUS QTL discovery utilizing genomic annotations Implement a computational procedure for discovering molecular QTLs incorporating genomic annotations. ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170801_Overview_TORUS_DAP.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170801_Overview_TORUS_DAP.html#TORUS:-QTL-discovery-utilizing-genomic-annotations;-Implement-a-computational-procedure-for-discovering-molecular-QTLs-incorporating-genomic-annotations."},
{"mainTitle": "Overview of DAP and TORUS", "title": "DAP: deterministic approximation of posterior; Enables highly efficient and accurate joint enrichment analysis and identification of multiple causal variants.", "text": "DAP deterministic approximation of posterior Enables highly efficient and accurate joint enrichment analysis and identification of multiple causal variants. ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170801_Overview_TORUS_DAP.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170801_Overview_TORUS_DAP.html#DAP:-deterministic-approximation-of-posterior;-Enables-highly-efficient-and-accurate-joint-enrichment-analysis-and-identification-of-multiple-causal-variants."},
{"mainTitle": "Overview of DAP and TORUS", "title": "Effect Size Grid File", "text": "Effect Size Grid File ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170801_Overview_TORUS_DAP.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170801_Overview_TORUS_DAP.html#Effect-Size-Grid-File"},
{"mainTitle": "Overview of DAP and TORUS", "title": "Prior file", "text": "Prior file ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170801_Overview_TORUS_DAP.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170801_Overview_TORUS_DAP.html#Prior-file"},
{"mainTitle": "Results of toy example for multi-causal-gene in a region", "title": "Results of toy example for multi-causal-gene in a region", "text": "Results of toy example for multi-causal-gene in a region ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170905_Results_toy_example_multi_causal_gene.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170905_Results_toy_example_multi_causal_gene.html#Results-of-toy-example-for-multi-causal-gene-in-a-region"},
{"mainTitle": "Results of toy example for multi-causal-gene in a region", "title": "Two or more causal genes in a region with multiple CNVs and genes", "text": "Two or more causal genes in a region with multiple CNVs and genes ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170905_Results_toy_example_multi_causal_gene.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170905_Results_toy_example_multi_causal_gene.html#Two-or-more-causal-genes-in-a-region-with-multiple-CNVs-and-genes"},
{"mainTitle": "DAP on simulation results for calcium pathway", "title": "DAP on simulation results for calcium pathway", "text": "DAP on simulation results for calcium pathway ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170726_Dap_TORUS_on_simulation_for_Calcium_Pathway.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170726_Dap_TORUS_on_simulation_for_Calcium_Pathway.html#DAP-on-simulation-results-for-calcium-pathway"},
{"mainTitle": "DAP on simulation results for calcium pathway", "title": "Data transform: $y \in \{0,1\}$ to $y \in \{0,t\}$", "text": "Data transform y \in \ 0,1\ to y \in \ 0,t\ ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170726_Dap_TORUS_on_simulation_for_Calcium_Pathway.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170726_Dap_TORUS_on_simulation_for_Calcium_Pathway.html#Data-transform:-$y-\in-\{0,1\}$-to-$y-\in-\{0,t\}$"},
{"mainTitle": "DAP on simulation results for calcium pathway", "title": "Run TORUS", "text": "Run TORUS ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170726_Dap_TORUS_on_simulation_for_Calcium_Pathway.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170726_Dap_TORUS_on_simulation_for_Calcium_Pathway.html#Run-TORUS"},
{"mainTitle": "DAP on simulation results for calcium pathway", "title": "Run DAP", "text": "Run DAP ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170726_Dap_TORUS_on_simulation_for_Calcium_Pathway.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170726_Dap_TORUS_on_simulation_for_Calcium_Pathway.html#Run-DAP"},
{"mainTitle": "Implement calcium signaling pathway", "title": "Implement calcium signaling pathway", "text": "Implement calcium signaling pathway ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170726_CalciumSignalingPathway.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170726_CalciumSignalingPathway.html#Implement-calcium-signaling-pathway"},
{"mainTitle": "Enrichment analysis of CNVs in schizophrenia", "title": "Enrichment analysis of CNVs in schizophrenia", "text": "Enrichment analysis of CNVs in schizophrenia ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170216_Enrichment_analysis_of_CNV_in_schizophrenia.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170216_Enrichment_analysis_of_CNV_in_schizophrenia.html#Enrichment-analysis-of-CNVs-in-schizophrenia"},
{"mainTitle": "Enrichment analysis of CNVs in schizophrenia", "title": "Steps", "text": "Steps ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170216_Enrichment_analysis_of_CNV_in_schizophrenia.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170216_Enrichment_analysis_of_CNV_in_schizophrenia.html#Steps"},
{"mainTitle": "Enrichment analysis of CNVs in schizophrenia", "title": "Load CNV and drop duplicates", "text": "Load CNV and drop duplicates ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170216_Enrichment_analysis_of_CNV_in_schizophrenia.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170216_Enrichment_analysis_of_CNV_in_schizophrenia.html#Load-CNV-and-drop-duplicates"},
{"mainTitle": "Enrichment analysis of CNVs in schizophrenia", "title": "Load reference gene database and drop duplicates", "text": "Load reference gene database and drop duplicates ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170216_Enrichment_analysis_of_CNV_in_schizophrenia.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170216_Enrichment_analysis_of_CNV_in_schizophrenia.html#Load-reference-gene-database-and-drop-duplicates"},
{"mainTitle": "Enrichment analysis of CNVs in schizophrenia", "title": "Calculate number of CNVs per gene", "text": "Calculate number of CNVs per gene ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170216_Enrichment_analysis_of_CNV_in_schizophrenia.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170216_Enrichment_analysis_of_CNV_in_schizophrenia.html#Calculate-number-of-CNVs-per-gene"},
{"mainTitle": "Enrichment analysis of CNVs in schizophrenia", "title": "Question", "text": "Question ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170216_Enrichment_analysis_of_CNV_in_schizophrenia.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170216_Enrichment_analysis_of_CNV_in_schizophrenia.html#Question"},
{"mainTitle": "Enrichment analysis of CNVs in schizophrenia", "title": "Enrichment analyses (Fisher Exact test)", "text": "Enrichment analyses Fisher Exact test ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170216_Enrichment_analysis_of_CNV_in_schizophrenia.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170216_Enrichment_analysis_of_CNV_in_schizophrenia.html#Enrichment-analyses-(Fisher-Exact-test)"},
{"mainTitle": "Enrichment analysis of CNVs in schizophrenia", "title": "Visualize enrichment analyses", "text": "Visualize enrichment analyses ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170216_Enrichment_analysis_of_CNV_in_schizophrenia.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170216_Enrichment_analysis_of_CNV_in_schizophrenia.html#Visualize-enrichment-analyses"},
{"mainTitle": "Enrichment analysis of CNVs in schizophrenia", "title": "Reference", "text": "Reference ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170216_Enrichment_analysis_of_CNV_in_schizophrenia.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170216_Enrichment_analysis_of_CNV_in_schizophrenia.html#Reference"},
{"mainTitle": "R: use package varbvs and feather to file to obtain .RDS file", "title": "R: use package varbvs and feather to file to obtain .RDS file", "text": "R use package varbvs and feather to file to obtain .RDS file ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170706_R_varbvs_toy.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170706_R_varbvs_toy.html#R:-use-package-varbvs-and-feather-to-file-to-obtain-.RDS-file"},
{"mainTitle": "Toy example for multi-genes and multi-CNVs in a region", "title": "Toy example for multi-genes and multi-CNVs in a region", "text": "Toy example for multi-genes and multi-CNVs in a region ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170906_toy_multiCNVs_multigenes_TORUS.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170906_toy_multiCNVs_multigenes_TORUS.html#Toy-example-for-multi-genes-and-multi-CNVs-in-a-region"},
{"mainTitle": "Toy example for multi-genes and multi-CNVs in a region", "title": "One or more causal genes", "text": "One or more causal genes ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170906_toy_multiCNVs_multigenes_TORUS.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170906_toy_multiCNVs_multigenes_TORUS.html#One-or-more-causal-genes"},
{"mainTitle": "Toy example for multi-genes and multi-CNVs in a region", "title": "Obtain summary statistics z_score and run TORUS", "text": "Obtain summary statistics z_score and run TORUS ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170906_toy_multiCNVs_multigenes_TORUS.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170906_toy_multiCNVs_multigenes_TORUS.html#Obtain-summary-statistics-z_score-and-run-TORUS"},
{"mainTitle": "Toy example for multi-genes and multi-CNVs in a region", "title": "command", "text": "command ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170906_toy_multiCNVs_multigenes_TORUS.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170906_toy_multiCNVs_multigenes_TORUS.html#command"},
{"mainTitle": "Obtain LFDR using .RDS and p-value using simulated dataset (.pkl)", "title": "Obtain LFDR using .RDS and p-value using simulated dataset (.pkl)", "text": "Obtain LFDR using .RDS and p-value using simulated dataset .pkl ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170504_lfdr_plot.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170504_lfdr_plot.html#Obtain-LFDR-using-.RDS-and-p-value-using-simulated-dataset-(.pkl)"},
{"mainTitle": "Simulation of Exome-wide CNV data for case control samples", "title": "Simulation of Exome-wide CNV data for case control samples", "text": "Simulation of Exome-wide CNV data for case control samples ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170526_Simulation.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170526_Simulation.html#Simulation-of-Exome-wide-CNV-data-for-case-control-samples"},
{"mainTitle": "Simulation of Exome-wide CNV data for case control samples", "title": "Simulation outline", "text": "Simulation outline ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170526_Simulation.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170526_Simulation.html#Simulation-outline"},
{"mainTitle": "Simulation of Exome-wide CNV data for case control samples", "title": "Parameters", "text": "Parameters ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170526_Simulation.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170526_Simulation.html#Parameters"},
{"mainTitle": "Simulation of Exome-wide CNV data for case control samples", "title": "Simulation steps", "text": "Simulation steps ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170526_Simulation.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170526_Simulation.html#Simulation-steps"},
{"mainTitle": "Simulation of Exome-wide CNV data for case control samples", "title": "Step 1: Simulate number of CNV per exome", "text": "Step 1 Simulate number of CNV per exome ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170526_Simulation.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170526_Simulation.html#Step-1:-Simulate-number-of-CNV-per-exome"},
{"mainTitle": "Simulation of Exome-wide CNV data for case control samples", "title": "Step 2: Simulate CNV for an exome", "text": "Step 2 Simulate CNV for an exome ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170526_Simulation.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170526_Simulation.html#Step-2:-Simulate-CNV-for-an-exome"},
{"mainTitle": "Simulation of Exome-wide CNV data for case control samples", "title": "Positions of CNVs", "text": "Positions of CNVs ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170526_Simulation.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170526_Simulation.html#Positions-of-CNVs"},
{"mainTitle": "Simulation of Exome-wide CNV data for case control samples", "title": "Length of CNVs", "text": "Length of CNVs ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170526_Simulation.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170526_Simulation.html#Length-of-CNVs"},
{"mainTitle": "Simulation of Exome-wide CNV data for case control samples", "title": "Step 3: Tag each simulated CNV by causal genes", "text": "Step 3 Tag each simulated CNV by causal genes ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170526_Simulation.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170526_Simulation.html#Step-3:-Tag-each-simulated-CNV-by-causal-genes"},
{"mainTitle": "Simulation of Exome-wide CNV data for case control samples", "title": "Step 4: Get the probability of a sample being case give all genes", "text": "Step 4 Get the probability of a sample being case give all genes ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170526_Simulation.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170526_Simulation.html#Step-4:-Get-the-probability-of-a-sample-being-case-give-all-genes"},
{"mainTitle": "Simulation of Exome-wide CNV data for case control samples", "title": "Implementation", "text": "Implementation ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170526_Simulation.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170526_Simulation.html#Implementation"},
{"mainTitle": "Simulation of Exome-wide CNV data for case control samples", "title": "Step 1 & 2", "text": "Step 1 2 ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170526_Simulation.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170526_Simulation.html#Step-1-&-2"},
{"mainTitle": "Simulation of Exome-wide CNV data for case control samples", "title": "Step 3", "text": "Step 3 ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170526_Simulation.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170526_Simulation.html#Step-3"},
{"mainTitle": "Simulation of Exome-wide CNV data for case control samples", "title": "Step 4", "text": "Step 4 ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170526_Simulation.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170526_Simulation.html#Step-4"},
{"mainTitle": "Simulation of Exome-wide CNV data for case control samples", "title": "Putting all together", "text": "Putting all together ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170526_Simulation.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170526_Simulation.html#Putting-all-together"},
{"mainTitle": "Simulation of Exome-wide CNV data for case control samples", "title": "sns.set(style="whitegrid")", "text": "sns.set style='whitegrid' ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170526_Simulation.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170526_Simulation.html#sns.set(style="whitegrid")"},
{"mainTitle": "Overview of DAP on Calcium Pathway Simulation", "title": "Overview of DAP on Calcium Pathway Simulation", "text": "Overview of DAP on Calcium Pathway Simulation ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170817_Overview_of_DAP_on_calcium_pathway_simulation.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170817_Overview_of_DAP_on_calcium_pathway_simulation.html#Overview-of-DAP-on-Calcium-Pathway-Simulation"},
{"mainTitle": "Overview of DAP on Calcium Pathway Simulation", "title": "DAP for calcium pathway simulation", "text": "DAP for calcium pathway simulation ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170817_Overview_of_DAP_on_calcium_pathway_simulation.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170817_Overview_of_DAP_on_calcium_pathway_simulation.html#DAP-for-calcium-pathway-simulation"},
{"mainTitle": "Overview of DAP on Calcium Pathway Simulation", "title": "sample size = 4000, gamma shape = 5", "text": "sample size = 4000, gamma shape = 5 ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170817_Overview_of_DAP_on_calcium_pathway_simulation.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170817_Overview_of_DAP_on_calcium_pathway_simulation.html#sample-size-=-4000,-gamma-shape-=-5"},
{"mainTitle": "Overview of DAP on Calcium Pathway Simulation", "title": "sample size = 1000, gamma shape = 25", "text": "sample size = 1000, gamma shape = 25 ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170817_Overview_of_DAP_on_calcium_pathway_simulation.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170817_Overview_of_DAP_on_calcium_pathway_simulation.html#sample-size-=-1000,-gamma-shape-=-25"},
{"mainTitle": "Overview of DAP on Calcium Pathway Simulation", "title": "DAP-greedy: improved Adaptive DAP Algorithm", "text": "DAP-greedy improved Adaptive DAP Algorithm ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170817_Overview_of_DAP_on_calcium_pathway_simulation.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170817_Overview_of_DAP_on_calcium_pathway_simulation.html#DAP-greedy:-improved-Adaptive-DAP-Algorithm"},
{"mainTitle": "Overview of DAP on Calcium Pathway Simulation", "title": "New features", "text": "New features ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170817_Overview_of_DAP_on_calcium_pathway_simulation.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170817_Overview_of_DAP_on_calcium_pathway_simulation.html#New-features"},
{"mainTitle": "Overview of DAP on Calcium Pathway Simulation", "title": "sample size = 4000, gamma shape = 5: the PIP is slightly larger than DAP", "text": "sample size = 4000, gamma shape = 5 the PIP is slightly larger than DAP ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170817_Overview_of_DAP_on_calcium_pathway_simulation.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170817_Overview_of_DAP_on_calcium_pathway_simulation.html#sample-size-=-4000,-gamma-shape-=-5:-the-PIP-is-slightly-larger-than-DAP"},
{"mainTitle": "Overview of DAP on Calcium Pathway Simulation", "title": "sample size = 1000, gamma shape = 25:  the PIP is slightly larger than DAP", "text": "sample size = 1000, gamma shape = 25 the PIP is slightly larger than DAP ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170817_Overview_of_DAP_on_calcium_pathway_simulation.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170817_Overview_of_DAP_on_calcium_pathway_simulation.html#sample-size-=-1000,-gamma-shape-=-25:--the-PIP-is-slightly-larger-than-DAP"},
{"mainTitle": "Overview of DAP on Calcium Pathway Simulation", "title": "Next step", "text": "Next step ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170817_Overview_of_DAP_on_calcium_pathway_simulation.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170817_Overview_of_DAP_on_calcium_pathway_simulation.html#Next-step"},
{"mainTitle": "Plots for enrichment analysis of CNV in Schizophrenia", "title": "Plots for enrichment analysis of CNV in Schizophrenia", "text": "Plots for enrichment analysis of CNV in Schizophrenia ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/analysis/20170413_Plot_enrichment_analysis_CNV.html", "url": "http://gaow.github.io/cnv-gene-mapping/analysis/20170413_Plot_enrichment_analysis_CNV.html#Plots-for-enrichment-analysis-of-CNV-in-Schizophrenia"},
{"mainTitle": "R: read .RDS file and draw histogram of "PIP"", "title": "R: read .RDS file and draw histogram of "PIP"", "text": "R read .RDS file and draw histogram of 'PIP' ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/prototype/20170504_R_useRDS.html", "url": "http://gaow.github.io/cnv-gene-mapping/prototype/20170504_R_useRDS.html#R:-read-.RDS-file-and-draw-histogram-of-"PIP""},
{"mainTitle": "Toy example of 4-gene configuration overlapped with one CNV", "title": "Toy example of 4-gene configuration overlapped with one CNV", "text": "Toy example of 4-gene configuration overlapped with one CNV ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/prototype/20170606_toy_4gene.html", "url": "http://gaow.github.io/cnv-gene-mapping/prototype/20170606_toy_4gene.html#Toy-example-of-4-gene-configuration-overlapped-with-one-CNV"},
{"mainTitle": "Exploratory of a toy CNV dataset in VCF format", "title": "Exploratory of a toy CNV dataset in VCF format", "text": "Exploratory of a toy CNV dataset in VCF format ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/prototype/20160817_VCF_Toy.html", "url": "http://gaow.github.io/cnv-gene-mapping/prototype/20160817_VCF_Toy.html#Exploratory-of-a-toy-CNV-dataset-in-VCF-format"},
{"mainTitle": "Toy example of 3-gene configuration overlapped with one CNV", "title": "Toy example of 3-gene configuration overlapped with one CNV", "text": "Toy example of 3-gene configuration overlapped with one CNV ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/prototype/20170605_toy_3gene.html", "url": "http://gaow.github.io/cnv-gene-mapping/prototype/20170605_toy_3gene.html#Toy-example-of-3-gene-configuration-overlapped-with-one-CNV"},
{"mainTitle": "Transfer old version pickle to new version pickle in "debug"", "title": "Transfer old version pickle to new version pickle in "debug"", "text": "Transfer old version pickle to new version pickle in 'debug' ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/prototype/20170820_Pickle_Block_Matrix.html", "url": "http://gaow.github.io/cnv-gene-mapping/prototype/20170820_Pickle_Block_Matrix.html#Transfer-old-version-pickle-to-new-version-pickle-in-"debug""},
{"mainTitle": "Analysis of toy data via DAP", "title": "Analysis of toy data via DAP", "text": "Analysis of toy data via DAP ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/prototype/20170710_dap_on_toy.html", "url": "http://gaow.github.io/cnv-gene-mapping/prototype/20170710_dap_on_toy.html#Analysis-of-toy-data-via-DAP"},
{"mainTitle": "Visualize genome-wide SCNA Profile", "title": "Visualize genome-wide SCNA Profile", "text": "Visualize genome-wide SCNA Profile ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/prototype/20170113_Toy_from_saasCNV.html", "url": "http://gaow.github.io/cnv-gene-mapping/prototype/20170113_Toy_from_saasCNV.html#Visualize-genome-wide-SCNA-Profile"},
{"mainTitle": "Visualize genome-wide SCNA Profile", "title": "Result", "text": "Result ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/prototype/20170113_Toy_from_saasCNV.html", "url": "http://gaow.github.io/cnv-gene-mapping/prototype/20170113_Toy_from_saasCNV.html#Result"},
{"mainTitle": "Visualize genome-wide SCNA Profile", "title": "Figure 1", "text": "Figure 1 ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/prototype/20170113_Toy_from_saasCNV.html", "url": "http://gaow.github.io/cnv-gene-mapping/prototype/20170113_Toy_from_saasCNV.html#Figure-1"},
{"mainTitle": "Visualize genome-wide SCNA Profile", "title": "Figure 2", "text": "Figure 2 ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/prototype/20170113_Toy_from_saasCNV.html", "url": "http://gaow.github.io/cnv-gene-mapping/prototype/20170113_Toy_from_saasCNV.html#Figure-2"},
{"mainTitle": "Use BoomSpikeSlab", "title": "Use BoomSpikeSlab", "text": "Use BoomSpikeSlab ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20190909_Boom_Spike_Slab.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20190909_Boom_Spike_Slab.html#Use-BoomSpikeSlab"},
{"mainTitle": "Use BoomSpikeSlab", "title": "First check the result of varbvs on whole genome", "text": "First check the result of varbvs on whole genome ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20190909_Boom_Spike_Slab.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20190909_Boom_Spike_Slab.html#First-check-the-result-of-varbvs-on-whole-genome"},
{"mainTitle": "Copy model simulation and analysis workflow", "title": "Copy model simulation and analysis workflow", "text": "Copy model simulation and analysis workflow ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20190717_workflow.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20190717_workflow.html#Copy-model-simulation-and-analysis-workflow"},
{"mainTitle": "Copy model simulation and analysis workflow", "title": "Run this workflow", "text": "Run this workflow ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20190717_workflow.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20190717_workflow.html#Run-this-workflow"},
{"mainTitle": "Copy model simulation and analysis workflow", "title": "Simulation:", "text": "Simulation ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20190717_workflow.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20190717_workflow.html#Simulation:"},
{"mainTitle": "Copy model simulation and analysis workflow", "title": "Get histogram", "text": "Get histogram ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20190717_workflow.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20190717_workflow.html#Get-histogram"},
{"mainTitle": "Copy model simulation and analysis workflow", "title": "Analyze", "text": "Analyze ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20190717_workflow.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20190717_workflow.html#Analyze"},
{"mainTitle": "Copy model simulation and analysis workflow", "title": "midway", "text": "midway ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20190717_workflow.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20190717_workflow.html#midway"},
{"mainTitle": "Copy model simulation and analysis workflow", "title": "n_gene_in_block = 20, sample_size = 200000", "text": "n_gene_in_block = 20, sample_size = 200000 ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20190717_workflow.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20190717_workflow.html#n_gene_in_block-=-20,-sample_size-=-200000"},
{"mainTitle": "Copy model simulation and analysis workflow", "title": "midway", "text": "midway ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20190717_workflow.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20190717_workflow.html#midway"},
{"mainTitle": "Copy model simulation and analysis workflow", "title": "local", "text": "local ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20190717_workflow.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20190717_workflow.html#local"},
{"mainTitle": "Copy model simulation and analysis workflow", "title": "n_gene_in_block = 30, sample_size = 200000, method = gamma, shape = 3, scale = 1", "text": "n_gene_in_block = 30, sample_size = 200000, method = gamma, shape = 3, scale = 1 ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20190717_workflow.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20190717_workflow.html#n_gene_in_block-=-30,-sample_size-=-200000,-method-=-gamma,-shape-=-3,-scale-=-1"},
{"mainTitle": "Copy model simulation and analysis workflow", "title": "midway 20200304", "text": "midway 20200304 ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20190717_workflow.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20190717_workflow.html#midway-20200304"},
{"mainTitle": "Copy model simulation and analysis workflow", "title": "new susie local 30300331", "text": "new susie local 30300331 ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20190717_workflow.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20190717_workflow.html#new-susie-local-30300331"},
{"mainTitle": "Copy model simulation and analysis workflow", "title": "Real deletion data", "text": "Real deletion data ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20190717_workflow.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20190717_workflow.html#Real-deletion-data"},
{"mainTitle": "Copy model simulation and analysis workflow", "title": "local, fisher test", "text": "local, fisher test ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20190717_workflow.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20190717_workflow.html#local,-fisher-test"},
{"mainTitle": "Copy model simulation and analysis workflow", "title": "midway", "text": "midway ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20190717_workflow.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20190717_workflow.html#midway"},
{"mainTitle": "Copy model simulation and analysis workflow", "title": "kill all jobs on midway", "text": "kill all jobs on midway ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20190717_workflow.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20190717_workflow.html#kill-all-jobs-on-midway"},
{"mainTitle": "Copy model simulation and analysis workflow", "title": "simulate_5 are not used in updated pipeline", "text": "simulate_5 are not used in updated pipeline ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20190717_workflow.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20190717_workflow.html#simulate_5-are-not-used-in-updated-pipeline"},
{"mainTitle": "Copy model simulation and analysis workflow", "title": "Note", "text": "Note ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20190717_workflow.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20190717_workflow.html#Note"},
{"mainTitle": "Simulation, analyze and plots", "title": "Simulation, analyze and plots", "text": "Simulation, analyze and plots ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20190819_workflow_report.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20190819_workflow_report.html#Simulation,-analyze-and-plots"},
{"mainTitle": "Simulation, analyze and plots", "title": "Previous work", "text": "Previous work ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20190819_workflow_report.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20190819_workflow_report.html#Previous-work"},
{"mainTitle": "Simulation, analyze and plots", "title": "Genes conflict with each other", "text": "Genes conflict with each other ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20190819_workflow_report.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20190819_workflow_report.html#Genes-conflict-with-each-other"},
{"mainTitle": "Simulation, analyze and plots", "title": "Simulation", "text": "Simulation ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20190819_workflow_report.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20190819_workflow_report.html#Simulation"},
{"mainTitle": "Simulation, analyze and plots", "title": "Histogram", "text": "Histogram ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20190819_workflow_report.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20190819_workflow_report.html#Histogram"},
{"mainTitle": "Simulation, analyze and plots", "title": "Analyze", "text": "Analyze ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20190819_workflow_report.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20190819_workflow_report.html#Analyze"},
{"mainTitle": "Simulation, analyze and plots", "title": "SuSiE and varbvs results comparison for block #505 as an example", "text": "SuSiE and varbvs results comparison for block 505 as an example ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20190819_workflow_report.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20190819_workflow_report.html#SuSiE-and-varbvs-results-comparison-for-block-#505-as-an-example"},
{"mainTitle": "Simulation, analyze and plots", "title": "Questions", "text": "Questions ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20190819_workflow_report.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20190819_workflow_report.html#Questions"},
{"mainTitle": "Answers for new comments in January 2020", "title": "Answers for new comments in January 2020", "text": "Answers for new comments in January 2020 ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20200214_report.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20200214_report.html#Answers-for-new-comments-in-January-2020"},
{"mainTitle": "Answers for new comments in January 2020", "title": "PIPs of logistic method in non-signal blocks are expected to be approximately equal", "text": "PIPs of logistic method in non-signal blocks are expected to be approximately equal ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20200214_report.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20200214_report.html#PIPs-of-logistic-method-in-non-signal-blocks-are-expected-to-be-approximately-equal"},
{"mainTitle": "Answers for new comments in January 2020", "title": "Compare logistic and pymc3 on simulated single causal gene block.", "text": "Compare logistic and pymc3 on simulated single causal gene block. ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20200214_report.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20200214_report.html#Compare-logistic-and-pymc3-on-simulated-single-causal-gene-block."},
{"mainTitle": "Answers for new comments in January 2020", "title": "Real deletion CNV data", "text": "Real deletion CNV data ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20200214_report.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20200214_report.html#Real-deletion-CNV-data"},
{"mainTitle": "Answers for new comments in January 2020", "title": "ROC curve issue", "text": "ROC curve issue ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20200214_report.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20200214_report.html#ROC-curve-issue"},
{"mainTitle": "PyMC3 report: Logistic regression spike slab prior", "title": "PyMC3 report: Logistic regression spike slab prior", "text": "PyMC3 report Logistic regression spike slab prior ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20191008_PyMC3_report.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20191008_PyMC3_report.html#PyMC3-report:-Logistic-regression-spike-slab-prior"},
{"mainTitle": "PyMC3 report: Logistic regression spike slab prior", "title": "PyMC3 introduction", "text": "PyMC3 introduction ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20191008_PyMC3_report.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20191008_PyMC3_report.html#PyMC3-introduction"},
{"mainTitle": "PyMC3 report: Logistic regression spike slab prior", "title": "PyMC3 performance", "text": "PyMC3 performance ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20191008_PyMC3_report.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20191008_PyMC3_report.html#PyMC3-performance"},
{"mainTitle": "PyMC3 report: Logistic regression spike slab prior", "title": "Single effect block", "text": "Single effect block ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20191008_PyMC3_report.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20191008_PyMC3_report.html#Single-effect-block"},
{"mainTitle": "PyMC3 report: Logistic regression spike slab prior", "title": "Block # 739_743", "text": "Block 739_743 ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20191008_PyMC3_report.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20191008_PyMC3_report.html#Block-#-739_743"},
{"mainTitle": "PyMC3 report: Logistic regression spike slab prior", "title": "varbvs summary", "text": "varbvs summary ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20191008_PyMC3_report.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20191008_PyMC3_report.html#varbvs-summary"},
{"mainTitle": "PyMC3 report: Logistic regression spike slab prior", "title": "Fisher test results and effect size for simulation", "text": "Fisher test results and effect size for simulation ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20191008_PyMC3_report.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20191008_PyMC3_report.html#Fisher-test-results-and-effect-size-for-simulation"},
{"mainTitle": "PyMC3 report: Logistic regression spike slab prior", "title": "PyMC3", "text": "PyMC3 ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20191008_PyMC3_report.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20191008_PyMC3_report.html#PyMC3"},
{"mainTitle": "PyMC3 report: Logistic regression spike slab prior", "title": "Block # 2122_2125", "text": "Block 2122_2125 ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20191008_PyMC3_report.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20191008_PyMC3_report.html#Block-#-2122_2125"},
{"mainTitle": "PyMC3 report: Logistic regression spike slab prior", "title": "varbvs summary", "text": "varbvs summary ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20191008_PyMC3_report.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20191008_PyMC3_report.html#varbvs-summary"},
{"mainTitle": "PyMC3 report: Logistic regression spike slab prior", "title": "Fisher test results and effect size for simulation", "text": "Fisher test results and effect size for simulation ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20191008_PyMC3_report.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20191008_PyMC3_report.html#Fisher-test-results-and-effect-size-for-simulation"},
{"mainTitle": "PyMC3 report: Logistic regression spike slab prior", "title": "PyMC3", "text": "PyMC3 ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20191008_PyMC3_report.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20191008_PyMC3_report.html#PyMC3"},
{"mainTitle": "PyMC3 report: Logistic regression spike slab prior", "title": "Block # 423_427", "text": "Block 423_427 ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20191008_PyMC3_report.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20191008_PyMC3_report.html#Block-#-423_427"},
{"mainTitle": "PyMC3 report: Logistic regression spike slab prior", "title": "varbvs summary", "text": "varbvs summary ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20191008_PyMC3_report.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20191008_PyMC3_report.html#varbvs-summary"},
{"mainTitle": "PyMC3 report: Logistic regression spike slab prior", "title": "Fisher test results and effect size for simulation", "text": "Fisher test results and effect size for simulation ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20191008_PyMC3_report.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20191008_PyMC3_report.html#Fisher-test-results-and-effect-size-for-simulation"},
{"mainTitle": "PyMC3 report: Logistic regression spike slab prior", "title": "PyMC3", "text": "PyMC3 ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20191008_PyMC3_report.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20191008_PyMC3_report.html#PyMC3"},
{"mainTitle": "PyMC3 report: Logistic regression spike slab prior", "title": "Block 1858_1875", "text": "Block 1858_1875 ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20191008_PyMC3_report.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20191008_PyMC3_report.html#Block-1858_1875"},
{"mainTitle": "PyMC3 report: Logistic regression spike slab prior", "title": "varbvs summary", "text": "varbvs summary ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20191008_PyMC3_report.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20191008_PyMC3_report.html#varbvs-summary"},
{"mainTitle": "PyMC3 report: Logistic regression spike slab prior", "title": "Fisher test results and effect size for simulation", "text": "Fisher test results and effect size for simulation ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20191008_PyMC3_report.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20191008_PyMC3_report.html#Fisher-test-results-and-effect-size-for-simulation"},
{"mainTitle": "PyMC3 report: Logistic regression spike slab prior", "title": "PyMC3", "text": "PyMC3 ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20191008_PyMC3_report.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20191008_PyMC3_report.html#PyMC3"},
{"mainTitle": "PyMC3 report: Logistic regression spike slab prior", "title": "Multiple effect block", "text": "Multiple effect block ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20191008_PyMC3_report.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20191008_PyMC3_report.html#Multiple-effect-block"},
{"mainTitle": "PyMC3 report: Logistic regression spike slab prior", "title": "Block # 556_583", "text": "Block 556_583 ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20191008_PyMC3_report.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20191008_PyMC3_report.html#Block-#-556_583"},
{"mainTitle": "PyMC3 report: Logistic regression spike slab prior", "title": "varbvs summary", "text": "varbvs summary ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20191008_PyMC3_report.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20191008_PyMC3_report.html#varbvs-summary"},
{"mainTitle": "PyMC3 report: Logistic regression spike slab prior", "title": "Fisher test results and effect size for simulation", "text": "Fisher test results and effect size for simulation ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20191008_PyMC3_report.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20191008_PyMC3_report.html#Fisher-test-results-and-effect-size-for-simulation"},
{"mainTitle": "PyMC3 report: Logistic regression spike slab prior", "title": "PyMC3 results", "text": "PyMC3 results ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20191008_PyMC3_report.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20191008_PyMC3_report.html#PyMC3-results"},
{"mainTitle": "PyMC3 report: Logistic regression spike slab prior", "title": "Block # 2092_2099", "text": "Block 2092_2099 ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20191008_PyMC3_report.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20191008_PyMC3_report.html#Block-#-2092_2099"},
{"mainTitle": "PyMC3 report: Logistic regression spike slab prior", "title": "varbvs summary", "text": "varbvs summary ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20191008_PyMC3_report.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20191008_PyMC3_report.html#varbvs-summary"},
{"mainTitle": "PyMC3 report: Logistic regression spike slab prior", "title": "Fisher test results and effect size for simulation", "text": "Fisher test results and effect size for simulation ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20191008_PyMC3_report.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20191008_PyMC3_report.html#Fisher-test-results-and-effect-size-for-simulation"},
{"mainTitle": "PyMC3 report: Logistic regression spike slab prior", "title": "PyMC3", "text": "PyMC3 ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20191008_PyMC3_report.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20191008_PyMC3_report.html#PyMC3"},
{"mainTitle": "PyMC3 report: Logistic regression spike slab prior", "title": "Block # 666_677", "text": "Block 666_677 ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20191008_PyMC3_report.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20191008_PyMC3_report.html#Block-#-666_677"},
{"mainTitle": "PyMC3 report: Logistic regression spike slab prior", "title": "varbvs summary", "text": "varbvs summary ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20191008_PyMC3_report.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20191008_PyMC3_report.html#varbvs-summary"},
{"mainTitle": "PyMC3 report: Logistic regression spike slab prior", "title": "Fisher test results and effect size for simulation", "text": "Fisher test results and effect size for simulation ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20191008_PyMC3_report.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20191008_PyMC3_report.html#Fisher-test-results-and-effect-size-for-simulation"},
{"mainTitle": "PyMC3 report: Logistic regression spike slab prior", "title": "PyMC3 results", "text": "PyMC3 results ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20191008_PyMC3_report.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20191008_PyMC3_report.html#PyMC3-results"},
{"mainTitle": "PyMC3 report: Logistic regression spike slab prior", "title": "Block # 1018_1031", "text": "Block 1018_1031 ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20191008_PyMC3_report.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20191008_PyMC3_report.html#Block-#-1018_1031"},
{"mainTitle": "PyMC3 report: Logistic regression spike slab prior", "title": "varbvs summary", "text": "varbvs summary ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20191008_PyMC3_report.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20191008_PyMC3_report.html#varbvs-summary"},
{"mainTitle": "PyMC3 report: Logistic regression spike slab prior", "title": "Fisher test results and effect size for simulation", "text": "Fisher test results and effect size for simulation ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20191008_PyMC3_report.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20191008_PyMC3_report.html#Fisher-test-results-and-effect-size-for-simulation"},
{"mainTitle": "PyMC3 report: Logistic regression spike slab prior", "title": "PyMC3 results", "text": "PyMC3 results ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20191008_PyMC3_report.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20191008_PyMC3_report.html#PyMC3-results"},
{"mainTitle": "PyMC3 report: Logistic regression spike slab prior", "title": "block 23_36", "text": "block 23_36 ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20191008_PyMC3_report.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20191008_PyMC3_report.html#block-23_36"},
{"mainTitle": "PyMC3 report: Logistic regression spike slab prior", "title": "varbvs summary", "text": "varbvs summary ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20191008_PyMC3_report.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20191008_PyMC3_report.html#varbvs-summary"},
{"mainTitle": "PyMC3 report: Logistic regression spike slab prior", "title": "Fisher test results and effect size for simulation", "text": "Fisher test results and effect size for simulation ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20191008_PyMC3_report.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20191008_PyMC3_report.html#Fisher-test-results-and-effect-size-for-simulation"},
{"mainTitle": "PyMC3 report: Logistic regression spike slab prior", "title": "PyMC3 results", "text": "PyMC3 results ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20191008_PyMC3_report.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20191008_PyMC3_report.html#PyMC3-results"},
{"mainTitle": "PyMC3 report: Logistic regression spike slab prior", "title": "PyMC3 results (perform 2nd time)", "text": "PyMC3 results perform 2nd time ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20191008_PyMC3_report.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20191008_PyMC3_report.html#PyMC3-results-(perform-2nd-time)"},
{"mainTitle": "Integrated fine-mapping of non-coding disease variants with functional genomics data", "title": "Integrated fine-mapping of non-coding disease variants with functional genomics data", "text": "Integrated fine-mapping of non-coding disease variants with functional genomics data ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20190908_proposal_manuscript.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20190908_proposal_manuscript.html#Integrated-fine-mapping-of-non-coding-disease-variants-with-functional-genomics-data"},
{"mainTitle": "Integrated fine-mapping of non-coding disease variants with functional genomics data", "title": "Background and Public Health Significance", "text": "Background and Public Health Significance ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20190908_proposal_manuscript.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20190908_proposal_manuscript.html#Background-and-Public-Health-Significance"},
{"mainTitle": "Integrated fine-mapping of non-coding disease variants with functional genomics data", "title": "Schizophrenia", "text": "Schizophrenia ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20190908_proposal_manuscript.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20190908_proposal_manuscript.html#Schizophrenia"},
{"mainTitle": "Integrated fine-mapping of non-coding disease variants with functional genomics data", "title": "The limitation of Genome-wide Association Studies (GWAS)", "text": "The limitation of Genome-wide Association Studies GWAS ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20190908_proposal_manuscript.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20190908_proposal_manuscript.html#The-limitation-of-Genome-wide-Association-Studies-(GWAS)"},
{"mainTitle": "Integrated fine-mapping of non-coding disease variants with functional genomics data", "title": "Open chromatin regions and allele-specific open chromatin variants", "text": "Open chromatin regions and allele-specific open chromatin variants ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20190908_proposal_manuscript.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20190908_proposal_manuscript.html#Open-chromatin-regions-and-allele-specific-open-chromatin-variants"},
{"mainTitle": "Integrated fine-mapping of non-coding disease variants with functional genomics data", "title": "Public Health Significance", "text": "Public Health Significance ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20190908_proposal_manuscript.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20190908_proposal_manuscript.html#Public-Health-Significance"},
{"mainTitle": "Integrated fine-mapping of non-coding disease variants with functional genomics data", "title": "Aim", "text": "Aim ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20190908_proposal_manuscript.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20190908_proposal_manuscript.html#Aim"},
{"mainTitle": "Integrated fine-mapping of non-coding disease variants with functional genomics data", "title": "Aim 1: To construct statistical model for GWAS enrichment analysis for genomic features", "text": "Aim 1 To construct statistical model for GWAS enrichment analysis for genomic features ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20190908_proposal_manuscript.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20190908_proposal_manuscript.html#Aim-1:-To-construct-statistical-model-for-GWAS-enrichment-analysis-for-genomic-features"},
{"mainTitle": "Integrated fine-mapping of non-coding disease variants with functional genomics data", "title": "Aim 2: To utilize newly developed statistical method to explore the enrichment of Schizophrenia GWAS signals in different regulatory regions", "text": "Aim 2 To utilize newly developed statistical method to explore the enrichment of Schizophrenia GWAS signals in different regulatory regions ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20190908_proposal_manuscript.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20190908_proposal_manuscript.html#Aim-2:-To-utilize-newly-developed-statistical-method-to-explore-the-enrichment-of-Schizophrenia-GWAS-signals-in-different-regulatory-regions"},
{"mainTitle": "Integrated fine-mapping of non-coding disease variants with functional genomics data", "title": "Aim 3: To perform fine-mapping of putative causal SNPs in 108 identified loci", "text": "Aim 3 To perform fine-mapping of putative causal SNPs in 108 identified loci ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20190908_proposal_manuscript.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20190908_proposal_manuscript.html#Aim-3:-To-perform-fine-mapping-of-putative-causal-SNPs-in-108-identified-loci"},
{"mainTitle": "Integrated fine-mapping of non-coding disease variants with functional genomics data", "title": "Data", "text": "Data ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20190908_proposal_manuscript.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20190908_proposal_manuscript.html#Data"},
{"mainTitle": "Integrated fine-mapping of non-coding disease variants with functional genomics data", "title": "Reference", "text": "Reference ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20190908_proposal_manuscript.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20190908_proposal_manuscript.html#Reference"},
{"mainTitle": "Integrated fine-mapping of non-coding disease variants with functional genomics data", "title": "Alzheimer's Disease", "text": "Alzheimer's Disease ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20190908_proposal_manuscript.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20190908_proposal_manuscript.html#Alzheimer's-Disease"},
{"mainTitle": "Integrated fine-mapping of non-coding disease variants with functional genomics data", "title": "Background", "text": "Background ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20190908_proposal_manuscript.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20190908_proposal_manuscript.html#Background"},
{"mainTitle": "Integrated fine-mapping of non-coding disease variants with functional genomics data", "title": "Aims and strategies (methods)", "text": "Aims and strategies methods ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20190908_proposal_manuscript.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20190908_proposal_manuscript.html#Aims-and-strategies-(methods)"},
{"mainTitle": "Integrated fine-mapping of non-coding disease variants with functional genomics data", "title": "Aim 1: To explore whether AD GWAS signals are enriched in different regulatory regions", "text": "Aim 1 To explore whether AD GWAS signals are enriched in different regulatory regions ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20190908_proposal_manuscript.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20190908_proposal_manuscript.html#Aim-1:-To-explore-whether-AD-GWAS-signals-are-enriched-in-different-regulatory-regions"},
{"mainTitle": "Integrated fine-mapping of non-coding disease variants with functional genomics data", "title": "Aim 2: To identify putative causal variants in 29 loci identified in AD GWAS", "text": "Aim 2 To identify putative causal variants in 29 loci identified in AD GWAS ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20190908_proposal_manuscript.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20190908_proposal_manuscript.html#Aim-2:-To-identify-putative-causal-variants-in-29-loci-identified-in-AD-GWAS"},
{"mainTitle": "Integrated fine-mapping of non-coding disease variants with functional genomics data", "title": "reference", "text": "reference ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20190908_proposal_manuscript.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20190908_proposal_manuscript.html#reference"},
{"mainTitle": "Integrated fine-mapping of non-coding disease variants with functional genomics data", "title": "A statistical method to identify causal genes in copy number variation data", "text": "A statistical method to identify causal genes in copy number variation data ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20190908_proposal_manuscript.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20190908_proposal_manuscript.html#A-statistical-method-to-identify-causal-genes-in-copy-number-variation-data"},
{"mainTitle": "Integrated fine-mapping of non-coding disease variants with functional genomics data", "title": "Background and Significance", "text": "Background and Significance ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20190908_proposal_manuscript.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20190908_proposal_manuscript.html#Background-and-Significance"},
{"mainTitle": "Integrated fine-mapping of non-coding disease variants with functional genomics data", "title": "Background", "text": "Background ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20190908_proposal_manuscript.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20190908_proposal_manuscript.html#Background"},
{"mainTitle": "Integrated fine-mapping of non-coding disease variants with functional genomics data", "title": "Aim and Public Health Significance", "text": "Aim and Public Health Significance ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20190908_proposal_manuscript.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20190908_proposal_manuscript.html#Aim-and-Public-Health-Significance"},
{"mainTitle": "Integrated fine-mapping of non-coding disease variants with functional genomics data", "title": "Aim", "text": "Aim ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20190908_proposal_manuscript.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20190908_proposal_manuscript.html#Aim"},
{"mainTitle": "Integrated fine-mapping of non-coding disease variants with functional genomics data", "title": "Public Health Significance", "text": "Public Health Significance ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20190908_proposal_manuscript.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20190908_proposal_manuscript.html#Public-Health-Significance"},
{"mainTitle": "Integrated fine-mapping of non-coding disease variants with functional genomics data", "title": "Methods", "text": "Methods ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20190908_proposal_manuscript.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20190908_proposal_manuscript.html#Methods"},
{"mainTitle": "Integrated fine-mapping of non-coding disease variants with functional genomics data", "title": "Integrated association analysis of CNV and SNP data for causal genes mapping in psychiatric disorders", "text": "Integrated association analysis of CNV and SNP data for causal genes mapping in psychiatric disorders ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20190908_proposal_manuscript.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20190908_proposal_manuscript.html#Integrated-association-analysis-of-CNV-and-SNP-data-for-causal-genes-mapping-in-psychiatric-disorders"},
{"mainTitle": "Integrated fine-mapping of non-coding disease variants with functional genomics data", "title": "Reference", "text": "Reference ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20190908_proposal_manuscript.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20190908_proposal_manuscript.html#Reference"},
{"mainTitle": "Methods discription", "title": "Methods discription", "text": "Methods discription ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/Untitled.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/Untitled.html#Methods-discription"},
{"mainTitle": "Methods discription", "title": "PyMC3", "text": "PyMC3 ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/Untitled.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/Untitled.html#PyMC3"},
{"mainTitle": "Single effect using logistic", "title": "Single effect using logistic", "text": "Single effect using logistic ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20191022_logistic_single_effect.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20191022_logistic_single_effect.html#Single-effect-using-logistic"},
{"mainTitle": "The results of varbvs result and logitBvs", "title": "The results of varbvs result and logitBvs", "text": "The results of varbvs result and logitBvs ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20190918_logitBvs_report.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20190918_logitBvs_report.html#The-results-of-varbvs-result-and-logitBvs"},
{"mainTitle": "The results of varbvs result and logitBvs", "title": "Previous simulation", "text": "Previous simulation ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20190918_logitBvs_report.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20190918_logitBvs_report.html#Previous-simulation"},
{"mainTitle": "The results of varbvs result and logitBvs", "title": "The histogram for the number of genes in simulated natural gene blocks", "text": "The histogram for the number of genes in simulated natural gene blocks ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20190918_logitBvs_report.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20190918_logitBvs_report.html#The-histogram-for-the-number-of-genes-in-simulated-natural-gene-blocks"},
{"mainTitle": "The results of varbvs result and logitBvs", "title": "varbvs results comparison with real beta", "text": "varbvs results comparison with real beta ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20190918_logitBvs_report.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20190918_logitBvs_report.html#varbvs-results-comparison-with-real-beta"},
{"mainTitle": "The results of varbvs result and logitBvs", "title": "logitBvs result", "text": "logitBvs result ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20190918_logitBvs_report.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20190918_logitBvs_report.html#logitBvs-result"},
{"mainTitle": "The results of varbvs result and logitBvs", "title": "BoomSpikeSlab results", "text": "BoomSpikeSlab results ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20190918_logitBvs_report.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20190918_logitBvs_report.html#BoomSpikeSlab-results"},
{"mainTitle": "The results of varbvs result and logitBvs", "title": "Use function LogitZellnerPrior to obtain prior", "text": "Use function LogitZellnerPrior to obtain prior ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20190918_logitBvs_report.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20190918_logitBvs_report.html#Use-function-LogitZellnerPrior-to-obtain-prior"},
{"mainTitle": "The results of varbvs result and logitBvs", "title": "Use function logit.spike for MCMC, 1000 iteration", "text": "Use function logit.spike for MCMC, 1000 iteration ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20190918_logitBvs_report.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20190918_logitBvs_report.html#Use-function-logit.spike-for-MCMC,-1000-iteration"},
{"mainTitle": "Solutions for discussion January", "title": "Solutions for discussion January", "text": "Solutions for discussion January ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20200214_new_comments.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20200214_new_comments.html#Solutions-for-discussion-January"},
{"mainTitle": "Solutions for discussion January", "title": "Single effect blocks for pymc3 and logistic", "text": "Single effect blocks for pymc3 and logistic ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20200214_new_comments.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20200214_new_comments.html#Single-effect-blocks-for-pymc3-and-logistic"},
{"mainTitle": "False Discovery Rate of method logit", "title": "False Discovery Rate of method logit", "text": "False Discovery Rate of method logit ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20200228_logit_FDR.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20200228_logit_FDR.html#False-Discovery-Rate-of-method-logit"},
{"mainTitle": "False Discovery Rate of method logit", "title": "non-signal block", "text": "non-signal block ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20200228_logit_FDR.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20200228_logit_FDR.html#non-signal-block"},
{"mainTitle": "Logistic regression spike slab prior using PyMC3", "title": "Logistic regression spike slab prior using PyMC3", "text": "Logistic regression spike slab prior using PyMC3 ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20191005_PyMC3_SpikeSlab.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20191005_PyMC3_SpikeSlab.html#Logistic-regression-spike-slab-prior-using-PyMC3"},
{"mainTitle": "Logistic regression spike slab prior using PyMC3", "title": "Software required", "text": "Software required ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20191005_PyMC3_SpikeSlab.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20191005_PyMC3_SpikeSlab.html#Software-required"},
{"mainTitle": "Logistic regression spike slab prior using PyMC3", "title": "Mixture normal distribution", "text": "Mixture normal distribution ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20191005_PyMC3_SpikeSlab.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20191005_PyMC3_SpikeSlab.html#Mixture-normal-distribution"},
{"mainTitle": "Logistic regression spike slab prior using PyMC3", "title": "Import X and y", "text": "Import X and y ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20191005_PyMC3_SpikeSlab.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20191005_PyMC3_SpikeSlab.html#Import-X-and-y"},
{"mainTitle": "Logistic regression spike slab prior using PyMC3", "title": "Model specification", "text": "Model specification ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20191005_PyMC3_SpikeSlab.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20191005_PyMC3_SpikeSlab.html#Model-specification"},
{"mainTitle": "Logistic regression spike slab prior using PyMC3", "title": "Spike and Slab model", "text": "Spike and Slab model ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20191005_PyMC3_SpikeSlab.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20191005_PyMC3_SpikeSlab.html#Spike-and-Slab-model"},
{"mainTitle": "Logistic regression spike slab prior using PyMC3", "title": "Sampling", "text": "Sampling ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20191005_PyMC3_SpikeSlab.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20191005_PyMC3_SpikeSlab.html#Sampling"},
{"mainTitle": "Logistic regression spike slab prior using PyMC3", "title": "Results", "text": "Results ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20191005_PyMC3_SpikeSlab.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20191005_PyMC3_SpikeSlab.html#Results"},
{"mainTitle": "Logistic regression spike slab prior using PyMC3", "title": "Mixture normal model", "text": "Mixture normal model ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20191005_PyMC3_SpikeSlab.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20191005_PyMC3_SpikeSlab.html#Mixture-normal-model"},
{"mainTitle": "Logistic regression spike slab prior using PyMC3", "title": "Sampling", "text": "Sampling ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20191005_PyMC3_SpikeSlab.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20191005_PyMC3_SpikeSlab.html#Sampling"},
{"mainTitle": "Logistic regression spike slab prior using PyMC3", "title": "Results", "text": "Results ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20191005_PyMC3_SpikeSlab.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20191005_PyMC3_SpikeSlab.html#Results"},
{"mainTitle": "Logistic regression spike slab prior using PyMC3", "title": "Mixture normal model 2", "text": "Mixture normal model 2 ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20191005_PyMC3_SpikeSlab.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20191005_PyMC3_SpikeSlab.html#Mixture-normal-model-2"},
{"mainTitle": "Logistic regression spike slab prior using PyMC3", "title": "Some sanity check", "text": "Some sanity check ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20191005_PyMC3_SpikeSlab.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20191005_PyMC3_SpikeSlab.html#Some-sanity-check"},
{"mainTitle": "Use varbvs result to compare", "title": "Use varbvs result to compare", "text": "Use varbvs result to compare ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20191009_varbvs_gene_block.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20191009_varbvs_gene_block.html#Use-varbvs-result-to-compare"},
{"mainTitle": "Use varbvs result to compare", "title": "FDR", "text": "FDR ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20191009_varbvs_gene_block.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20191009_varbvs_gene_block.html#FDR"},
{"mainTitle": "Use varbvs result to compare", "title": "Power", "text": "Power ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20191009_varbvs_gene_block.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20191009_varbvs_gene_block.html#Power"},
{"mainTitle": "ROC curve comparison PyMC3, susie, varbvs, logit", "title": "ROC curve comparison PyMC3, susie, varbvs, logit", "text": "ROC curve comparison PyMC3, susie, varbvs, logit ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20200109_FDR_power_curve.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20200109_FDR_power_curve.html#ROC-curve-comparison-PyMC3,-susie,-varbvs,-logit"},
{"mainTitle": "ROC curve comparison PyMC3, susie, varbvs, logit", "title": "block20", "text": "block20 ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20200109_FDR_power_curve.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20200109_FDR_power_curve.html#block20"},
{"mainTitle": "ROC curve comparison PyMC3, susie, varbvs, logit", "title": "block 30", "text": "block 30 ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20200109_FDR_power_curve.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20200109_FDR_power_curve.html#block-30"},
{"mainTitle": "ROC curve comparison PyMC3, susie, varbvs, logit", "title": "Drawing", "text": "Drawing ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20200109_FDR_power_curve.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20200109_FDR_power_curve.html#Drawing"},
{"mainTitle": "ROC curve comparison PyMC3, susie, varbvs, logit", "title": "FDR power curve", "text": "FDR power curve ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20200109_FDR_power_curve.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20200109_FDR_power_curve.html#FDR-power-curve"},
{"mainTitle": "Simulation process, analysis and result", "title": "Simulation process, analysis and result", "text": "Simulation process, analysis and result ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/Simulation_summary.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/Simulation_summary.html#Simulation-process,-analysis-and-result"},
{"mainTitle": "Simulation process, analysis and result", "title": "Introduction and aims", "text": "Introduction and aims ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/Simulation_summary.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/Simulation_summary.html#Introduction-and-aims"},
{"mainTitle": "Simulation process, analysis and result", "title": "Simulation steps", "text": "Simulation steps ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/Simulation_summary.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/Simulation_summary.html#Simulation-steps"},
{"mainTitle": "Simulation process, analysis and result", "title": "Simulation parameters", "text": "Simulation parameters ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/Simulation_summary.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/Simulation_summary.html#Simulation-parameters"},
{"mainTitle": "Simulation process, analysis and result", "title": "Analysis methods", "text": "Analysis methods ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/Simulation_summary.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/Simulation_summary.html#Analysis-methods"},
{"mainTitle": "Simulation process, analysis and result", "title": "PyMC3", "text": "PyMC3 ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/Simulation_summary.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/Simulation_summary.html#PyMC3"},
{"mainTitle": "Simulation process, analysis and result", "title": "Results", "text": "Results ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/Simulation_summary.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/Simulation_summary.html#Results"},
{"mainTitle": "Useful Notes and links", "title": "Useful Notes and links", "text": "Useful Notes and links ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/Installation_notes_links.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/Installation_notes_links.html#Useful-Notes-and-links"},
{"mainTitle": "Useful Notes and links", "title": "intall SoS, sos-notebook, jupyter-lab", "text": "intall SoS, sos-notebook, jupyter-lab ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/Installation_notes_links.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/Installation_notes_links.html#intall-SoS,-sos-notebook,-jupyter-lab"},
{"mainTitle": "Useful Notes and links", "title": "install susieR", "text": "install susieR ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/Installation_notes_links.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/Installation_notes_links.html#install-susieR"},
{"mainTitle": "Simulation and analysis result overview", "title": "Simulation and analysis result overview", "text": "Simulation and analysis result overview ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20190827_varbvs_whole_genome.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20190827_varbvs_whole_genome.html#Simulation-and-analysis-result-overview"},
{"mainTitle": "Simulation and analysis result overview", "title": "Run varbvs on whole simulated genome for specifying priors", "text": "Run varbvs on whole simulated genome for specifying priors ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20190827_varbvs_whole_genome.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20190827_varbvs_whole_genome.html#Run-varbvs-on-whole-simulated-genome-for-specifying-priors"},
{"mainTitle": "Simulation and analysis result overview", "title": "SuSiE", "text": "SuSiE ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20190827_varbvs_whole_genome.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20190827_varbvs_whole_genome.html#SuSiE"},
{"mainTitle": "Simulation and analysis result overview", "title": "varbvs", "text": "varbvs ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20190827_varbvs_whole_genome.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20190827_varbvs_whole_genome.html#varbvs"},
{"mainTitle": "Obtain clean reference gene", "title": "Obtain clean reference gene", "text": "Obtain clean reference gene ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20190627_Clean_RefGene.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20190627_Clean_RefGene.html#Obtain-clean-reference-gene"},
{"mainTitle": "Obtain clean reference gene", "title": "Import original genes/exons", "text": "Import original genes exons ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20190627_Clean_RefGene.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20190627_Clean_RefGene.html#Import-original-genes/exons"},
{"mainTitle": "Obtain clean reference gene", "title": "Remove chromosome Y, M and unmatched", "text": "Remove chromosome Y, M and unmatched ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20190627_Clean_RefGene.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20190627_Clean_RefGene.html#Remove-chromosome-Y,-M-and-unmatched"},
{"mainTitle": "Obtain clean reference gene", "title": "Drop genes that share same chr, start, end and exon/gene name (44322 left)", "text": "Drop genes that share same chr, start, end and exon gene name 44322 left ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20190627_Clean_RefGene.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20190627_Clean_RefGene.html#Drop-genes-that-share-same-chr,-start,-end-and-exon/gene-name-(44322-left)"},
{"mainTitle": "Obtain clean reference gene", "title": "Mark the genes and adjust gene names", "text": "Mark the genes and adjust gene names ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20190627_Clean_RefGene.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20190627_Clean_RefGene.html#Mark-the-genes-and-adjust-gene-names"},
{"mainTitle": "Obtain clean reference gene", "title": "Rename duplicated genes", "text": "Rename duplicated genes ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20190627_Clean_RefGene.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20190627_Clean_RefGene.html#Rename-duplicated-genes"},
{"mainTitle": "Obtain clean reference gene", "title": "Drop genes that share same chr, start and end (43442 left)", "text": "Drop genes that share same chr, start and end 43442 left ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20190627_Clean_RefGene.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20190627_Clean_RefGene.html#Drop-genes-that-share-same-chr,-start-and-end-(43442-left)"},
{"mainTitle": "Obtain clean reference gene", "title": "Check for validity", "text": "Check for validity ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20190627_Clean_RefGene.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20190627_Clean_RefGene.html#Check-for-validity"},
{"mainTitle": "Single effect CNV-gene pattern report using new logistic and other methods", "title": "Single effect CNV-gene pattern report using new logistic and other methods", "text": "Single effect CNV-gene pattern report using new logistic and other methods ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20191023_logistic_report.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20191023_logistic_report.html#Single-effect-CNV-gene-pattern-report-using-new-logistic-and-other-methods"},
{"mainTitle": "Single effect CNV-gene pattern report using new logistic and other methods", "title": "Single effect result summary", "text": "Single effect result summary ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20191023_logistic_report.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20191023_logistic_report.html#Single-effect-result-summary"},
{"mainTitle": "Single effect CNV-gene pattern report using new logistic and other methods", "title": "Questions", "text": "Questions ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20191023_logistic_report.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20191023_logistic_report.html#Questions"},
{"mainTitle": "Single effect CNV-gene pattern report using new logistic and other methods", "title": "Single effect results", "text": "Single effect results ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20191023_logistic_report.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20191023_logistic_report.html#Single-effect-results"},
{"mainTitle": "Single effect CNV-gene pattern report using new logistic and other methods", "title": "Block # 739_743", "text": "Block 739_743 ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20191023_logistic_report.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20191023_logistic_report.html#Block-#-739_743"},
{"mainTitle": "Single effect CNV-gene pattern report using new logistic and other methods", "title": "Correlation (Table 1)", "text": "Correlation Table 1 ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20191023_logistic_report.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20191023_logistic_report.html#Correlation-(Table-1)"},
{"mainTitle": "Single effect CNV-gene pattern report using new logistic and other methods", "title": "Fisher test (Table 2)", "text": "Fisher test Table 2 ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20191023_logistic_report.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20191023_logistic_report.html#Fisher-test-(Table-2)"},
{"mainTitle": "Single effect CNV-gene pattern report using new logistic and other methods", "title": "logistic (newly implemented by Peter) posterior inclusion probabilities, and the posterior, means and standard deviations (Table 3)", "text": "logistic newly implemented by Peter posterior inclusion probabilities, and the posterior, means and standard deviations Table 3 ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20191023_logistic_report.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20191023_logistic_report.html#logistic-(newly-implemented-by-Peter)-posterior-inclusion-probabilities,-and-the-posterior,-means-and-standard-deviations-(Table-3)"},
{"mainTitle": "Single effect CNV-gene pattern report using new logistic and other methods", "title": "SuSiE (Table 4)", "text": "SuSiE Table 4 ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20191023_logistic_report.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20191023_logistic_report.html#SuSiE-(Table-4)"},
{"mainTitle": "Single effect CNV-gene pattern report using new logistic and other methods", "title": "varbvs (Table 5)", "text": "varbvs Table 5 ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20191023_logistic_report.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20191023_logistic_report.html#varbvs-(Table-5)"},
{"mainTitle": "Single effect CNV-gene pattern report using new logistic and other methods", "title": "PyMC3 Spike-and-slab model (Table 6)", "text": "PyMC3 Spike-and-slab model Table 6 ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20191023_logistic_report.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20191023_logistic_report.html#PyMC3-Spike-and-slab-model-(Table-6)"},
{"mainTitle": "Single effect CNV-gene pattern report using new logistic and other methods", "title": "PyMC3 Mixture normal model (Table 7)", "text": "PyMC3 Mixture normal model Table 7 ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20191023_logistic_report.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20191023_logistic_report.html#PyMC3-Mixture-normal-model-(Table-7)"},
{"mainTitle": "Obtain Genome Window (40~100 genes per window)", "title": "Obtain Genome Window (40~100 genes per window)", "text": "Obtain Genome Window 40 100 genes per window ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20190710_dele_Genome_window.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20190710_dele_Genome_window.html#Obtain-Genome-Window-(40~100-genes-per-window)"},
{"mainTitle": "Introduction (Use dot for numbering)", "title": "Introduction (Use dot for numbering)", "text": "Introduction Use dot for numbering ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20191012_ASHG_poster.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20191012_ASHG_poster.html#Introduction-(Use-dot-for-numbering)"},
{"mainTitle": "Introduction (Use dot for numbering)", "title": "Motivation and Aim", "text": "Motivation and Aim ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20191012_ASHG_poster.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20191012_ASHG_poster.html#Motivation-and-Aim"},
{"mainTitle": "Introduction (Use dot for numbering)", "title": "Data available", "text": "Data available ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20191012_ASHG_poster.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20191012_ASHG_poster.html#Data-available"},
{"mainTitle": "Introduction (Use dot for numbering)", "title": "Approach and algorighm", "text": "Approach and algorighm ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20191012_ASHG_poster.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20191012_ASHG_poster.html#Approach-and-algorighm"},
{"mainTitle": "Introduction (Use dot for numbering)", "title": "Simulation processes", "text": "Simulation processes ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20191012_ASHG_poster.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20191012_ASHG_poster.html#Simulation-processes"},
{"mainTitle": "Introduction (Use dot for numbering)", "title": "Results", "text": "Results ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20191012_ASHG_poster.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20191012_ASHG_poster.html#Results"},
{"mainTitle": "Introduction (Use dot for numbering)", "title": "Reference", "text": "Reference ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20191012_ASHG_poster.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20191012_ASHG_poster.html#Reference"},
{"mainTitle": "Introduction (Use dot for numbering)", "title": "ASHG talks", "text": "ASHG talks ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20191012_ASHG_poster.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20191012_ASHG_poster.html#ASHG-talks"},
{"mainTitle": "Introduction (Use dot for numbering)", "title": "ASHG posters", "text": "ASHG posters ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20191012_ASHG_poster.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20191012_ASHG_poster.html#ASHG-posters"},
{"mainTitle": "Obtain X matrix", "title": "Obtain X matrix", "text": "Obtain X matrix ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20190628_Xmatrix.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20190628_Xmatrix.html#Obtain-X-matrix"},
{"mainTitle": "Obtain X matrix", "title": "Import phenotype", "text": "Import phenotype ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20190628_Xmatrix.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20190628_Xmatrix.html#Import-phenotype"},
{"mainTitle": "Obtain X matrix", "title": "Import individual CNV", "text": "Import individual CNV ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20190628_Xmatrix.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20190628_Xmatrix.html#Import-individual-CNV"},
{"mainTitle": "Obtain X matrix", "title": "Import reference genes", "text": "Import reference genes ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/dsc/20190628_Xmatrix.html", "url": "http://gaow.github.io/cnv-gene-mapping/dsc/20190628_Xmatrix.html#Import-reference-genes"},
{"mainTitle": "plink", "title": "plink", "text": "plink ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/setup/plink.html", "url": "http://gaow.github.io/cnv-gene-mapping/setup/plink.html#plink"},
{"mainTitle": "install saasCNV", "title": "install saasCNV", "text": "install saasCNV ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/setup/index.html", "url": "http://gaow.github.io/cnv-gene-mapping/setup/index.html#install-saasCNV"},
{"mainTitle": "Literature notes", "title": "Literature notes", "text": "Literature notes ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/writeup/Literature_Notes.html", "url": "http://gaow.github.io/cnv-gene-mapping/writeup/Literature_Notes.html#Literature-notes"},
{"mainTitle": "Literature notes", "title": "Pinto 2014 AJHG", "text": "Pinto 2014 AJHG ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/writeup/Literature_Notes.html", "url": "http://gaow.github.io/cnv-gene-mapping/writeup/Literature_Notes.html#Pinto-2014-AJHG"},
{"mainTitle": "Literature notes", "title": "Copy-number variations (CNVs)", "text": "Copy-number variations CNVs ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/writeup/Literature_Notes.html", "url": "http://gaow.github.io/cnv-gene-mapping/writeup/Literature_Notes.html#Copy-number-variations-(CNVs)"},
{"mainTitle": "Literature notes", "title": "Autism spectrum disorders (ASDs)", "text": "Autism spectrum disorders ASDs ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/writeup/Literature_Notes.html", "url": "http://gaow.github.io/cnv-gene-mapping/writeup/Literature_Notes.html#Autism-spectrum-disorders-(ASDs)"},
{"mainTitle": "Literature notes", "title": "Subjects of the study", "text": "Subjects of the study ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/writeup/Literature_Notes.html", "url": "http://gaow.github.io/cnv-gene-mapping/writeup/Literature_Notes.html#Subjects-of-the-study"},
{"mainTitle": "Literature notes", "title": "Aim/goal", "text": "Aim goal ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/writeup/Literature_Notes.html", "url": "http://gaow.github.io/cnv-gene-mapping/writeup/Literature_Notes.html#Aim/goal"},
{"mainTitle": "Literature notes", "title": "Data/CNV analysis", "text": "Data CNV analysis ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/writeup/Literature_Notes.html", "url": "http://gaow.github.io/cnv-gene-mapping/writeup/Literature_Notes.html#Data/CNV-analysis"},
{"mainTitle": "Literature notes", "title": "One/multiple-gene-hit burden analysis", "text": "One multiple-gene-hit burden analysis ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/writeup/Literature_Notes.html", "url": "http://gaow.github.io/cnv-gene-mapping/writeup/Literature_Notes.html#One/multiple-gene-hit-burden-analysis"},
{"mainTitle": "Literature notes", "title": "Results", "text": "Results ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/writeup/Literature_Notes.html", "url": "http://gaow.github.io/cnv-gene-mapping/writeup/Literature_Notes.html#Results"},
{"mainTitle": "Literature notes", "title": "Excess burden of rare and de novo CNVs", "text": "Excess burden of rare and de novo CNVs ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/writeup/Literature_Notes.html", "url": "http://gaow.github.io/cnv-gene-mapping/writeup/Literature_Notes.html#Excess-burden-of-rare-and-de-novo-CNVs"},
{"mainTitle": "Literature notes", "title": "CNV burden in autosomal-dominant or X-linked genes implicated in ASD/ID", "text": "CNV burden in autosomal-dominant or X-linked genes implicated in ASD ID ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/writeup/Literature_Notes.html", "url": "http://gaow.github.io/cnv-gene-mapping/writeup/Literature_Notes.html#CNV-burden-in-autosomal-dominant-or-X-linked-genes-implicated-in-ASD/ID"},
{"mainTitle": "Literature notes", "title": "Increased multigene burden in ASD-affected subjects", "text": "Increased multigene burden in ASD-affected subjects ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/writeup/Literature_Notes.html", "url": "http://gaow.github.io/cnv-gene-mapping/writeup/Literature_Notes.html#Increased-multigene-burden-in-ASD-affected-subjects"},
{"mainTitle": "Literature notes", "title": "De novo CNVs and Loss-of-Function SNVs converge on functional gene networks", "text": "De novo CNVs and Loss-of-Function SNVs converge on functional gene networks ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/writeup/Literature_Notes.html", "url": "http://gaow.github.io/cnv-gene-mapping/writeup/Literature_Notes.html#De-novo-CNVs-and-Loss-of-Function-SNVs-converge-on-functional-gene-networks"},
{"mainTitle": "Literature notes", "title": "Girirajan ARG 2011", "text": "Girirajan ARG 2011 ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/writeup/Literature_Notes.html", "url": "http://gaow.github.io/cnv-gene-mapping/writeup/Literature_Notes.html#Girirajan-ARG-2011"},
{"mainTitle": "Learning resources", "title": "Learning resources", "text": "Learning resources ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/writeup/Learning_Resources.html", "url": "http://gaow.github.io/cnv-gene-mapping/writeup/Learning_Resources.html#Learning-resources"},
{"mainTitle": "Learning resources", "title": "NGS", "text": "NGS ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/writeup/Learning_Resources.html", "url": "http://gaow.github.io/cnv-gene-mapping/writeup/Learning_Resources.html#NGS"},
{"mainTitle": "Learning resources", "title": "CNV lecture notes: identify CNV", "text": "CNV lecture notes identify CNV ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/writeup/Learning_Resources.html", "url": "http://gaow.github.io/cnv-gene-mapping/writeup/Learning_Resources.html#CNV-lecture-notes:-identify-CNV"},
{"mainTitle": "Learning resources", "title": "Bioinformatics course", "text": "Bioinformatics course ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/writeup/Learning_Resources.html", "url": "http://gaow.github.io/cnv-gene-mapping/writeup/Learning_Resources.html#Bioinformatics-course"},
{"mainTitle": "Learning resources", "title": "AWK", "text": "AWK ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/writeup/Learning_Resources.html", "url": "http://gaow.github.io/cnv-gene-mapping/writeup/Learning_Resources.html#AWK"},
{"mainTitle": "Learning resources", "title": "BDA book: Bayesian Data analysis / Gelman et al.", "text": "BDA book Bayesian Data analysis Gelman et al. ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/writeup/Learning_Resources.html", "url": "http://gaow.github.io/cnv-gene-mapping/writeup/Learning_Resources.html#BDA-book:-Bayesian-Data-analysis-/-Gelman-et-al."},
{"mainTitle": "Learning resources", "title": "Machine Learning: a probabilistic perspective", "text": "Machine Learning a probabilistic perspective ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/writeup/Learning_Resources.html", "url": "http://gaow.github.io/cnv-gene-mapping/writeup/Learning_Resources.html#Machine-Learning:-a-probabilistic-perspective"},
{"mainTitle": "Learning resources", "title": "stephens999/stat302", "text": "stephens999 stat302 ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/writeup/Learning_Resources.html", "url": "http://gaow.github.io/cnv-gene-mapping/writeup/Learning_Resources.html#stephens999/stat302"},
{"mainTitle": "Learning resources", "title": "The Art of R Programming: A Tour of Statistical Software Design", "text": "The Art of R Programming A Tour of Statistical Software Design ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/writeup/Learning_Resources.html", "url": "http://gaow.github.io/cnv-gene-mapping/writeup/Learning_Resources.html#The-Art-of-R-Programming:-A-Tour-of-Statistical-Software-Design"},
{"mainTitle": "Glossary", "title": "Glossary", "text": "Glossary ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/writeup/Glossary.html", "url": "http://gaow.github.io/cnv-gene-mapping/writeup/Glossary.html#Glossary"},
{"mainTitle": "Glossary", "title": "Bioinformatics", "text": "Bioinformatics ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/writeup/Glossary.html", "url": "http://gaow.github.io/cnv-gene-mapping/writeup/Glossary.html#Bioinformatics"},
{"mainTitle": "Glossary", "title": "what is bioinformatics?", "text": "what is bioinformatics ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/writeup/Glossary.html", "url": "http://gaow.github.io/cnv-gene-mapping/writeup/Glossary.html#what-is-bioinformatics?"},
{"mainTitle": "Glossary", "title": "alignment", "text": "alignment ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/writeup/Glossary.html", "url": "http://gaow.github.io/cnv-gene-mapping/writeup/Glossary.html#alignment"},
{"mainTitle": "Glossary", "title": "DBMS/database management system", "text": "DBMS database management system ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/writeup/Glossary.html", "url": "http://gaow.github.io/cnv-gene-mapping/writeup/Glossary.html#DBMS/database-management-system"},
{"mainTitle": "Glossary", "title": "Bioinformatics resource", "text": "Bioinformatics resource ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/writeup/Glossary.html", "url": "http://gaow.github.io/cnv-gene-mapping/writeup/Glossary.html#Bioinformatics-resource"},
{"mainTitle": "Glossary", "title": "CNV notes", "text": "CNV notes ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/writeup/Glossary.html", "url": "http://gaow.github.io/cnv-gene-mapping/writeup/Glossary.html#CNV-notes"},
{"mainTitle": "Glossary", "title": "Structural variation (SV)", "text": "Structural variation SV ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/writeup/Glossary.html", "url": "http://gaow.github.io/cnv-gene-mapping/writeup/Glossary.html#Structural-variation-(SV)"},
{"mainTitle": "Glossary", "title": "copy number variation", "text": "copy number variation ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/writeup/Glossary.html", "url": "http://gaow.github.io/cnv-gene-mapping/writeup/Glossary.html#copy-number-variation"},
{"mainTitle": "Glossary", "title": "De novo mutation", "text": "De novo mutation ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/writeup/Glossary.html", "url": "http://gaow.github.io/cnv-gene-mapping/writeup/Glossary.html#De-novo-mutation"},
{"mainTitle": "Glossary", "title": "SNV with coding regions", "text": "SNV with coding regions ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/writeup/Glossary.html", "url": "http://gaow.github.io/cnv-gene-mapping/writeup/Glossary.html#SNV-with-coding-regions"},
{"mainTitle": "Glossary", "title": "NGS", "text": "NGS ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/writeup/Glossary.html", "url": "http://gaow.github.io/cnv-gene-mapping/writeup/Glossary.html#NGS"},
{"mainTitle": "Glossary", "title": "Sanger sequencing", "text": "Sanger sequencing ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/writeup/Glossary.html", "url": "http://gaow.github.io/cnv-gene-mapping/writeup/Glossary.html#Sanger-sequencing"},
{"mainTitle": "Glossary", "title": "Next generation sequencing/deep sequencing", "text": "Next generation sequencing deep sequencing ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/writeup/Glossary.html", "url": "http://gaow.github.io/cnv-gene-mapping/writeup/Glossary.html#Next-generation-sequencing/deep-sequencing"},
{"mainTitle": "Glossary", "title": "NGS Applications", "text": "NGS Applications ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/writeup/Glossary.html", "url": "http://gaow.github.io/cnv-gene-mapping/writeup/Glossary.html#NGS-Applications"},
{"mainTitle": "Glossary", "title": "Genomics", "text": "Genomics ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/writeup/Glossary.html", "url": "http://gaow.github.io/cnv-gene-mapping/writeup/Glossary.html#Genomics"},
{"mainTitle": "Glossary", "title": "Main Applications of NGS in Genetics", "text": "Main Applications of NGS in Genetics ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/writeup/Glossary.html", "url": "http://gaow.github.io/cnv-gene-mapping/writeup/Glossary.html#Main-Applications-of-NGS-in-Genetics"},
{"mainTitle": "Glossary", "title": "Structural Variation", "text": "Structural Variation ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/writeup/Glossary.html", "url": "http://gaow.github.io/cnv-gene-mapping/writeup/Glossary.html#Structural-Variation"},
{"mainTitle": "Glossary", "title": "Methods for discovery and then for genotyping", "text": "Methods for discovery and then for genotyping ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/writeup/Glossary.html", "url": "http://gaow.github.io/cnv-gene-mapping/writeup/Glossary.html#Methods-for-discovery-and-then-for-genotyping"},
{"mainTitle": "Glossary", "title": "Hybridization-based microarray approaches", "text": "Hybridization-based microarray approaches ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/writeup/Glossary.html", "url": "http://gaow.github.io/cnv-gene-mapping/writeup/Glossary.html#Hybridization-based-microarray-approaches"},
{"mainTitle": "Glossary", "title": "Array CGH", "text": "Array CGH ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/writeup/Glossary.html", "url": "http://gaow.github.io/cnv-gene-mapping/writeup/Glossary.html#Array-CGH"},
{"mainTitle": "Glossary", "title": "SNP arrays", "text": "SNP arrays ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/writeup/Glossary.html", "url": "http://gaow.github.io/cnv-gene-mapping/writeup/Glossary.html#SNP-arrays"},
{"mainTitle": "Integration of CNV and SNV for gene mapping of psychiatric disorders: methods and data analysis", "title": "Integration of CNV and SNV for gene mapping of psychiatric disorders: methods and data analysis", "text": "Integration of CNV and SNV for gene mapping of psychiatric disorders methods and data analysis ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/writeup/dbGaP_Proposal.html", "url": "http://gaow.github.io/cnv-gene-mapping/writeup/dbGaP_Proposal.html#Integration-of-CNV-and-SNV-for-gene-mapping-of-psychiatric-disorders:-methods-and-data-analysis"},
{"mainTitle": "Integration of CNV and SNV for gene mapping of psychiatric disorders: methods and data analysis", "title": "Non-technical summary", "text": "Non-technical summary ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/writeup/dbGaP_Proposal.html", "url": "http://gaow.github.io/cnv-gene-mapping/writeup/dbGaP_Proposal.html#Non-technical-summary"},
{"mainTitle": "Integration of CNV and SNV for gene mapping of psychiatric disorders: methods and data analysis", "title": "Research Use Statement (RUS)", "text": "Research Use Statement RUS ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/writeup/dbGaP_Proposal.html", "url": "http://gaow.github.io/cnv-gene-mapping/writeup/dbGaP_Proposal.html#Research-Use-Statement-(RUS)"},
{"mainTitle": "CNV Abstract", "title": "CNV Abstract", "text": "CNV Abstract ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/writeup/20190603_ASHG_Abstract.html", "url": "http://gaow.github.io/cnv-gene-mapping/writeup/20190603_ASHG_Abstract.html#CNV-Abstract"},
{"mainTitle": "CNV Abstract", "title": "Me", "text": "Me ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/writeup/20190603_ASHG_Abstract.html", "url": "http://gaow.github.io/cnv-gene-mapping/writeup/20190603_ASHG_Abstract.html#Me"},
{"mainTitle": "CNV Abstract", "title": "Gao", "text": "Gao ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/writeup/20190603_ASHG_Abstract.html", "url": "http://gaow.github.io/cnv-gene-mapping/writeup/20190603_ASHG_Abstract.html#Gao"},
{"mainTitle": "CNV Abstract", "title": "Dr. He", "text": "Dr. He ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/writeup/20190603_ASHG_Abstract.html", "url": "http://gaow.github.io/cnv-gene-mapping/writeup/20190603_ASHG_Abstract.html#Dr.-He"},
{"mainTitle": "CNV Abstract", "title": "Final", "text": "Final ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/writeup/20190603_ASHG_Abstract.html", "url": "http://gaow.github.io/cnv-gene-mapping/writeup/20190603_ASHG_Abstract.html#Final"},
{"mainTitle": "CNV Abstract", "title": "RCC MindBytes", "text": "RCC MindBytes ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/writeup/20190603_ASHG_Abstract.html", "url": "http://gaow.github.io/cnv-gene-mapping/writeup/20190603_ASHG_Abstract.html#RCC-MindBytes"},
{"mainTitle": "Gene Mapping with CNV", "title": "CNV meeting notes starting from June 2017", "text": "CNV meeting notes starting from June 2017 ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/writeup/CNV_meeting_2017.html", "url": "http://gaow.github.io/cnv-gene-mapping/writeup/CNV_meeting_2017.html#CNV-meeting-notes-starting-from-June-2017"},
{"mainTitle": "Gene Mapping with CNV", "title": "Notes for meeting 06/02/2017", "text": "Notes for meeting 06 02 2017 ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/writeup/CNV_meeting_2017.html", "url": "http://gaow.github.io/cnv-gene-mapping/writeup/CNV_meeting_2017.html#Notes-for-meeting-06/02/2017"},
{"mainTitle": "Gene Mapping with CNV", "title": "Notes for meeting 06/15/2017", "text": "Notes for meeting 06 15 2017 ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/writeup/CNV_meeting_2017.html", "url": "http://gaow.github.io/cnv-gene-mapping/writeup/CNV_meeting_2017.html#Notes-for-meeting-06/15/2017"},
{"mainTitle": "Gene Mapping with CNV", "title": "Notes for meeting 07/21/2017", "text": "Notes for meeting 07 21 2017 ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/writeup/CNV_meeting_2017.html", "url": "http://gaow.github.io/cnv-gene-mapping/writeup/CNV_meeting_2017.html#Notes-for-meeting-07/21/2017"},
{"mainTitle": "Gene Mapping with CNV", "title": "Notes for meeting 08/22/2017", "text": "Notes for meeting 08 22 2017 ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/writeup/CNV_meeting_2017.html", "url": "http://gaow.github.io/cnv-gene-mapping/writeup/CNV_meeting_2017.html#Notes-for-meeting-08/22/2017"},
{"mainTitle": "Gene Mapping with CNV", "title": "Notes for metting 09/19/2017", "text": "Notes for metting 09 19 2017 ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/writeup/CNV_meeting_2017.html", "url": "http://gaow.github.io/cnv-gene-mapping/writeup/CNV_meeting_2017.html#Notes-for-metting-09/19/2017"},
{"mainTitle": "Gene Mapping with CNV", "title": "Notes for meeting 11/17/17", "text": "Notes for meeting 11 17 17 ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/writeup/CNV_meeting_2017.html", "url": "http://gaow.github.io/cnv-gene-mapping/writeup/CNV_meeting_2017.html#Notes-for-meeting-11/17/17"},
{"mainTitle": "Project meeting 20161208", "title": "Project meeting 20161208", "text": "Project meeting 20161208 ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/writeup/CNV_meeting_2016.html", "url": "http://gaow.github.io/cnv-gene-mapping/writeup/CNV_meeting_2016.html#Project-meeting-20161208"},
{"mainTitle": "Project meeting 20161208", "title": "Modify enrichment analysis", "text": "Modify enrichment analysis ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/writeup/CNV_meeting_2016.html", "url": "http://gaow.github.io/cnv-gene-mapping/writeup/CNV_meeting_2016.html#Modify-enrichment-analysis"},
{"mainTitle": "Project meeting 20161208", "title": "Simulation steps", "text": "Simulation steps ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/writeup/CNV_meeting_2016.html", "url": "http://gaow.github.io/cnv-gene-mapping/writeup/CNV_meeting_2016.html#Simulation-steps"},
{"mainTitle": "Project meeting 20161208", "title": "Project meeting 20160919", "text": "Project meeting 20160919 ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/writeup/CNV_meeting_2016.html", "url": "http://gaow.github.io/cnv-gene-mapping/writeup/CNV_meeting_2016.html#Project-meeting-20160919"},
{"mainTitle": "Project meeting 20161208", "title": "Status", "text": "Status ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/writeup/CNV_meeting_2016.html", "url": "http://gaow.github.io/cnv-gene-mapping/writeup/CNV_meeting_2016.html#Status"},
{"mainTitle": "Project meeting 20161208", "title": "Done", "text": "Done ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/writeup/CNV_meeting_2016.html", "url": "http://gaow.github.io/cnv-gene-mapping/writeup/CNV_meeting_2016.html#Done"},
{"mainTitle": "Project meeting 20161208", "title": "Doing", "text": "Doing ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/writeup/CNV_meeting_2016.html", "url": "http://gaow.github.io/cnv-gene-mapping/writeup/CNV_meeting_2016.html#Doing"},
{"mainTitle": "Project meeting 20161208", "title": "Todo", "text": "Todo ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/writeup/CNV_meeting_2016.html", "url": "http://gaow.github.io/cnv-gene-mapping/writeup/CNV_meeting_2016.html#Todo"},
{"mainTitle": "CNV datasets", "title": "CNV datasets", "text": "CNV datasets ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/writeup/Dataset_List.html", "url": "http://gaow.github.io/cnv-gene-mapping/writeup/Dataset_List.html#CNV-datasets"},
{"mainTitle": "Useful Info", "title": "Useful Info", "text": "Useful Info ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/writeup/Useful_Info.html", "url": "http://gaow.github.io/cnv-gene-mapping/writeup/Useful_Info.html#Useful-Info"},
{"mainTitle": "Useful Info", "title": "dbGaP Data access requests", "text": "dbGaP Data access requests ", "tags": "", "mainUrl": "http://gaow.github.io/cnv-gene-mapping/writeup/Useful_Info.html", "url": "http://gaow.github.io/cnv-gene-mapping/writeup/Useful_Info.html#dbGaP-Data-access-requests"}
]};
